// out/eks/Cluster.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.eks.Cluster {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.EKS();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/eks-2017-11-01.normal.json"), this.client);
    }
    associateEncryptionConfig(partialParams) {
        return this.client.associateEncryptionConfig(this.ops["AssociateEncryptionConfig"].apply(partialParams));
    }
    associateIdentityProviderConfig(partialParams) {
        return this.client.associateIdentityProviderConfig(this.ops["AssociateIdentityProviderConfig"].apply(partialParams));
    }
    createAddon(partialParams) {
        return this.client.createAddon(this.ops["CreateAddon"].apply(partialParams));
    }
    createCluster(partialParams) {
        return this.client.createCluster(this.ops["CreateCluster"].apply(partialParams));
    }
    createFargateProfile(partialParams) {
        return this.client.createFargateProfile(this.ops["CreateFargateProfile"].apply(partialParams));
    }
    createNodegroup(partialParams) {
        return this.client.createNodegroup(this.ops["CreateNodegroup"].apply(partialParams));
    }
    deleteAddon(partialParams) {
        return this.client.deleteAddon(this.ops["DeleteAddon"].apply(partialParams));
    }
    deleteCluster(partialParams) {
        return this.client.deleteCluster(this.ops["DeleteCluster"].apply(partialParams));
    }
    deleteFargateProfile(partialParams) {
        return this.client.deleteFargateProfile(this.ops["DeleteFargateProfile"].apply(partialParams));
    }
    deleteNodegroup(partialParams) {
        return this.client.deleteNodegroup(this.ops["DeleteNodegroup"].apply(partialParams));
    }
    deregisterCluster(partialParams) {
        return this.client.deregisterCluster(this.ops["DeregisterCluster"].apply(partialParams));
    }
    describeAddon(partialParams) {
        return this.client.describeAddon(this.ops["DescribeAddon"].apply(partialParams));
    }
    describeCluster(partialParams) {
        return this.client.describeCluster(this.ops["DescribeCluster"].apply(partialParams));
    }
    describeFargateProfile(partialParams) {
        return this.client.describeFargateProfile(this.ops["DescribeFargateProfile"].apply(partialParams));
    }
    describeIdentityProviderConfig(partialParams) {
        return this.client.describeIdentityProviderConfig(this.ops["DescribeIdentityProviderConfig"].apply(partialParams));
    }
    describeNodegroup(partialParams) {
        return this.client.describeNodegroup(this.ops["DescribeNodegroup"].apply(partialParams));
    }
    describeUpdate(partialParams) {
        return this.client.describeUpdate(this.ops["DescribeUpdate"].apply(partialParams));
    }
    disassociateIdentityProviderConfig(partialParams) {
        return this.client.disassociateIdentityProviderConfig(this.ops["DisassociateIdentityProviderConfig"].apply(partialParams));
    }
    listAddons(partialParams) {
        return this.client.listAddons(this.ops["ListAddons"].apply(partialParams));
    }
    listFargateProfiles(partialParams) {
        return this.client.listFargateProfiles(this.ops["ListFargateProfiles"].apply(partialParams));
    }
    listIdentityProviderConfigs(partialParams) {
        return this.client.listIdentityProviderConfigs(this.ops["ListIdentityProviderConfigs"].apply(partialParams));
    }
    listNodegroups(partialParams) {
        return this.client.listNodegroups(this.ops["ListNodegroups"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listUpdates(partialParams) {
        return this.client.listUpdates(this.ops["ListUpdates"].apply(partialParams));
    }
    registerCluster(partialParams) {
        return this.client.registerCluster(this.ops["RegisterCluster"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAddon(partialParams) {
        return this.client.updateAddon(this.ops["UpdateAddon"].apply(partialParams));
    }
    updateClusterConfig(partialParams) {
        return this.client.updateClusterConfig(this.ops["UpdateClusterConfig"].apply(partialParams));
    }
    updateClusterVersion(partialParams) {
        return this.client.updateClusterVersion(this.ops["UpdateClusterVersion"].apply(partialParams));
    }
    updateNodegroupConfig(partialParams) {
        return this.client.updateNodegroupConfig(this.ops["UpdateNodegroupConfig"].apply(partialParams));
    }
    updateNodegroupVersion(partialParams) {
        return this.client.updateNodegroupVersion(this.ops["UpdateNodegroupVersion"].apply(partialParams));
    }
}
exports.default = Resource;
