// out/worklink/WebsiteCertificateAuthorityAssociation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.worklink.WebsiteCertificateAuthorityAssociation {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.WorkLink();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/worklink-2018-09-25.normal.json"), this.client);
    }
    associateDomain(partialParams) {
        return this.client.associateDomain(this.ops["AssociateDomain"].apply(partialParams));
    }
    associateWebsiteAuthorizationProvider(partialParams) {
        return this.client.associateWebsiteAuthorizationProvider(this.ops["AssociateWebsiteAuthorizationProvider"].apply(partialParams));
    }
    associateWebsiteCertificateAuthority(partialParams) {
        return this.client.associateWebsiteCertificateAuthority(this.ops["AssociateWebsiteCertificateAuthority"].apply(partialParams));
    }
    createFleet(partialParams) {
        return this.client.createFleet(this.ops["CreateFleet"].apply(partialParams));
    }
    deleteFleet(partialParams) {
        return this.client.deleteFleet(this.ops["DeleteFleet"].apply(partialParams));
    }
    describeAuditStreamConfiguration(partialParams) {
        return this.client.describeAuditStreamConfiguration(this.ops["DescribeAuditStreamConfiguration"].apply(partialParams));
    }
    describeCompanyNetworkConfiguration(partialParams) {
        return this.client.describeCompanyNetworkConfiguration(this.ops["DescribeCompanyNetworkConfiguration"].apply(partialParams));
    }
    describeDevice(partialParams) {
        return this.client.describeDevice(this.ops["DescribeDevice"].apply(partialParams));
    }
    describeDevicePolicyConfiguration(partialParams) {
        return this.client.describeDevicePolicyConfiguration(this.ops["DescribeDevicePolicyConfiguration"].apply(partialParams));
    }
    describeDomain(partialParams) {
        return this.client.describeDomain(this.ops["DescribeDomain"].apply(partialParams));
    }
    describeFleetMetadata(partialParams) {
        return this.client.describeFleetMetadata(this.ops["DescribeFleetMetadata"].apply(partialParams));
    }
    describeIdentityProviderConfiguration(partialParams) {
        return this.client.describeIdentityProviderConfiguration(this.ops["DescribeIdentityProviderConfiguration"].apply(partialParams));
    }
    describeWebsiteCertificateAuthority(partialParams) {
        return this.client.describeWebsiteCertificateAuthority(this.ops["DescribeWebsiteCertificateAuthority"].apply(partialParams));
    }
    disassociateDomain(partialParams) {
        return this.client.disassociateDomain(this.ops["DisassociateDomain"].apply(partialParams));
    }
    disassociateWebsiteAuthorizationProvider(partialParams) {
        return this.client.disassociateWebsiteAuthorizationProvider(this.ops["DisassociateWebsiteAuthorizationProvider"].apply(partialParams));
    }
    disassociateWebsiteCertificateAuthority(partialParams) {
        return this.client.disassociateWebsiteCertificateAuthority(this.ops["DisassociateWebsiteCertificateAuthority"].apply(partialParams));
    }
    listDevices(partialParams) {
        return this.client.listDevices(this.ops["ListDevices"].apply(partialParams));
    }
    listDomains(partialParams) {
        return this.client.listDomains(this.ops["ListDomains"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listWebsiteAuthorizationProviders(partialParams) {
        return this.client.listWebsiteAuthorizationProviders(this.ops["ListWebsiteAuthorizationProviders"].apply(partialParams));
    }
    listWebsiteCertificateAuthorities(partialParams) {
        return this.client.listWebsiteCertificateAuthorities(this.ops["ListWebsiteCertificateAuthorities"].apply(partialParams));
    }
    restoreDomainAccess(partialParams) {
        return this.client.restoreDomainAccess(this.ops["RestoreDomainAccess"].apply(partialParams));
    }
    revokeDomainAccess(partialParams) {
        return this.client.revokeDomainAccess(this.ops["RevokeDomainAccess"].apply(partialParams));
    }
    signOutUser(partialParams) {
        return this.client.signOutUser(this.ops["SignOutUser"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAuditStreamConfiguration(partialParams) {
        return this.client.updateAuditStreamConfiguration(this.ops["UpdateAuditStreamConfiguration"].apply(partialParams));
    }
    updateCompanyNetworkConfiguration(partialParams) {
        return this.client.updateCompanyNetworkConfiguration(this.ops["UpdateCompanyNetworkConfiguration"].apply(partialParams));
    }
    updateDevicePolicyConfiguration(partialParams) {
        return this.client.updateDevicePolicyConfiguration(this.ops["UpdateDevicePolicyConfiguration"].apply(partialParams));
    }
    updateDomainMetadata(partialParams) {
        return this.client.updateDomainMetadata(this.ops["UpdateDomainMetadata"].apply(partialParams));
    }
    updateFleetMetadata(partialParams) {
        return this.client.updateFleetMetadata(this.ops["UpdateFleetMetadata"].apply(partialParams));
    }
    updateIdentityProviderConfiguration(partialParams) {
        return this.client.updateIdentityProviderConfiguration(this.ops["UpdateIdentityProviderConfiguration"].apply(partialParams));
    }
}
exports.default = Resource;
