// out/appconfig/DeploymentStrategy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.appconfig.DeploymentStrategy {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.AppConfig();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/appconfig-2019-10-09.normal.json"), this.client);
    }
    createApplication(partialParams) {
        return this.client.createApplication(this.ops["CreateApplication"].apply(partialParams));
    }
    createConfigurationProfile(partialParams) {
        return this.client.createConfigurationProfile(this.ops["CreateConfigurationProfile"].apply(partialParams));
    }
    createDeploymentStrategy(partialParams) {
        return this.client.createDeploymentStrategy(this.ops["CreateDeploymentStrategy"].apply(partialParams));
    }
    createEnvironment(partialParams) {
        return this.client.createEnvironment(this.ops["CreateEnvironment"].apply(partialParams));
    }
    createHostedConfigurationVersion(partialParams) {
        return this.client.createHostedConfigurationVersion(this.ops["CreateHostedConfigurationVersion"].apply(partialParams));
    }
    deleteApplication(partialParams) {
        return this.client.deleteApplication(this.ops["DeleteApplication"].apply(partialParams));
    }
    deleteConfigurationProfile(partialParams) {
        return this.client.deleteConfigurationProfile(this.ops["DeleteConfigurationProfile"].apply(partialParams));
    }
    deleteDeploymentStrategy(partialParams) {
        return this.client.deleteDeploymentStrategy(this.ops["DeleteDeploymentStrategy"].apply(partialParams));
    }
    deleteEnvironment(partialParams) {
        return this.client.deleteEnvironment(this.ops["DeleteEnvironment"].apply(partialParams));
    }
    deleteHostedConfigurationVersion(partialParams) {
        return this.client.deleteHostedConfigurationVersion(this.ops["DeleteHostedConfigurationVersion"].apply(partialParams));
    }
    getApplication(partialParams) {
        return this.client.getApplication(this.ops["GetApplication"].apply(partialParams));
    }
    getConfiguration(partialParams) {
        return this.client.getConfiguration(this.ops["GetConfiguration"].apply(partialParams));
    }
    getConfigurationProfile(partialParams) {
        return this.client.getConfigurationProfile(this.ops["GetConfigurationProfile"].apply(partialParams));
    }
    getDeployment(partialParams) {
        return this.client.getDeployment(this.ops["GetDeployment"].apply(partialParams));
    }
    getDeploymentStrategy(partialParams) {
        return this.client.getDeploymentStrategy(this.ops["GetDeploymentStrategy"].apply(partialParams));
    }
    getEnvironment(partialParams) {
        return this.client.getEnvironment(this.ops["GetEnvironment"].apply(partialParams));
    }
    getHostedConfigurationVersion(partialParams) {
        return this.client.getHostedConfigurationVersion(this.ops["GetHostedConfigurationVersion"].apply(partialParams));
    }
    listConfigurationProfiles(partialParams) {
        return this.client.listConfigurationProfiles(this.ops["ListConfigurationProfiles"].apply(partialParams));
    }
    listDeployments(partialParams) {
        return this.client.listDeployments(this.ops["ListDeployments"].apply(partialParams));
    }
    listEnvironments(partialParams) {
        return this.client.listEnvironments(this.ops["ListEnvironments"].apply(partialParams));
    }
    listHostedConfigurationVersions(partialParams) {
        return this.client.listHostedConfigurationVersions(this.ops["ListHostedConfigurationVersions"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    startDeployment(partialParams) {
        return this.client.startDeployment(this.ops["StartDeployment"].apply(partialParams));
    }
    stopDeployment(partialParams) {
        return this.client.stopDeployment(this.ops["StopDeployment"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApplication(partialParams) {
        return this.client.updateApplication(this.ops["UpdateApplication"].apply(partialParams));
    }
    updateConfigurationProfile(partialParams) {
        return this.client.updateConfigurationProfile(this.ops["UpdateConfigurationProfile"].apply(partialParams));
    }
    updateDeploymentStrategy(partialParams) {
        return this.client.updateDeploymentStrategy(this.ops["UpdateDeploymentStrategy"].apply(partialParams));
    }
    updateEnvironment(partialParams) {
        return this.client.updateEnvironment(this.ops["UpdateEnvironment"].apply(partialParams));
    }
    validateConfiguration(partialParams) {
        return this.client.validateConfiguration(this.ops["ValidateConfiguration"].apply(partialParams));
    }
}
exports.default = Resource;
