// out/directconnect/Connection.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.directconnect.Connection {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.DirectConnect();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/directconnect-2012-10-25.normal.json"), this.client);
    }
    acceptDirectConnectGatewayAssociationProposal(partialParams) {
        return this.client.acceptDirectConnectGatewayAssociationProposal(this.ops["AcceptDirectConnectGatewayAssociationProposal"].apply(partialParams));
    }
    allocateConnectionOnInterconnect(partialParams) {
        return this.client.allocateConnectionOnInterconnect(this.ops["AllocateConnectionOnInterconnect"].apply(partialParams));
    }
    allocateHostedConnection(partialParams) {
        return this.client.allocateHostedConnection(this.ops["AllocateHostedConnection"].apply(partialParams));
    }
    allocatePrivateVirtualInterface(partialParams) {
        return this.client.allocatePrivateVirtualInterface(this.ops["AllocatePrivateVirtualInterface"].apply(partialParams));
    }
    allocatePublicVirtualInterface(partialParams) {
        return this.client.allocatePublicVirtualInterface(this.ops["AllocatePublicVirtualInterface"].apply(partialParams));
    }
    allocateTransitVirtualInterface(partialParams) {
        return this.client.allocateTransitVirtualInterface(this.ops["AllocateTransitVirtualInterface"].apply(partialParams));
    }
    associateConnectionWithLag(partialParams) {
        return this.client.associateConnectionWithLag(this.ops["AssociateConnectionWithLag"].apply(partialParams));
    }
    associateHostedConnection(partialParams) {
        return this.client.associateHostedConnection(this.ops["AssociateHostedConnection"].apply(partialParams));
    }
    associateMacSecKey(partialParams) {
        return this.client.associateMacSecKey(this.ops["AssociateMacSecKey"].apply(partialParams));
    }
    associateVirtualInterface(partialParams) {
        return this.client.associateVirtualInterface(this.ops["AssociateVirtualInterface"].apply(partialParams));
    }
    confirmConnection(partialParams) {
        return this.client.confirmConnection(this.ops["ConfirmConnection"].apply(partialParams));
    }
    confirmPrivateVirtualInterface(partialParams) {
        return this.client.confirmPrivateVirtualInterface(this.ops["ConfirmPrivateVirtualInterface"].apply(partialParams));
    }
    confirmPublicVirtualInterface(partialParams) {
        return this.client.confirmPublicVirtualInterface(this.ops["ConfirmPublicVirtualInterface"].apply(partialParams));
    }
    confirmTransitVirtualInterface(partialParams) {
        return this.client.confirmTransitVirtualInterface(this.ops["ConfirmTransitVirtualInterface"].apply(partialParams));
    }
    createConnection(partialParams) {
        return this.client.createConnection(this.ops["CreateConnection"].apply(partialParams));
    }
    createDirectConnectGateway(partialParams) {
        return this.client.createDirectConnectGateway(this.ops["CreateDirectConnectGateway"].apply(partialParams));
    }
    createDirectConnectGatewayAssociation(partialParams) {
        return this.client.createDirectConnectGatewayAssociation(this.ops["CreateDirectConnectGatewayAssociation"].apply(partialParams));
    }
    createDirectConnectGatewayAssociationProposal(partialParams) {
        return this.client.createDirectConnectGatewayAssociationProposal(this.ops["CreateDirectConnectGatewayAssociationProposal"].apply(partialParams));
    }
    createInterconnect(partialParams) {
        return this.client.createInterconnect(this.ops["CreateInterconnect"].apply(partialParams));
    }
    createLag(partialParams) {
        return this.client.createLag(this.ops["CreateLag"].apply(partialParams));
    }
    createPrivateVirtualInterface(partialParams) {
        return this.client.createPrivateVirtualInterface(this.ops["CreatePrivateVirtualInterface"].apply(partialParams));
    }
    createPublicVirtualInterface(partialParams) {
        return this.client.createPublicVirtualInterface(this.ops["CreatePublicVirtualInterface"].apply(partialParams));
    }
    createTransitVirtualInterface(partialParams) {
        return this.client.createTransitVirtualInterface(this.ops["CreateTransitVirtualInterface"].apply(partialParams));
    }
    deleteConnection(partialParams) {
        return this.client.deleteConnection(this.ops["DeleteConnection"].apply(partialParams));
    }
    deleteDirectConnectGateway(partialParams) {
        return this.client.deleteDirectConnectGateway(this.ops["DeleteDirectConnectGateway"].apply(partialParams));
    }
    deleteDirectConnectGatewayAssociationProposal(partialParams) {
        return this.client.deleteDirectConnectGatewayAssociationProposal(this.ops["DeleteDirectConnectGatewayAssociationProposal"].apply(partialParams));
    }
    deleteInterconnect(partialParams) {
        return this.client.deleteInterconnect(this.ops["DeleteInterconnect"].apply(partialParams));
    }
    deleteLag(partialParams) {
        return this.client.deleteLag(this.ops["DeleteLag"].apply(partialParams));
    }
    deleteVirtualInterface(partialParams) {
        return this.client.deleteVirtualInterface(this.ops["DeleteVirtualInterface"].apply(partialParams));
    }
    describeConnectionLoa(partialParams) {
        return this.client.describeConnectionLoa(this.ops["DescribeConnectionLoa"].apply(partialParams));
    }
    describeConnectionsOnInterconnect(partialParams) {
        return this.client.describeConnectionsOnInterconnect(this.ops["DescribeConnectionsOnInterconnect"].apply(partialParams));
    }
    describeHostedConnections(partialParams) {
        return this.client.describeHostedConnections(this.ops["DescribeHostedConnections"].apply(partialParams));
    }
    describeInterconnectLoa(partialParams) {
        return this.client.describeInterconnectLoa(this.ops["DescribeInterconnectLoa"].apply(partialParams));
    }
    describeLoa(partialParams) {
        return this.client.describeLoa(this.ops["DescribeLoa"].apply(partialParams));
    }
    describeRouterConfiguration(partialParams) {
        return this.client.describeRouterConfiguration(this.ops["DescribeRouterConfiguration"].apply(partialParams));
    }
    describeTags(partialParams) {
        return this.client.describeTags(this.ops["DescribeTags"].apply(partialParams));
    }
    disassociateConnectionFromLag(partialParams) {
        return this.client.disassociateConnectionFromLag(this.ops["DisassociateConnectionFromLag"].apply(partialParams));
    }
    disassociateMacSecKey(partialParams) {
        return this.client.disassociateMacSecKey(this.ops["DisassociateMacSecKey"].apply(partialParams));
    }
    startBgpFailoverTest(partialParams) {
        return this.client.startBgpFailoverTest(this.ops["StartBgpFailoverTest"].apply(partialParams));
    }
    stopBgpFailoverTest(partialParams) {
        return this.client.stopBgpFailoverTest(this.ops["StopBgpFailoverTest"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateConnection(partialParams) {
        return this.client.updateConnection(this.ops["UpdateConnection"].apply(partialParams));
    }
    updateDirectConnectGateway(partialParams) {
        return this.client.updateDirectConnectGateway(this.ops["UpdateDirectConnectGateway"].apply(partialParams));
    }
    updateLag(partialParams) {
        return this.client.updateLag(this.ops["UpdateLag"].apply(partialParams));
    }
    updateVirtualInterfaceAttributes(partialParams) {
        return this.client.updateVirtualInterfaceAttributes(this.ops["UpdateVirtualInterfaceAttributes"].apply(partialParams));
    }
}
exports.default = Resource;
