// out/waf/Rule.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.waf.Rule {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.WAF();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/waf-2015-08-24.normal.json"), this.client);
    }
    createByteMatchSet(partialParams) {
        return this.client.createByteMatchSet(this.ops["CreateByteMatchSet"].apply(partialParams));
    }
    createGeoMatchSet(partialParams) {
        return this.client.createGeoMatchSet(this.ops["CreateGeoMatchSet"].apply(partialParams));
    }
    createIPSet(partialParams) {
        return this.client.createIPSet(this.ops["CreateIPSet"].apply(partialParams));
    }
    createRateBasedRule(partialParams) {
        return this.client.createRateBasedRule(this.ops["CreateRateBasedRule"].apply(partialParams));
    }
    createRegexMatchSet(partialParams) {
        return this.client.createRegexMatchSet(this.ops["CreateRegexMatchSet"].apply(partialParams));
    }
    createRegexPatternSet(partialParams) {
        return this.client.createRegexPatternSet(this.ops["CreateRegexPatternSet"].apply(partialParams));
    }
    createRule(partialParams) {
        return this.client.createRule(this.ops["CreateRule"].apply(partialParams));
    }
    createRuleGroup(partialParams) {
        return this.client.createRuleGroup(this.ops["CreateRuleGroup"].apply(partialParams));
    }
    createSizeConstraintSet(partialParams) {
        return this.client.createSizeConstraintSet(this.ops["CreateSizeConstraintSet"].apply(partialParams));
    }
    createSqlInjectionMatchSet(partialParams) {
        return this.client.createSqlInjectionMatchSet(this.ops["CreateSqlInjectionMatchSet"].apply(partialParams));
    }
    createWebACL(partialParams) {
        return this.client.createWebACL(this.ops["CreateWebACL"].apply(partialParams));
    }
    createWebACLMigrationStack(partialParams) {
        return this.client.createWebACLMigrationStack(this.ops["CreateWebACLMigrationStack"].apply(partialParams));
    }
    createXssMatchSet(partialParams) {
        return this.client.createXssMatchSet(this.ops["CreateXssMatchSet"].apply(partialParams));
    }
    deleteByteMatchSet(partialParams) {
        return this.client.deleteByteMatchSet(this.ops["DeleteByteMatchSet"].apply(partialParams));
    }
    deleteGeoMatchSet(partialParams) {
        return this.client.deleteGeoMatchSet(this.ops["DeleteGeoMatchSet"].apply(partialParams));
    }
    deleteIPSet(partialParams) {
        return this.client.deleteIPSet(this.ops["DeleteIPSet"].apply(partialParams));
    }
    deleteLoggingConfiguration(partialParams) {
        return this.client.deleteLoggingConfiguration(this.ops["DeleteLoggingConfiguration"].apply(partialParams));
    }
    deletePermissionPolicy(partialParams) {
        return this.client.deletePermissionPolicy(this.ops["DeletePermissionPolicy"].apply(partialParams));
    }
    deleteRateBasedRule(partialParams) {
        return this.client.deleteRateBasedRule(this.ops["DeleteRateBasedRule"].apply(partialParams));
    }
    deleteRegexMatchSet(partialParams) {
        return this.client.deleteRegexMatchSet(this.ops["DeleteRegexMatchSet"].apply(partialParams));
    }
    deleteRegexPatternSet(partialParams) {
        return this.client.deleteRegexPatternSet(this.ops["DeleteRegexPatternSet"].apply(partialParams));
    }
    deleteRule(partialParams) {
        return this.client.deleteRule(this.ops["DeleteRule"].apply(partialParams));
    }
    deleteRuleGroup(partialParams) {
        return this.client.deleteRuleGroup(this.ops["DeleteRuleGroup"].apply(partialParams));
    }
    deleteSizeConstraintSet(partialParams) {
        return this.client.deleteSizeConstraintSet(this.ops["DeleteSizeConstraintSet"].apply(partialParams));
    }
    deleteSqlInjectionMatchSet(partialParams) {
        return this.client.deleteSqlInjectionMatchSet(this.ops["DeleteSqlInjectionMatchSet"].apply(partialParams));
    }
    deleteWebACL(partialParams) {
        return this.client.deleteWebACL(this.ops["DeleteWebACL"].apply(partialParams));
    }
    deleteXssMatchSet(partialParams) {
        return this.client.deleteXssMatchSet(this.ops["DeleteXssMatchSet"].apply(partialParams));
    }
    getByteMatchSet(partialParams) {
        return this.client.getByteMatchSet(this.ops["GetByteMatchSet"].apply(partialParams));
    }
    getChangeTokenStatus(partialParams) {
        return this.client.getChangeTokenStatus(this.ops["GetChangeTokenStatus"].apply(partialParams));
    }
    getGeoMatchSet(partialParams) {
        return this.client.getGeoMatchSet(this.ops["GetGeoMatchSet"].apply(partialParams));
    }
    getIPSet(partialParams) {
        return this.client.getIPSet(this.ops["GetIPSet"].apply(partialParams));
    }
    getLoggingConfiguration(partialParams) {
        return this.client.getLoggingConfiguration(this.ops["GetLoggingConfiguration"].apply(partialParams));
    }
    getPermissionPolicy(partialParams) {
        return this.client.getPermissionPolicy(this.ops["GetPermissionPolicy"].apply(partialParams));
    }
    getRateBasedRule(partialParams) {
        return this.client.getRateBasedRule(this.ops["GetRateBasedRule"].apply(partialParams));
    }
    getRateBasedRuleManagedKeys(partialParams) {
        return this.client.getRateBasedRuleManagedKeys(this.ops["GetRateBasedRuleManagedKeys"].apply(partialParams));
    }
    getRegexMatchSet(partialParams) {
        return this.client.getRegexMatchSet(this.ops["GetRegexMatchSet"].apply(partialParams));
    }
    getRegexPatternSet(partialParams) {
        return this.client.getRegexPatternSet(this.ops["GetRegexPatternSet"].apply(partialParams));
    }
    getRule(partialParams) {
        return this.client.getRule(this.ops["GetRule"].apply(partialParams));
    }
    getRuleGroup(partialParams) {
        return this.client.getRuleGroup(this.ops["GetRuleGroup"].apply(partialParams));
    }
    getSampledRequests(partialParams) {
        return this.client.getSampledRequests(this.ops["GetSampledRequests"].apply(partialParams));
    }
    getSizeConstraintSet(partialParams) {
        return this.client.getSizeConstraintSet(this.ops["GetSizeConstraintSet"].apply(partialParams));
    }
    getSqlInjectionMatchSet(partialParams) {
        return this.client.getSqlInjectionMatchSet(this.ops["GetSqlInjectionMatchSet"].apply(partialParams));
    }
    getWebACL(partialParams) {
        return this.client.getWebACL(this.ops["GetWebACL"].apply(partialParams));
    }
    getXssMatchSet(partialParams) {
        return this.client.getXssMatchSet(this.ops["GetXssMatchSet"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putLoggingConfiguration(partialParams) {
        return this.client.putLoggingConfiguration(this.ops["PutLoggingConfiguration"].apply(partialParams));
    }
    putPermissionPolicy(partialParams) {
        return this.client.putPermissionPolicy(this.ops["PutPermissionPolicy"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateByteMatchSet(partialParams) {
        return this.client.updateByteMatchSet(this.ops["UpdateByteMatchSet"].apply(partialParams));
    }
    updateGeoMatchSet(partialParams) {
        return this.client.updateGeoMatchSet(this.ops["UpdateGeoMatchSet"].apply(partialParams));
    }
    updateIPSet(partialParams) {
        return this.client.updateIPSet(this.ops["UpdateIPSet"].apply(partialParams));
    }
    updateRateBasedRule(partialParams) {
        return this.client.updateRateBasedRule(this.ops["UpdateRateBasedRule"].apply(partialParams));
    }
    updateRegexMatchSet(partialParams) {
        return this.client.updateRegexMatchSet(this.ops["UpdateRegexMatchSet"].apply(partialParams));
    }
    updateRegexPatternSet(partialParams) {
        return this.client.updateRegexPatternSet(this.ops["UpdateRegexPatternSet"].apply(partialParams));
    }
    updateRule(partialParams) {
        return this.client.updateRule(this.ops["UpdateRule"].apply(partialParams));
    }
    updateRuleGroup(partialParams) {
        return this.client.updateRuleGroup(this.ops["UpdateRuleGroup"].apply(partialParams));
    }
    updateSizeConstraintSet(partialParams) {
        return this.client.updateSizeConstraintSet(this.ops["UpdateSizeConstraintSet"].apply(partialParams));
    }
    updateSqlInjectionMatchSet(partialParams) {
        return this.client.updateSqlInjectionMatchSet(this.ops["UpdateSqlInjectionMatchSet"].apply(partialParams));
    }
    updateWebACL(partialParams) {
        return this.client.updateWebACL(this.ops["UpdateWebACL"].apply(partialParams));
    }
    updateXssMatchSet(partialParams) {
        return this.client.updateXssMatchSet(this.ops["UpdateXssMatchSet"].apply(partialParams));
    }
}
exports.default = Resource;
