// out/globalaccelerator/Listener.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.globalaccelerator.Listener {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.GlobalAccelerator();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/globalaccelerator-2018-08-08.normal.json"), this.client);
    }
    addCustomRoutingEndpoints(partialParams) {
        return this.client.addCustomRoutingEndpoints(this.ops["AddCustomRoutingEndpoints"].apply(partialParams));
    }
    advertiseByoipCidr(partialParams) {
        return this.client.advertiseByoipCidr(this.ops["AdvertiseByoipCidr"].apply(partialParams));
    }
    allowCustomRoutingTraffic(partialParams) {
        return this.client.allowCustomRoutingTraffic(this.ops["AllowCustomRoutingTraffic"].apply(partialParams));
    }
    createAccelerator(partialParams) {
        return this.client.createAccelerator(this.ops["CreateAccelerator"].apply(partialParams));
    }
    createCustomRoutingAccelerator(partialParams) {
        return this.client.createCustomRoutingAccelerator(this.ops["CreateCustomRoutingAccelerator"].apply(partialParams));
    }
    createCustomRoutingEndpointGroup(partialParams) {
        return this.client.createCustomRoutingEndpointGroup(this.ops["CreateCustomRoutingEndpointGroup"].apply(partialParams));
    }
    createCustomRoutingListener(partialParams) {
        return this.client.createCustomRoutingListener(this.ops["CreateCustomRoutingListener"].apply(partialParams));
    }
    createEndpointGroup(partialParams) {
        return this.client.createEndpointGroup(this.ops["CreateEndpointGroup"].apply(partialParams));
    }
    createListener(partialParams) {
        return this.client.createListener(this.ops["CreateListener"].apply(partialParams));
    }
    deleteAccelerator(partialParams) {
        return this.client.deleteAccelerator(this.ops["DeleteAccelerator"].apply(partialParams));
    }
    deleteCustomRoutingAccelerator(partialParams) {
        return this.client.deleteCustomRoutingAccelerator(this.ops["DeleteCustomRoutingAccelerator"].apply(partialParams));
    }
    deleteCustomRoutingEndpointGroup(partialParams) {
        return this.client.deleteCustomRoutingEndpointGroup(this.ops["DeleteCustomRoutingEndpointGroup"].apply(partialParams));
    }
    deleteCustomRoutingListener(partialParams) {
        return this.client.deleteCustomRoutingListener(this.ops["DeleteCustomRoutingListener"].apply(partialParams));
    }
    deleteEndpointGroup(partialParams) {
        return this.client.deleteEndpointGroup(this.ops["DeleteEndpointGroup"].apply(partialParams));
    }
    deleteListener(partialParams) {
        return this.client.deleteListener(this.ops["DeleteListener"].apply(partialParams));
    }
    denyCustomRoutingTraffic(partialParams) {
        return this.client.denyCustomRoutingTraffic(this.ops["DenyCustomRoutingTraffic"].apply(partialParams));
    }
    deprovisionByoipCidr(partialParams) {
        return this.client.deprovisionByoipCidr(this.ops["DeprovisionByoipCidr"].apply(partialParams));
    }
    describeAccelerator(partialParams) {
        return this.client.describeAccelerator(this.ops["DescribeAccelerator"].apply(partialParams));
    }
    describeAcceleratorAttributes(partialParams) {
        return this.client.describeAcceleratorAttributes(this.ops["DescribeAcceleratorAttributes"].apply(partialParams));
    }
    describeCustomRoutingAccelerator(partialParams) {
        return this.client.describeCustomRoutingAccelerator(this.ops["DescribeCustomRoutingAccelerator"].apply(partialParams));
    }
    describeCustomRoutingAcceleratorAttributes(partialParams) {
        return this.client.describeCustomRoutingAcceleratorAttributes(this.ops["DescribeCustomRoutingAcceleratorAttributes"].apply(partialParams));
    }
    describeCustomRoutingEndpointGroup(partialParams) {
        return this.client.describeCustomRoutingEndpointGroup(this.ops["DescribeCustomRoutingEndpointGroup"].apply(partialParams));
    }
    describeCustomRoutingListener(partialParams) {
        return this.client.describeCustomRoutingListener(this.ops["DescribeCustomRoutingListener"].apply(partialParams));
    }
    describeEndpointGroup(partialParams) {
        return this.client.describeEndpointGroup(this.ops["DescribeEndpointGroup"].apply(partialParams));
    }
    describeListener(partialParams) {
        return this.client.describeListener(this.ops["DescribeListener"].apply(partialParams));
    }
    listCustomRoutingEndpointGroups(partialParams) {
        return this.client.listCustomRoutingEndpointGroups(this.ops["ListCustomRoutingEndpointGroups"].apply(partialParams));
    }
    listCustomRoutingListeners(partialParams) {
        return this.client.listCustomRoutingListeners(this.ops["ListCustomRoutingListeners"].apply(partialParams));
    }
    listCustomRoutingPortMappings(partialParams) {
        return this.client.listCustomRoutingPortMappings(this.ops["ListCustomRoutingPortMappings"].apply(partialParams));
    }
    listCustomRoutingPortMappingsByDestination(partialParams) {
        return this.client.listCustomRoutingPortMappingsByDestination(this.ops["ListCustomRoutingPortMappingsByDestination"].apply(partialParams));
    }
    listEndpointGroups(partialParams) {
        return this.client.listEndpointGroups(this.ops["ListEndpointGroups"].apply(partialParams));
    }
    listListeners(partialParams) {
        return this.client.listListeners(this.ops["ListListeners"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    provisionByoipCidr(partialParams) {
        return this.client.provisionByoipCidr(this.ops["ProvisionByoipCidr"].apply(partialParams));
    }
    removeCustomRoutingEndpoints(partialParams) {
        return this.client.removeCustomRoutingEndpoints(this.ops["RemoveCustomRoutingEndpoints"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAccelerator(partialParams) {
        return this.client.updateAccelerator(this.ops["UpdateAccelerator"].apply(partialParams));
    }
    updateAcceleratorAttributes(partialParams) {
        return this.client.updateAcceleratorAttributes(this.ops["UpdateAcceleratorAttributes"].apply(partialParams));
    }
    updateCustomRoutingAccelerator(partialParams) {
        return this.client.updateCustomRoutingAccelerator(this.ops["UpdateCustomRoutingAccelerator"].apply(partialParams));
    }
    updateCustomRoutingAcceleratorAttributes(partialParams) {
        return this.client.updateCustomRoutingAcceleratorAttributes(this.ops["UpdateCustomRoutingAcceleratorAttributes"].apply(partialParams));
    }
    updateCustomRoutingListener(partialParams) {
        return this.client.updateCustomRoutingListener(this.ops["UpdateCustomRoutingListener"].apply(partialParams));
    }
    updateEndpointGroup(partialParams) {
        return this.client.updateEndpointGroup(this.ops["UpdateEndpointGroup"].apply(partialParams));
    }
    updateListener(partialParams) {
        return this.client.updateListener(this.ops["UpdateListener"].apply(partialParams));
    }
    withdrawByoipCidr(partialParams) {
        return this.client.withdrawByoipCidr(this.ops["WithdrawByoipCidr"].apply(partialParams));
    }
}
exports.default = Resource;
