// out/organizations/DelegatedAdministrator.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.organizations.DelegatedAdministrator {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Organizations();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/organizations-2016-11-28.normal.json"), this.client);
    }
    acceptHandshake(partialParams) {
        return this.client.acceptHandshake(this.ops["AcceptHandshake"].apply(partialParams));
    }
    attachPolicy(partialParams) {
        return this.client.attachPolicy(this.ops["AttachPolicy"].apply(partialParams));
    }
    cancelHandshake(partialParams) {
        return this.client.cancelHandshake(this.ops["CancelHandshake"].apply(partialParams));
    }
    createAccount(partialParams) {
        return this.client.createAccount(this.ops["CreateAccount"].apply(partialParams));
    }
    createGovCloudAccount(partialParams) {
        return this.client.createGovCloudAccount(this.ops["CreateGovCloudAccount"].apply(partialParams));
    }
    createOrganizationalUnit(partialParams) {
        return this.client.createOrganizationalUnit(this.ops["CreateOrganizationalUnit"].apply(partialParams));
    }
    createPolicy(partialParams) {
        return this.client.createPolicy(this.ops["CreatePolicy"].apply(partialParams));
    }
    declineHandshake(partialParams) {
        return this.client.declineHandshake(this.ops["DeclineHandshake"].apply(partialParams));
    }
    deleteOrganizationalUnit(partialParams) {
        return this.client.deleteOrganizationalUnit(this.ops["DeleteOrganizationalUnit"].apply(partialParams));
    }
    deletePolicy(partialParams) {
        return this.client.deletePolicy(this.ops["DeletePolicy"].apply(partialParams));
    }
    deregisterDelegatedAdministrator(partialParams) {
        return this.client.deregisterDelegatedAdministrator(this.ops["DeregisterDelegatedAdministrator"].apply(partialParams));
    }
    describeAccount(partialParams) {
        return this.client.describeAccount(this.ops["DescribeAccount"].apply(partialParams));
    }
    describeCreateAccountStatus(partialParams) {
        return this.client.describeCreateAccountStatus(this.ops["DescribeCreateAccountStatus"].apply(partialParams));
    }
    describeEffectivePolicy(partialParams) {
        return this.client.describeEffectivePolicy(this.ops["DescribeEffectivePolicy"].apply(partialParams));
    }
    describeHandshake(partialParams) {
        return this.client.describeHandshake(this.ops["DescribeHandshake"].apply(partialParams));
    }
    describeOrganizationalUnit(partialParams) {
        return this.client.describeOrganizationalUnit(this.ops["DescribeOrganizationalUnit"].apply(partialParams));
    }
    describePolicy(partialParams) {
        return this.client.describePolicy(this.ops["DescribePolicy"].apply(partialParams));
    }
    detachPolicy(partialParams) {
        return this.client.detachPolicy(this.ops["DetachPolicy"].apply(partialParams));
    }
    disableAWSServiceAccess(partialParams) {
        return this.client.disableAWSServiceAccess(this.ops["DisableAWSServiceAccess"].apply(partialParams));
    }
    disablePolicyType(partialParams) {
        return this.client.disablePolicyType(this.ops["DisablePolicyType"].apply(partialParams));
    }
    enableAWSServiceAccess(partialParams) {
        return this.client.enableAWSServiceAccess(this.ops["EnableAWSServiceAccess"].apply(partialParams));
    }
    enablePolicyType(partialParams) {
        return this.client.enablePolicyType(this.ops["EnablePolicyType"].apply(partialParams));
    }
    inviteAccountToOrganization(partialParams) {
        return this.client.inviteAccountToOrganization(this.ops["InviteAccountToOrganization"].apply(partialParams));
    }
    listAccountsForParent(partialParams) {
        return this.client.listAccountsForParent(this.ops["ListAccountsForParent"].apply(partialParams));
    }
    listChildren(partialParams) {
        return this.client.listChildren(this.ops["ListChildren"].apply(partialParams));
    }
    listDelegatedServicesForAccount(partialParams) {
        return this.client.listDelegatedServicesForAccount(this.ops["ListDelegatedServicesForAccount"].apply(partialParams));
    }
    listOrganizationalUnitsForParent(partialParams) {
        return this.client.listOrganizationalUnitsForParent(this.ops["ListOrganizationalUnitsForParent"].apply(partialParams));
    }
    listParents(partialParams) {
        return this.client.listParents(this.ops["ListParents"].apply(partialParams));
    }
    listPolicies(partialParams) {
        return this.client.listPolicies(this.ops["ListPolicies"].apply(partialParams));
    }
    listPoliciesForTarget(partialParams) {
        return this.client.listPoliciesForTarget(this.ops["ListPoliciesForTarget"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTargetsForPolicy(partialParams) {
        return this.client.listTargetsForPolicy(this.ops["ListTargetsForPolicy"].apply(partialParams));
    }
    moveAccount(partialParams) {
        return this.client.moveAccount(this.ops["MoveAccount"].apply(partialParams));
    }
    registerDelegatedAdministrator(partialParams) {
        return this.client.registerDelegatedAdministrator(this.ops["RegisterDelegatedAdministrator"].apply(partialParams));
    }
    removeAccountFromOrganization(partialParams) {
        return this.client.removeAccountFromOrganization(this.ops["RemoveAccountFromOrganization"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateOrganizationalUnit(partialParams) {
        return this.client.updateOrganizationalUnit(this.ops["UpdateOrganizationalUnit"].apply(partialParams));
    }
    updatePolicy(partialParams) {
        return this.client.updatePolicy(this.ops["UpdatePolicy"].apply(partialParams));
    }
}
exports.default = Resource;
