// out/mediaconvert/Queue.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.mediaconvert.Queue {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.MediaConvert();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/mediaconvert-2017-08-29.normal.json"), this.client);
    }
    associateCertificate(partialParams) {
        return this.client.associateCertificate(this.ops["AssociateCertificate"].apply(partialParams));
    }
    cancelJob(partialParams) {
        return this.client.cancelJob(this.ops["CancelJob"].apply(partialParams));
    }
    createJob(partialParams) {
        return this.client.createJob(this.ops["CreateJob"].apply(partialParams));
    }
    createJobTemplate(partialParams) {
        return this.client.createJobTemplate(this.ops["CreateJobTemplate"].apply(partialParams));
    }
    createPreset(partialParams) {
        return this.client.createPreset(this.ops["CreatePreset"].apply(partialParams));
    }
    createQueue(partialParams) {
        return this.client.createQueue(this.ops["CreateQueue"].apply(partialParams));
    }
    deleteJobTemplate(partialParams) {
        return this.client.deleteJobTemplate(this.ops["DeleteJobTemplate"].apply(partialParams));
    }
    deletePreset(partialParams) {
        return this.client.deletePreset(this.ops["DeletePreset"].apply(partialParams));
    }
    deleteQueue(partialParams) {
        return this.client.deleteQueue(this.ops["DeleteQueue"].apply(partialParams));
    }
    disassociateCertificate(partialParams) {
        return this.client.disassociateCertificate(this.ops["DisassociateCertificate"].apply(partialParams));
    }
    getJob(partialParams) {
        return this.client.getJob(this.ops["GetJob"].apply(partialParams));
    }
    getJobTemplate(partialParams) {
        return this.client.getJobTemplate(this.ops["GetJobTemplate"].apply(partialParams));
    }
    getPreset(partialParams) {
        return this.client.getPreset(this.ops["GetPreset"].apply(partialParams));
    }
    getQueue(partialParams) {
        return this.client.getQueue(this.ops["GetQueue"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putPolicy(partialParams) {
        return this.client.putPolicy(this.ops["PutPolicy"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateJobTemplate(partialParams) {
        return this.client.updateJobTemplate(this.ops["UpdateJobTemplate"].apply(partialParams));
    }
    updatePreset(partialParams) {
        return this.client.updatePreset(this.ops["UpdatePreset"].apply(partialParams));
    }
    updateQueue(partialParams) {
        return this.client.updateQueue(this.ops["UpdateQueue"].apply(partialParams));
    }
}
exports.default = Resource;
