// out/pinpoint/BaiduChannel.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.pinpoint.BaiduChannel {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Pinpoint();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/pinpoint-2016-12-01.normal.json"), this.client);
    }
    createApp(partialParams) {
        return this.client.createApp(this.ops["CreateApp"].apply(partialParams));
    }
    createCampaign(partialParams) {
        return this.client.createCampaign(this.ops["CreateCampaign"].apply(partialParams));
    }
    createEmailTemplate(partialParams) {
        return this.client.createEmailTemplate(this.ops["CreateEmailTemplate"].apply(partialParams));
    }
    createExportJob(partialParams) {
        return this.client.createExportJob(this.ops["CreateExportJob"].apply(partialParams));
    }
    createImportJob(partialParams) {
        return this.client.createImportJob(this.ops["CreateImportJob"].apply(partialParams));
    }
    createInAppTemplate(partialParams) {
        return this.client.createInAppTemplate(this.ops["CreateInAppTemplate"].apply(partialParams));
    }
    createJourney(partialParams) {
        return this.client.createJourney(this.ops["CreateJourney"].apply(partialParams));
    }
    createPushTemplate(partialParams) {
        return this.client.createPushTemplate(this.ops["CreatePushTemplate"].apply(partialParams));
    }
    createRecommenderConfiguration(partialParams) {
        return this.client.createRecommenderConfiguration(this.ops["CreateRecommenderConfiguration"].apply(partialParams));
    }
    createSegment(partialParams) {
        return this.client.createSegment(this.ops["CreateSegment"].apply(partialParams));
    }
    createSmsTemplate(partialParams) {
        return this.client.createSmsTemplate(this.ops["CreateSmsTemplate"].apply(partialParams));
    }
    createVoiceTemplate(partialParams) {
        return this.client.createVoiceTemplate(this.ops["CreateVoiceTemplate"].apply(partialParams));
    }
    deleteAdmChannel(partialParams) {
        return this.client.deleteAdmChannel(this.ops["DeleteAdmChannel"].apply(partialParams));
    }
    deleteApnsChannel(partialParams) {
        return this.client.deleteApnsChannel(this.ops["DeleteApnsChannel"].apply(partialParams));
    }
    deleteApnsSandboxChannel(partialParams) {
        return this.client.deleteApnsSandboxChannel(this.ops["DeleteApnsSandboxChannel"].apply(partialParams));
    }
    deleteApnsVoipChannel(partialParams) {
        return this.client.deleteApnsVoipChannel(this.ops["DeleteApnsVoipChannel"].apply(partialParams));
    }
    deleteApnsVoipSandboxChannel(partialParams) {
        return this.client.deleteApnsVoipSandboxChannel(this.ops["DeleteApnsVoipSandboxChannel"].apply(partialParams));
    }
    deleteApp(partialParams) {
        return this.client.deleteApp(this.ops["DeleteApp"].apply(partialParams));
    }
    deleteBaiduChannel(partialParams) {
        return this.client.deleteBaiduChannel(this.ops["DeleteBaiduChannel"].apply(partialParams));
    }
    deleteCampaign(partialParams) {
        return this.client.deleteCampaign(this.ops["DeleteCampaign"].apply(partialParams));
    }
    deleteEmailChannel(partialParams) {
        return this.client.deleteEmailChannel(this.ops["DeleteEmailChannel"].apply(partialParams));
    }
    deleteEmailTemplate(partialParams) {
        return this.client.deleteEmailTemplate(this.ops["DeleteEmailTemplate"].apply(partialParams));
    }
    deleteEndpoint(partialParams) {
        return this.client.deleteEndpoint(this.ops["DeleteEndpoint"].apply(partialParams));
    }
    deleteEventStream(partialParams) {
        return this.client.deleteEventStream(this.ops["DeleteEventStream"].apply(partialParams));
    }
    deleteGcmChannel(partialParams) {
        return this.client.deleteGcmChannel(this.ops["DeleteGcmChannel"].apply(partialParams));
    }
    deleteInAppTemplate(partialParams) {
        return this.client.deleteInAppTemplate(this.ops["DeleteInAppTemplate"].apply(partialParams));
    }
    deleteJourney(partialParams) {
        return this.client.deleteJourney(this.ops["DeleteJourney"].apply(partialParams));
    }
    deletePushTemplate(partialParams) {
        return this.client.deletePushTemplate(this.ops["DeletePushTemplate"].apply(partialParams));
    }
    deleteRecommenderConfiguration(partialParams) {
        return this.client.deleteRecommenderConfiguration(this.ops["DeleteRecommenderConfiguration"].apply(partialParams));
    }
    deleteSegment(partialParams) {
        return this.client.deleteSegment(this.ops["DeleteSegment"].apply(partialParams));
    }
    deleteSmsChannel(partialParams) {
        return this.client.deleteSmsChannel(this.ops["DeleteSmsChannel"].apply(partialParams));
    }
    deleteSmsTemplate(partialParams) {
        return this.client.deleteSmsTemplate(this.ops["DeleteSmsTemplate"].apply(partialParams));
    }
    deleteUserEndpoints(partialParams) {
        return this.client.deleteUserEndpoints(this.ops["DeleteUserEndpoints"].apply(partialParams));
    }
    deleteVoiceChannel(partialParams) {
        return this.client.deleteVoiceChannel(this.ops["DeleteVoiceChannel"].apply(partialParams));
    }
    deleteVoiceTemplate(partialParams) {
        return this.client.deleteVoiceTemplate(this.ops["DeleteVoiceTemplate"].apply(partialParams));
    }
    getAdmChannel(partialParams) {
        return this.client.getAdmChannel(this.ops["GetAdmChannel"].apply(partialParams));
    }
    getApnsChannel(partialParams) {
        return this.client.getApnsChannel(this.ops["GetApnsChannel"].apply(partialParams));
    }
    getApnsSandboxChannel(partialParams) {
        return this.client.getApnsSandboxChannel(this.ops["GetApnsSandboxChannel"].apply(partialParams));
    }
    getApnsVoipChannel(partialParams) {
        return this.client.getApnsVoipChannel(this.ops["GetApnsVoipChannel"].apply(partialParams));
    }
    getApnsVoipSandboxChannel(partialParams) {
        return this.client.getApnsVoipSandboxChannel(this.ops["GetApnsVoipSandboxChannel"].apply(partialParams));
    }
    getApp(partialParams) {
        return this.client.getApp(this.ops["GetApp"].apply(partialParams));
    }
    getApplicationDateRangeKpi(partialParams) {
        return this.client.getApplicationDateRangeKpi(this.ops["GetApplicationDateRangeKpi"].apply(partialParams));
    }
    getApplicationSettings(partialParams) {
        return this.client.getApplicationSettings(this.ops["GetApplicationSettings"].apply(partialParams));
    }
    getBaiduChannel(partialParams) {
        return this.client.getBaiduChannel(this.ops["GetBaiduChannel"].apply(partialParams));
    }
    getCampaign(partialParams) {
        return this.client.getCampaign(this.ops["GetCampaign"].apply(partialParams));
    }
    getCampaignActivities(partialParams) {
        return this.client.getCampaignActivities(this.ops["GetCampaignActivities"].apply(partialParams));
    }
    getCampaignDateRangeKpi(partialParams) {
        return this.client.getCampaignDateRangeKpi(this.ops["GetCampaignDateRangeKpi"].apply(partialParams));
    }
    getCampaignVersion(partialParams) {
        return this.client.getCampaignVersion(this.ops["GetCampaignVersion"].apply(partialParams));
    }
    getCampaignVersions(partialParams) {
        return this.client.getCampaignVersions(this.ops["GetCampaignVersions"].apply(partialParams));
    }
    getCampaigns(partialParams) {
        return this.client.getCampaigns(this.ops["GetCampaigns"].apply(partialParams));
    }
    getChannels(partialParams) {
        return this.client.getChannels(this.ops["GetChannels"].apply(partialParams));
    }
    getEmailChannel(partialParams) {
        return this.client.getEmailChannel(this.ops["GetEmailChannel"].apply(partialParams));
    }
    getEmailTemplate(partialParams) {
        return this.client.getEmailTemplate(this.ops["GetEmailTemplate"].apply(partialParams));
    }
    getEndpoint(partialParams) {
        return this.client.getEndpoint(this.ops["GetEndpoint"].apply(partialParams));
    }
    getEventStream(partialParams) {
        return this.client.getEventStream(this.ops["GetEventStream"].apply(partialParams));
    }
    getExportJob(partialParams) {
        return this.client.getExportJob(this.ops["GetExportJob"].apply(partialParams));
    }
    getExportJobs(partialParams) {
        return this.client.getExportJobs(this.ops["GetExportJobs"].apply(partialParams));
    }
    getGcmChannel(partialParams) {
        return this.client.getGcmChannel(this.ops["GetGcmChannel"].apply(partialParams));
    }
    getImportJob(partialParams) {
        return this.client.getImportJob(this.ops["GetImportJob"].apply(partialParams));
    }
    getImportJobs(partialParams) {
        return this.client.getImportJobs(this.ops["GetImportJobs"].apply(partialParams));
    }
    getInAppMessages(partialParams) {
        return this.client.getInAppMessages(this.ops["GetInAppMessages"].apply(partialParams));
    }
    getInAppTemplate(partialParams) {
        return this.client.getInAppTemplate(this.ops["GetInAppTemplate"].apply(partialParams));
    }
    getJourney(partialParams) {
        return this.client.getJourney(this.ops["GetJourney"].apply(partialParams));
    }
    getJourneyDateRangeKpi(partialParams) {
        return this.client.getJourneyDateRangeKpi(this.ops["GetJourneyDateRangeKpi"].apply(partialParams));
    }
    getJourneyExecutionActivityMetrics(partialParams) {
        return this.client.getJourneyExecutionActivityMetrics(this.ops["GetJourneyExecutionActivityMetrics"].apply(partialParams));
    }
    getJourneyExecutionMetrics(partialParams) {
        return this.client.getJourneyExecutionMetrics(this.ops["GetJourneyExecutionMetrics"].apply(partialParams));
    }
    getPushTemplate(partialParams) {
        return this.client.getPushTemplate(this.ops["GetPushTemplate"].apply(partialParams));
    }
    getRecommenderConfiguration(partialParams) {
        return this.client.getRecommenderConfiguration(this.ops["GetRecommenderConfiguration"].apply(partialParams));
    }
    getSegment(partialParams) {
        return this.client.getSegment(this.ops["GetSegment"].apply(partialParams));
    }
    getSegmentExportJobs(partialParams) {
        return this.client.getSegmentExportJobs(this.ops["GetSegmentExportJobs"].apply(partialParams));
    }
    getSegmentImportJobs(partialParams) {
        return this.client.getSegmentImportJobs(this.ops["GetSegmentImportJobs"].apply(partialParams));
    }
    getSegmentVersion(partialParams) {
        return this.client.getSegmentVersion(this.ops["GetSegmentVersion"].apply(partialParams));
    }
    getSegmentVersions(partialParams) {
        return this.client.getSegmentVersions(this.ops["GetSegmentVersions"].apply(partialParams));
    }
    getSegments(partialParams) {
        return this.client.getSegments(this.ops["GetSegments"].apply(partialParams));
    }
    getSmsChannel(partialParams) {
        return this.client.getSmsChannel(this.ops["GetSmsChannel"].apply(partialParams));
    }
    getSmsTemplate(partialParams) {
        return this.client.getSmsTemplate(this.ops["GetSmsTemplate"].apply(partialParams));
    }
    getUserEndpoints(partialParams) {
        return this.client.getUserEndpoints(this.ops["GetUserEndpoints"].apply(partialParams));
    }
    getVoiceChannel(partialParams) {
        return this.client.getVoiceChannel(this.ops["GetVoiceChannel"].apply(partialParams));
    }
    getVoiceTemplate(partialParams) {
        return this.client.getVoiceTemplate(this.ops["GetVoiceTemplate"].apply(partialParams));
    }
    listJourneys(partialParams) {
        return this.client.listJourneys(this.ops["ListJourneys"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTemplateVersions(partialParams) {
        return this.client.listTemplateVersions(this.ops["ListTemplateVersions"].apply(partialParams));
    }
    phoneNumberValidate(partialParams) {
        return this.client.phoneNumberValidate(this.ops["PhoneNumberValidate"].apply(partialParams));
    }
    putEventStream(partialParams) {
        return this.client.putEventStream(this.ops["PutEventStream"].apply(partialParams));
    }
    putEvents(partialParams) {
        return this.client.putEvents(this.ops["PutEvents"].apply(partialParams));
    }
    removeAttributes(partialParams) {
        return this.client.removeAttributes(this.ops["RemoveAttributes"].apply(partialParams));
    }
    sendMessages(partialParams) {
        return this.client.sendMessages(this.ops["SendMessages"].apply(partialParams));
    }
    sendOTPMessage(partialParams) {
        return this.client.sendOTPMessage(this.ops["SendOTPMessage"].apply(partialParams));
    }
    sendUsersMessages(partialParams) {
        return this.client.sendUsersMessages(this.ops["SendUsersMessages"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAdmChannel(partialParams) {
        return this.client.updateAdmChannel(this.ops["UpdateAdmChannel"].apply(partialParams));
    }
    updateApnsChannel(partialParams) {
        return this.client.updateApnsChannel(this.ops["UpdateApnsChannel"].apply(partialParams));
    }
    updateApnsSandboxChannel(partialParams) {
        return this.client.updateApnsSandboxChannel(this.ops["UpdateApnsSandboxChannel"].apply(partialParams));
    }
    updateApnsVoipChannel(partialParams) {
        return this.client.updateApnsVoipChannel(this.ops["UpdateApnsVoipChannel"].apply(partialParams));
    }
    updateApnsVoipSandboxChannel(partialParams) {
        return this.client.updateApnsVoipSandboxChannel(this.ops["UpdateApnsVoipSandboxChannel"].apply(partialParams));
    }
    updateApplicationSettings(partialParams) {
        return this.client.updateApplicationSettings(this.ops["UpdateApplicationSettings"].apply(partialParams));
    }
    updateBaiduChannel(partialParams) {
        return this.client.updateBaiduChannel(this.ops["UpdateBaiduChannel"].apply(partialParams));
    }
    updateCampaign(partialParams) {
        return this.client.updateCampaign(this.ops["UpdateCampaign"].apply(partialParams));
    }
    updateEmailChannel(partialParams) {
        return this.client.updateEmailChannel(this.ops["UpdateEmailChannel"].apply(partialParams));
    }
    updateEmailTemplate(partialParams) {
        return this.client.updateEmailTemplate(this.ops["UpdateEmailTemplate"].apply(partialParams));
    }
    updateEndpoint(partialParams) {
        return this.client.updateEndpoint(this.ops["UpdateEndpoint"].apply(partialParams));
    }
    updateEndpointsBatch(partialParams) {
        return this.client.updateEndpointsBatch(this.ops["UpdateEndpointsBatch"].apply(partialParams));
    }
    updateGcmChannel(partialParams) {
        return this.client.updateGcmChannel(this.ops["UpdateGcmChannel"].apply(partialParams));
    }
    updateInAppTemplate(partialParams) {
        return this.client.updateInAppTemplate(this.ops["UpdateInAppTemplate"].apply(partialParams));
    }
    updateJourney(partialParams) {
        return this.client.updateJourney(this.ops["UpdateJourney"].apply(partialParams));
    }
    updateJourneyState(partialParams) {
        return this.client.updateJourneyState(this.ops["UpdateJourneyState"].apply(partialParams));
    }
    updatePushTemplate(partialParams) {
        return this.client.updatePushTemplate(this.ops["UpdatePushTemplate"].apply(partialParams));
    }
    updateRecommenderConfiguration(partialParams) {
        return this.client.updateRecommenderConfiguration(this.ops["UpdateRecommenderConfiguration"].apply(partialParams));
    }
    updateSegment(partialParams) {
        return this.client.updateSegment(this.ops["UpdateSegment"].apply(partialParams));
    }
    updateSmsChannel(partialParams) {
        return this.client.updateSmsChannel(this.ops["UpdateSmsChannel"].apply(partialParams));
    }
    updateSmsTemplate(partialParams) {
        return this.client.updateSmsTemplate(this.ops["UpdateSmsTemplate"].apply(partialParams));
    }
    updateTemplateActiveVersion(partialParams) {
        return this.client.updateTemplateActiveVersion(this.ops["UpdateTemplateActiveVersion"].apply(partialParams));
    }
    updateVoiceChannel(partialParams) {
        return this.client.updateVoiceChannel(this.ops["UpdateVoiceChannel"].apply(partialParams));
    }
    updateVoiceTemplate(partialParams) {
        return this.client.updateVoiceTemplate(this.ops["UpdateVoiceTemplate"].apply(partialParams));
    }
    verifyOTPMessage(partialParams) {
        return this.client.verifyOTPMessage(this.ops["VerifyOTPMessage"].apply(partialParams));
    }
}
exports.default = Resource;
