// out/imagebuilder/Component.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.imagebuilder.Component {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Imagebuilder();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/imagebuilder-2019-12-02.normal.json"), this.client);
    }
    cancelImageCreation(partialParams) {
        return this.client.cancelImageCreation(this.ops["CancelImageCreation"].apply(partialParams));
    }
    createComponent(partialParams) {
        return this.client.createComponent(this.ops["CreateComponent"].apply(partialParams));
    }
    createContainerRecipe(partialParams) {
        return this.client.createContainerRecipe(this.ops["CreateContainerRecipe"].apply(partialParams));
    }
    createDistributionConfiguration(partialParams) {
        return this.client.createDistributionConfiguration(this.ops["CreateDistributionConfiguration"].apply(partialParams));
    }
    createImage(partialParams) {
        return this.client.createImage(this.ops["CreateImage"].apply(partialParams));
    }
    createImagePipeline(partialParams) {
        return this.client.createImagePipeline(this.ops["CreateImagePipeline"].apply(partialParams));
    }
    createImageRecipe(partialParams) {
        return this.client.createImageRecipe(this.ops["CreateImageRecipe"].apply(partialParams));
    }
    createInfrastructureConfiguration(partialParams) {
        return this.client.createInfrastructureConfiguration(this.ops["CreateInfrastructureConfiguration"].apply(partialParams));
    }
    deleteComponent(partialParams) {
        return this.client.deleteComponent(this.ops["DeleteComponent"].apply(partialParams));
    }
    deleteContainerRecipe(partialParams) {
        return this.client.deleteContainerRecipe(this.ops["DeleteContainerRecipe"].apply(partialParams));
    }
    deleteDistributionConfiguration(partialParams) {
        return this.client.deleteDistributionConfiguration(this.ops["DeleteDistributionConfiguration"].apply(partialParams));
    }
    deleteImage(partialParams) {
        return this.client.deleteImage(this.ops["DeleteImage"].apply(partialParams));
    }
    deleteImagePipeline(partialParams) {
        return this.client.deleteImagePipeline(this.ops["DeleteImagePipeline"].apply(partialParams));
    }
    deleteImageRecipe(partialParams) {
        return this.client.deleteImageRecipe(this.ops["DeleteImageRecipe"].apply(partialParams));
    }
    deleteInfrastructureConfiguration(partialParams) {
        return this.client.deleteInfrastructureConfiguration(this.ops["DeleteInfrastructureConfiguration"].apply(partialParams));
    }
    getComponent(partialParams) {
        return this.client.getComponent(this.ops["GetComponent"].apply(partialParams));
    }
    getComponentPolicy(partialParams) {
        return this.client.getComponentPolicy(this.ops["GetComponentPolicy"].apply(partialParams));
    }
    getContainerRecipe(partialParams) {
        return this.client.getContainerRecipe(this.ops["GetContainerRecipe"].apply(partialParams));
    }
    getContainerRecipePolicy(partialParams) {
        return this.client.getContainerRecipePolicy(this.ops["GetContainerRecipePolicy"].apply(partialParams));
    }
    getDistributionConfiguration(partialParams) {
        return this.client.getDistributionConfiguration(this.ops["GetDistributionConfiguration"].apply(partialParams));
    }
    getImage(partialParams) {
        return this.client.getImage(this.ops["GetImage"].apply(partialParams));
    }
    getImagePipeline(partialParams) {
        return this.client.getImagePipeline(this.ops["GetImagePipeline"].apply(partialParams));
    }
    getImagePolicy(partialParams) {
        return this.client.getImagePolicy(this.ops["GetImagePolicy"].apply(partialParams));
    }
    getImageRecipe(partialParams) {
        return this.client.getImageRecipe(this.ops["GetImageRecipe"].apply(partialParams));
    }
    getImageRecipePolicy(partialParams) {
        return this.client.getImageRecipePolicy(this.ops["GetImageRecipePolicy"].apply(partialParams));
    }
    getInfrastructureConfiguration(partialParams) {
        return this.client.getInfrastructureConfiguration(this.ops["GetInfrastructureConfiguration"].apply(partialParams));
    }
    importComponent(partialParams) {
        return this.client.importComponent(this.ops["ImportComponent"].apply(partialParams));
    }
    importVmImage(partialParams) {
        return this.client.importVmImage(this.ops["ImportVmImage"].apply(partialParams));
    }
    listComponentBuildVersions(partialParams) {
        return this.client.listComponentBuildVersions(this.ops["ListComponentBuildVersions"].apply(partialParams));
    }
    listImageBuildVersions(partialParams) {
        return this.client.listImageBuildVersions(this.ops["ListImageBuildVersions"].apply(partialParams));
    }
    listImagePackages(partialParams) {
        return this.client.listImagePackages(this.ops["ListImagePackages"].apply(partialParams));
    }
    listImagePipelineImages(partialParams) {
        return this.client.listImagePipelineImages(this.ops["ListImagePipelineImages"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putComponentPolicy(partialParams) {
        return this.client.putComponentPolicy(this.ops["PutComponentPolicy"].apply(partialParams));
    }
    putContainerRecipePolicy(partialParams) {
        return this.client.putContainerRecipePolicy(this.ops["PutContainerRecipePolicy"].apply(partialParams));
    }
    putImagePolicy(partialParams) {
        return this.client.putImagePolicy(this.ops["PutImagePolicy"].apply(partialParams));
    }
    putImageRecipePolicy(partialParams) {
        return this.client.putImageRecipePolicy(this.ops["PutImageRecipePolicy"].apply(partialParams));
    }
    startImagePipelineExecution(partialParams) {
        return this.client.startImagePipelineExecution(this.ops["StartImagePipelineExecution"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateDistributionConfiguration(partialParams) {
        return this.client.updateDistributionConfiguration(this.ops["UpdateDistributionConfiguration"].apply(partialParams));
    }
    updateImagePipeline(partialParams) {
        return this.client.updateImagePipeline(this.ops["UpdateImagePipeline"].apply(partialParams));
    }
    updateInfrastructureConfiguration(partialParams) {
        return this.client.updateInfrastructureConfiguration(this.ops["UpdateInfrastructureConfiguration"].apply(partialParams));
    }
}
exports.default = Resource;
