// out/chime/VoiceConnectorOrganization.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.chime.VoiceConnectorOrganization {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Chime();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/chime-2018-05-01.normal.json"), this.client);
    }
    associatePhoneNumberWithUser(partialParams) {
        return this.client.associatePhoneNumberWithUser(this.ops["AssociatePhoneNumberWithUser"].apply(partialParams));
    }
    associatePhoneNumbersWithVoiceConnector(partialParams) {
        return this.client.associatePhoneNumbersWithVoiceConnector(this.ops["AssociatePhoneNumbersWithVoiceConnector"].apply(partialParams));
    }
    associatePhoneNumbersWithVoiceConnectorGroup(partialParams) {
        return this.client.associatePhoneNumbersWithVoiceConnectorGroup(this.ops["AssociatePhoneNumbersWithVoiceConnectorGroup"].apply(partialParams));
    }
    associateSigninDelegateGroupsWithAccount(partialParams) {
        return this.client.associateSigninDelegateGroupsWithAccount(this.ops["AssociateSigninDelegateGroupsWithAccount"].apply(partialParams));
    }
    batchCreateAttendee(partialParams) {
        return this.client.batchCreateAttendee(this.ops["BatchCreateAttendee"].apply(partialParams));
    }
    batchCreateChannelMembership(partialParams) {
        return this.client.batchCreateChannelMembership(this.ops["BatchCreateChannelMembership"].apply(partialParams));
    }
    batchCreateRoomMembership(partialParams) {
        return this.client.batchCreateRoomMembership(this.ops["BatchCreateRoomMembership"].apply(partialParams));
    }
    batchDeletePhoneNumber(partialParams) {
        return this.client.batchDeletePhoneNumber(this.ops["BatchDeletePhoneNumber"].apply(partialParams));
    }
    batchSuspendUser(partialParams) {
        return this.client.batchSuspendUser(this.ops["BatchSuspendUser"].apply(partialParams));
    }
    batchUnsuspendUser(partialParams) {
        return this.client.batchUnsuspendUser(this.ops["BatchUnsuspendUser"].apply(partialParams));
    }
    batchUpdatePhoneNumber(partialParams) {
        return this.client.batchUpdatePhoneNumber(this.ops["BatchUpdatePhoneNumber"].apply(partialParams));
    }
    batchUpdateUser(partialParams) {
        return this.client.batchUpdateUser(this.ops["BatchUpdateUser"].apply(partialParams));
    }
    createAccount(partialParams) {
        return this.client.createAccount(this.ops["CreateAccount"].apply(partialParams));
    }
    createAppInstance(partialParams) {
        return this.client.createAppInstance(this.ops["CreateAppInstance"].apply(partialParams));
    }
    createAppInstanceAdmin(partialParams) {
        return this.client.createAppInstanceAdmin(this.ops["CreateAppInstanceAdmin"].apply(partialParams));
    }
    createAppInstanceUser(partialParams) {
        return this.client.createAppInstanceUser(this.ops["CreateAppInstanceUser"].apply(partialParams));
    }
    createAttendee(partialParams) {
        return this.client.createAttendee(this.ops["CreateAttendee"].apply(partialParams));
    }
    createBot(partialParams) {
        return this.client.createBot(this.ops["CreateBot"].apply(partialParams));
    }
    createChannel(partialParams) {
        return this.client.createChannel(this.ops["CreateChannel"].apply(partialParams));
    }
    createChannelBan(partialParams) {
        return this.client.createChannelBan(this.ops["CreateChannelBan"].apply(partialParams));
    }
    createChannelMembership(partialParams) {
        return this.client.createChannelMembership(this.ops["CreateChannelMembership"].apply(partialParams));
    }
    createChannelModerator(partialParams) {
        return this.client.createChannelModerator(this.ops["CreateChannelModerator"].apply(partialParams));
    }
    createMediaCapturePipeline(partialParams) {
        return this.client.createMediaCapturePipeline(this.ops["CreateMediaCapturePipeline"].apply(partialParams));
    }
    createMeeting(partialParams) {
        return this.client.createMeeting(this.ops["CreateMeeting"].apply(partialParams));
    }
    createMeetingDialOut(partialParams) {
        return this.client.createMeetingDialOut(this.ops["CreateMeetingDialOut"].apply(partialParams));
    }
    createMeetingWithAttendees(partialParams) {
        return this.client.createMeetingWithAttendees(this.ops["CreateMeetingWithAttendees"].apply(partialParams));
    }
    createPhoneNumberOrder(partialParams) {
        return this.client.createPhoneNumberOrder(this.ops["CreatePhoneNumberOrder"].apply(partialParams));
    }
    createProxySession(partialParams) {
        return this.client.createProxySession(this.ops["CreateProxySession"].apply(partialParams));
    }
    createRoom(partialParams) {
        return this.client.createRoom(this.ops["CreateRoom"].apply(partialParams));
    }
    createRoomMembership(partialParams) {
        return this.client.createRoomMembership(this.ops["CreateRoomMembership"].apply(partialParams));
    }
    createSipMediaApplication(partialParams) {
        return this.client.createSipMediaApplication(this.ops["CreateSipMediaApplication"].apply(partialParams));
    }
    createSipMediaApplicationCall(partialParams) {
        return this.client.createSipMediaApplicationCall(this.ops["CreateSipMediaApplicationCall"].apply(partialParams));
    }
    createSipRule(partialParams) {
        return this.client.createSipRule(this.ops["CreateSipRule"].apply(partialParams));
    }
    createUser(partialParams) {
        return this.client.createUser(this.ops["CreateUser"].apply(partialParams));
    }
    createVoiceConnector(partialParams) {
        return this.client.createVoiceConnector(this.ops["CreateVoiceConnector"].apply(partialParams));
    }
    createVoiceConnectorGroup(partialParams) {
        return this.client.createVoiceConnectorGroup(this.ops["CreateVoiceConnectorGroup"].apply(partialParams));
    }
    deleteAccount(partialParams) {
        return this.client.deleteAccount(this.ops["DeleteAccount"].apply(partialParams));
    }
    deleteAppInstance(partialParams) {
        return this.client.deleteAppInstance(this.ops["DeleteAppInstance"].apply(partialParams));
    }
    deleteAppInstanceAdmin(partialParams) {
        return this.client.deleteAppInstanceAdmin(this.ops["DeleteAppInstanceAdmin"].apply(partialParams));
    }
    deleteAppInstanceStreamingConfigurations(partialParams) {
        return this.client.deleteAppInstanceStreamingConfigurations(this.ops["DeleteAppInstanceStreamingConfigurations"].apply(partialParams));
    }
    deleteAppInstanceUser(partialParams) {
        return this.client.deleteAppInstanceUser(this.ops["DeleteAppInstanceUser"].apply(partialParams));
    }
    deleteAttendee(partialParams) {
        return this.client.deleteAttendee(this.ops["DeleteAttendee"].apply(partialParams));
    }
    deleteChannel(partialParams) {
        return this.client.deleteChannel(this.ops["DeleteChannel"].apply(partialParams));
    }
    deleteChannelBan(partialParams) {
        return this.client.deleteChannelBan(this.ops["DeleteChannelBan"].apply(partialParams));
    }
    deleteChannelMembership(partialParams) {
        return this.client.deleteChannelMembership(this.ops["DeleteChannelMembership"].apply(partialParams));
    }
    deleteChannelMessage(partialParams) {
        return this.client.deleteChannelMessage(this.ops["DeleteChannelMessage"].apply(partialParams));
    }
    deleteChannelModerator(partialParams) {
        return this.client.deleteChannelModerator(this.ops["DeleteChannelModerator"].apply(partialParams));
    }
    deleteEventsConfiguration(partialParams) {
        return this.client.deleteEventsConfiguration(this.ops["DeleteEventsConfiguration"].apply(partialParams));
    }
    deleteMediaCapturePipeline(partialParams) {
        return this.client.deleteMediaCapturePipeline(this.ops["DeleteMediaCapturePipeline"].apply(partialParams));
    }
    deleteMeeting(partialParams) {
        return this.client.deleteMeeting(this.ops["DeleteMeeting"].apply(partialParams));
    }
    deletePhoneNumber(partialParams) {
        return this.client.deletePhoneNumber(this.ops["DeletePhoneNumber"].apply(partialParams));
    }
    deleteProxySession(partialParams) {
        return this.client.deleteProxySession(this.ops["DeleteProxySession"].apply(partialParams));
    }
    deleteRoom(partialParams) {
        return this.client.deleteRoom(this.ops["DeleteRoom"].apply(partialParams));
    }
    deleteRoomMembership(partialParams) {
        return this.client.deleteRoomMembership(this.ops["DeleteRoomMembership"].apply(partialParams));
    }
    deleteSipMediaApplication(partialParams) {
        return this.client.deleteSipMediaApplication(this.ops["DeleteSipMediaApplication"].apply(partialParams));
    }
    deleteSipRule(partialParams) {
        return this.client.deleteSipRule(this.ops["DeleteSipRule"].apply(partialParams));
    }
    deleteVoiceConnector(partialParams) {
        return this.client.deleteVoiceConnector(this.ops["DeleteVoiceConnector"].apply(partialParams));
    }
    deleteVoiceConnectorEmergencyCallingConfiguration(partialParams) {
        return this.client.deleteVoiceConnectorEmergencyCallingConfiguration(this.ops["DeleteVoiceConnectorEmergencyCallingConfiguration"].apply(partialParams));
    }
    deleteVoiceConnectorGroup(partialParams) {
        return this.client.deleteVoiceConnectorGroup(this.ops["DeleteVoiceConnectorGroup"].apply(partialParams));
    }
    deleteVoiceConnectorOrigination(partialParams) {
        return this.client.deleteVoiceConnectorOrigination(this.ops["DeleteVoiceConnectorOrigination"].apply(partialParams));
    }
    deleteVoiceConnectorProxy(partialParams) {
        return this.client.deleteVoiceConnectorProxy(this.ops["DeleteVoiceConnectorProxy"].apply(partialParams));
    }
    deleteVoiceConnectorStreamingConfiguration(partialParams) {
        return this.client.deleteVoiceConnectorStreamingConfiguration(this.ops["DeleteVoiceConnectorStreamingConfiguration"].apply(partialParams));
    }
    deleteVoiceConnectorTermination(partialParams) {
        return this.client.deleteVoiceConnectorTermination(this.ops["DeleteVoiceConnectorTermination"].apply(partialParams));
    }
    deleteVoiceConnectorTerminationCredentials(partialParams) {
        return this.client.deleteVoiceConnectorTerminationCredentials(this.ops["DeleteVoiceConnectorTerminationCredentials"].apply(partialParams));
    }
    describeAppInstance(partialParams) {
        return this.client.describeAppInstance(this.ops["DescribeAppInstance"].apply(partialParams));
    }
    describeAppInstanceAdmin(partialParams) {
        return this.client.describeAppInstanceAdmin(this.ops["DescribeAppInstanceAdmin"].apply(partialParams));
    }
    describeAppInstanceUser(partialParams) {
        return this.client.describeAppInstanceUser(this.ops["DescribeAppInstanceUser"].apply(partialParams));
    }
    describeChannel(partialParams) {
        return this.client.describeChannel(this.ops["DescribeChannel"].apply(partialParams));
    }
    describeChannelBan(partialParams) {
        return this.client.describeChannelBan(this.ops["DescribeChannelBan"].apply(partialParams));
    }
    describeChannelMembership(partialParams) {
        return this.client.describeChannelMembership(this.ops["DescribeChannelMembership"].apply(partialParams));
    }
    describeChannelMembershipForAppInstanceUser(partialParams) {
        return this.client.describeChannelMembershipForAppInstanceUser(this.ops["DescribeChannelMembershipForAppInstanceUser"].apply(partialParams));
    }
    describeChannelModeratedByAppInstanceUser(partialParams) {
        return this.client.describeChannelModeratedByAppInstanceUser(this.ops["DescribeChannelModeratedByAppInstanceUser"].apply(partialParams));
    }
    describeChannelModerator(partialParams) {
        return this.client.describeChannelModerator(this.ops["DescribeChannelModerator"].apply(partialParams));
    }
    disassociatePhoneNumberFromUser(partialParams) {
        return this.client.disassociatePhoneNumberFromUser(this.ops["DisassociatePhoneNumberFromUser"].apply(partialParams));
    }
    disassociatePhoneNumbersFromVoiceConnector(partialParams) {
        return this.client.disassociatePhoneNumbersFromVoiceConnector(this.ops["DisassociatePhoneNumbersFromVoiceConnector"].apply(partialParams));
    }
    disassociatePhoneNumbersFromVoiceConnectorGroup(partialParams) {
        return this.client.disassociatePhoneNumbersFromVoiceConnectorGroup(this.ops["DisassociatePhoneNumbersFromVoiceConnectorGroup"].apply(partialParams));
    }
    disassociateSigninDelegateGroupsFromAccount(partialParams) {
        return this.client.disassociateSigninDelegateGroupsFromAccount(this.ops["DisassociateSigninDelegateGroupsFromAccount"].apply(partialParams));
    }
    getAccount(partialParams) {
        return this.client.getAccount(this.ops["GetAccount"].apply(partialParams));
    }
    getAccountSettings(partialParams) {
        return this.client.getAccountSettings(this.ops["GetAccountSettings"].apply(partialParams));
    }
    getAppInstanceRetentionSettings(partialParams) {
        return this.client.getAppInstanceRetentionSettings(this.ops["GetAppInstanceRetentionSettings"].apply(partialParams));
    }
    getAppInstanceStreamingConfigurations(partialParams) {
        return this.client.getAppInstanceStreamingConfigurations(this.ops["GetAppInstanceStreamingConfigurations"].apply(partialParams));
    }
    getAttendee(partialParams) {
        return this.client.getAttendee(this.ops["GetAttendee"].apply(partialParams));
    }
    getBot(partialParams) {
        return this.client.getBot(this.ops["GetBot"].apply(partialParams));
    }
    getChannelMessage(partialParams) {
        return this.client.getChannelMessage(this.ops["GetChannelMessage"].apply(partialParams));
    }
    getEventsConfiguration(partialParams) {
        return this.client.getEventsConfiguration(this.ops["GetEventsConfiguration"].apply(partialParams));
    }
    getMediaCapturePipeline(partialParams) {
        return this.client.getMediaCapturePipeline(this.ops["GetMediaCapturePipeline"].apply(partialParams));
    }
    getMeeting(partialParams) {
        return this.client.getMeeting(this.ops["GetMeeting"].apply(partialParams));
    }
    getPhoneNumber(partialParams) {
        return this.client.getPhoneNumber(this.ops["GetPhoneNumber"].apply(partialParams));
    }
    getPhoneNumberOrder(partialParams) {
        return this.client.getPhoneNumberOrder(this.ops["GetPhoneNumberOrder"].apply(partialParams));
    }
    getProxySession(partialParams) {
        return this.client.getProxySession(this.ops["GetProxySession"].apply(partialParams));
    }
    getRetentionSettings(partialParams) {
        return this.client.getRetentionSettings(this.ops["GetRetentionSettings"].apply(partialParams));
    }
    getRoom(partialParams) {
        return this.client.getRoom(this.ops["GetRoom"].apply(partialParams));
    }
    getSipMediaApplication(partialParams) {
        return this.client.getSipMediaApplication(this.ops["GetSipMediaApplication"].apply(partialParams));
    }
    getSipMediaApplicationLoggingConfiguration(partialParams) {
        return this.client.getSipMediaApplicationLoggingConfiguration(this.ops["GetSipMediaApplicationLoggingConfiguration"].apply(partialParams));
    }
    getSipRule(partialParams) {
        return this.client.getSipRule(this.ops["GetSipRule"].apply(partialParams));
    }
    getUser(partialParams) {
        return this.client.getUser(this.ops["GetUser"].apply(partialParams));
    }
    getUserSettings(partialParams) {
        return this.client.getUserSettings(this.ops["GetUserSettings"].apply(partialParams));
    }
    getVoiceConnector(partialParams) {
        return this.client.getVoiceConnector(this.ops["GetVoiceConnector"].apply(partialParams));
    }
    getVoiceConnectorEmergencyCallingConfiguration(partialParams) {
        return this.client.getVoiceConnectorEmergencyCallingConfiguration(this.ops["GetVoiceConnectorEmergencyCallingConfiguration"].apply(partialParams));
    }
    getVoiceConnectorGroup(partialParams) {
        return this.client.getVoiceConnectorGroup(this.ops["GetVoiceConnectorGroup"].apply(partialParams));
    }
    getVoiceConnectorLoggingConfiguration(partialParams) {
        return this.client.getVoiceConnectorLoggingConfiguration(this.ops["GetVoiceConnectorLoggingConfiguration"].apply(partialParams));
    }
    getVoiceConnectorOrigination(partialParams) {
        return this.client.getVoiceConnectorOrigination(this.ops["GetVoiceConnectorOrigination"].apply(partialParams));
    }
    getVoiceConnectorProxy(partialParams) {
        return this.client.getVoiceConnectorProxy(this.ops["GetVoiceConnectorProxy"].apply(partialParams));
    }
    getVoiceConnectorStreamingConfiguration(partialParams) {
        return this.client.getVoiceConnectorStreamingConfiguration(this.ops["GetVoiceConnectorStreamingConfiguration"].apply(partialParams));
    }
    getVoiceConnectorTermination(partialParams) {
        return this.client.getVoiceConnectorTermination(this.ops["GetVoiceConnectorTermination"].apply(partialParams));
    }
    getVoiceConnectorTerminationHealth(partialParams) {
        return this.client.getVoiceConnectorTerminationHealth(this.ops["GetVoiceConnectorTerminationHealth"].apply(partialParams));
    }
    inviteUsers(partialParams) {
        return this.client.inviteUsers(this.ops["InviteUsers"].apply(partialParams));
    }
    listAppInstanceAdmins(partialParams) {
        return this.client.listAppInstanceAdmins(this.ops["ListAppInstanceAdmins"].apply(partialParams));
    }
    listAppInstanceUsers(partialParams) {
        return this.client.listAppInstanceUsers(this.ops["ListAppInstanceUsers"].apply(partialParams));
    }
    listAttendeeTags(partialParams) {
        return this.client.listAttendeeTags(this.ops["ListAttendeeTags"].apply(partialParams));
    }
    listAttendees(partialParams) {
        return this.client.listAttendees(this.ops["ListAttendees"].apply(partialParams));
    }
    listBots(partialParams) {
        return this.client.listBots(this.ops["ListBots"].apply(partialParams));
    }
    listChannelBans(partialParams) {
        return this.client.listChannelBans(this.ops["ListChannelBans"].apply(partialParams));
    }
    listChannelMemberships(partialParams) {
        return this.client.listChannelMemberships(this.ops["ListChannelMemberships"].apply(partialParams));
    }
    listChannelMessages(partialParams) {
        return this.client.listChannelMessages(this.ops["ListChannelMessages"].apply(partialParams));
    }
    listChannelModerators(partialParams) {
        return this.client.listChannelModerators(this.ops["ListChannelModerators"].apply(partialParams));
    }
    listChannels(partialParams) {
        return this.client.listChannels(this.ops["ListChannels"].apply(partialParams));
    }
    listMeetingTags(partialParams) {
        return this.client.listMeetingTags(this.ops["ListMeetingTags"].apply(partialParams));
    }
    listProxySessions(partialParams) {
        return this.client.listProxySessions(this.ops["ListProxySessions"].apply(partialParams));
    }
    listRoomMemberships(partialParams) {
        return this.client.listRoomMemberships(this.ops["ListRoomMemberships"].apply(partialParams));
    }
    listRooms(partialParams) {
        return this.client.listRooms(this.ops["ListRooms"].apply(partialParams));
    }
    listSupportedPhoneNumberCountries(partialParams) {
        return this.client.listSupportedPhoneNumberCountries(this.ops["ListSupportedPhoneNumberCountries"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listUsers(partialParams) {
        return this.client.listUsers(this.ops["ListUsers"].apply(partialParams));
    }
    listVoiceConnectorTerminationCredentials(partialParams) {
        return this.client.listVoiceConnectorTerminationCredentials(this.ops["ListVoiceConnectorTerminationCredentials"].apply(partialParams));
    }
    logoutUser(partialParams) {
        return this.client.logoutUser(this.ops["LogoutUser"].apply(partialParams));
    }
    putAppInstanceRetentionSettings(partialParams) {
        return this.client.putAppInstanceRetentionSettings(this.ops["PutAppInstanceRetentionSettings"].apply(partialParams));
    }
    putAppInstanceStreamingConfigurations(partialParams) {
        return this.client.putAppInstanceStreamingConfigurations(this.ops["PutAppInstanceStreamingConfigurations"].apply(partialParams));
    }
    putEventsConfiguration(partialParams) {
        return this.client.putEventsConfiguration(this.ops["PutEventsConfiguration"].apply(partialParams));
    }
    putRetentionSettings(partialParams) {
        return this.client.putRetentionSettings(this.ops["PutRetentionSettings"].apply(partialParams));
    }
    putSipMediaApplicationLoggingConfiguration(partialParams) {
        return this.client.putSipMediaApplicationLoggingConfiguration(this.ops["PutSipMediaApplicationLoggingConfiguration"].apply(partialParams));
    }
    putVoiceConnectorEmergencyCallingConfiguration(partialParams) {
        return this.client.putVoiceConnectorEmergencyCallingConfiguration(this.ops["PutVoiceConnectorEmergencyCallingConfiguration"].apply(partialParams));
    }
    putVoiceConnectorLoggingConfiguration(partialParams) {
        return this.client.putVoiceConnectorLoggingConfiguration(this.ops["PutVoiceConnectorLoggingConfiguration"].apply(partialParams));
    }
    putVoiceConnectorOrigination(partialParams) {
        return this.client.putVoiceConnectorOrigination(this.ops["PutVoiceConnectorOrigination"].apply(partialParams));
    }
    putVoiceConnectorProxy(partialParams) {
        return this.client.putVoiceConnectorProxy(this.ops["PutVoiceConnectorProxy"].apply(partialParams));
    }
    putVoiceConnectorStreamingConfiguration(partialParams) {
        return this.client.putVoiceConnectorStreamingConfiguration(this.ops["PutVoiceConnectorStreamingConfiguration"].apply(partialParams));
    }
    putVoiceConnectorTermination(partialParams) {
        return this.client.putVoiceConnectorTermination(this.ops["PutVoiceConnectorTermination"].apply(partialParams));
    }
    putVoiceConnectorTerminationCredentials(partialParams) {
        return this.client.putVoiceConnectorTerminationCredentials(this.ops["PutVoiceConnectorTerminationCredentials"].apply(partialParams));
    }
    redactChannelMessage(partialParams) {
        return this.client.redactChannelMessage(this.ops["RedactChannelMessage"].apply(partialParams));
    }
    redactConversationMessage(partialParams) {
        return this.client.redactConversationMessage(this.ops["RedactConversationMessage"].apply(partialParams));
    }
    redactRoomMessage(partialParams) {
        return this.client.redactRoomMessage(this.ops["RedactRoomMessage"].apply(partialParams));
    }
    regenerateSecurityToken(partialParams) {
        return this.client.regenerateSecurityToken(this.ops["RegenerateSecurityToken"].apply(partialParams));
    }
    resetPersonalPIN(partialParams) {
        return this.client.resetPersonalPIN(this.ops["ResetPersonalPIN"].apply(partialParams));
    }
    restorePhoneNumber(partialParams) {
        return this.client.restorePhoneNumber(this.ops["RestorePhoneNumber"].apply(partialParams));
    }
    sendChannelMessage(partialParams) {
        return this.client.sendChannelMessage(this.ops["SendChannelMessage"].apply(partialParams));
    }
    startMeetingTranscription(partialParams) {
        return this.client.startMeetingTranscription(this.ops["StartMeetingTranscription"].apply(partialParams));
    }
    stopMeetingTranscription(partialParams) {
        return this.client.stopMeetingTranscription(this.ops["StopMeetingTranscription"].apply(partialParams));
    }
    tagAttendee(partialParams) {
        return this.client.tagAttendee(this.ops["TagAttendee"].apply(partialParams));
    }
    tagMeeting(partialParams) {
        return this.client.tagMeeting(this.ops["TagMeeting"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagAttendee(partialParams) {
        return this.client.untagAttendee(this.ops["UntagAttendee"].apply(partialParams));
    }
    untagMeeting(partialParams) {
        return this.client.untagMeeting(this.ops["UntagMeeting"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAccount(partialParams) {
        return this.client.updateAccount(this.ops["UpdateAccount"].apply(partialParams));
    }
    updateAccountSettings(partialParams) {
        return this.client.updateAccountSettings(this.ops["UpdateAccountSettings"].apply(partialParams));
    }
    updateAppInstance(partialParams) {
        return this.client.updateAppInstance(this.ops["UpdateAppInstance"].apply(partialParams));
    }
    updateAppInstanceUser(partialParams) {
        return this.client.updateAppInstanceUser(this.ops["UpdateAppInstanceUser"].apply(partialParams));
    }
    updateBot(partialParams) {
        return this.client.updateBot(this.ops["UpdateBot"].apply(partialParams));
    }
    updateChannel(partialParams) {
        return this.client.updateChannel(this.ops["UpdateChannel"].apply(partialParams));
    }
    updateChannelMessage(partialParams) {
        return this.client.updateChannelMessage(this.ops["UpdateChannelMessage"].apply(partialParams));
    }
    updateChannelReadMarker(partialParams) {
        return this.client.updateChannelReadMarker(this.ops["UpdateChannelReadMarker"].apply(partialParams));
    }
    updatePhoneNumber(partialParams) {
        return this.client.updatePhoneNumber(this.ops["UpdatePhoneNumber"].apply(partialParams));
    }
    updatePhoneNumberSettings(partialParams) {
        return this.client.updatePhoneNumberSettings(this.ops["UpdatePhoneNumberSettings"].apply(partialParams));
    }
    updateProxySession(partialParams) {
        return this.client.updateProxySession(this.ops["UpdateProxySession"].apply(partialParams));
    }
    updateRoom(partialParams) {
        return this.client.updateRoom(this.ops["UpdateRoom"].apply(partialParams));
    }
    updateRoomMembership(partialParams) {
        return this.client.updateRoomMembership(this.ops["UpdateRoomMembership"].apply(partialParams));
    }
    updateSipMediaApplication(partialParams) {
        return this.client.updateSipMediaApplication(this.ops["UpdateSipMediaApplication"].apply(partialParams));
    }
    updateSipMediaApplicationCall(partialParams) {
        return this.client.updateSipMediaApplicationCall(this.ops["UpdateSipMediaApplicationCall"].apply(partialParams));
    }
    updateSipRule(partialParams) {
        return this.client.updateSipRule(this.ops["UpdateSipRule"].apply(partialParams));
    }
    updateUser(partialParams) {
        return this.client.updateUser(this.ops["UpdateUser"].apply(partialParams));
    }
    updateUserSettings(partialParams) {
        return this.client.updateUserSettings(this.ops["UpdateUserSettings"].apply(partialParams));
    }
    updateVoiceConnector(partialParams) {
        return this.client.updateVoiceConnector(this.ops["UpdateVoiceConnector"].apply(partialParams));
    }
    updateVoiceConnectorGroup(partialParams) {
        return this.client.updateVoiceConnectorGroup(this.ops["UpdateVoiceConnectorGroup"].apply(partialParams));
    }
}
exports.default = Resource;
