// out/servicediscovery/PublicDnsNamespace.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.servicediscovery.PublicDnsNamespace {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ServiceDiscovery();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/servicediscovery-2017-03-14.normal.json"), this.client);
    }
    createHttpNamespace(partialParams) {
        return this.client.createHttpNamespace(this.ops["CreateHttpNamespace"].apply(partialParams));
    }
    createPrivateDnsNamespace(partialParams) {
        return this.client.createPrivateDnsNamespace(this.ops["CreatePrivateDnsNamespace"].apply(partialParams));
    }
    createPublicDnsNamespace(partialParams) {
        return this.client.createPublicDnsNamespace(this.ops["CreatePublicDnsNamespace"].apply(partialParams));
    }
    createService(partialParams) {
        return this.client.createService(this.ops["CreateService"].apply(partialParams));
    }
    deleteNamespace(partialParams) {
        return this.client.deleteNamespace(this.ops["DeleteNamespace"].apply(partialParams));
    }
    deleteService(partialParams) {
        return this.client.deleteService(this.ops["DeleteService"].apply(partialParams));
    }
    deregisterInstance(partialParams) {
        return this.client.deregisterInstance(this.ops["DeregisterInstance"].apply(partialParams));
    }
    discoverInstances(partialParams) {
        return this.client.discoverInstances(this.ops["DiscoverInstances"].apply(partialParams));
    }
    getInstance(partialParams) {
        return this.client.getInstance(this.ops["GetInstance"].apply(partialParams));
    }
    getInstancesHealthStatus(partialParams) {
        return this.client.getInstancesHealthStatus(this.ops["GetInstancesHealthStatus"].apply(partialParams));
    }
    getNamespace(partialParams) {
        return this.client.getNamespace(this.ops["GetNamespace"].apply(partialParams));
    }
    getOperation(partialParams) {
        return this.client.getOperation(this.ops["GetOperation"].apply(partialParams));
    }
    getService(partialParams) {
        return this.client.getService(this.ops["GetService"].apply(partialParams));
    }
    listInstances(partialParams) {
        return this.client.listInstances(this.ops["ListInstances"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    registerInstance(partialParams) {
        return this.client.registerInstance(this.ops["RegisterInstance"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateHttpNamespace(partialParams) {
        return this.client.updateHttpNamespace(this.ops["UpdateHttpNamespace"].apply(partialParams));
    }
    updateInstanceCustomHealthStatus(partialParams) {
        return this.client.updateInstanceCustomHealthStatus(this.ops["UpdateInstanceCustomHealthStatus"].apply(partialParams));
    }
    updatePrivateDnsNamespace(partialParams) {
        return this.client.updatePrivateDnsNamespace(this.ops["UpdatePrivateDnsNamespace"].apply(partialParams));
    }
    updatePublicDnsNamespace(partialParams) {
        return this.client.updatePublicDnsNamespace(this.ops["UpdatePublicDnsNamespace"].apply(partialParams));
    }
    updateService(partialParams) {
        return this.client.updateService(this.ops["UpdateService"].apply(partialParams));
    }
}
exports.default = Resource;
