// out/securityhub/Member.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.securityhub.Member {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.SecurityHub();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/securityhub-2018-10-26.normal.json"), this.client);
    }
    acceptAdministratorInvitation(partialParams) {
        return this.client.acceptAdministratorInvitation(this.ops["AcceptAdministratorInvitation"].apply(partialParams));
    }
    acceptInvitation(partialParams) {
        return this.client.acceptInvitation(this.ops["AcceptInvitation"].apply(partialParams));
    }
    batchDisableStandards(partialParams) {
        return this.client.batchDisableStandards(this.ops["BatchDisableStandards"].apply(partialParams));
    }
    batchEnableStandards(partialParams) {
        return this.client.batchEnableStandards(this.ops["BatchEnableStandards"].apply(partialParams));
    }
    batchImportFindings(partialParams) {
        return this.client.batchImportFindings(this.ops["BatchImportFindings"].apply(partialParams));
    }
    batchUpdateFindings(partialParams) {
        return this.client.batchUpdateFindings(this.ops["BatchUpdateFindings"].apply(partialParams));
    }
    createActionTarget(partialParams) {
        return this.client.createActionTarget(this.ops["CreateActionTarget"].apply(partialParams));
    }
    createFindingAggregator(partialParams) {
        return this.client.createFindingAggregator(this.ops["CreateFindingAggregator"].apply(partialParams));
    }
    createInsight(partialParams) {
        return this.client.createInsight(this.ops["CreateInsight"].apply(partialParams));
    }
    createMembers(partialParams) {
        return this.client.createMembers(this.ops["CreateMembers"].apply(partialParams));
    }
    declineInvitations(partialParams) {
        return this.client.declineInvitations(this.ops["DeclineInvitations"].apply(partialParams));
    }
    deleteActionTarget(partialParams) {
        return this.client.deleteActionTarget(this.ops["DeleteActionTarget"].apply(partialParams));
    }
    deleteFindingAggregator(partialParams) {
        return this.client.deleteFindingAggregator(this.ops["DeleteFindingAggregator"].apply(partialParams));
    }
    deleteInsight(partialParams) {
        return this.client.deleteInsight(this.ops["DeleteInsight"].apply(partialParams));
    }
    deleteInvitations(partialParams) {
        return this.client.deleteInvitations(this.ops["DeleteInvitations"].apply(partialParams));
    }
    deleteMembers(partialParams) {
        return this.client.deleteMembers(this.ops["DeleteMembers"].apply(partialParams));
    }
    describeStandardsControls(partialParams) {
        return this.client.describeStandardsControls(this.ops["DescribeStandardsControls"].apply(partialParams));
    }
    disableImportFindingsForProduct(partialParams) {
        return this.client.disableImportFindingsForProduct(this.ops["DisableImportFindingsForProduct"].apply(partialParams));
    }
    disableOrganizationAdminAccount(partialParams) {
        return this.client.disableOrganizationAdminAccount(this.ops["DisableOrganizationAdminAccount"].apply(partialParams));
    }
    disassociateMembers(partialParams) {
        return this.client.disassociateMembers(this.ops["DisassociateMembers"].apply(partialParams));
    }
    enableImportFindingsForProduct(partialParams) {
        return this.client.enableImportFindingsForProduct(this.ops["EnableImportFindingsForProduct"].apply(partialParams));
    }
    enableOrganizationAdminAccount(partialParams) {
        return this.client.enableOrganizationAdminAccount(this.ops["EnableOrganizationAdminAccount"].apply(partialParams));
    }
    getFindingAggregator(partialParams) {
        return this.client.getFindingAggregator(this.ops["GetFindingAggregator"].apply(partialParams));
    }
    getInsightResults(partialParams) {
        return this.client.getInsightResults(this.ops["GetInsightResults"].apply(partialParams));
    }
    getMembers(partialParams) {
        return this.client.getMembers(this.ops["GetMembers"].apply(partialParams));
    }
    inviteMembers(partialParams) {
        return this.client.inviteMembers(this.ops["InviteMembers"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateActionTarget(partialParams) {
        return this.client.updateActionTarget(this.ops["UpdateActionTarget"].apply(partialParams));
    }
    updateFindingAggregator(partialParams) {
        return this.client.updateFindingAggregator(this.ops["UpdateFindingAggregator"].apply(partialParams));
    }
    updateFindings(partialParams) {
        return this.client.updateFindings(this.ops["UpdateFindings"].apply(partialParams));
    }
    updateInsight(partialParams) {
        return this.client.updateInsight(this.ops["UpdateInsight"].apply(partialParams));
    }
    updateOrganizationConfiguration(partialParams) {
        return this.client.updateOrganizationConfiguration(this.ops["UpdateOrganizationConfiguration"].apply(partialParams));
    }
    updateStandardsControl(partialParams) {
        return this.client.updateStandardsControl(this.ops["UpdateStandardsControl"].apply(partialParams));
    }
}
exports.default = Resource;
