// out/appsync/ApiCache.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.appsync.ApiCache {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.AppSync();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/appsync-2017-07-25.normal.json"), this.client);
    }
    associateApi(partialParams) {
        return this.client.associateApi(this.ops["AssociateApi"].apply(partialParams));
    }
    createApiCache(partialParams) {
        return this.client.createApiCache(this.ops["CreateApiCache"].apply(partialParams));
    }
    createApiKey(partialParams) {
        return this.client.createApiKey(this.ops["CreateApiKey"].apply(partialParams));
    }
    createDataSource(partialParams) {
        return this.client.createDataSource(this.ops["CreateDataSource"].apply(partialParams));
    }
    createDomainName(partialParams) {
        return this.client.createDomainName(this.ops["CreateDomainName"].apply(partialParams));
    }
    createFunction(partialParams) {
        return this.client.createFunction(this.ops["CreateFunction"].apply(partialParams));
    }
    createGraphqlApi(partialParams) {
        return this.client.createGraphqlApi(this.ops["CreateGraphqlApi"].apply(partialParams));
    }
    createResolver(partialParams) {
        return this.client.createResolver(this.ops["CreateResolver"].apply(partialParams));
    }
    createType(partialParams) {
        return this.client.createType(this.ops["CreateType"].apply(partialParams));
    }
    deleteApiCache(partialParams) {
        return this.client.deleteApiCache(this.ops["DeleteApiCache"].apply(partialParams));
    }
    deleteApiKey(partialParams) {
        return this.client.deleteApiKey(this.ops["DeleteApiKey"].apply(partialParams));
    }
    deleteDataSource(partialParams) {
        return this.client.deleteDataSource(this.ops["DeleteDataSource"].apply(partialParams));
    }
    deleteDomainName(partialParams) {
        return this.client.deleteDomainName(this.ops["DeleteDomainName"].apply(partialParams));
    }
    deleteFunction(partialParams) {
        return this.client.deleteFunction(this.ops["DeleteFunction"].apply(partialParams));
    }
    deleteGraphqlApi(partialParams) {
        return this.client.deleteGraphqlApi(this.ops["DeleteGraphqlApi"].apply(partialParams));
    }
    deleteResolver(partialParams) {
        return this.client.deleteResolver(this.ops["DeleteResolver"].apply(partialParams));
    }
    deleteType(partialParams) {
        return this.client.deleteType(this.ops["DeleteType"].apply(partialParams));
    }
    disassociateApi(partialParams) {
        return this.client.disassociateApi(this.ops["DisassociateApi"].apply(partialParams));
    }
    flushApiCache(partialParams) {
        return this.client.flushApiCache(this.ops["FlushApiCache"].apply(partialParams));
    }
    getApiAssociation(partialParams) {
        return this.client.getApiAssociation(this.ops["GetApiAssociation"].apply(partialParams));
    }
    getApiCache(partialParams) {
        return this.client.getApiCache(this.ops["GetApiCache"].apply(partialParams));
    }
    getDataSource(partialParams) {
        return this.client.getDataSource(this.ops["GetDataSource"].apply(partialParams));
    }
    getDomainName(partialParams) {
        return this.client.getDomainName(this.ops["GetDomainName"].apply(partialParams));
    }
    getFunction(partialParams) {
        return this.client.getFunction(this.ops["GetFunction"].apply(partialParams));
    }
    getGraphqlApi(partialParams) {
        return this.client.getGraphqlApi(this.ops["GetGraphqlApi"].apply(partialParams));
    }
    getIntrospectionSchema(partialParams) {
        return this.client.getIntrospectionSchema(this.ops["GetIntrospectionSchema"].apply(partialParams));
    }
    getResolver(partialParams) {
        return this.client.getResolver(this.ops["GetResolver"].apply(partialParams));
    }
    getSchemaCreationStatus(partialParams) {
        return this.client.getSchemaCreationStatus(this.ops["GetSchemaCreationStatus"].apply(partialParams));
    }
    getType(partialParams) {
        return this.client.getType(this.ops["GetType"].apply(partialParams));
    }
    listApiKeys(partialParams) {
        return this.client.listApiKeys(this.ops["ListApiKeys"].apply(partialParams));
    }
    listDataSources(partialParams) {
        return this.client.listDataSources(this.ops["ListDataSources"].apply(partialParams));
    }
    listFunctions(partialParams) {
        return this.client.listFunctions(this.ops["ListFunctions"].apply(partialParams));
    }
    listResolvers(partialParams) {
        return this.client.listResolvers(this.ops["ListResolvers"].apply(partialParams));
    }
    listResolversByFunction(partialParams) {
        return this.client.listResolversByFunction(this.ops["ListResolversByFunction"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTypes(partialParams) {
        return this.client.listTypes(this.ops["ListTypes"].apply(partialParams));
    }
    startSchemaCreation(partialParams) {
        return this.client.startSchemaCreation(this.ops["StartSchemaCreation"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApiCache(partialParams) {
        return this.client.updateApiCache(this.ops["UpdateApiCache"].apply(partialParams));
    }
    updateApiKey(partialParams) {
        return this.client.updateApiKey(this.ops["UpdateApiKey"].apply(partialParams));
    }
    updateDataSource(partialParams) {
        return this.client.updateDataSource(this.ops["UpdateDataSource"].apply(partialParams));
    }
    updateDomainName(partialParams) {
        return this.client.updateDomainName(this.ops["UpdateDomainName"].apply(partialParams));
    }
    updateFunction(partialParams) {
        return this.client.updateFunction(this.ops["UpdateFunction"].apply(partialParams));
    }
    updateGraphqlApi(partialParams) {
        return this.client.updateGraphqlApi(this.ops["UpdateGraphqlApi"].apply(partialParams));
    }
    updateResolver(partialParams) {
        return this.client.updateResolver(this.ops["UpdateResolver"].apply(partialParams));
    }
    updateType(partialParams) {
        return this.client.updateType(this.ops["UpdateType"].apply(partialParams));
    }
}
exports.default = Resource;
