// out/glacier/Vault.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.glacier.Vault {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Glacier();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/glacier-2012-06-01.normal.json"), this.client);
    }
    abortMultipartUpload(partialParams) {
        return this.client.abortMultipartUpload(this.ops["AbortMultipartUpload"].apply(partialParams));
    }
    abortVaultLock(partialParams) {
        return this.client.abortVaultLock(this.ops["AbortVaultLock"].apply(partialParams));
    }
    addTagsToVault(partialParams) {
        return this.client.addTagsToVault(this.ops["AddTagsToVault"].apply(partialParams));
    }
    completeMultipartUpload(partialParams) {
        return this.client.completeMultipartUpload(this.ops["CompleteMultipartUpload"].apply(partialParams));
    }
    completeVaultLock(partialParams) {
        return this.client.completeVaultLock(this.ops["CompleteVaultLock"].apply(partialParams));
    }
    createVault(partialParams) {
        return this.client.createVault(this.ops["CreateVault"].apply(partialParams));
    }
    deleteArchive(partialParams) {
        return this.client.deleteArchive(this.ops["DeleteArchive"].apply(partialParams));
    }
    deleteVault(partialParams) {
        return this.client.deleteVault(this.ops["DeleteVault"].apply(partialParams));
    }
    deleteVaultAccessPolicy(partialParams) {
        return this.client.deleteVaultAccessPolicy(this.ops["DeleteVaultAccessPolicy"].apply(partialParams));
    }
    deleteVaultNotifications(partialParams) {
        return this.client.deleteVaultNotifications(this.ops["DeleteVaultNotifications"].apply(partialParams));
    }
    describeJob(partialParams) {
        return this.client.describeJob(this.ops["DescribeJob"].apply(partialParams));
    }
    describeVault(partialParams) {
        return this.client.describeVault(this.ops["DescribeVault"].apply(partialParams));
    }
    getDataRetrievalPolicy(partialParams) {
        return this.client.getDataRetrievalPolicy(this.ops["GetDataRetrievalPolicy"].apply(partialParams));
    }
    getJobOutput(partialParams) {
        return this.client.getJobOutput(this.ops["GetJobOutput"].apply(partialParams));
    }
    getVaultAccessPolicy(partialParams) {
        return this.client.getVaultAccessPolicy(this.ops["GetVaultAccessPolicy"].apply(partialParams));
    }
    getVaultLock(partialParams) {
        return this.client.getVaultLock(this.ops["GetVaultLock"].apply(partialParams));
    }
    getVaultNotifications(partialParams) {
        return this.client.getVaultNotifications(this.ops["GetVaultNotifications"].apply(partialParams));
    }
    initiateJob(partialParams) {
        return this.client.initiateJob(this.ops["InitiateJob"].apply(partialParams));
    }
    initiateMultipartUpload(partialParams) {
        return this.client.initiateMultipartUpload(this.ops["InitiateMultipartUpload"].apply(partialParams));
    }
    initiateVaultLock(partialParams) {
        return this.client.initiateVaultLock(this.ops["InitiateVaultLock"].apply(partialParams));
    }
    listJobs(partialParams) {
        return this.client.listJobs(this.ops["ListJobs"].apply(partialParams));
    }
    listMultipartUploads(partialParams) {
        return this.client.listMultipartUploads(this.ops["ListMultipartUploads"].apply(partialParams));
    }
    listParts(partialParams) {
        return this.client.listParts(this.ops["ListParts"].apply(partialParams));
    }
    listProvisionedCapacity(partialParams) {
        return this.client.listProvisionedCapacity(this.ops["ListProvisionedCapacity"].apply(partialParams));
    }
    listTagsForVault(partialParams) {
        return this.client.listTagsForVault(this.ops["ListTagsForVault"].apply(partialParams));
    }
    listVaults(partialParams) {
        return this.client.listVaults(this.ops["ListVaults"].apply(partialParams));
    }
    purchaseProvisionedCapacity(partialParams) {
        return this.client.purchaseProvisionedCapacity(this.ops["PurchaseProvisionedCapacity"].apply(partialParams));
    }
    removeTagsFromVault(partialParams) {
        return this.client.removeTagsFromVault(this.ops["RemoveTagsFromVault"].apply(partialParams));
    }
    setDataRetrievalPolicy(partialParams) {
        return this.client.setDataRetrievalPolicy(this.ops["SetDataRetrievalPolicy"].apply(partialParams));
    }
    setVaultAccessPolicy(partialParams) {
        return this.client.setVaultAccessPolicy(this.ops["SetVaultAccessPolicy"].apply(partialParams));
    }
    setVaultNotifications(partialParams) {
        return this.client.setVaultNotifications(this.ops["SetVaultNotifications"].apply(partialParams));
    }
    uploadArchive(partialParams) {
        return this.client.uploadArchive(this.ops["UploadArchive"].apply(partialParams));
    }
    uploadMultipartPart(partialParams) {
        return this.client.uploadMultipartPart(this.ops["UploadMultipartPart"].apply(partialParams));
    }
}
exports.default = Resource;
