// out/connect/BotAssociation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.connect.BotAssociation {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Connect();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/connect-2017-08-08.normal.json"), this.client);
    }
    associateApprovedOrigin(partialParams) {
        return this.client.associateApprovedOrigin(this.ops["AssociateApprovedOrigin"].apply(partialParams));
    }
    associateBot(partialParams) {
        return this.client.associateBot(this.ops["AssociateBot"].apply(partialParams));
    }
    associateDefaultVocabulary(partialParams) {
        return this.client.associateDefaultVocabulary(this.ops["AssociateDefaultVocabulary"].apply(partialParams));
    }
    associateInstanceStorageConfig(partialParams) {
        return this.client.associateInstanceStorageConfig(this.ops["AssociateInstanceStorageConfig"].apply(partialParams));
    }
    associateLambdaFunction(partialParams) {
        return this.client.associateLambdaFunction(this.ops["AssociateLambdaFunction"].apply(partialParams));
    }
    associateLexBot(partialParams) {
        return this.client.associateLexBot(this.ops["AssociateLexBot"].apply(partialParams));
    }
    associateQueueQuickConnects(partialParams) {
        return this.client.associateQueueQuickConnects(this.ops["AssociateQueueQuickConnects"].apply(partialParams));
    }
    associateRoutingProfileQueues(partialParams) {
        return this.client.associateRoutingProfileQueues(this.ops["AssociateRoutingProfileQueues"].apply(partialParams));
    }
    associateSecurityKey(partialParams) {
        return this.client.associateSecurityKey(this.ops["AssociateSecurityKey"].apply(partialParams));
    }
    createAgentStatus(partialParams) {
        return this.client.createAgentStatus(this.ops["CreateAgentStatus"].apply(partialParams));
    }
    createContactFlow(partialParams) {
        return this.client.createContactFlow(this.ops["CreateContactFlow"].apply(partialParams));
    }
    createContactFlowModule(partialParams) {
        return this.client.createContactFlowModule(this.ops["CreateContactFlowModule"].apply(partialParams));
    }
    createHoursOfOperation(partialParams) {
        return this.client.createHoursOfOperation(this.ops["CreateHoursOfOperation"].apply(partialParams));
    }
    createInstance(partialParams) {
        return this.client.createInstance(this.ops["CreateInstance"].apply(partialParams));
    }
    createIntegrationAssociation(partialParams) {
        return this.client.createIntegrationAssociation(this.ops["CreateIntegrationAssociation"].apply(partialParams));
    }
    createQueue(partialParams) {
        return this.client.createQueue(this.ops["CreateQueue"].apply(partialParams));
    }
    createQuickConnect(partialParams) {
        return this.client.createQuickConnect(this.ops["CreateQuickConnect"].apply(partialParams));
    }
    createRoutingProfile(partialParams) {
        return this.client.createRoutingProfile(this.ops["CreateRoutingProfile"].apply(partialParams));
    }
    createSecurityProfile(partialParams) {
        return this.client.createSecurityProfile(this.ops["CreateSecurityProfile"].apply(partialParams));
    }
    createUseCase(partialParams) {
        return this.client.createUseCase(this.ops["CreateUseCase"].apply(partialParams));
    }
    createUser(partialParams) {
        return this.client.createUser(this.ops["CreateUser"].apply(partialParams));
    }
    createUserHierarchyGroup(partialParams) {
        return this.client.createUserHierarchyGroup(this.ops["CreateUserHierarchyGroup"].apply(partialParams));
    }
    createVocabulary(partialParams) {
        return this.client.createVocabulary(this.ops["CreateVocabulary"].apply(partialParams));
    }
    deleteContactFlow(partialParams) {
        return this.client.deleteContactFlow(this.ops["DeleteContactFlow"].apply(partialParams));
    }
    deleteContactFlowModule(partialParams) {
        return this.client.deleteContactFlowModule(this.ops["DeleteContactFlowModule"].apply(partialParams));
    }
    deleteHoursOfOperation(partialParams) {
        return this.client.deleteHoursOfOperation(this.ops["DeleteHoursOfOperation"].apply(partialParams));
    }
    deleteInstance(partialParams) {
        return this.client.deleteInstance(this.ops["DeleteInstance"].apply(partialParams));
    }
    deleteIntegrationAssociation(partialParams) {
        return this.client.deleteIntegrationAssociation(this.ops["DeleteIntegrationAssociation"].apply(partialParams));
    }
    deleteQuickConnect(partialParams) {
        return this.client.deleteQuickConnect(this.ops["DeleteQuickConnect"].apply(partialParams));
    }
    deleteSecurityProfile(partialParams) {
        return this.client.deleteSecurityProfile(this.ops["DeleteSecurityProfile"].apply(partialParams));
    }
    deleteUseCase(partialParams) {
        return this.client.deleteUseCase(this.ops["DeleteUseCase"].apply(partialParams));
    }
    deleteUser(partialParams) {
        return this.client.deleteUser(this.ops["DeleteUser"].apply(partialParams));
    }
    deleteUserHierarchyGroup(partialParams) {
        return this.client.deleteUserHierarchyGroup(this.ops["DeleteUserHierarchyGroup"].apply(partialParams));
    }
    deleteVocabulary(partialParams) {
        return this.client.deleteVocabulary(this.ops["DeleteVocabulary"].apply(partialParams));
    }
    describeAgentStatus(partialParams) {
        return this.client.describeAgentStatus(this.ops["DescribeAgentStatus"].apply(partialParams));
    }
    describeContact(partialParams) {
        return this.client.describeContact(this.ops["DescribeContact"].apply(partialParams));
    }
    describeContactFlow(partialParams) {
        return this.client.describeContactFlow(this.ops["DescribeContactFlow"].apply(partialParams));
    }
    describeContactFlowModule(partialParams) {
        return this.client.describeContactFlowModule(this.ops["DescribeContactFlowModule"].apply(partialParams));
    }
    describeHoursOfOperation(partialParams) {
        return this.client.describeHoursOfOperation(this.ops["DescribeHoursOfOperation"].apply(partialParams));
    }
    describeInstance(partialParams) {
        return this.client.describeInstance(this.ops["DescribeInstance"].apply(partialParams));
    }
    describeInstanceAttribute(partialParams) {
        return this.client.describeInstanceAttribute(this.ops["DescribeInstanceAttribute"].apply(partialParams));
    }
    describeInstanceStorageConfig(partialParams) {
        return this.client.describeInstanceStorageConfig(this.ops["DescribeInstanceStorageConfig"].apply(partialParams));
    }
    describeQueue(partialParams) {
        return this.client.describeQueue(this.ops["DescribeQueue"].apply(partialParams));
    }
    describeQuickConnect(partialParams) {
        return this.client.describeQuickConnect(this.ops["DescribeQuickConnect"].apply(partialParams));
    }
    describeRoutingProfile(partialParams) {
        return this.client.describeRoutingProfile(this.ops["DescribeRoutingProfile"].apply(partialParams));
    }
    describeSecurityProfile(partialParams) {
        return this.client.describeSecurityProfile(this.ops["DescribeSecurityProfile"].apply(partialParams));
    }
    describeUser(partialParams) {
        return this.client.describeUser(this.ops["DescribeUser"].apply(partialParams));
    }
    describeUserHierarchyGroup(partialParams) {
        return this.client.describeUserHierarchyGroup(this.ops["DescribeUserHierarchyGroup"].apply(partialParams));
    }
    describeUserHierarchyStructure(partialParams) {
        return this.client.describeUserHierarchyStructure(this.ops["DescribeUserHierarchyStructure"].apply(partialParams));
    }
    describeVocabulary(partialParams) {
        return this.client.describeVocabulary(this.ops["DescribeVocabulary"].apply(partialParams));
    }
    disassociateApprovedOrigin(partialParams) {
        return this.client.disassociateApprovedOrigin(this.ops["DisassociateApprovedOrigin"].apply(partialParams));
    }
    disassociateBot(partialParams) {
        return this.client.disassociateBot(this.ops["DisassociateBot"].apply(partialParams));
    }
    disassociateInstanceStorageConfig(partialParams) {
        return this.client.disassociateInstanceStorageConfig(this.ops["DisassociateInstanceStorageConfig"].apply(partialParams));
    }
    disassociateLambdaFunction(partialParams) {
        return this.client.disassociateLambdaFunction(this.ops["DisassociateLambdaFunction"].apply(partialParams));
    }
    disassociateLexBot(partialParams) {
        return this.client.disassociateLexBot(this.ops["DisassociateLexBot"].apply(partialParams));
    }
    disassociateQueueQuickConnects(partialParams) {
        return this.client.disassociateQueueQuickConnects(this.ops["DisassociateQueueQuickConnects"].apply(partialParams));
    }
    disassociateRoutingProfileQueues(partialParams) {
        return this.client.disassociateRoutingProfileQueues(this.ops["DisassociateRoutingProfileQueues"].apply(partialParams));
    }
    disassociateSecurityKey(partialParams) {
        return this.client.disassociateSecurityKey(this.ops["DisassociateSecurityKey"].apply(partialParams));
    }
    getContactAttributes(partialParams) {
        return this.client.getContactAttributes(this.ops["GetContactAttributes"].apply(partialParams));
    }
    getCurrentMetricData(partialParams) {
        return this.client.getCurrentMetricData(this.ops["GetCurrentMetricData"].apply(partialParams));
    }
    getFederationToken(partialParams) {
        return this.client.getFederationToken(this.ops["GetFederationToken"].apply(partialParams));
    }
    getMetricData(partialParams) {
        return this.client.getMetricData(this.ops["GetMetricData"].apply(partialParams));
    }
    listAgentStatuses(partialParams) {
        return this.client.listAgentStatuses(this.ops["ListAgentStatuses"].apply(partialParams));
    }
    listApprovedOrigins(partialParams) {
        return this.client.listApprovedOrigins(this.ops["ListApprovedOrigins"].apply(partialParams));
    }
    listBots(partialParams) {
        return this.client.listBots(this.ops["ListBots"].apply(partialParams));
    }
    listContactFlowModules(partialParams) {
        return this.client.listContactFlowModules(this.ops["ListContactFlowModules"].apply(partialParams));
    }
    listContactFlows(partialParams) {
        return this.client.listContactFlows(this.ops["ListContactFlows"].apply(partialParams));
    }
    listContactReferences(partialParams) {
        return this.client.listContactReferences(this.ops["ListContactReferences"].apply(partialParams));
    }
    listDefaultVocabularies(partialParams) {
        return this.client.listDefaultVocabularies(this.ops["ListDefaultVocabularies"].apply(partialParams));
    }
    listHoursOfOperations(partialParams) {
        return this.client.listHoursOfOperations(this.ops["ListHoursOfOperations"].apply(partialParams));
    }
    listInstanceAttributes(partialParams) {
        return this.client.listInstanceAttributes(this.ops["ListInstanceAttributes"].apply(partialParams));
    }
    listInstanceStorageConfigs(partialParams) {
        return this.client.listInstanceStorageConfigs(this.ops["ListInstanceStorageConfigs"].apply(partialParams));
    }
    listIntegrationAssociations(partialParams) {
        return this.client.listIntegrationAssociations(this.ops["ListIntegrationAssociations"].apply(partialParams));
    }
    listLambdaFunctions(partialParams) {
        return this.client.listLambdaFunctions(this.ops["ListLambdaFunctions"].apply(partialParams));
    }
    listLexBots(partialParams) {
        return this.client.listLexBots(this.ops["ListLexBots"].apply(partialParams));
    }
    listPhoneNumbers(partialParams) {
        return this.client.listPhoneNumbers(this.ops["ListPhoneNumbers"].apply(partialParams));
    }
    listPrompts(partialParams) {
        return this.client.listPrompts(this.ops["ListPrompts"].apply(partialParams));
    }
    listQueueQuickConnects(partialParams) {
        return this.client.listQueueQuickConnects(this.ops["ListQueueQuickConnects"].apply(partialParams));
    }
    listQueues(partialParams) {
        return this.client.listQueues(this.ops["ListQueues"].apply(partialParams));
    }
    listQuickConnects(partialParams) {
        return this.client.listQuickConnects(this.ops["ListQuickConnects"].apply(partialParams));
    }
    listRoutingProfileQueues(partialParams) {
        return this.client.listRoutingProfileQueues(this.ops["ListRoutingProfileQueues"].apply(partialParams));
    }
    listRoutingProfiles(partialParams) {
        return this.client.listRoutingProfiles(this.ops["ListRoutingProfiles"].apply(partialParams));
    }
    listSecurityKeys(partialParams) {
        return this.client.listSecurityKeys(this.ops["ListSecurityKeys"].apply(partialParams));
    }
    listSecurityProfilePermissions(partialParams) {
        return this.client.listSecurityProfilePermissions(this.ops["ListSecurityProfilePermissions"].apply(partialParams));
    }
    listSecurityProfiles(partialParams) {
        return this.client.listSecurityProfiles(this.ops["ListSecurityProfiles"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listUseCases(partialParams) {
        return this.client.listUseCases(this.ops["ListUseCases"].apply(partialParams));
    }
    listUserHierarchyGroups(partialParams) {
        return this.client.listUserHierarchyGroups(this.ops["ListUserHierarchyGroups"].apply(partialParams));
    }
    listUsers(partialParams) {
        return this.client.listUsers(this.ops["ListUsers"].apply(partialParams));
    }
    resumeContactRecording(partialParams) {
        return this.client.resumeContactRecording(this.ops["ResumeContactRecording"].apply(partialParams));
    }
    searchVocabularies(partialParams) {
        return this.client.searchVocabularies(this.ops["SearchVocabularies"].apply(partialParams));
    }
    startChatContact(partialParams) {
        return this.client.startChatContact(this.ops["StartChatContact"].apply(partialParams));
    }
    startContactRecording(partialParams) {
        return this.client.startContactRecording(this.ops["StartContactRecording"].apply(partialParams));
    }
    startContactStreaming(partialParams) {
        return this.client.startContactStreaming(this.ops["StartContactStreaming"].apply(partialParams));
    }
    startOutboundVoiceContact(partialParams) {
        return this.client.startOutboundVoiceContact(this.ops["StartOutboundVoiceContact"].apply(partialParams));
    }
    startTaskContact(partialParams) {
        return this.client.startTaskContact(this.ops["StartTaskContact"].apply(partialParams));
    }
    stopContact(partialParams) {
        return this.client.stopContact(this.ops["StopContact"].apply(partialParams));
    }
    stopContactRecording(partialParams) {
        return this.client.stopContactRecording(this.ops["StopContactRecording"].apply(partialParams));
    }
    stopContactStreaming(partialParams) {
        return this.client.stopContactStreaming(this.ops["StopContactStreaming"].apply(partialParams));
    }
    suspendContactRecording(partialParams) {
        return this.client.suspendContactRecording(this.ops["SuspendContactRecording"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAgentStatus(partialParams) {
        return this.client.updateAgentStatus(this.ops["UpdateAgentStatus"].apply(partialParams));
    }
    updateContact(partialParams) {
        return this.client.updateContact(this.ops["UpdateContact"].apply(partialParams));
    }
    updateContactAttributes(partialParams) {
        return this.client.updateContactAttributes(this.ops["UpdateContactAttributes"].apply(partialParams));
    }
    updateContactFlowContent(partialParams) {
        return this.client.updateContactFlowContent(this.ops["UpdateContactFlowContent"].apply(partialParams));
    }
    updateContactFlowMetadata(partialParams) {
        return this.client.updateContactFlowMetadata(this.ops["UpdateContactFlowMetadata"].apply(partialParams));
    }
    updateContactFlowModuleContent(partialParams) {
        return this.client.updateContactFlowModuleContent(this.ops["UpdateContactFlowModuleContent"].apply(partialParams));
    }
    updateContactFlowModuleMetadata(partialParams) {
        return this.client.updateContactFlowModuleMetadata(this.ops["UpdateContactFlowModuleMetadata"].apply(partialParams));
    }
    updateContactFlowName(partialParams) {
        return this.client.updateContactFlowName(this.ops["UpdateContactFlowName"].apply(partialParams));
    }
    updateContactSchedule(partialParams) {
        return this.client.updateContactSchedule(this.ops["UpdateContactSchedule"].apply(partialParams));
    }
    updateHoursOfOperation(partialParams) {
        return this.client.updateHoursOfOperation(this.ops["UpdateHoursOfOperation"].apply(partialParams));
    }
    updateInstanceAttribute(partialParams) {
        return this.client.updateInstanceAttribute(this.ops["UpdateInstanceAttribute"].apply(partialParams));
    }
    updateInstanceStorageConfig(partialParams) {
        return this.client.updateInstanceStorageConfig(this.ops["UpdateInstanceStorageConfig"].apply(partialParams));
    }
    updateQueueHoursOfOperation(partialParams) {
        return this.client.updateQueueHoursOfOperation(this.ops["UpdateQueueHoursOfOperation"].apply(partialParams));
    }
    updateQueueMaxContacts(partialParams) {
        return this.client.updateQueueMaxContacts(this.ops["UpdateQueueMaxContacts"].apply(partialParams));
    }
    updateQueueName(partialParams) {
        return this.client.updateQueueName(this.ops["UpdateQueueName"].apply(partialParams));
    }
    updateQueueOutboundCallerConfig(partialParams) {
        return this.client.updateQueueOutboundCallerConfig(this.ops["UpdateQueueOutboundCallerConfig"].apply(partialParams));
    }
    updateQueueStatus(partialParams) {
        return this.client.updateQueueStatus(this.ops["UpdateQueueStatus"].apply(partialParams));
    }
    updateQuickConnectConfig(partialParams) {
        return this.client.updateQuickConnectConfig(this.ops["UpdateQuickConnectConfig"].apply(partialParams));
    }
    updateQuickConnectName(partialParams) {
        return this.client.updateQuickConnectName(this.ops["UpdateQuickConnectName"].apply(partialParams));
    }
    updateRoutingProfileConcurrency(partialParams) {
        return this.client.updateRoutingProfileConcurrency(this.ops["UpdateRoutingProfileConcurrency"].apply(partialParams));
    }
    updateRoutingProfileDefaultOutboundQueue(partialParams) {
        return this.client.updateRoutingProfileDefaultOutboundQueue(this.ops["UpdateRoutingProfileDefaultOutboundQueue"].apply(partialParams));
    }
    updateRoutingProfileName(partialParams) {
        return this.client.updateRoutingProfileName(this.ops["UpdateRoutingProfileName"].apply(partialParams));
    }
    updateRoutingProfileQueues(partialParams) {
        return this.client.updateRoutingProfileQueues(this.ops["UpdateRoutingProfileQueues"].apply(partialParams));
    }
    updateSecurityProfile(partialParams) {
        return this.client.updateSecurityProfile(this.ops["UpdateSecurityProfile"].apply(partialParams));
    }
    updateUserHierarchy(partialParams) {
        return this.client.updateUserHierarchy(this.ops["UpdateUserHierarchy"].apply(partialParams));
    }
    updateUserHierarchyGroupName(partialParams) {
        return this.client.updateUserHierarchyGroupName(this.ops["UpdateUserHierarchyGroupName"].apply(partialParams));
    }
    updateUserHierarchyStructure(partialParams) {
        return this.client.updateUserHierarchyStructure(this.ops["UpdateUserHierarchyStructure"].apply(partialParams));
    }
    updateUserIdentityInfo(partialParams) {
        return this.client.updateUserIdentityInfo(this.ops["UpdateUserIdentityInfo"].apply(partialParams));
    }
    updateUserPhoneConfig(partialParams) {
        return this.client.updateUserPhoneConfig(this.ops["UpdateUserPhoneConfig"].apply(partialParams));
    }
    updateUserRoutingProfile(partialParams) {
        return this.client.updateUserRoutingProfile(this.ops["UpdateUserRoutingProfile"].apply(partialParams));
    }
    updateUserSecurityProfiles(partialParams) {
        return this.client.updateUserSecurityProfiles(this.ops["UpdateUserSecurityProfiles"].apply(partialParams));
    }
}
exports.default = Resource;
