// out/macie/MemberAccountAssociation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.macie.MemberAccountAssociation {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Macie();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/macie-2017-12-19.normal.json"), this.client);
    }
    associateMemberAccount(partialParams) {
        return this.client.associateMemberAccount(this.ops["AssociateMemberAccount"].apply(partialParams));
    }
    associateS3Resources(partialParams) {
        return this.client.associateS3Resources(this.ops["AssociateS3Resources"].apply(partialParams));
    }
    disassociateMemberAccount(partialParams) {
        return this.client.disassociateMemberAccount(this.ops["DisassociateMemberAccount"].apply(partialParams));
    }
    disassociateS3Resources(partialParams) {
        return this.client.disassociateS3Resources(this.ops["DisassociateS3Resources"].apply(partialParams));
    }
    updateS3Resources(partialParams) {
        return this.client.updateS3Resources(this.ops["UpdateS3Resources"].apply(partialParams));
    }
}
exports.default = Resource;
