// out/fms/Policy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.fms.Policy {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.FMS();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/fms-2018-01-01.normal.json"), this.client);
    }
    associateAdminAccount(partialParams) {
        return this.client.associateAdminAccount(this.ops["AssociateAdminAccount"].apply(partialParams));
    }
    deleteAppsList(partialParams) {
        return this.client.deleteAppsList(this.ops["DeleteAppsList"].apply(partialParams));
    }
    deletePolicy(partialParams) {
        return this.client.deletePolicy(this.ops["DeletePolicy"].apply(partialParams));
    }
    deleteProtocolsList(partialParams) {
        return this.client.deleteProtocolsList(this.ops["DeleteProtocolsList"].apply(partialParams));
    }
    getAppsList(partialParams) {
        return this.client.getAppsList(this.ops["GetAppsList"].apply(partialParams));
    }
    getComplianceDetail(partialParams) {
        return this.client.getComplianceDetail(this.ops["GetComplianceDetail"].apply(partialParams));
    }
    getPolicy(partialParams) {
        return this.client.getPolicy(this.ops["GetPolicy"].apply(partialParams));
    }
    getProtectionStatus(partialParams) {
        return this.client.getProtectionStatus(this.ops["GetProtectionStatus"].apply(partialParams));
    }
    getProtocolsList(partialParams) {
        return this.client.getProtocolsList(this.ops["GetProtocolsList"].apply(partialParams));
    }
    getViolationDetails(partialParams) {
        return this.client.getViolationDetails(this.ops["GetViolationDetails"].apply(partialParams));
    }
    listAppsLists(partialParams) {
        return this.client.listAppsLists(this.ops["ListAppsLists"].apply(partialParams));
    }
    listComplianceStatus(partialParams) {
        return this.client.listComplianceStatus(this.ops["ListComplianceStatus"].apply(partialParams));
    }
    listProtocolsLists(partialParams) {
        return this.client.listProtocolsLists(this.ops["ListProtocolsLists"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putAppsList(partialParams) {
        return this.client.putAppsList(this.ops["PutAppsList"].apply(partialParams));
    }
    putNotificationChannel(partialParams) {
        return this.client.putNotificationChannel(this.ops["PutNotificationChannel"].apply(partialParams));
    }
    putPolicy(partialParams) {
        return this.client.putPolicy(this.ops["PutPolicy"].apply(partialParams));
    }
    putProtocolsList(partialParams) {
        return this.client.putProtocolsList(this.ops["PutProtocolsList"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
}
exports.default = Resource;
