// out/workspaces/Workspace.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.workspaces.Workspace {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.WorkSpaces();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/workspaces-2015-04-08.normal.json"), this.client);
    }
    associateConnectionAlias(partialParams) {
        return this.client.associateConnectionAlias(this.ops["AssociateConnectionAlias"].apply(partialParams));
    }
    associateIpGroups(partialParams) {
        return this.client.associateIpGroups(this.ops["AssociateIpGroups"].apply(partialParams));
    }
    authorizeIpRules(partialParams) {
        return this.client.authorizeIpRules(this.ops["AuthorizeIpRules"].apply(partialParams));
    }
    copyWorkspaceImage(partialParams) {
        return this.client.copyWorkspaceImage(this.ops["CopyWorkspaceImage"].apply(partialParams));
    }
    createConnectClientAddIn(partialParams) {
        return this.client.createConnectClientAddIn(this.ops["CreateConnectClientAddIn"].apply(partialParams));
    }
    createConnectionAlias(partialParams) {
        return this.client.createConnectionAlias(this.ops["CreateConnectionAlias"].apply(partialParams));
    }
    createIpGroup(partialParams) {
        return this.client.createIpGroup(this.ops["CreateIpGroup"].apply(partialParams));
    }
    createTags(partialParams) {
        return this.client.createTags(this.ops["CreateTags"].apply(partialParams));
    }
    createUpdatedWorkspaceImage(partialParams) {
        return this.client.createUpdatedWorkspaceImage(this.ops["CreateUpdatedWorkspaceImage"].apply(partialParams));
    }
    createWorkspaceBundle(partialParams) {
        return this.client.createWorkspaceBundle(this.ops["CreateWorkspaceBundle"].apply(partialParams));
    }
    createWorkspaces(partialParams) {
        return this.client.createWorkspaces(this.ops["CreateWorkspaces"].apply(partialParams));
    }
    deleteConnectClientAddIn(partialParams) {
        return this.client.deleteConnectClientAddIn(this.ops["DeleteConnectClientAddIn"].apply(partialParams));
    }
    deleteConnectionAlias(partialParams) {
        return this.client.deleteConnectionAlias(this.ops["DeleteConnectionAlias"].apply(partialParams));
    }
    deleteIpGroup(partialParams) {
        return this.client.deleteIpGroup(this.ops["DeleteIpGroup"].apply(partialParams));
    }
    deleteTags(partialParams) {
        return this.client.deleteTags(this.ops["DeleteTags"].apply(partialParams));
    }
    deleteWorkspaceImage(partialParams) {
        return this.client.deleteWorkspaceImage(this.ops["DeleteWorkspaceImage"].apply(partialParams));
    }
    deregisterWorkspaceDirectory(partialParams) {
        return this.client.deregisterWorkspaceDirectory(this.ops["DeregisterWorkspaceDirectory"].apply(partialParams));
    }
    describeClientProperties(partialParams) {
        return this.client.describeClientProperties(this.ops["DescribeClientProperties"].apply(partialParams));
    }
    describeConnectClientAddIns(partialParams) {
        return this.client.describeConnectClientAddIns(this.ops["DescribeConnectClientAddIns"].apply(partialParams));
    }
    describeConnectionAliasPermissions(partialParams) {
        return this.client.describeConnectionAliasPermissions(this.ops["DescribeConnectionAliasPermissions"].apply(partialParams));
    }
    describeTags(partialParams) {
        return this.client.describeTags(this.ops["DescribeTags"].apply(partialParams));
    }
    describeWorkspaceImagePermissions(partialParams) {
        return this.client.describeWorkspaceImagePermissions(this.ops["DescribeWorkspaceImagePermissions"].apply(partialParams));
    }
    describeWorkspaceSnapshots(partialParams) {
        return this.client.describeWorkspaceSnapshots(this.ops["DescribeWorkspaceSnapshots"].apply(partialParams));
    }
    disassociateConnectionAlias(partialParams) {
        return this.client.disassociateConnectionAlias(this.ops["DisassociateConnectionAlias"].apply(partialParams));
    }
    disassociateIpGroups(partialParams) {
        return this.client.disassociateIpGroups(this.ops["DisassociateIpGroups"].apply(partialParams));
    }
    importWorkspaceImage(partialParams) {
        return this.client.importWorkspaceImage(this.ops["ImportWorkspaceImage"].apply(partialParams));
    }
    listAvailableManagementCidrRanges(partialParams) {
        return this.client.listAvailableManagementCidrRanges(this.ops["ListAvailableManagementCidrRanges"].apply(partialParams));
    }
    migrateWorkspace(partialParams) {
        return this.client.migrateWorkspace(this.ops["MigrateWorkspace"].apply(partialParams));
    }
    modifyClientProperties(partialParams) {
        return this.client.modifyClientProperties(this.ops["ModifyClientProperties"].apply(partialParams));
    }
    modifySelfservicePermissions(partialParams) {
        return this.client.modifySelfservicePermissions(this.ops["ModifySelfservicePermissions"].apply(partialParams));
    }
    modifyWorkspaceAccessProperties(partialParams) {
        return this.client.modifyWorkspaceAccessProperties(this.ops["ModifyWorkspaceAccessProperties"].apply(partialParams));
    }
    modifyWorkspaceCreationProperties(partialParams) {
        return this.client.modifyWorkspaceCreationProperties(this.ops["ModifyWorkspaceCreationProperties"].apply(partialParams));
    }
    modifyWorkspaceProperties(partialParams) {
        return this.client.modifyWorkspaceProperties(this.ops["ModifyWorkspaceProperties"].apply(partialParams));
    }
    modifyWorkspaceState(partialParams) {
        return this.client.modifyWorkspaceState(this.ops["ModifyWorkspaceState"].apply(partialParams));
    }
    rebootWorkspaces(partialParams) {
        return this.client.rebootWorkspaces(this.ops["RebootWorkspaces"].apply(partialParams));
    }
    rebuildWorkspaces(partialParams) {
        return this.client.rebuildWorkspaces(this.ops["RebuildWorkspaces"].apply(partialParams));
    }
    registerWorkspaceDirectory(partialParams) {
        return this.client.registerWorkspaceDirectory(this.ops["RegisterWorkspaceDirectory"].apply(partialParams));
    }
    restoreWorkspace(partialParams) {
        return this.client.restoreWorkspace(this.ops["RestoreWorkspace"].apply(partialParams));
    }
    revokeIpRules(partialParams) {
        return this.client.revokeIpRules(this.ops["RevokeIpRules"].apply(partialParams));
    }
    startWorkspaces(partialParams) {
        return this.client.startWorkspaces(this.ops["StartWorkspaces"].apply(partialParams));
    }
    stopWorkspaces(partialParams) {
        return this.client.stopWorkspaces(this.ops["StopWorkspaces"].apply(partialParams));
    }
    terminateWorkspaces(partialParams) {
        return this.client.terminateWorkspaces(this.ops["TerminateWorkspaces"].apply(partialParams));
    }
    updateConnectClientAddIn(partialParams) {
        return this.client.updateConnectClientAddIn(this.ops["UpdateConnectClientAddIn"].apply(partialParams));
    }
    updateConnectionAliasPermission(partialParams) {
        return this.client.updateConnectionAliasPermission(this.ops["UpdateConnectionAliasPermission"].apply(partialParams));
    }
    updateRulesOfIpGroup(partialParams) {
        return this.client.updateRulesOfIpGroup(this.ops["UpdateRulesOfIpGroup"].apply(partialParams));
    }
    updateWorkspaceImagePermission(partialParams) {
        return this.client.updateWorkspaceImagePermission(this.ops["UpdateWorkspaceImagePermission"].apply(partialParams));
    }
}
exports.default = Resource;
