// out/athena/Workgroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.athena.Workgroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Athena();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/athena-2017-05-18.normal.json"), this.client);
    }
    batchGetNamedQuery(partialParams) {
        return this.client.batchGetNamedQuery(this.ops["BatchGetNamedQuery"].apply(partialParams));
    }
    batchGetQueryExecution(partialParams) {
        return this.client.batchGetQueryExecution(this.ops["BatchGetQueryExecution"].apply(partialParams));
    }
    createDataCatalog(partialParams) {
        return this.client.createDataCatalog(this.ops["CreateDataCatalog"].apply(partialParams));
    }
    createNamedQuery(partialParams) {
        return this.client.createNamedQuery(this.ops["CreateNamedQuery"].apply(partialParams));
    }
    createPreparedStatement(partialParams) {
        return this.client.createPreparedStatement(this.ops["CreatePreparedStatement"].apply(partialParams));
    }
    createWorkGroup(partialParams) {
        return this.client.createWorkGroup(this.ops["CreateWorkGroup"].apply(partialParams));
    }
    deleteDataCatalog(partialParams) {
        return this.client.deleteDataCatalog(this.ops["DeleteDataCatalog"].apply(partialParams));
    }
    deleteNamedQuery(partialParams) {
        return this.client.deleteNamedQuery(this.ops["DeleteNamedQuery"].apply(partialParams));
    }
    deletePreparedStatement(partialParams) {
        return this.client.deletePreparedStatement(this.ops["DeletePreparedStatement"].apply(partialParams));
    }
    deleteWorkGroup(partialParams) {
        return this.client.deleteWorkGroup(this.ops["DeleteWorkGroup"].apply(partialParams));
    }
    getDataCatalog(partialParams) {
        return this.client.getDataCatalog(this.ops["GetDataCatalog"].apply(partialParams));
    }
    getDatabase(partialParams) {
        return this.client.getDatabase(this.ops["GetDatabase"].apply(partialParams));
    }
    getNamedQuery(partialParams) {
        return this.client.getNamedQuery(this.ops["GetNamedQuery"].apply(partialParams));
    }
    getPreparedStatement(partialParams) {
        return this.client.getPreparedStatement(this.ops["GetPreparedStatement"].apply(partialParams));
    }
    getQueryExecution(partialParams) {
        return this.client.getQueryExecution(this.ops["GetQueryExecution"].apply(partialParams));
    }
    getQueryResults(partialParams) {
        return this.client.getQueryResults(this.ops["GetQueryResults"].apply(partialParams));
    }
    getTableMetadata(partialParams) {
        return this.client.getTableMetadata(this.ops["GetTableMetadata"].apply(partialParams));
    }
    getWorkGroup(partialParams) {
        return this.client.getWorkGroup(this.ops["GetWorkGroup"].apply(partialParams));
    }
    listDatabases(partialParams) {
        return this.client.listDatabases(this.ops["ListDatabases"].apply(partialParams));
    }
    listPreparedStatements(partialParams) {
        return this.client.listPreparedStatements(this.ops["ListPreparedStatements"].apply(partialParams));
    }
    listTableMetadata(partialParams) {
        return this.client.listTableMetadata(this.ops["ListTableMetadata"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    startQueryExecution(partialParams) {
        return this.client.startQueryExecution(this.ops["StartQueryExecution"].apply(partialParams));
    }
    stopQueryExecution(partialParams) {
        return this.client.stopQueryExecution(this.ops["StopQueryExecution"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateDataCatalog(partialParams) {
        return this.client.updateDataCatalog(this.ops["UpdateDataCatalog"].apply(partialParams));
    }
    updatePreparedStatement(partialParams) {
        return this.client.updatePreparedStatement(this.ops["UpdatePreparedStatement"].apply(partialParams));
    }
    updateWorkGroup(partialParams) {
        return this.client.updateWorkGroup(this.ops["UpdateWorkGroup"].apply(partialParams));
    }
}
exports.default = Resource;
