// out/cloudfront/Function.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.cloudfront.Function {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.CloudFront();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/cloudfront-2020-05-31.normal.json"), this.client);
    }
    associateAlias(partialParams) {
        return this.client.associateAlias(this.ops["AssociateAlias"].apply(partialParams));
    }
    createCachePolicy(partialParams) {
        return this.client.createCachePolicy(this.ops["CreateCachePolicy"].apply(partialParams));
    }
    createCloudFrontOriginAccessIdentity(partialParams) {
        return this.client.createCloudFrontOriginAccessIdentity(this.ops["CreateCloudFrontOriginAccessIdentity"].apply(partialParams));
    }
    createDistribution(partialParams) {
        return this.client.createDistribution(this.ops["CreateDistribution"].apply(partialParams));
    }
    createDistributionWithTags(partialParams) {
        return this.client.createDistributionWithTags(this.ops["CreateDistributionWithTags"].apply(partialParams));
    }
    createFieldLevelEncryptionConfig(partialParams) {
        return this.client.createFieldLevelEncryptionConfig(this.ops["CreateFieldLevelEncryptionConfig"].apply(partialParams));
    }
    createFieldLevelEncryptionProfile(partialParams) {
        return this.client.createFieldLevelEncryptionProfile(this.ops["CreateFieldLevelEncryptionProfile"].apply(partialParams));
    }
    createFunction(partialParams) {
        return this.client.createFunction(this.ops["CreateFunction"].apply(partialParams));
    }
    createInvalidation(partialParams) {
        return this.client.createInvalidation(this.ops["CreateInvalidation"].apply(partialParams));
    }
    createKeyGroup(partialParams) {
        return this.client.createKeyGroup(this.ops["CreateKeyGroup"].apply(partialParams));
    }
    createMonitoringSubscription(partialParams) {
        return this.client.createMonitoringSubscription(this.ops["CreateMonitoringSubscription"].apply(partialParams));
    }
    createOriginRequestPolicy(partialParams) {
        return this.client.createOriginRequestPolicy(this.ops["CreateOriginRequestPolicy"].apply(partialParams));
    }
    createPublicKey(partialParams) {
        return this.client.createPublicKey(this.ops["CreatePublicKey"].apply(partialParams));
    }
    createRealtimeLogConfig(partialParams) {
        return this.client.createRealtimeLogConfig(this.ops["CreateRealtimeLogConfig"].apply(partialParams));
    }
    createResponseHeadersPolicy(partialParams) {
        return this.client.createResponseHeadersPolicy(this.ops["CreateResponseHeadersPolicy"].apply(partialParams));
    }
    createStreamingDistribution(partialParams) {
        return this.client.createStreamingDistribution(this.ops["CreateStreamingDistribution"].apply(partialParams));
    }
    createStreamingDistributionWithTags(partialParams) {
        return this.client.createStreamingDistributionWithTags(this.ops["CreateStreamingDistributionWithTags"].apply(partialParams));
    }
    deleteCachePolicy(partialParams) {
        return this.client.deleteCachePolicy(this.ops["DeleteCachePolicy"].apply(partialParams));
    }
    deleteCloudFrontOriginAccessIdentity(partialParams) {
        return this.client.deleteCloudFrontOriginAccessIdentity(this.ops["DeleteCloudFrontOriginAccessIdentity"].apply(partialParams));
    }
    deleteDistribution(partialParams) {
        return this.client.deleteDistribution(this.ops["DeleteDistribution"].apply(partialParams));
    }
    deleteFieldLevelEncryptionConfig(partialParams) {
        return this.client.deleteFieldLevelEncryptionConfig(this.ops["DeleteFieldLevelEncryptionConfig"].apply(partialParams));
    }
    deleteFieldLevelEncryptionProfile(partialParams) {
        return this.client.deleteFieldLevelEncryptionProfile(this.ops["DeleteFieldLevelEncryptionProfile"].apply(partialParams));
    }
    deleteFunction(partialParams) {
        return this.client.deleteFunction(this.ops["DeleteFunction"].apply(partialParams));
    }
    deleteKeyGroup(partialParams) {
        return this.client.deleteKeyGroup(this.ops["DeleteKeyGroup"].apply(partialParams));
    }
    deleteMonitoringSubscription(partialParams) {
        return this.client.deleteMonitoringSubscription(this.ops["DeleteMonitoringSubscription"].apply(partialParams));
    }
    deleteOriginRequestPolicy(partialParams) {
        return this.client.deleteOriginRequestPolicy(this.ops["DeleteOriginRequestPolicy"].apply(partialParams));
    }
    deletePublicKey(partialParams) {
        return this.client.deletePublicKey(this.ops["DeletePublicKey"].apply(partialParams));
    }
    deleteResponseHeadersPolicy(partialParams) {
        return this.client.deleteResponseHeadersPolicy(this.ops["DeleteResponseHeadersPolicy"].apply(partialParams));
    }
    deleteStreamingDistribution(partialParams) {
        return this.client.deleteStreamingDistribution(this.ops["DeleteStreamingDistribution"].apply(partialParams));
    }
    describeFunction(partialParams) {
        return this.client.describeFunction(this.ops["DescribeFunction"].apply(partialParams));
    }
    getCachePolicy(partialParams) {
        return this.client.getCachePolicy(this.ops["GetCachePolicy"].apply(partialParams));
    }
    getCachePolicyConfig(partialParams) {
        return this.client.getCachePolicyConfig(this.ops["GetCachePolicyConfig"].apply(partialParams));
    }
    getCloudFrontOriginAccessIdentity(partialParams) {
        return this.client.getCloudFrontOriginAccessIdentity(this.ops["GetCloudFrontOriginAccessIdentity"].apply(partialParams));
    }
    getCloudFrontOriginAccessIdentityConfig(partialParams) {
        return this.client.getCloudFrontOriginAccessIdentityConfig(this.ops["GetCloudFrontOriginAccessIdentityConfig"].apply(partialParams));
    }
    getDistribution(partialParams) {
        return this.client.getDistribution(this.ops["GetDistribution"].apply(partialParams));
    }
    getDistributionConfig(partialParams) {
        return this.client.getDistributionConfig(this.ops["GetDistributionConfig"].apply(partialParams));
    }
    getFieldLevelEncryption(partialParams) {
        return this.client.getFieldLevelEncryption(this.ops["GetFieldLevelEncryption"].apply(partialParams));
    }
    getFieldLevelEncryptionConfig(partialParams) {
        return this.client.getFieldLevelEncryptionConfig(this.ops["GetFieldLevelEncryptionConfig"].apply(partialParams));
    }
    getFieldLevelEncryptionProfile(partialParams) {
        return this.client.getFieldLevelEncryptionProfile(this.ops["GetFieldLevelEncryptionProfile"].apply(partialParams));
    }
    getFieldLevelEncryptionProfileConfig(partialParams) {
        return this.client.getFieldLevelEncryptionProfileConfig(this.ops["GetFieldLevelEncryptionProfileConfig"].apply(partialParams));
    }
    getFunction(partialParams) {
        return this.client.getFunction(this.ops["GetFunction"].apply(partialParams));
    }
    getInvalidation(partialParams) {
        return this.client.getInvalidation(this.ops["GetInvalidation"].apply(partialParams));
    }
    getKeyGroup(partialParams) {
        return this.client.getKeyGroup(this.ops["GetKeyGroup"].apply(partialParams));
    }
    getKeyGroupConfig(partialParams) {
        return this.client.getKeyGroupConfig(this.ops["GetKeyGroupConfig"].apply(partialParams));
    }
    getMonitoringSubscription(partialParams) {
        return this.client.getMonitoringSubscription(this.ops["GetMonitoringSubscription"].apply(partialParams));
    }
    getOriginRequestPolicy(partialParams) {
        return this.client.getOriginRequestPolicy(this.ops["GetOriginRequestPolicy"].apply(partialParams));
    }
    getOriginRequestPolicyConfig(partialParams) {
        return this.client.getOriginRequestPolicyConfig(this.ops["GetOriginRequestPolicyConfig"].apply(partialParams));
    }
    getPublicKey(partialParams) {
        return this.client.getPublicKey(this.ops["GetPublicKey"].apply(partialParams));
    }
    getPublicKeyConfig(partialParams) {
        return this.client.getPublicKeyConfig(this.ops["GetPublicKeyConfig"].apply(partialParams));
    }
    getResponseHeadersPolicy(partialParams) {
        return this.client.getResponseHeadersPolicy(this.ops["GetResponseHeadersPolicy"].apply(partialParams));
    }
    getResponseHeadersPolicyConfig(partialParams) {
        return this.client.getResponseHeadersPolicyConfig(this.ops["GetResponseHeadersPolicyConfig"].apply(partialParams));
    }
    getStreamingDistribution(partialParams) {
        return this.client.getStreamingDistribution(this.ops["GetStreamingDistribution"].apply(partialParams));
    }
    getStreamingDistributionConfig(partialParams) {
        return this.client.getStreamingDistributionConfig(this.ops["GetStreamingDistributionConfig"].apply(partialParams));
    }
    listConflictingAliases(partialParams) {
        return this.client.listConflictingAliases(this.ops["ListConflictingAliases"].apply(partialParams));
    }
    listDistributionsByCachePolicyId(partialParams) {
        return this.client.listDistributionsByCachePolicyId(this.ops["ListDistributionsByCachePolicyId"].apply(partialParams));
    }
    listDistributionsByKeyGroup(partialParams) {
        return this.client.listDistributionsByKeyGroup(this.ops["ListDistributionsByKeyGroup"].apply(partialParams));
    }
    listDistributionsByOriginRequestPolicyId(partialParams) {
        return this.client.listDistributionsByOriginRequestPolicyId(this.ops["ListDistributionsByOriginRequestPolicyId"].apply(partialParams));
    }
    listDistributionsByResponseHeadersPolicyId(partialParams) {
        return this.client.listDistributionsByResponseHeadersPolicyId(this.ops["ListDistributionsByResponseHeadersPolicyId"].apply(partialParams));
    }
    listDistributionsByWebACLId(partialParams) {
        return this.client.listDistributionsByWebACLId(this.ops["ListDistributionsByWebACLId"].apply(partialParams));
    }
    listInvalidations(partialParams) {
        return this.client.listInvalidations(this.ops["ListInvalidations"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    publishFunction(partialParams) {
        return this.client.publishFunction(this.ops["PublishFunction"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    testFunction(partialParams) {
        return this.client.testFunction(this.ops["TestFunction"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateCachePolicy(partialParams) {
        return this.client.updateCachePolicy(this.ops["UpdateCachePolicy"].apply(partialParams));
    }
    updateCloudFrontOriginAccessIdentity(partialParams) {
        return this.client.updateCloudFrontOriginAccessIdentity(this.ops["UpdateCloudFrontOriginAccessIdentity"].apply(partialParams));
    }
    updateDistribution(partialParams) {
        return this.client.updateDistribution(this.ops["UpdateDistribution"].apply(partialParams));
    }
    updateFieldLevelEncryptionConfig(partialParams) {
        return this.client.updateFieldLevelEncryptionConfig(this.ops["UpdateFieldLevelEncryptionConfig"].apply(partialParams));
    }
    updateFieldLevelEncryptionProfile(partialParams) {
        return this.client.updateFieldLevelEncryptionProfile(this.ops["UpdateFieldLevelEncryptionProfile"].apply(partialParams));
    }
    updateFunction(partialParams) {
        return this.client.updateFunction(this.ops["UpdateFunction"].apply(partialParams));
    }
    updateKeyGroup(partialParams) {
        return this.client.updateKeyGroup(this.ops["UpdateKeyGroup"].apply(partialParams));
    }
    updateOriginRequestPolicy(partialParams) {
        return this.client.updateOriginRequestPolicy(this.ops["UpdateOriginRequestPolicy"].apply(partialParams));
    }
    updatePublicKey(partialParams) {
        return this.client.updatePublicKey(this.ops["UpdatePublicKey"].apply(partialParams));
    }
    updateResponseHeadersPolicy(partialParams) {
        return this.client.updateResponseHeadersPolicy(this.ops["UpdateResponseHeadersPolicy"].apply(partialParams));
    }
    updateStreamingDistribution(partialParams) {
        return this.client.updateStreamingDistribution(this.ops["UpdateStreamingDistribution"].apply(partialParams));
    }
}
exports.default = Resource;
