// out/iam/UserPolicy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.iam.UserPolicy {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.IAM();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/iam-2010-05-08.normal.json"), this.client);
    }
    addClientIDToOpenIDConnectProvider(partialParams) {
        return this.client.addClientIDToOpenIDConnectProvider(this.ops["AddClientIDToOpenIDConnectProvider"].apply(partialParams));
    }
    addRoleToInstanceProfile(partialParams) {
        return this.client.addRoleToInstanceProfile(this.ops["AddRoleToInstanceProfile"].apply(partialParams));
    }
    addUserToGroup(partialParams) {
        return this.client.addUserToGroup(this.ops["AddUserToGroup"].apply(partialParams));
    }
    attachGroupPolicy(partialParams) {
        return this.client.attachGroupPolicy(this.ops["AttachGroupPolicy"].apply(partialParams));
    }
    attachRolePolicy(partialParams) {
        return this.client.attachRolePolicy(this.ops["AttachRolePolicy"].apply(partialParams));
    }
    attachUserPolicy(partialParams) {
        return this.client.attachUserPolicy(this.ops["AttachUserPolicy"].apply(partialParams));
    }
    changePassword(partialParams) {
        return this.client.changePassword(this.ops["ChangePassword"].apply(partialParams));
    }
    createAccountAlias(partialParams) {
        return this.client.createAccountAlias(this.ops["CreateAccountAlias"].apply(partialParams));
    }
    createGroup(partialParams) {
        return this.client.createGroup(this.ops["CreateGroup"].apply(partialParams));
    }
    createInstanceProfile(partialParams) {
        return this.client.createInstanceProfile(this.ops["CreateInstanceProfile"].apply(partialParams));
    }
    createLoginProfile(partialParams) {
        return this.client.createLoginProfile(this.ops["CreateLoginProfile"].apply(partialParams));
    }
    createOpenIDConnectProvider(partialParams) {
        return this.client.createOpenIDConnectProvider(this.ops["CreateOpenIDConnectProvider"].apply(partialParams));
    }
    createPolicy(partialParams) {
        return this.client.createPolicy(this.ops["CreatePolicy"].apply(partialParams));
    }
    createPolicyVersion(partialParams) {
        return this.client.createPolicyVersion(this.ops["CreatePolicyVersion"].apply(partialParams));
    }
    createRole(partialParams) {
        return this.client.createRole(this.ops["CreateRole"].apply(partialParams));
    }
    createSAMLProvider(partialParams) {
        return this.client.createSAMLProvider(this.ops["CreateSAMLProvider"].apply(partialParams));
    }
    createServiceLinkedRole(partialParams) {
        return this.client.createServiceLinkedRole(this.ops["CreateServiceLinkedRole"].apply(partialParams));
    }
    createServiceSpecificCredential(partialParams) {
        return this.client.createServiceSpecificCredential(this.ops["CreateServiceSpecificCredential"].apply(partialParams));
    }
    createUser(partialParams) {
        return this.client.createUser(this.ops["CreateUser"].apply(partialParams));
    }
    createVirtualMFADevice(partialParams) {
        return this.client.createVirtualMFADevice(this.ops["CreateVirtualMFADevice"].apply(partialParams));
    }
    deactivateMFADevice(partialParams) {
        return this.client.deactivateMFADevice(this.ops["DeactivateMFADevice"].apply(partialParams));
    }
    deleteAccessKey(partialParams) {
        return this.client.deleteAccessKey(this.ops["DeleteAccessKey"].apply(partialParams));
    }
    deleteAccountAlias(partialParams) {
        return this.client.deleteAccountAlias(this.ops["DeleteAccountAlias"].apply(partialParams));
    }
    deleteGroup(partialParams) {
        return this.client.deleteGroup(this.ops["DeleteGroup"].apply(partialParams));
    }
    deleteGroupPolicy(partialParams) {
        return this.client.deleteGroupPolicy(this.ops["DeleteGroupPolicy"].apply(partialParams));
    }
    deleteInstanceProfile(partialParams) {
        return this.client.deleteInstanceProfile(this.ops["DeleteInstanceProfile"].apply(partialParams));
    }
    deleteLoginProfile(partialParams) {
        return this.client.deleteLoginProfile(this.ops["DeleteLoginProfile"].apply(partialParams));
    }
    deleteOpenIDConnectProvider(partialParams) {
        return this.client.deleteOpenIDConnectProvider(this.ops["DeleteOpenIDConnectProvider"].apply(partialParams));
    }
    deletePolicy(partialParams) {
        return this.client.deletePolicy(this.ops["DeletePolicy"].apply(partialParams));
    }
    deletePolicyVersion(partialParams) {
        return this.client.deletePolicyVersion(this.ops["DeletePolicyVersion"].apply(partialParams));
    }
    deleteRole(partialParams) {
        return this.client.deleteRole(this.ops["DeleteRole"].apply(partialParams));
    }
    deleteRolePermissionsBoundary(partialParams) {
        return this.client.deleteRolePermissionsBoundary(this.ops["DeleteRolePermissionsBoundary"].apply(partialParams));
    }
    deleteRolePolicy(partialParams) {
        return this.client.deleteRolePolicy(this.ops["DeleteRolePolicy"].apply(partialParams));
    }
    deleteSAMLProvider(partialParams) {
        return this.client.deleteSAMLProvider(this.ops["DeleteSAMLProvider"].apply(partialParams));
    }
    deleteSSHPublicKey(partialParams) {
        return this.client.deleteSSHPublicKey(this.ops["DeleteSSHPublicKey"].apply(partialParams));
    }
    deleteServerCertificate(partialParams) {
        return this.client.deleteServerCertificate(this.ops["DeleteServerCertificate"].apply(partialParams));
    }
    deleteServiceLinkedRole(partialParams) {
        return this.client.deleteServiceLinkedRole(this.ops["DeleteServiceLinkedRole"].apply(partialParams));
    }
    deleteServiceSpecificCredential(partialParams) {
        return this.client.deleteServiceSpecificCredential(this.ops["DeleteServiceSpecificCredential"].apply(partialParams));
    }
    deleteSigningCertificate(partialParams) {
        return this.client.deleteSigningCertificate(this.ops["DeleteSigningCertificate"].apply(partialParams));
    }
    deleteUser(partialParams) {
        return this.client.deleteUser(this.ops["DeleteUser"].apply(partialParams));
    }
    deleteUserPermissionsBoundary(partialParams) {
        return this.client.deleteUserPermissionsBoundary(this.ops["DeleteUserPermissionsBoundary"].apply(partialParams));
    }
    deleteUserPolicy(partialParams) {
        return this.client.deleteUserPolicy(this.ops["DeleteUserPolicy"].apply(partialParams));
    }
    deleteVirtualMFADevice(partialParams) {
        return this.client.deleteVirtualMFADevice(this.ops["DeleteVirtualMFADevice"].apply(partialParams));
    }
    detachGroupPolicy(partialParams) {
        return this.client.detachGroupPolicy(this.ops["DetachGroupPolicy"].apply(partialParams));
    }
    detachRolePolicy(partialParams) {
        return this.client.detachRolePolicy(this.ops["DetachRolePolicy"].apply(partialParams));
    }
    detachUserPolicy(partialParams) {
        return this.client.detachUserPolicy(this.ops["DetachUserPolicy"].apply(partialParams));
    }
    enableMFADevice(partialParams) {
        return this.client.enableMFADevice(this.ops["EnableMFADevice"].apply(partialParams));
    }
    generateOrganizationsAccessReport(partialParams) {
        return this.client.generateOrganizationsAccessReport(this.ops["GenerateOrganizationsAccessReport"].apply(partialParams));
    }
    generateServiceLastAccessedDetails(partialParams) {
        return this.client.generateServiceLastAccessedDetails(this.ops["GenerateServiceLastAccessedDetails"].apply(partialParams));
    }
    getAccessKeyLastUsed(partialParams) {
        return this.client.getAccessKeyLastUsed(this.ops["GetAccessKeyLastUsed"].apply(partialParams));
    }
    getContextKeysForCustomPolicy(partialParams) {
        return this.client.getContextKeysForCustomPolicy(this.ops["GetContextKeysForCustomPolicy"].apply(partialParams));
    }
    getContextKeysForPrincipalPolicy(partialParams) {
        return this.client.getContextKeysForPrincipalPolicy(this.ops["GetContextKeysForPrincipalPolicy"].apply(partialParams));
    }
    getGroup(partialParams) {
        return this.client.getGroup(this.ops["GetGroup"].apply(partialParams));
    }
    getGroupPolicy(partialParams) {
        return this.client.getGroupPolicy(this.ops["GetGroupPolicy"].apply(partialParams));
    }
    getInstanceProfile(partialParams) {
        return this.client.getInstanceProfile(this.ops["GetInstanceProfile"].apply(partialParams));
    }
    getLoginProfile(partialParams) {
        return this.client.getLoginProfile(this.ops["GetLoginProfile"].apply(partialParams));
    }
    getOpenIDConnectProvider(partialParams) {
        return this.client.getOpenIDConnectProvider(this.ops["GetOpenIDConnectProvider"].apply(partialParams));
    }
    getOrganizationsAccessReport(partialParams) {
        return this.client.getOrganizationsAccessReport(this.ops["GetOrganizationsAccessReport"].apply(partialParams));
    }
    getPolicy(partialParams) {
        return this.client.getPolicy(this.ops["GetPolicy"].apply(partialParams));
    }
    getPolicyVersion(partialParams) {
        return this.client.getPolicyVersion(this.ops["GetPolicyVersion"].apply(partialParams));
    }
    getRole(partialParams) {
        return this.client.getRole(this.ops["GetRole"].apply(partialParams));
    }
    getRolePolicy(partialParams) {
        return this.client.getRolePolicy(this.ops["GetRolePolicy"].apply(partialParams));
    }
    getSAMLProvider(partialParams) {
        return this.client.getSAMLProvider(this.ops["GetSAMLProvider"].apply(partialParams));
    }
    getSSHPublicKey(partialParams) {
        return this.client.getSSHPublicKey(this.ops["GetSSHPublicKey"].apply(partialParams));
    }
    getServerCertificate(partialParams) {
        return this.client.getServerCertificate(this.ops["GetServerCertificate"].apply(partialParams));
    }
    getServiceLastAccessedDetails(partialParams) {
        return this.client.getServiceLastAccessedDetails(this.ops["GetServiceLastAccessedDetails"].apply(partialParams));
    }
    getServiceLastAccessedDetailsWithEntities(partialParams) {
        return this.client.getServiceLastAccessedDetailsWithEntities(this.ops["GetServiceLastAccessedDetailsWithEntities"].apply(partialParams));
    }
    getServiceLinkedRoleDeletionStatus(partialParams) {
        return this.client.getServiceLinkedRoleDeletionStatus(this.ops["GetServiceLinkedRoleDeletionStatus"].apply(partialParams));
    }
    getUserPolicy(partialParams) {
        return this.client.getUserPolicy(this.ops["GetUserPolicy"].apply(partialParams));
    }
    listAttachedGroupPolicies(partialParams) {
        return this.client.listAttachedGroupPolicies(this.ops["ListAttachedGroupPolicies"].apply(partialParams));
    }
    listAttachedRolePolicies(partialParams) {
        return this.client.listAttachedRolePolicies(this.ops["ListAttachedRolePolicies"].apply(partialParams));
    }
    listAttachedUserPolicies(partialParams) {
        return this.client.listAttachedUserPolicies(this.ops["ListAttachedUserPolicies"].apply(partialParams));
    }
    listEntitiesForPolicy(partialParams) {
        return this.client.listEntitiesForPolicy(this.ops["ListEntitiesForPolicy"].apply(partialParams));
    }
    listGroupPolicies(partialParams) {
        return this.client.listGroupPolicies(this.ops["ListGroupPolicies"].apply(partialParams));
    }
    listGroupsForUser(partialParams) {
        return this.client.listGroupsForUser(this.ops["ListGroupsForUser"].apply(partialParams));
    }
    listInstanceProfileTags(partialParams) {
        return this.client.listInstanceProfileTags(this.ops["ListInstanceProfileTags"].apply(partialParams));
    }
    listInstanceProfilesForRole(partialParams) {
        return this.client.listInstanceProfilesForRole(this.ops["ListInstanceProfilesForRole"].apply(partialParams));
    }
    listMFADeviceTags(partialParams) {
        return this.client.listMFADeviceTags(this.ops["ListMFADeviceTags"].apply(partialParams));
    }
    listOpenIDConnectProviderTags(partialParams) {
        return this.client.listOpenIDConnectProviderTags(this.ops["ListOpenIDConnectProviderTags"].apply(partialParams));
    }
    listPoliciesGrantingServiceAccess(partialParams) {
        return this.client.listPoliciesGrantingServiceAccess(this.ops["ListPoliciesGrantingServiceAccess"].apply(partialParams));
    }
    listPolicyTags(partialParams) {
        return this.client.listPolicyTags(this.ops["ListPolicyTags"].apply(partialParams));
    }
    listPolicyVersions(partialParams) {
        return this.client.listPolicyVersions(this.ops["ListPolicyVersions"].apply(partialParams));
    }
    listRolePolicies(partialParams) {
        return this.client.listRolePolicies(this.ops["ListRolePolicies"].apply(partialParams));
    }
    listRoleTags(partialParams) {
        return this.client.listRoleTags(this.ops["ListRoleTags"].apply(partialParams));
    }
    listSAMLProviderTags(partialParams) {
        return this.client.listSAMLProviderTags(this.ops["ListSAMLProviderTags"].apply(partialParams));
    }
    listServerCertificateTags(partialParams) {
        return this.client.listServerCertificateTags(this.ops["ListServerCertificateTags"].apply(partialParams));
    }
    listUserPolicies(partialParams) {
        return this.client.listUserPolicies(this.ops["ListUserPolicies"].apply(partialParams));
    }
    listUserTags(partialParams) {
        return this.client.listUserTags(this.ops["ListUserTags"].apply(partialParams));
    }
    putGroupPolicy(partialParams) {
        return this.client.putGroupPolicy(this.ops["PutGroupPolicy"].apply(partialParams));
    }
    putRolePermissionsBoundary(partialParams) {
        return this.client.putRolePermissionsBoundary(this.ops["PutRolePermissionsBoundary"].apply(partialParams));
    }
    putRolePolicy(partialParams) {
        return this.client.putRolePolicy(this.ops["PutRolePolicy"].apply(partialParams));
    }
    putUserPermissionsBoundary(partialParams) {
        return this.client.putUserPermissionsBoundary(this.ops["PutUserPermissionsBoundary"].apply(partialParams));
    }
    putUserPolicy(partialParams) {
        return this.client.putUserPolicy(this.ops["PutUserPolicy"].apply(partialParams));
    }
    removeClientIDFromOpenIDConnectProvider(partialParams) {
        return this.client.removeClientIDFromOpenIDConnectProvider(this.ops["RemoveClientIDFromOpenIDConnectProvider"].apply(partialParams));
    }
    removeRoleFromInstanceProfile(partialParams) {
        return this.client.removeRoleFromInstanceProfile(this.ops["RemoveRoleFromInstanceProfile"].apply(partialParams));
    }
    removeUserFromGroup(partialParams) {
        return this.client.removeUserFromGroup(this.ops["RemoveUserFromGroup"].apply(partialParams));
    }
    resetServiceSpecificCredential(partialParams) {
        return this.client.resetServiceSpecificCredential(this.ops["ResetServiceSpecificCredential"].apply(partialParams));
    }
    resyncMFADevice(partialParams) {
        return this.client.resyncMFADevice(this.ops["ResyncMFADevice"].apply(partialParams));
    }
    setDefaultPolicyVersion(partialParams) {
        return this.client.setDefaultPolicyVersion(this.ops["SetDefaultPolicyVersion"].apply(partialParams));
    }
    setSecurityTokenServicePreferences(partialParams) {
        return this.client.setSecurityTokenServicePreferences(this.ops["SetSecurityTokenServicePreferences"].apply(partialParams));
    }
    simulateCustomPolicy(partialParams) {
        return this.client.simulateCustomPolicy(this.ops["SimulateCustomPolicy"].apply(partialParams));
    }
    simulatePrincipalPolicy(partialParams) {
        return this.client.simulatePrincipalPolicy(this.ops["SimulatePrincipalPolicy"].apply(partialParams));
    }
    tagInstanceProfile(partialParams) {
        return this.client.tagInstanceProfile(this.ops["TagInstanceProfile"].apply(partialParams));
    }
    tagMFADevice(partialParams) {
        return this.client.tagMFADevice(this.ops["TagMFADevice"].apply(partialParams));
    }
    tagOpenIDConnectProvider(partialParams) {
        return this.client.tagOpenIDConnectProvider(this.ops["TagOpenIDConnectProvider"].apply(partialParams));
    }
    tagPolicy(partialParams) {
        return this.client.tagPolicy(this.ops["TagPolicy"].apply(partialParams));
    }
    tagRole(partialParams) {
        return this.client.tagRole(this.ops["TagRole"].apply(partialParams));
    }
    tagSAMLProvider(partialParams) {
        return this.client.tagSAMLProvider(this.ops["TagSAMLProvider"].apply(partialParams));
    }
    tagServerCertificate(partialParams) {
        return this.client.tagServerCertificate(this.ops["TagServerCertificate"].apply(partialParams));
    }
    tagUser(partialParams) {
        return this.client.tagUser(this.ops["TagUser"].apply(partialParams));
    }
    untagInstanceProfile(partialParams) {
        return this.client.untagInstanceProfile(this.ops["UntagInstanceProfile"].apply(partialParams));
    }
    untagMFADevice(partialParams) {
        return this.client.untagMFADevice(this.ops["UntagMFADevice"].apply(partialParams));
    }
    untagOpenIDConnectProvider(partialParams) {
        return this.client.untagOpenIDConnectProvider(this.ops["UntagOpenIDConnectProvider"].apply(partialParams));
    }
    untagPolicy(partialParams) {
        return this.client.untagPolicy(this.ops["UntagPolicy"].apply(partialParams));
    }
    untagRole(partialParams) {
        return this.client.untagRole(this.ops["UntagRole"].apply(partialParams));
    }
    untagSAMLProvider(partialParams) {
        return this.client.untagSAMLProvider(this.ops["UntagSAMLProvider"].apply(partialParams));
    }
    untagServerCertificate(partialParams) {
        return this.client.untagServerCertificate(this.ops["UntagServerCertificate"].apply(partialParams));
    }
    untagUser(partialParams) {
        return this.client.untagUser(this.ops["UntagUser"].apply(partialParams));
    }
    updateAccessKey(partialParams) {
        return this.client.updateAccessKey(this.ops["UpdateAccessKey"].apply(partialParams));
    }
    updateAssumeRolePolicy(partialParams) {
        return this.client.updateAssumeRolePolicy(this.ops["UpdateAssumeRolePolicy"].apply(partialParams));
    }
    updateGroup(partialParams) {
        return this.client.updateGroup(this.ops["UpdateGroup"].apply(partialParams));
    }
    updateLoginProfile(partialParams) {
        return this.client.updateLoginProfile(this.ops["UpdateLoginProfile"].apply(partialParams));
    }
    updateOpenIDConnectProviderThumbprint(partialParams) {
        return this.client.updateOpenIDConnectProviderThumbprint(this.ops["UpdateOpenIDConnectProviderThumbprint"].apply(partialParams));
    }
    updateRole(partialParams) {
        return this.client.updateRole(this.ops["UpdateRole"].apply(partialParams));
    }
    updateRoleDescription(partialParams) {
        return this.client.updateRoleDescription(this.ops["UpdateRoleDescription"].apply(partialParams));
    }
    updateSAMLProvider(partialParams) {
        return this.client.updateSAMLProvider(this.ops["UpdateSAMLProvider"].apply(partialParams));
    }
    updateSSHPublicKey(partialParams) {
        return this.client.updateSSHPublicKey(this.ops["UpdateSSHPublicKey"].apply(partialParams));
    }
    updateServerCertificate(partialParams) {
        return this.client.updateServerCertificate(this.ops["UpdateServerCertificate"].apply(partialParams));
    }
    updateServiceSpecificCredential(partialParams) {
        return this.client.updateServiceSpecificCredential(this.ops["UpdateServiceSpecificCredential"].apply(partialParams));
    }
    updateSigningCertificate(partialParams) {
        return this.client.updateSigningCertificate(this.ops["UpdateSigningCertificate"].apply(partialParams));
    }
    updateUser(partialParams) {
        return this.client.updateUser(this.ops["UpdateUser"].apply(partialParams));
    }
    uploadSSHPublicKey(partialParams) {
        return this.client.uploadSSHPublicKey(this.ops["UploadSSHPublicKey"].apply(partialParams));
    }
    uploadServerCertificate(partialParams) {
        return this.client.uploadServerCertificate(this.ops["UploadServerCertificate"].apply(partialParams));
    }
    uploadSigningCertificate(partialParams) {
        return this.client.uploadSigningCertificate(this.ops["UploadSigningCertificate"].apply(partialParams));
    }
}
exports.default = Resource;
