// out/route53/Record.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.route53.Record {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Route53();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/route53-2013-04-01.normal.json"), this.client);
    }
    activateKeySigningKey(partialParams) {
        return this.client.activateKeySigningKey(this.ops["ActivateKeySigningKey"].apply(partialParams));
    }
    associateVPCWithHostedZone(partialParams) {
        return this.client.associateVPCWithHostedZone(this.ops["AssociateVPCWithHostedZone"].apply(partialParams));
    }
    changeResourceRecordSets(partialParams) {
        return this.client.changeResourceRecordSets(this.ops["ChangeResourceRecordSets"].apply(partialParams));
    }
    changeTagsForResource(partialParams) {
        return this.client.changeTagsForResource(this.ops["ChangeTagsForResource"].apply(partialParams));
    }
    createHealthCheck(partialParams) {
        return this.client.createHealthCheck(this.ops["CreateHealthCheck"].apply(partialParams));
    }
    createHostedZone(partialParams) {
        return this.client.createHostedZone(this.ops["CreateHostedZone"].apply(partialParams));
    }
    createKeySigningKey(partialParams) {
        return this.client.createKeySigningKey(this.ops["CreateKeySigningKey"].apply(partialParams));
    }
    createQueryLoggingConfig(partialParams) {
        return this.client.createQueryLoggingConfig(this.ops["CreateQueryLoggingConfig"].apply(partialParams));
    }
    createReusableDelegationSet(partialParams) {
        return this.client.createReusableDelegationSet(this.ops["CreateReusableDelegationSet"].apply(partialParams));
    }
    createTrafficPolicy(partialParams) {
        return this.client.createTrafficPolicy(this.ops["CreateTrafficPolicy"].apply(partialParams));
    }
    createTrafficPolicyInstance(partialParams) {
        return this.client.createTrafficPolicyInstance(this.ops["CreateTrafficPolicyInstance"].apply(partialParams));
    }
    createTrafficPolicyVersion(partialParams) {
        return this.client.createTrafficPolicyVersion(this.ops["CreateTrafficPolicyVersion"].apply(partialParams));
    }
    createVPCAssociationAuthorization(partialParams) {
        return this.client.createVPCAssociationAuthorization(this.ops["CreateVPCAssociationAuthorization"].apply(partialParams));
    }
    deactivateKeySigningKey(partialParams) {
        return this.client.deactivateKeySigningKey(this.ops["DeactivateKeySigningKey"].apply(partialParams));
    }
    deleteHealthCheck(partialParams) {
        return this.client.deleteHealthCheck(this.ops["DeleteHealthCheck"].apply(partialParams));
    }
    deleteHostedZone(partialParams) {
        return this.client.deleteHostedZone(this.ops["DeleteHostedZone"].apply(partialParams));
    }
    deleteKeySigningKey(partialParams) {
        return this.client.deleteKeySigningKey(this.ops["DeleteKeySigningKey"].apply(partialParams));
    }
    deleteQueryLoggingConfig(partialParams) {
        return this.client.deleteQueryLoggingConfig(this.ops["DeleteQueryLoggingConfig"].apply(partialParams));
    }
    deleteReusableDelegationSet(partialParams) {
        return this.client.deleteReusableDelegationSet(this.ops["DeleteReusableDelegationSet"].apply(partialParams));
    }
    deleteTrafficPolicy(partialParams) {
        return this.client.deleteTrafficPolicy(this.ops["DeleteTrafficPolicy"].apply(partialParams));
    }
    deleteTrafficPolicyInstance(partialParams) {
        return this.client.deleteTrafficPolicyInstance(this.ops["DeleteTrafficPolicyInstance"].apply(partialParams));
    }
    deleteVPCAssociationAuthorization(partialParams) {
        return this.client.deleteVPCAssociationAuthorization(this.ops["DeleteVPCAssociationAuthorization"].apply(partialParams));
    }
    disableHostedZoneDNSSEC(partialParams) {
        return this.client.disableHostedZoneDNSSEC(this.ops["DisableHostedZoneDNSSEC"].apply(partialParams));
    }
    disassociateVPCFromHostedZone(partialParams) {
        return this.client.disassociateVPCFromHostedZone(this.ops["DisassociateVPCFromHostedZone"].apply(partialParams));
    }
    enableHostedZoneDNSSEC(partialParams) {
        return this.client.enableHostedZoneDNSSEC(this.ops["EnableHostedZoneDNSSEC"].apply(partialParams));
    }
    getAccountLimit(partialParams) {
        return this.client.getAccountLimit(this.ops["GetAccountLimit"].apply(partialParams));
    }
    getChange(partialParams) {
        return this.client.getChange(this.ops["GetChange"].apply(partialParams));
    }
    getDNSSEC(partialParams) {
        return this.client.getDNSSEC(this.ops["GetDNSSEC"].apply(partialParams));
    }
    getHealthCheck(partialParams) {
        return this.client.getHealthCheck(this.ops["GetHealthCheck"].apply(partialParams));
    }
    getHealthCheckLastFailureReason(partialParams) {
        return this.client.getHealthCheckLastFailureReason(this.ops["GetHealthCheckLastFailureReason"].apply(partialParams));
    }
    getHealthCheckStatus(partialParams) {
        return this.client.getHealthCheckStatus(this.ops["GetHealthCheckStatus"].apply(partialParams));
    }
    getHostedZone(partialParams) {
        return this.client.getHostedZone(this.ops["GetHostedZone"].apply(partialParams));
    }
    getHostedZoneLimit(partialParams) {
        return this.client.getHostedZoneLimit(this.ops["GetHostedZoneLimit"].apply(partialParams));
    }
    getQueryLoggingConfig(partialParams) {
        return this.client.getQueryLoggingConfig(this.ops["GetQueryLoggingConfig"].apply(partialParams));
    }
    getReusableDelegationSet(partialParams) {
        return this.client.getReusableDelegationSet(this.ops["GetReusableDelegationSet"].apply(partialParams));
    }
    getReusableDelegationSetLimit(partialParams) {
        return this.client.getReusableDelegationSetLimit(this.ops["GetReusableDelegationSetLimit"].apply(partialParams));
    }
    getTrafficPolicy(partialParams) {
        return this.client.getTrafficPolicy(this.ops["GetTrafficPolicy"].apply(partialParams));
    }
    getTrafficPolicyInstance(partialParams) {
        return this.client.getTrafficPolicyInstance(this.ops["GetTrafficPolicyInstance"].apply(partialParams));
    }
    listHostedZonesByVPC(partialParams) {
        return this.client.listHostedZonesByVPC(this.ops["ListHostedZonesByVPC"].apply(partialParams));
    }
    listResourceRecordSets(partialParams) {
        return this.client.listResourceRecordSets(this.ops["ListResourceRecordSets"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTagsForResources(partialParams) {
        return this.client.listTagsForResources(this.ops["ListTagsForResources"].apply(partialParams));
    }
    listTrafficPolicyInstancesByHostedZone(partialParams) {
        return this.client.listTrafficPolicyInstancesByHostedZone(this.ops["ListTrafficPolicyInstancesByHostedZone"].apply(partialParams));
    }
    listTrafficPolicyInstancesByPolicy(partialParams) {
        return this.client.listTrafficPolicyInstancesByPolicy(this.ops["ListTrafficPolicyInstancesByPolicy"].apply(partialParams));
    }
    listTrafficPolicyVersions(partialParams) {
        return this.client.listTrafficPolicyVersions(this.ops["ListTrafficPolicyVersions"].apply(partialParams));
    }
    listVPCAssociationAuthorizations(partialParams) {
        return this.client.listVPCAssociationAuthorizations(this.ops["ListVPCAssociationAuthorizations"].apply(partialParams));
    }
    testDNSAnswer(partialParams) {
        return this.client.testDNSAnswer(this.ops["TestDNSAnswer"].apply(partialParams));
    }
    updateHealthCheck(partialParams) {
        return this.client.updateHealthCheck(this.ops["UpdateHealthCheck"].apply(partialParams));
    }
    updateHostedZoneComment(partialParams) {
        return this.client.updateHostedZoneComment(this.ops["UpdateHostedZoneComment"].apply(partialParams));
    }
    updateTrafficPolicyComment(partialParams) {
        return this.client.updateTrafficPolicyComment(this.ops["UpdateTrafficPolicyComment"].apply(partialParams));
    }
    updateTrafficPolicyInstance(partialParams) {
        return this.client.updateTrafficPolicyInstance(this.ops["UpdateTrafficPolicyInstance"].apply(partialParams));
    }
}
exports.default = Resource;
