// out/route53domains/RegisteredDomain.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.route53domains.RegisteredDomain {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Route53Domains();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/route53domains-2014-05-15.normal.json"), this.client);
    }
    acceptDomainTransferFromAnotherAwsAccount(partialParams) {
        return this.client.acceptDomainTransferFromAnotherAwsAccount(this.ops["AcceptDomainTransferFromAnotherAwsAccount"].apply(partialParams));
    }
    cancelDomainTransferToAnotherAwsAccount(partialParams) {
        return this.client.cancelDomainTransferToAnotherAwsAccount(this.ops["CancelDomainTransferToAnotherAwsAccount"].apply(partialParams));
    }
    checkDomainAvailability(partialParams) {
        return this.client.checkDomainAvailability(this.ops["CheckDomainAvailability"].apply(partialParams));
    }
    checkDomainTransferability(partialParams) {
        return this.client.checkDomainTransferability(this.ops["CheckDomainTransferability"].apply(partialParams));
    }
    deleteDomain(partialParams) {
        return this.client.deleteDomain(this.ops["DeleteDomain"].apply(partialParams));
    }
    deleteTagsForDomain(partialParams) {
        return this.client.deleteTagsForDomain(this.ops["DeleteTagsForDomain"].apply(partialParams));
    }
    disableDomainAutoRenew(partialParams) {
        return this.client.disableDomainAutoRenew(this.ops["DisableDomainAutoRenew"].apply(partialParams));
    }
    disableDomainTransferLock(partialParams) {
        return this.client.disableDomainTransferLock(this.ops["DisableDomainTransferLock"].apply(partialParams));
    }
    enableDomainAutoRenew(partialParams) {
        return this.client.enableDomainAutoRenew(this.ops["EnableDomainAutoRenew"].apply(partialParams));
    }
    enableDomainTransferLock(partialParams) {
        return this.client.enableDomainTransferLock(this.ops["EnableDomainTransferLock"].apply(partialParams));
    }
    getDomainDetail(partialParams) {
        return this.client.getDomainDetail(this.ops["GetDomainDetail"].apply(partialParams));
    }
    getDomainSuggestions(partialParams) {
        return this.client.getDomainSuggestions(this.ops["GetDomainSuggestions"].apply(partialParams));
    }
    getOperationDetail(partialParams) {
        return this.client.getOperationDetail(this.ops["GetOperationDetail"].apply(partialParams));
    }
    listTagsForDomain(partialParams) {
        return this.client.listTagsForDomain(this.ops["ListTagsForDomain"].apply(partialParams));
    }
    registerDomain(partialParams) {
        return this.client.registerDomain(this.ops["RegisterDomain"].apply(partialParams));
    }
    rejectDomainTransferFromAnotherAwsAccount(partialParams) {
        return this.client.rejectDomainTransferFromAnotherAwsAccount(this.ops["RejectDomainTransferFromAnotherAwsAccount"].apply(partialParams));
    }
    renewDomain(partialParams) {
        return this.client.renewDomain(this.ops["RenewDomain"].apply(partialParams));
    }
    retrieveDomainAuthCode(partialParams) {
        return this.client.retrieveDomainAuthCode(this.ops["RetrieveDomainAuthCode"].apply(partialParams));
    }
    transferDomain(partialParams) {
        return this.client.transferDomain(this.ops["TransferDomain"].apply(partialParams));
    }
    transferDomainToAnotherAwsAccount(partialParams) {
        return this.client.transferDomainToAnotherAwsAccount(this.ops["TransferDomainToAnotherAwsAccount"].apply(partialParams));
    }
    updateDomainContact(partialParams) {
        return this.client.updateDomainContact(this.ops["UpdateDomainContact"].apply(partialParams));
    }
    updateDomainContactPrivacy(partialParams) {
        return this.client.updateDomainContactPrivacy(this.ops["UpdateDomainContactPrivacy"].apply(partialParams));
    }
    updateDomainNameservers(partialParams) {
        return this.client.updateDomainNameservers(this.ops["UpdateDomainNameservers"].apply(partialParams));
    }
    updateTagsForDomain(partialParams) {
        return this.client.updateTagsForDomain(this.ops["UpdateTagsForDomain"].apply(partialParams));
    }
}
exports.default = Resource;
