// out/rds/OptionGroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.rds.OptionGroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.RDS();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/rds-2014-10-31.normal.json"), this.client);
    }
    addRoleToDBCluster(partialParams) {
        return this.client.addRoleToDBCluster(this.ops["AddRoleToDBCluster"].apply(partialParams));
    }
    addRoleToDBInstance(partialParams) {
        return this.client.addRoleToDBInstance(this.ops["AddRoleToDBInstance"].apply(partialParams));
    }
    addSourceIdentifierToSubscription(partialParams) {
        return this.client.addSourceIdentifierToSubscription(this.ops["AddSourceIdentifierToSubscription"].apply(partialParams));
    }
    addTagsToResource(partialParams) {
        return this.client.addTagsToResource(this.ops["AddTagsToResource"].apply(partialParams));
    }
    applyPendingMaintenanceAction(partialParams) {
        return this.client.applyPendingMaintenanceAction(this.ops["ApplyPendingMaintenanceAction"].apply(partialParams));
    }
    authorizeDBSecurityGroupIngress(partialParams) {
        return this.client.authorizeDBSecurityGroupIngress(this.ops["AuthorizeDBSecurityGroupIngress"].apply(partialParams));
    }
    backtrackDBCluster(partialParams) {
        return this.client.backtrackDBCluster(this.ops["BacktrackDBCluster"].apply(partialParams));
    }
    cancelExportTask(partialParams) {
        return this.client.cancelExportTask(this.ops["CancelExportTask"].apply(partialParams));
    }
    copyDBClusterParameterGroup(partialParams) {
        return this.client.copyDBClusterParameterGroup(this.ops["CopyDBClusterParameterGroup"].apply(partialParams));
    }
    copyDBClusterSnapshot(partialParams) {
        return this.client.copyDBClusterSnapshot(this.ops["CopyDBClusterSnapshot"].apply(partialParams));
    }
    copyDBParameterGroup(partialParams) {
        return this.client.copyDBParameterGroup(this.ops["CopyDBParameterGroup"].apply(partialParams));
    }
    copyDBSnapshot(partialParams) {
        return this.client.copyDBSnapshot(this.ops["CopyDBSnapshot"].apply(partialParams));
    }
    copyOptionGroup(partialParams) {
        return this.client.copyOptionGroup(this.ops["CopyOptionGroup"].apply(partialParams));
    }
    createCustomAvailabilityZone(partialParams) {
        return this.client.createCustomAvailabilityZone(this.ops["CreateCustomAvailabilityZone"].apply(partialParams));
    }
    createCustomDBEngineVersion(partialParams) {
        return this.client.createCustomDBEngineVersion(this.ops["CreateCustomDBEngineVersion"].apply(partialParams));
    }
    createDBCluster(partialParams) {
        return this.client.createDBCluster(this.ops["CreateDBCluster"].apply(partialParams));
    }
    createDBClusterEndpoint(partialParams) {
        return this.client.createDBClusterEndpoint(this.ops["CreateDBClusterEndpoint"].apply(partialParams));
    }
    createDBClusterParameterGroup(partialParams) {
        return this.client.createDBClusterParameterGroup(this.ops["CreateDBClusterParameterGroup"].apply(partialParams));
    }
    createDBClusterSnapshot(partialParams) {
        return this.client.createDBClusterSnapshot(this.ops["CreateDBClusterSnapshot"].apply(partialParams));
    }
    createDBInstance(partialParams) {
        return this.client.createDBInstance(this.ops["CreateDBInstance"].apply(partialParams));
    }
    createDBInstanceReadReplica(partialParams) {
        return this.client.createDBInstanceReadReplica(this.ops["CreateDBInstanceReadReplica"].apply(partialParams));
    }
    createDBParameterGroup(partialParams) {
        return this.client.createDBParameterGroup(this.ops["CreateDBParameterGroup"].apply(partialParams));
    }
    createDBProxy(partialParams) {
        return this.client.createDBProxy(this.ops["CreateDBProxy"].apply(partialParams));
    }
    createDBProxyEndpoint(partialParams) {
        return this.client.createDBProxyEndpoint(this.ops["CreateDBProxyEndpoint"].apply(partialParams));
    }
    createDBSecurityGroup(partialParams) {
        return this.client.createDBSecurityGroup(this.ops["CreateDBSecurityGroup"].apply(partialParams));
    }
    createDBSnapshot(partialParams) {
        return this.client.createDBSnapshot(this.ops["CreateDBSnapshot"].apply(partialParams));
    }
    createDBSubnetGroup(partialParams) {
        return this.client.createDBSubnetGroup(this.ops["CreateDBSubnetGroup"].apply(partialParams));
    }
    createEventSubscription(partialParams) {
        return this.client.createEventSubscription(this.ops["CreateEventSubscription"].apply(partialParams));
    }
    createOptionGroup(partialParams) {
        return this.client.createOptionGroup(this.ops["CreateOptionGroup"].apply(partialParams));
    }
    deleteCustomAvailabilityZone(partialParams) {
        return this.client.deleteCustomAvailabilityZone(this.ops["DeleteCustomAvailabilityZone"].apply(partialParams));
    }
    deleteCustomDBEngineVersion(partialParams) {
        return this.client.deleteCustomDBEngineVersion(this.ops["DeleteCustomDBEngineVersion"].apply(partialParams));
    }
    deleteDBCluster(partialParams) {
        return this.client.deleteDBCluster(this.ops["DeleteDBCluster"].apply(partialParams));
    }
    deleteDBClusterEndpoint(partialParams) {
        return this.client.deleteDBClusterEndpoint(this.ops["DeleteDBClusterEndpoint"].apply(partialParams));
    }
    deleteDBClusterParameterGroup(partialParams) {
        return this.client.deleteDBClusterParameterGroup(this.ops["DeleteDBClusterParameterGroup"].apply(partialParams));
    }
    deleteDBClusterSnapshot(partialParams) {
        return this.client.deleteDBClusterSnapshot(this.ops["DeleteDBClusterSnapshot"].apply(partialParams));
    }
    deleteDBInstance(partialParams) {
        return this.client.deleteDBInstance(this.ops["DeleteDBInstance"].apply(partialParams));
    }
    deleteDBParameterGroup(partialParams) {
        return this.client.deleteDBParameterGroup(this.ops["DeleteDBParameterGroup"].apply(partialParams));
    }
    deleteDBProxy(partialParams) {
        return this.client.deleteDBProxy(this.ops["DeleteDBProxy"].apply(partialParams));
    }
    deleteDBProxyEndpoint(partialParams) {
        return this.client.deleteDBProxyEndpoint(this.ops["DeleteDBProxyEndpoint"].apply(partialParams));
    }
    deleteDBSecurityGroup(partialParams) {
        return this.client.deleteDBSecurityGroup(this.ops["DeleteDBSecurityGroup"].apply(partialParams));
    }
    deleteDBSnapshot(partialParams) {
        return this.client.deleteDBSnapshot(this.ops["DeleteDBSnapshot"].apply(partialParams));
    }
    deleteDBSubnetGroup(partialParams) {
        return this.client.deleteDBSubnetGroup(this.ops["DeleteDBSubnetGroup"].apply(partialParams));
    }
    deleteEventSubscription(partialParams) {
        return this.client.deleteEventSubscription(this.ops["DeleteEventSubscription"].apply(partialParams));
    }
    deleteGlobalCluster(partialParams) {
        return this.client.deleteGlobalCluster(this.ops["DeleteGlobalCluster"].apply(partialParams));
    }
    deleteInstallationMedia(partialParams) {
        return this.client.deleteInstallationMedia(this.ops["DeleteInstallationMedia"].apply(partialParams));
    }
    deleteOptionGroup(partialParams) {
        return this.client.deleteOptionGroup(this.ops["DeleteOptionGroup"].apply(partialParams));
    }
    deregisterDBProxyTargets(partialParams) {
        return this.client.deregisterDBProxyTargets(this.ops["DeregisterDBProxyTargets"].apply(partialParams));
    }
    describeDBClusterBacktracks(partialParams) {
        return this.client.describeDBClusterBacktracks(this.ops["DescribeDBClusterBacktracks"].apply(partialParams));
    }
    describeDBClusterParameters(partialParams) {
        return this.client.describeDBClusterParameters(this.ops["DescribeDBClusterParameters"].apply(partialParams));
    }
    describeDBClusterSnapshotAttributes(partialParams) {
        return this.client.describeDBClusterSnapshotAttributes(this.ops["DescribeDBClusterSnapshotAttributes"].apply(partialParams));
    }
    describeDBLogFiles(partialParams) {
        return this.client.describeDBLogFiles(this.ops["DescribeDBLogFiles"].apply(partialParams));
    }
    describeDBParameters(partialParams) {
        return this.client.describeDBParameters(this.ops["DescribeDBParameters"].apply(partialParams));
    }
    describeDBProxyTargetGroups(partialParams) {
        return this.client.describeDBProxyTargetGroups(this.ops["DescribeDBProxyTargetGroups"].apply(partialParams));
    }
    describeDBProxyTargets(partialParams) {
        return this.client.describeDBProxyTargets(this.ops["DescribeDBProxyTargets"].apply(partialParams));
    }
    describeDBSnapshotAttributes(partialParams) {
        return this.client.describeDBSnapshotAttributes(this.ops["DescribeDBSnapshotAttributes"].apply(partialParams));
    }
    describeEngineDefaultClusterParameters(partialParams) {
        return this.client.describeEngineDefaultClusterParameters(this.ops["DescribeEngineDefaultClusterParameters"].apply(partialParams));
    }
    describeEngineDefaultParameters(partialParams) {
        return this.client.describeEngineDefaultParameters(this.ops["DescribeEngineDefaultParameters"].apply(partialParams));
    }
    describeOptionGroupOptions(partialParams) {
        return this.client.describeOptionGroupOptions(this.ops["DescribeOptionGroupOptions"].apply(partialParams));
    }
    describeOrderableDBInstanceOptions(partialParams) {
        return this.client.describeOrderableDBInstanceOptions(this.ops["DescribeOrderableDBInstanceOptions"].apply(partialParams));
    }
    describeValidDBInstanceModifications(partialParams) {
        return this.client.describeValidDBInstanceModifications(this.ops["DescribeValidDBInstanceModifications"].apply(partialParams));
    }
    downloadDBLogFilePortion(partialParams) {
        return this.client.downloadDBLogFilePortion(this.ops["DownloadDBLogFilePortion"].apply(partialParams));
    }
    failoverDBCluster(partialParams) {
        return this.client.failoverDBCluster(this.ops["FailoverDBCluster"].apply(partialParams));
    }
    failoverGlobalCluster(partialParams) {
        return this.client.failoverGlobalCluster(this.ops["FailoverGlobalCluster"].apply(partialParams));
    }
    importInstallationMedia(partialParams) {
        return this.client.importInstallationMedia(this.ops["ImportInstallationMedia"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    modifyCurrentDBClusterCapacity(partialParams) {
        return this.client.modifyCurrentDBClusterCapacity(this.ops["ModifyCurrentDBClusterCapacity"].apply(partialParams));
    }
    modifyCustomDBEngineVersion(partialParams) {
        return this.client.modifyCustomDBEngineVersion(this.ops["ModifyCustomDBEngineVersion"].apply(partialParams));
    }
    modifyDBCluster(partialParams) {
        return this.client.modifyDBCluster(this.ops["ModifyDBCluster"].apply(partialParams));
    }
    modifyDBClusterEndpoint(partialParams) {
        return this.client.modifyDBClusterEndpoint(this.ops["ModifyDBClusterEndpoint"].apply(partialParams));
    }
    modifyDBClusterParameterGroup(partialParams) {
        return this.client.modifyDBClusterParameterGroup(this.ops["ModifyDBClusterParameterGroup"].apply(partialParams));
    }
    modifyDBClusterSnapshotAttribute(partialParams) {
        return this.client.modifyDBClusterSnapshotAttribute(this.ops["ModifyDBClusterSnapshotAttribute"].apply(partialParams));
    }
    modifyDBInstance(partialParams) {
        return this.client.modifyDBInstance(this.ops["ModifyDBInstance"].apply(partialParams));
    }
    modifyDBParameterGroup(partialParams) {
        return this.client.modifyDBParameterGroup(this.ops["ModifyDBParameterGroup"].apply(partialParams));
    }
    modifyDBProxy(partialParams) {
        return this.client.modifyDBProxy(this.ops["ModifyDBProxy"].apply(partialParams));
    }
    modifyDBProxyEndpoint(partialParams) {
        return this.client.modifyDBProxyEndpoint(this.ops["ModifyDBProxyEndpoint"].apply(partialParams));
    }
    modifyDBProxyTargetGroup(partialParams) {
        return this.client.modifyDBProxyTargetGroup(this.ops["ModifyDBProxyTargetGroup"].apply(partialParams));
    }
    modifyDBSnapshot(partialParams) {
        return this.client.modifyDBSnapshot(this.ops["ModifyDBSnapshot"].apply(partialParams));
    }
    modifyDBSnapshotAttribute(partialParams) {
        return this.client.modifyDBSnapshotAttribute(this.ops["ModifyDBSnapshotAttribute"].apply(partialParams));
    }
    modifyDBSubnetGroup(partialParams) {
        return this.client.modifyDBSubnetGroup(this.ops["ModifyDBSubnetGroup"].apply(partialParams));
    }
    modifyEventSubscription(partialParams) {
        return this.client.modifyEventSubscription(this.ops["ModifyEventSubscription"].apply(partialParams));
    }
    modifyOptionGroup(partialParams) {
        return this.client.modifyOptionGroup(this.ops["ModifyOptionGroup"].apply(partialParams));
    }
    promoteReadReplica(partialParams) {
        return this.client.promoteReadReplica(this.ops["PromoteReadReplica"].apply(partialParams));
    }
    promoteReadReplicaDBCluster(partialParams) {
        return this.client.promoteReadReplicaDBCluster(this.ops["PromoteReadReplicaDBCluster"].apply(partialParams));
    }
    purchaseReservedDBInstancesOffering(partialParams) {
        return this.client.purchaseReservedDBInstancesOffering(this.ops["PurchaseReservedDBInstancesOffering"].apply(partialParams));
    }
    rebootDBCluster(partialParams) {
        return this.client.rebootDBCluster(this.ops["RebootDBCluster"].apply(partialParams));
    }
    rebootDBInstance(partialParams) {
        return this.client.rebootDBInstance(this.ops["RebootDBInstance"].apply(partialParams));
    }
    registerDBProxyTargets(partialParams) {
        return this.client.registerDBProxyTargets(this.ops["RegisterDBProxyTargets"].apply(partialParams));
    }
    removeRoleFromDBCluster(partialParams) {
        return this.client.removeRoleFromDBCluster(this.ops["RemoveRoleFromDBCluster"].apply(partialParams));
    }
    removeRoleFromDBInstance(partialParams) {
        return this.client.removeRoleFromDBInstance(this.ops["RemoveRoleFromDBInstance"].apply(partialParams));
    }
    removeSourceIdentifierFromSubscription(partialParams) {
        return this.client.removeSourceIdentifierFromSubscription(this.ops["RemoveSourceIdentifierFromSubscription"].apply(partialParams));
    }
    removeTagsFromResource(partialParams) {
        return this.client.removeTagsFromResource(this.ops["RemoveTagsFromResource"].apply(partialParams));
    }
    resetDBClusterParameterGroup(partialParams) {
        return this.client.resetDBClusterParameterGroup(this.ops["ResetDBClusterParameterGroup"].apply(partialParams));
    }
    resetDBParameterGroup(partialParams) {
        return this.client.resetDBParameterGroup(this.ops["ResetDBParameterGroup"].apply(partialParams));
    }
    restoreDBClusterFromS3(partialParams) {
        return this.client.restoreDBClusterFromS3(this.ops["RestoreDBClusterFromS3"].apply(partialParams));
    }
    restoreDBClusterFromSnapshot(partialParams) {
        return this.client.restoreDBClusterFromSnapshot(this.ops["RestoreDBClusterFromSnapshot"].apply(partialParams));
    }
    restoreDBClusterToPointInTime(partialParams) {
        return this.client.restoreDBClusterToPointInTime(this.ops["RestoreDBClusterToPointInTime"].apply(partialParams));
    }
    restoreDBInstanceFromDBSnapshot(partialParams) {
        return this.client.restoreDBInstanceFromDBSnapshot(this.ops["RestoreDBInstanceFromDBSnapshot"].apply(partialParams));
    }
    restoreDBInstanceFromS3(partialParams) {
        return this.client.restoreDBInstanceFromS3(this.ops["RestoreDBInstanceFromS3"].apply(partialParams));
    }
    restoreDBInstanceToPointInTime(partialParams) {
        return this.client.restoreDBInstanceToPointInTime(this.ops["RestoreDBInstanceToPointInTime"].apply(partialParams));
    }
    revokeDBSecurityGroupIngress(partialParams) {
        return this.client.revokeDBSecurityGroupIngress(this.ops["RevokeDBSecurityGroupIngress"].apply(partialParams));
    }
    startActivityStream(partialParams) {
        return this.client.startActivityStream(this.ops["StartActivityStream"].apply(partialParams));
    }
    startDBCluster(partialParams) {
        return this.client.startDBCluster(this.ops["StartDBCluster"].apply(partialParams));
    }
    startDBInstance(partialParams) {
        return this.client.startDBInstance(this.ops["StartDBInstance"].apply(partialParams));
    }
    startDBInstanceAutomatedBackupsReplication(partialParams) {
        return this.client.startDBInstanceAutomatedBackupsReplication(this.ops["StartDBInstanceAutomatedBackupsReplication"].apply(partialParams));
    }
    startExportTask(partialParams) {
        return this.client.startExportTask(this.ops["StartExportTask"].apply(partialParams));
    }
    stopActivityStream(partialParams) {
        return this.client.stopActivityStream(this.ops["StopActivityStream"].apply(partialParams));
    }
    stopDBCluster(partialParams) {
        return this.client.stopDBCluster(this.ops["StopDBCluster"].apply(partialParams));
    }
    stopDBInstance(partialParams) {
        return this.client.stopDBInstance(this.ops["StopDBInstance"].apply(partialParams));
    }
    stopDBInstanceAutomatedBackupsReplication(partialParams) {
        return this.client.stopDBInstanceAutomatedBackupsReplication(this.ops["StopDBInstanceAutomatedBackupsReplication"].apply(partialParams));
    }
}
exports.default = Resource;
