// out/apigatewayv2/Deployment.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.apigatewayv2.Deployment {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ApiGatewayV2();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/apigatewayv2-2018-11-29.normal.json"), this.client);
    }
    createApi(partialParams) {
        return this.client.createApi(this.ops["CreateApi"].apply(partialParams));
    }
    createApiMapping(partialParams) {
        return this.client.createApiMapping(this.ops["CreateApiMapping"].apply(partialParams));
    }
    createAuthorizer(partialParams) {
        return this.client.createAuthorizer(this.ops["CreateAuthorizer"].apply(partialParams));
    }
    createDeployment(partialParams) {
        return this.client.createDeployment(this.ops["CreateDeployment"].apply(partialParams));
    }
    createDomainName(partialParams) {
        return this.client.createDomainName(this.ops["CreateDomainName"].apply(partialParams));
    }
    createIntegration(partialParams) {
        return this.client.createIntegration(this.ops["CreateIntegration"].apply(partialParams));
    }
    createIntegrationResponse(partialParams) {
        return this.client.createIntegrationResponse(this.ops["CreateIntegrationResponse"].apply(partialParams));
    }
    createModel(partialParams) {
        return this.client.createModel(this.ops["CreateModel"].apply(partialParams));
    }
    createRoute(partialParams) {
        return this.client.createRoute(this.ops["CreateRoute"].apply(partialParams));
    }
    createRouteResponse(partialParams) {
        return this.client.createRouteResponse(this.ops["CreateRouteResponse"].apply(partialParams));
    }
    createStage(partialParams) {
        return this.client.createStage(this.ops["CreateStage"].apply(partialParams));
    }
    createVpcLink(partialParams) {
        return this.client.createVpcLink(this.ops["CreateVpcLink"].apply(partialParams));
    }
    deleteAccessLogSettings(partialParams) {
        return this.client.deleteAccessLogSettings(this.ops["DeleteAccessLogSettings"].apply(partialParams));
    }
    deleteApi(partialParams) {
        return this.client.deleteApi(this.ops["DeleteApi"].apply(partialParams));
    }
    deleteApiMapping(partialParams) {
        return this.client.deleteApiMapping(this.ops["DeleteApiMapping"].apply(partialParams));
    }
    deleteAuthorizer(partialParams) {
        return this.client.deleteAuthorizer(this.ops["DeleteAuthorizer"].apply(partialParams));
    }
    deleteCorsConfiguration(partialParams) {
        return this.client.deleteCorsConfiguration(this.ops["DeleteCorsConfiguration"].apply(partialParams));
    }
    deleteDeployment(partialParams) {
        return this.client.deleteDeployment(this.ops["DeleteDeployment"].apply(partialParams));
    }
    deleteDomainName(partialParams) {
        return this.client.deleteDomainName(this.ops["DeleteDomainName"].apply(partialParams));
    }
    deleteIntegration(partialParams) {
        return this.client.deleteIntegration(this.ops["DeleteIntegration"].apply(partialParams));
    }
    deleteIntegrationResponse(partialParams) {
        return this.client.deleteIntegrationResponse(this.ops["DeleteIntegrationResponse"].apply(partialParams));
    }
    deleteModel(partialParams) {
        return this.client.deleteModel(this.ops["DeleteModel"].apply(partialParams));
    }
    deleteRoute(partialParams) {
        return this.client.deleteRoute(this.ops["DeleteRoute"].apply(partialParams));
    }
    deleteRouteRequestParameter(partialParams) {
        return this.client.deleteRouteRequestParameter(this.ops["DeleteRouteRequestParameter"].apply(partialParams));
    }
    deleteRouteResponse(partialParams) {
        return this.client.deleteRouteResponse(this.ops["DeleteRouteResponse"].apply(partialParams));
    }
    deleteRouteSettings(partialParams) {
        return this.client.deleteRouteSettings(this.ops["DeleteRouteSettings"].apply(partialParams));
    }
    deleteStage(partialParams) {
        return this.client.deleteStage(this.ops["DeleteStage"].apply(partialParams));
    }
    deleteVpcLink(partialParams) {
        return this.client.deleteVpcLink(this.ops["DeleteVpcLink"].apply(partialParams));
    }
    exportApi(partialParams) {
        return this.client.exportApi(this.ops["ExportApi"].apply(partialParams));
    }
    resetAuthorizersCache(partialParams) {
        return this.client.resetAuthorizersCache(this.ops["ResetAuthorizersCache"].apply(partialParams));
    }
    getApi(partialParams) {
        return this.client.getApi(this.ops["GetApi"].apply(partialParams));
    }
    getApiMapping(partialParams) {
        return this.client.getApiMapping(this.ops["GetApiMapping"].apply(partialParams));
    }
    getApiMappings(partialParams) {
        return this.client.getApiMappings(this.ops["GetApiMappings"].apply(partialParams));
    }
    getAuthorizer(partialParams) {
        return this.client.getAuthorizer(this.ops["GetAuthorizer"].apply(partialParams));
    }
    getAuthorizers(partialParams) {
        return this.client.getAuthorizers(this.ops["GetAuthorizers"].apply(partialParams));
    }
    getDeployment(partialParams) {
        return this.client.getDeployment(this.ops["GetDeployment"].apply(partialParams));
    }
    getDeployments(partialParams) {
        return this.client.getDeployments(this.ops["GetDeployments"].apply(partialParams));
    }
    getDomainName(partialParams) {
        return this.client.getDomainName(this.ops["GetDomainName"].apply(partialParams));
    }
    getIntegration(partialParams) {
        return this.client.getIntegration(this.ops["GetIntegration"].apply(partialParams));
    }
    getIntegrationResponse(partialParams) {
        return this.client.getIntegrationResponse(this.ops["GetIntegrationResponse"].apply(partialParams));
    }
    getIntegrationResponses(partialParams) {
        return this.client.getIntegrationResponses(this.ops["GetIntegrationResponses"].apply(partialParams));
    }
    getIntegrations(partialParams) {
        return this.client.getIntegrations(this.ops["GetIntegrations"].apply(partialParams));
    }
    getModel(partialParams) {
        return this.client.getModel(this.ops["GetModel"].apply(partialParams));
    }
    getModelTemplate(partialParams) {
        return this.client.getModelTemplate(this.ops["GetModelTemplate"].apply(partialParams));
    }
    getModels(partialParams) {
        return this.client.getModels(this.ops["GetModels"].apply(partialParams));
    }
    getRoute(partialParams) {
        return this.client.getRoute(this.ops["GetRoute"].apply(partialParams));
    }
    getRouteResponse(partialParams) {
        return this.client.getRouteResponse(this.ops["GetRouteResponse"].apply(partialParams));
    }
    getRouteResponses(partialParams) {
        return this.client.getRouteResponses(this.ops["GetRouteResponses"].apply(partialParams));
    }
    getRoutes(partialParams) {
        return this.client.getRoutes(this.ops["GetRoutes"].apply(partialParams));
    }
    getStage(partialParams) {
        return this.client.getStage(this.ops["GetStage"].apply(partialParams));
    }
    getStages(partialParams) {
        return this.client.getStages(this.ops["GetStages"].apply(partialParams));
    }
    getTags(partialParams) {
        return this.client.getTags(this.ops["GetTags"].apply(partialParams));
    }
    getVpcLink(partialParams) {
        return this.client.getVpcLink(this.ops["GetVpcLink"].apply(partialParams));
    }
    importApi(partialParams) {
        return this.client.importApi(this.ops["ImportApi"].apply(partialParams));
    }
    reimportApi(partialParams) {
        return this.client.reimportApi(this.ops["ReimportApi"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApi(partialParams) {
        return this.client.updateApi(this.ops["UpdateApi"].apply(partialParams));
    }
    updateApiMapping(partialParams) {
        return this.client.updateApiMapping(this.ops["UpdateApiMapping"].apply(partialParams));
    }
    updateAuthorizer(partialParams) {
        return this.client.updateAuthorizer(this.ops["UpdateAuthorizer"].apply(partialParams));
    }
    updateDeployment(partialParams) {
        return this.client.updateDeployment(this.ops["UpdateDeployment"].apply(partialParams));
    }
    updateDomainName(partialParams) {
        return this.client.updateDomainName(this.ops["UpdateDomainName"].apply(partialParams));
    }
    updateIntegration(partialParams) {
        return this.client.updateIntegration(this.ops["UpdateIntegration"].apply(partialParams));
    }
    updateIntegrationResponse(partialParams) {
        return this.client.updateIntegrationResponse(this.ops["UpdateIntegrationResponse"].apply(partialParams));
    }
    updateModel(partialParams) {
        return this.client.updateModel(this.ops["UpdateModel"].apply(partialParams));
    }
    updateRoute(partialParams) {
        return this.client.updateRoute(this.ops["UpdateRoute"].apply(partialParams));
    }
    updateRouteResponse(partialParams) {
        return this.client.updateRouteResponse(this.ops["UpdateRouteResponse"].apply(partialParams));
    }
    updateStage(partialParams) {
        return this.client.updateStage(this.ops["UpdateStage"].apply(partialParams));
    }
    updateVpcLink(partialParams) {
        return this.client.updateVpcLink(this.ops["UpdateVpcLink"].apply(partialParams));
    }
}
exports.default = Resource;
