// out/iot/Authorizer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.iot.Authorizer {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Iot();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/iot-2015-05-28.normal.json"), this.client);
    }
    acceptCertificateTransfer(partialParams) {
        return this.client.acceptCertificateTransfer(this.ops["AcceptCertificateTransfer"].apply(partialParams));
    }
    associateTargetsWithJob(partialParams) {
        return this.client.associateTargetsWithJob(this.ops["AssociateTargetsWithJob"].apply(partialParams));
    }
    attachPolicy(partialParams) {
        return this.client.attachPolicy(this.ops["AttachPolicy"].apply(partialParams));
    }
    attachPrincipalPolicy(partialParams) {
        return this.client.attachPrincipalPolicy(this.ops["AttachPrincipalPolicy"].apply(partialParams));
    }
    attachSecurityProfile(partialParams) {
        return this.client.attachSecurityProfile(this.ops["AttachSecurityProfile"].apply(partialParams));
    }
    attachThingPrincipal(partialParams) {
        return this.client.attachThingPrincipal(this.ops["AttachThingPrincipal"].apply(partialParams));
    }
    cancelAuditMitigationActionsTask(partialParams) {
        return this.client.cancelAuditMitigationActionsTask(this.ops["CancelAuditMitigationActionsTask"].apply(partialParams));
    }
    cancelAuditTask(partialParams) {
        return this.client.cancelAuditTask(this.ops["CancelAuditTask"].apply(partialParams));
    }
    cancelCertificateTransfer(partialParams) {
        return this.client.cancelCertificateTransfer(this.ops["CancelCertificateTransfer"].apply(partialParams));
    }
    cancelDetectMitigationActionsTask(partialParams) {
        return this.client.cancelDetectMitigationActionsTask(this.ops["CancelDetectMitigationActionsTask"].apply(partialParams));
    }
    cancelJob(partialParams) {
        return this.client.cancelJob(this.ops["CancelJob"].apply(partialParams));
    }
    cancelJobExecution(partialParams) {
        return this.client.cancelJobExecution(this.ops["CancelJobExecution"].apply(partialParams));
    }
    confirmTopicRuleDestination(partialParams) {
        return this.client.confirmTopicRuleDestination(this.ops["ConfirmTopicRuleDestination"].apply(partialParams));
    }
    createAuditSuppression(partialParams) {
        return this.client.createAuditSuppression(this.ops["CreateAuditSuppression"].apply(partialParams));
    }
    createAuthorizer(partialParams) {
        return this.client.createAuthorizer(this.ops["CreateAuthorizer"].apply(partialParams));
    }
    createBillingGroup(partialParams) {
        return this.client.createBillingGroup(this.ops["CreateBillingGroup"].apply(partialParams));
    }
    createCertificateFromCsr(partialParams) {
        return this.client.createCertificateFromCsr(this.ops["CreateCertificateFromCsr"].apply(partialParams));
    }
    createCustomMetric(partialParams) {
        return this.client.createCustomMetric(this.ops["CreateCustomMetric"].apply(partialParams));
    }
    createDimension(partialParams) {
        return this.client.createDimension(this.ops["CreateDimension"].apply(partialParams));
    }
    createDomainConfiguration(partialParams) {
        return this.client.createDomainConfiguration(this.ops["CreateDomainConfiguration"].apply(partialParams));
    }
    createDynamicThingGroup(partialParams) {
        return this.client.createDynamicThingGroup(this.ops["CreateDynamicThingGroup"].apply(partialParams));
    }
    createFleetMetric(partialParams) {
        return this.client.createFleetMetric(this.ops["CreateFleetMetric"].apply(partialParams));
    }
    createJob(partialParams) {
        return this.client.createJob(this.ops["CreateJob"].apply(partialParams));
    }
    createJobTemplate(partialParams) {
        return this.client.createJobTemplate(this.ops["CreateJobTemplate"].apply(partialParams));
    }
    createMitigationAction(partialParams) {
        return this.client.createMitigationAction(this.ops["CreateMitigationAction"].apply(partialParams));
    }
    createOTAUpdate(partialParams) {
        return this.client.createOTAUpdate(this.ops["CreateOTAUpdate"].apply(partialParams));
    }
    createPolicy(partialParams) {
        return this.client.createPolicy(this.ops["CreatePolicy"].apply(partialParams));
    }
    createPolicyVersion(partialParams) {
        return this.client.createPolicyVersion(this.ops["CreatePolicyVersion"].apply(partialParams));
    }
    createProvisioningClaim(partialParams) {
        return this.client.createProvisioningClaim(this.ops["CreateProvisioningClaim"].apply(partialParams));
    }
    createProvisioningTemplate(partialParams) {
        return this.client.createProvisioningTemplate(this.ops["CreateProvisioningTemplate"].apply(partialParams));
    }
    createProvisioningTemplateVersion(partialParams) {
        return this.client.createProvisioningTemplateVersion(this.ops["CreateProvisioningTemplateVersion"].apply(partialParams));
    }
    createRoleAlias(partialParams) {
        return this.client.createRoleAlias(this.ops["CreateRoleAlias"].apply(partialParams));
    }
    createScheduledAudit(partialParams) {
        return this.client.createScheduledAudit(this.ops["CreateScheduledAudit"].apply(partialParams));
    }
    createSecurityProfile(partialParams) {
        return this.client.createSecurityProfile(this.ops["CreateSecurityProfile"].apply(partialParams));
    }
    createStream(partialParams) {
        return this.client.createStream(this.ops["CreateStream"].apply(partialParams));
    }
    createThing(partialParams) {
        return this.client.createThing(this.ops["CreateThing"].apply(partialParams));
    }
    createThingGroup(partialParams) {
        return this.client.createThingGroup(this.ops["CreateThingGroup"].apply(partialParams));
    }
    createThingType(partialParams) {
        return this.client.createThingType(this.ops["CreateThingType"].apply(partialParams));
    }
    createTopicRule(partialParams) {
        return this.client.createTopicRule(this.ops["CreateTopicRule"].apply(partialParams));
    }
    createTopicRuleDestination(partialParams) {
        return this.client.createTopicRuleDestination(this.ops["CreateTopicRuleDestination"].apply(partialParams));
    }
    deleteAuditSuppression(partialParams) {
        return this.client.deleteAuditSuppression(this.ops["DeleteAuditSuppression"].apply(partialParams));
    }
    deleteAuthorizer(partialParams) {
        return this.client.deleteAuthorizer(this.ops["DeleteAuthorizer"].apply(partialParams));
    }
    deleteBillingGroup(partialParams) {
        return this.client.deleteBillingGroup(this.ops["DeleteBillingGroup"].apply(partialParams));
    }
    deleteCACertificate(partialParams) {
        return this.client.deleteCACertificate(this.ops["DeleteCACertificate"].apply(partialParams));
    }
    deleteCertificate(partialParams) {
        return this.client.deleteCertificate(this.ops["DeleteCertificate"].apply(partialParams));
    }
    deleteCustomMetric(partialParams) {
        return this.client.deleteCustomMetric(this.ops["DeleteCustomMetric"].apply(partialParams));
    }
    deleteDimension(partialParams) {
        return this.client.deleteDimension(this.ops["DeleteDimension"].apply(partialParams));
    }
    deleteDomainConfiguration(partialParams) {
        return this.client.deleteDomainConfiguration(this.ops["DeleteDomainConfiguration"].apply(partialParams));
    }
    deleteDynamicThingGroup(partialParams) {
        return this.client.deleteDynamicThingGroup(this.ops["DeleteDynamicThingGroup"].apply(partialParams));
    }
    deleteFleetMetric(partialParams) {
        return this.client.deleteFleetMetric(this.ops["DeleteFleetMetric"].apply(partialParams));
    }
    deleteJob(partialParams) {
        return this.client.deleteJob(this.ops["DeleteJob"].apply(partialParams));
    }
    deleteJobExecution(partialParams) {
        return this.client.deleteJobExecution(this.ops["DeleteJobExecution"].apply(partialParams));
    }
    deleteJobTemplate(partialParams) {
        return this.client.deleteJobTemplate(this.ops["DeleteJobTemplate"].apply(partialParams));
    }
    deleteMitigationAction(partialParams) {
        return this.client.deleteMitigationAction(this.ops["DeleteMitigationAction"].apply(partialParams));
    }
    deleteOTAUpdate(partialParams) {
        return this.client.deleteOTAUpdate(this.ops["DeleteOTAUpdate"].apply(partialParams));
    }
    deletePolicy(partialParams) {
        return this.client.deletePolicy(this.ops["DeletePolicy"].apply(partialParams));
    }
    deletePolicyVersion(partialParams) {
        return this.client.deletePolicyVersion(this.ops["DeletePolicyVersion"].apply(partialParams));
    }
    deleteProvisioningTemplate(partialParams) {
        return this.client.deleteProvisioningTemplate(this.ops["DeleteProvisioningTemplate"].apply(partialParams));
    }
    deleteProvisioningTemplateVersion(partialParams) {
        return this.client.deleteProvisioningTemplateVersion(this.ops["DeleteProvisioningTemplateVersion"].apply(partialParams));
    }
    deleteRoleAlias(partialParams) {
        return this.client.deleteRoleAlias(this.ops["DeleteRoleAlias"].apply(partialParams));
    }
    deleteScheduledAudit(partialParams) {
        return this.client.deleteScheduledAudit(this.ops["DeleteScheduledAudit"].apply(partialParams));
    }
    deleteSecurityProfile(partialParams) {
        return this.client.deleteSecurityProfile(this.ops["DeleteSecurityProfile"].apply(partialParams));
    }
    deleteStream(partialParams) {
        return this.client.deleteStream(this.ops["DeleteStream"].apply(partialParams));
    }
    deleteThing(partialParams) {
        return this.client.deleteThing(this.ops["DeleteThing"].apply(partialParams));
    }
    deleteThingGroup(partialParams) {
        return this.client.deleteThingGroup(this.ops["DeleteThingGroup"].apply(partialParams));
    }
    deleteThingType(partialParams) {
        return this.client.deleteThingType(this.ops["DeleteThingType"].apply(partialParams));
    }
    deleteTopicRule(partialParams) {
        return this.client.deleteTopicRule(this.ops["DeleteTopicRule"].apply(partialParams));
    }
    deleteTopicRuleDestination(partialParams) {
        return this.client.deleteTopicRuleDestination(this.ops["DeleteTopicRuleDestination"].apply(partialParams));
    }
    deleteV2LoggingLevel(partialParams) {
        return this.client.deleteV2LoggingLevel(this.ops["DeleteV2LoggingLevel"].apply(partialParams));
    }
    deprecateThingType(partialParams) {
        return this.client.deprecateThingType(this.ops["DeprecateThingType"].apply(partialParams));
    }
    describeAuditFinding(partialParams) {
        return this.client.describeAuditFinding(this.ops["DescribeAuditFinding"].apply(partialParams));
    }
    describeAuditMitigationActionsTask(partialParams) {
        return this.client.describeAuditMitigationActionsTask(this.ops["DescribeAuditMitigationActionsTask"].apply(partialParams));
    }
    describeAuditSuppression(partialParams) {
        return this.client.describeAuditSuppression(this.ops["DescribeAuditSuppression"].apply(partialParams));
    }
    describeAuditTask(partialParams) {
        return this.client.describeAuditTask(this.ops["DescribeAuditTask"].apply(partialParams));
    }
    describeAuthorizer(partialParams) {
        return this.client.describeAuthorizer(this.ops["DescribeAuthorizer"].apply(partialParams));
    }
    describeBillingGroup(partialParams) {
        return this.client.describeBillingGroup(this.ops["DescribeBillingGroup"].apply(partialParams));
    }
    describeCACertificate(partialParams) {
        return this.client.describeCACertificate(this.ops["DescribeCACertificate"].apply(partialParams));
    }
    describeCertificate(partialParams) {
        return this.client.describeCertificate(this.ops["DescribeCertificate"].apply(partialParams));
    }
    describeCustomMetric(partialParams) {
        return this.client.describeCustomMetric(this.ops["DescribeCustomMetric"].apply(partialParams));
    }
    describeDetectMitigationActionsTask(partialParams) {
        return this.client.describeDetectMitigationActionsTask(this.ops["DescribeDetectMitigationActionsTask"].apply(partialParams));
    }
    describeDimension(partialParams) {
        return this.client.describeDimension(this.ops["DescribeDimension"].apply(partialParams));
    }
    describeDomainConfiguration(partialParams) {
        return this.client.describeDomainConfiguration(this.ops["DescribeDomainConfiguration"].apply(partialParams));
    }
    describeFleetMetric(partialParams) {
        return this.client.describeFleetMetric(this.ops["DescribeFleetMetric"].apply(partialParams));
    }
    describeIndex(partialParams) {
        return this.client.describeIndex(this.ops["DescribeIndex"].apply(partialParams));
    }
    describeJob(partialParams) {
        return this.client.describeJob(this.ops["DescribeJob"].apply(partialParams));
    }
    describeJobExecution(partialParams) {
        return this.client.describeJobExecution(this.ops["DescribeJobExecution"].apply(partialParams));
    }
    describeJobTemplate(partialParams) {
        return this.client.describeJobTemplate(this.ops["DescribeJobTemplate"].apply(partialParams));
    }
    describeManagedJobTemplate(partialParams) {
        return this.client.describeManagedJobTemplate(this.ops["DescribeManagedJobTemplate"].apply(partialParams));
    }
    describeMitigationAction(partialParams) {
        return this.client.describeMitigationAction(this.ops["DescribeMitigationAction"].apply(partialParams));
    }
    describeProvisioningTemplate(partialParams) {
        return this.client.describeProvisioningTemplate(this.ops["DescribeProvisioningTemplate"].apply(partialParams));
    }
    describeProvisioningTemplateVersion(partialParams) {
        return this.client.describeProvisioningTemplateVersion(this.ops["DescribeProvisioningTemplateVersion"].apply(partialParams));
    }
    describeRoleAlias(partialParams) {
        return this.client.describeRoleAlias(this.ops["DescribeRoleAlias"].apply(partialParams));
    }
    describeScheduledAudit(partialParams) {
        return this.client.describeScheduledAudit(this.ops["DescribeScheduledAudit"].apply(partialParams));
    }
    describeSecurityProfile(partialParams) {
        return this.client.describeSecurityProfile(this.ops["DescribeSecurityProfile"].apply(partialParams));
    }
    describeStream(partialParams) {
        return this.client.describeStream(this.ops["DescribeStream"].apply(partialParams));
    }
    describeThing(partialParams) {
        return this.client.describeThing(this.ops["DescribeThing"].apply(partialParams));
    }
    describeThingGroup(partialParams) {
        return this.client.describeThingGroup(this.ops["DescribeThingGroup"].apply(partialParams));
    }
    describeThingRegistrationTask(partialParams) {
        return this.client.describeThingRegistrationTask(this.ops["DescribeThingRegistrationTask"].apply(partialParams));
    }
    describeThingType(partialParams) {
        return this.client.describeThingType(this.ops["DescribeThingType"].apply(partialParams));
    }
    detachPolicy(partialParams) {
        return this.client.detachPolicy(this.ops["DetachPolicy"].apply(partialParams));
    }
    detachPrincipalPolicy(partialParams) {
        return this.client.detachPrincipalPolicy(this.ops["DetachPrincipalPolicy"].apply(partialParams));
    }
    detachSecurityProfile(partialParams) {
        return this.client.detachSecurityProfile(this.ops["DetachSecurityProfile"].apply(partialParams));
    }
    detachThingPrincipal(partialParams) {
        return this.client.detachThingPrincipal(this.ops["DetachThingPrincipal"].apply(partialParams));
    }
    disableTopicRule(partialParams) {
        return this.client.disableTopicRule(this.ops["DisableTopicRule"].apply(partialParams));
    }
    enableTopicRule(partialParams) {
        return this.client.enableTopicRule(this.ops["EnableTopicRule"].apply(partialParams));
    }
    getBucketsAggregation(partialParams) {
        return this.client.getBucketsAggregation(this.ops["GetBucketsAggregation"].apply(partialParams));
    }
    getCardinality(partialParams) {
        return this.client.getCardinality(this.ops["GetCardinality"].apply(partialParams));
    }
    getJobDocument(partialParams) {
        return this.client.getJobDocument(this.ops["GetJobDocument"].apply(partialParams));
    }
    getOTAUpdate(partialParams) {
        return this.client.getOTAUpdate(this.ops["GetOTAUpdate"].apply(partialParams));
    }
    getPercentiles(partialParams) {
        return this.client.getPercentiles(this.ops["GetPercentiles"].apply(partialParams));
    }
    getPolicy(partialParams) {
        return this.client.getPolicy(this.ops["GetPolicy"].apply(partialParams));
    }
    getPolicyVersion(partialParams) {
        return this.client.getPolicyVersion(this.ops["GetPolicyVersion"].apply(partialParams));
    }
    getStatistics(partialParams) {
        return this.client.getStatistics(this.ops["GetStatistics"].apply(partialParams));
    }
    getTopicRule(partialParams) {
        return this.client.getTopicRule(this.ops["GetTopicRule"].apply(partialParams));
    }
    getTopicRuleDestination(partialParams) {
        return this.client.getTopicRuleDestination(this.ops["GetTopicRuleDestination"].apply(partialParams));
    }
    listAttachedPolicies(partialParams) {
        return this.client.listAttachedPolicies(this.ops["ListAttachedPolicies"].apply(partialParams));
    }
    listAuditMitigationActionsExecutions(partialParams) {
        return this.client.listAuditMitigationActionsExecutions(this.ops["ListAuditMitigationActionsExecutions"].apply(partialParams));
    }
    listAuditMitigationActionsTasks(partialParams) {
        return this.client.listAuditMitigationActionsTasks(this.ops["ListAuditMitigationActionsTasks"].apply(partialParams));
    }
    listAuditTasks(partialParams) {
        return this.client.listAuditTasks(this.ops["ListAuditTasks"].apply(partialParams));
    }
    listCertificatesByCA(partialParams) {
        return this.client.listCertificatesByCA(this.ops["ListCertificatesByCA"].apply(partialParams));
    }
    listDetectMitigationActionsTasks(partialParams) {
        return this.client.listDetectMitigationActionsTasks(this.ops["ListDetectMitigationActionsTasks"].apply(partialParams));
    }
    listJobExecutionsForJob(partialParams) {
        return this.client.listJobExecutionsForJob(this.ops["ListJobExecutionsForJob"].apply(partialParams));
    }
    listJobExecutionsForThing(partialParams) {
        return this.client.listJobExecutionsForThing(this.ops["ListJobExecutionsForThing"].apply(partialParams));
    }
    listPolicyPrincipals(partialParams) {
        return this.client.listPolicyPrincipals(this.ops["ListPolicyPrincipals"].apply(partialParams));
    }
    listPolicyVersions(partialParams) {
        return this.client.listPolicyVersions(this.ops["ListPolicyVersions"].apply(partialParams));
    }
    listPrincipalPolicies(partialParams) {
        return this.client.listPrincipalPolicies(this.ops["ListPrincipalPolicies"].apply(partialParams));
    }
    listPrincipalThings(partialParams) {
        return this.client.listPrincipalThings(this.ops["ListPrincipalThings"].apply(partialParams));
    }
    listProvisioningTemplateVersions(partialParams) {
        return this.client.listProvisioningTemplateVersions(this.ops["ListProvisioningTemplateVersions"].apply(partialParams));
    }
    listSecurityProfilesForTarget(partialParams) {
        return this.client.listSecurityProfilesForTarget(this.ops["ListSecurityProfilesForTarget"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTargetsForPolicy(partialParams) {
        return this.client.listTargetsForPolicy(this.ops["ListTargetsForPolicy"].apply(partialParams));
    }
    listTargetsForSecurityProfile(partialParams) {
        return this.client.listTargetsForSecurityProfile(this.ops["ListTargetsForSecurityProfile"].apply(partialParams));
    }
    listThingGroupsForThing(partialParams) {
        return this.client.listThingGroupsForThing(this.ops["ListThingGroupsForThing"].apply(partialParams));
    }
    listThingPrincipals(partialParams) {
        return this.client.listThingPrincipals(this.ops["ListThingPrincipals"].apply(partialParams));
    }
    listThingRegistrationTaskReports(partialParams) {
        return this.client.listThingRegistrationTaskReports(this.ops["ListThingRegistrationTaskReports"].apply(partialParams));
    }
    listThingsInBillingGroup(partialParams) {
        return this.client.listThingsInBillingGroup(this.ops["ListThingsInBillingGroup"].apply(partialParams));
    }
    listThingsInThingGroup(partialParams) {
        return this.client.listThingsInThingGroup(this.ops["ListThingsInThingGroup"].apply(partialParams));
    }
    listViolationEvents(partialParams) {
        return this.client.listViolationEvents(this.ops["ListViolationEvents"].apply(partialParams));
    }
    putVerificationStateOnViolation(partialParams) {
        return this.client.putVerificationStateOnViolation(this.ops["PutVerificationStateOnViolation"].apply(partialParams));
    }
    registerCACertificate(partialParams) {
        return this.client.registerCACertificate(this.ops["RegisterCACertificate"].apply(partialParams));
    }
    registerCertificate(partialParams) {
        return this.client.registerCertificate(this.ops["RegisterCertificate"].apply(partialParams));
    }
    registerCertificateWithoutCA(partialParams) {
        return this.client.registerCertificateWithoutCA(this.ops["RegisterCertificateWithoutCA"].apply(partialParams));
    }
    registerThing(partialParams) {
        return this.client.registerThing(this.ops["RegisterThing"].apply(partialParams));
    }
    rejectCertificateTransfer(partialParams) {
        return this.client.rejectCertificateTransfer(this.ops["RejectCertificateTransfer"].apply(partialParams));
    }
    replaceTopicRule(partialParams) {
        return this.client.replaceTopicRule(this.ops["ReplaceTopicRule"].apply(partialParams));
    }
    searchIndex(partialParams) {
        return this.client.searchIndex(this.ops["SearchIndex"].apply(partialParams));
    }
    setDefaultAuthorizer(partialParams) {
        return this.client.setDefaultAuthorizer(this.ops["SetDefaultAuthorizer"].apply(partialParams));
    }
    setDefaultPolicyVersion(partialParams) {
        return this.client.setDefaultPolicyVersion(this.ops["SetDefaultPolicyVersion"].apply(partialParams));
    }
    setLoggingOptions(partialParams) {
        return this.client.setLoggingOptions(this.ops["SetLoggingOptions"].apply(partialParams));
    }
    setV2LoggingLevel(partialParams) {
        return this.client.setV2LoggingLevel(this.ops["SetV2LoggingLevel"].apply(partialParams));
    }
    startAuditMitigationActionsTask(partialParams) {
        return this.client.startAuditMitigationActionsTask(this.ops["StartAuditMitigationActionsTask"].apply(partialParams));
    }
    startDetectMitigationActionsTask(partialParams) {
        return this.client.startDetectMitigationActionsTask(this.ops["StartDetectMitigationActionsTask"].apply(partialParams));
    }
    startOnDemandAuditTask(partialParams) {
        return this.client.startOnDemandAuditTask(this.ops["StartOnDemandAuditTask"].apply(partialParams));
    }
    startThingRegistrationTask(partialParams) {
        return this.client.startThingRegistrationTask(this.ops["StartThingRegistrationTask"].apply(partialParams));
    }
    stopThingRegistrationTask(partialParams) {
        return this.client.stopThingRegistrationTask(this.ops["StopThingRegistrationTask"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    testAuthorization(partialParams) {
        return this.client.testAuthorization(this.ops["TestAuthorization"].apply(partialParams));
    }
    testInvokeAuthorizer(partialParams) {
        return this.client.testInvokeAuthorizer(this.ops["TestInvokeAuthorizer"].apply(partialParams));
    }
    transferCertificate(partialParams) {
        return this.client.transferCertificate(this.ops["TransferCertificate"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAuditSuppression(partialParams) {
        return this.client.updateAuditSuppression(this.ops["UpdateAuditSuppression"].apply(partialParams));
    }
    updateAuthorizer(partialParams) {
        return this.client.updateAuthorizer(this.ops["UpdateAuthorizer"].apply(partialParams));
    }
    updateBillingGroup(partialParams) {
        return this.client.updateBillingGroup(this.ops["UpdateBillingGroup"].apply(partialParams));
    }
    updateCACertificate(partialParams) {
        return this.client.updateCACertificate(this.ops["UpdateCACertificate"].apply(partialParams));
    }
    updateCertificate(partialParams) {
        return this.client.updateCertificate(this.ops["UpdateCertificate"].apply(partialParams));
    }
    updateCustomMetric(partialParams) {
        return this.client.updateCustomMetric(this.ops["UpdateCustomMetric"].apply(partialParams));
    }
    updateDimension(partialParams) {
        return this.client.updateDimension(this.ops["UpdateDimension"].apply(partialParams));
    }
    updateDomainConfiguration(partialParams) {
        return this.client.updateDomainConfiguration(this.ops["UpdateDomainConfiguration"].apply(partialParams));
    }
    updateDynamicThingGroup(partialParams) {
        return this.client.updateDynamicThingGroup(this.ops["UpdateDynamicThingGroup"].apply(partialParams));
    }
    updateFleetMetric(partialParams) {
        return this.client.updateFleetMetric(this.ops["UpdateFleetMetric"].apply(partialParams));
    }
    updateJob(partialParams) {
        return this.client.updateJob(this.ops["UpdateJob"].apply(partialParams));
    }
    updateMitigationAction(partialParams) {
        return this.client.updateMitigationAction(this.ops["UpdateMitigationAction"].apply(partialParams));
    }
    updateProvisioningTemplate(partialParams) {
        return this.client.updateProvisioningTemplate(this.ops["UpdateProvisioningTemplate"].apply(partialParams));
    }
    updateRoleAlias(partialParams) {
        return this.client.updateRoleAlias(this.ops["UpdateRoleAlias"].apply(partialParams));
    }
    updateScheduledAudit(partialParams) {
        return this.client.updateScheduledAudit(this.ops["UpdateScheduledAudit"].apply(partialParams));
    }
    updateSecurityProfile(partialParams) {
        return this.client.updateSecurityProfile(this.ops["UpdateSecurityProfile"].apply(partialParams));
    }
    updateStream(partialParams) {
        return this.client.updateStream(this.ops["UpdateStream"].apply(partialParams));
    }
    updateThing(partialParams) {
        return this.client.updateThing(this.ops["UpdateThing"].apply(partialParams));
    }
    updateThingGroup(partialParams) {
        return this.client.updateThingGroup(this.ops["UpdateThingGroup"].apply(partialParams));
    }
    updateTopicRuleDestination(partialParams) {
        return this.client.updateTopicRuleDestination(this.ops["UpdateTopicRuleDestination"].apply(partialParams));
    }
    validateSecurityProfileBehaviors(partialParams) {
        return this.client.validateSecurityProfileBehaviors(this.ops["ValidateSecurityProfileBehaviors"].apply(partialParams));
    }
}
exports.default = Resource;
