// out/lambda/Alias.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.lambda.Alias {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Lambda();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/lambda-2015-03-31.normal.json"), this.client);
    }
    addLayerVersionPermission(partialParams) {
        return this.client.addLayerVersionPermission(this.ops["AddLayerVersionPermission"].apply(partialParams));
    }
    addPermission(partialParams) {
        return this.client.addPermission(this.ops["AddPermission"].apply(partialParams));
    }
    createAlias(partialParams) {
        return this.client.createAlias(this.ops["CreateAlias"].apply(partialParams));
    }
    createCodeSigningConfig(partialParams) {
        return this.client.createCodeSigningConfig(this.ops["CreateCodeSigningConfig"].apply(partialParams));
    }
    createEventSourceMapping(partialParams) {
        return this.client.createEventSourceMapping(this.ops["CreateEventSourceMapping"].apply(partialParams));
    }
    createFunction(partialParams) {
        return this.client.createFunction(this.ops["CreateFunction"].apply(partialParams));
    }
    deleteAlias(partialParams) {
        return this.client.deleteAlias(this.ops["DeleteAlias"].apply(partialParams));
    }
    deleteCodeSigningConfig(partialParams) {
        return this.client.deleteCodeSigningConfig(this.ops["DeleteCodeSigningConfig"].apply(partialParams));
    }
    deleteEventSourceMapping(partialParams) {
        return this.client.deleteEventSourceMapping(this.ops["DeleteEventSourceMapping"].apply(partialParams));
    }
    deleteFunction(partialParams) {
        return this.client.deleteFunction(this.ops["DeleteFunction"].apply(partialParams));
    }
    deleteFunctionCodeSigningConfig(partialParams) {
        return this.client.deleteFunctionCodeSigningConfig(this.ops["DeleteFunctionCodeSigningConfig"].apply(partialParams));
    }
    deleteFunctionConcurrency(partialParams) {
        return this.client.deleteFunctionConcurrency(this.ops["DeleteFunctionConcurrency"].apply(partialParams));
    }
    deleteFunctionEventInvokeConfig(partialParams) {
        return this.client.deleteFunctionEventInvokeConfig(this.ops["DeleteFunctionEventInvokeConfig"].apply(partialParams));
    }
    deleteLayerVersion(partialParams) {
        return this.client.deleteLayerVersion(this.ops["DeleteLayerVersion"].apply(partialParams));
    }
    deleteProvisionedConcurrencyConfig(partialParams) {
        return this.client.deleteProvisionedConcurrencyConfig(this.ops["DeleteProvisionedConcurrencyConfig"].apply(partialParams));
    }
    getAlias(partialParams) {
        return this.client.getAlias(this.ops["GetAlias"].apply(partialParams));
    }
    getCodeSigningConfig(partialParams) {
        return this.client.getCodeSigningConfig(this.ops["GetCodeSigningConfig"].apply(partialParams));
    }
    getEventSourceMapping(partialParams) {
        return this.client.getEventSourceMapping(this.ops["GetEventSourceMapping"].apply(partialParams));
    }
    getFunction(partialParams) {
        return this.client.getFunction(this.ops["GetFunction"].apply(partialParams));
    }
    getFunctionCodeSigningConfig(partialParams) {
        return this.client.getFunctionCodeSigningConfig(this.ops["GetFunctionCodeSigningConfig"].apply(partialParams));
    }
    getFunctionConcurrency(partialParams) {
        return this.client.getFunctionConcurrency(this.ops["GetFunctionConcurrency"].apply(partialParams));
    }
    getFunctionConfiguration(partialParams) {
        return this.client.getFunctionConfiguration(this.ops["GetFunctionConfiguration"].apply(partialParams));
    }
    getFunctionEventInvokeConfig(partialParams) {
        return this.client.getFunctionEventInvokeConfig(this.ops["GetFunctionEventInvokeConfig"].apply(partialParams));
    }
    getLayerVersion(partialParams) {
        return this.client.getLayerVersion(this.ops["GetLayerVersion"].apply(partialParams));
    }
    getLayerVersionByArn(partialParams) {
        return this.client.getLayerVersionByArn(this.ops["GetLayerVersionByArn"].apply(partialParams));
    }
    getLayerVersionPolicy(partialParams) {
        return this.client.getLayerVersionPolicy(this.ops["GetLayerVersionPolicy"].apply(partialParams));
    }
    getPolicy(partialParams) {
        return this.client.getPolicy(this.ops["GetPolicy"].apply(partialParams));
    }
    getProvisionedConcurrencyConfig(partialParams) {
        return this.client.getProvisionedConcurrencyConfig(this.ops["GetProvisionedConcurrencyConfig"].apply(partialParams));
    }
    invoke(partialParams) {
        return this.client.invoke(this.ops["Invoke"].apply(partialParams));
    }
    invokeAsync(partialParams) {
        return this.client.invokeAsync(this.ops["InvokeAsync"].apply(partialParams));
    }
    listAliases(partialParams) {
        return this.client.listAliases(this.ops["ListAliases"].apply(partialParams));
    }
    listFunctionEventInvokeConfigs(partialParams) {
        return this.client.listFunctionEventInvokeConfigs(this.ops["ListFunctionEventInvokeConfigs"].apply(partialParams));
    }
    listFunctionsByCodeSigningConfig(partialParams) {
        return this.client.listFunctionsByCodeSigningConfig(this.ops["ListFunctionsByCodeSigningConfig"].apply(partialParams));
    }
    listLayerVersions(partialParams) {
        return this.client.listLayerVersions(this.ops["ListLayerVersions"].apply(partialParams));
    }
    listProvisionedConcurrencyConfigs(partialParams) {
        return this.client.listProvisionedConcurrencyConfigs(this.ops["ListProvisionedConcurrencyConfigs"].apply(partialParams));
    }
    listTags(partialParams) {
        return this.client.listTags(this.ops["ListTags"].apply(partialParams));
    }
    listVersionsByFunction(partialParams) {
        return this.client.listVersionsByFunction(this.ops["ListVersionsByFunction"].apply(partialParams));
    }
    publishLayerVersion(partialParams) {
        return this.client.publishLayerVersion(this.ops["PublishLayerVersion"].apply(partialParams));
    }
    publishVersion(partialParams) {
        return this.client.publishVersion(this.ops["PublishVersion"].apply(partialParams));
    }
    putFunctionCodeSigningConfig(partialParams) {
        return this.client.putFunctionCodeSigningConfig(this.ops["PutFunctionCodeSigningConfig"].apply(partialParams));
    }
    putFunctionConcurrency(partialParams) {
        return this.client.putFunctionConcurrency(this.ops["PutFunctionConcurrency"].apply(partialParams));
    }
    putFunctionEventInvokeConfig(partialParams) {
        return this.client.putFunctionEventInvokeConfig(this.ops["PutFunctionEventInvokeConfig"].apply(partialParams));
    }
    putProvisionedConcurrencyConfig(partialParams) {
        return this.client.putProvisionedConcurrencyConfig(this.ops["PutProvisionedConcurrencyConfig"].apply(partialParams));
    }
    removeLayerVersionPermission(partialParams) {
        return this.client.removeLayerVersionPermission(this.ops["RemoveLayerVersionPermission"].apply(partialParams));
    }
    removePermission(partialParams) {
        return this.client.removePermission(this.ops["RemovePermission"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAlias(partialParams) {
        return this.client.updateAlias(this.ops["UpdateAlias"].apply(partialParams));
    }
    updateCodeSigningConfig(partialParams) {
        return this.client.updateCodeSigningConfig(this.ops["UpdateCodeSigningConfig"].apply(partialParams));
    }
    updateEventSourceMapping(partialParams) {
        return this.client.updateEventSourceMapping(this.ops["UpdateEventSourceMapping"].apply(partialParams));
    }
    updateFunctionCode(partialParams) {
        return this.client.updateFunctionCode(this.ops["UpdateFunctionCode"].apply(partialParams));
    }
    updateFunctionConfiguration(partialParams) {
        return this.client.updateFunctionConfiguration(this.ops["UpdateFunctionConfiguration"].apply(partialParams));
    }
    updateFunctionEventInvokeConfig(partialParams) {
        return this.client.updateFunctionEventInvokeConfig(this.ops["UpdateFunctionEventInvokeConfig"].apply(partialParams));
    }
}
exports.default = Resource;
