// out/autoscaling/Attachment.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.autoscaling.Attachment {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.AutoScaling();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/autoscaling-2011-01-01.normal.json"), this.client);
    }
    attachInstances(partialParams) {
        return this.client.attachInstances(this.ops["AttachInstances"].apply(partialParams));
    }
    attachLoadBalancerTargetGroups(partialParams) {
        return this.client.attachLoadBalancerTargetGroups(this.ops["AttachLoadBalancerTargetGroups"].apply(partialParams));
    }
    attachLoadBalancers(partialParams) {
        return this.client.attachLoadBalancers(this.ops["AttachLoadBalancers"].apply(partialParams));
    }
    batchDeleteScheduledAction(partialParams) {
        return this.client.batchDeleteScheduledAction(this.ops["BatchDeleteScheduledAction"].apply(partialParams));
    }
    batchPutScheduledUpdateGroupAction(partialParams) {
        return this.client.batchPutScheduledUpdateGroupAction(this.ops["BatchPutScheduledUpdateGroupAction"].apply(partialParams));
    }
    cancelInstanceRefresh(partialParams) {
        return this.client.cancelInstanceRefresh(this.ops["CancelInstanceRefresh"].apply(partialParams));
    }
    completeLifecycleAction(partialParams) {
        return this.client.completeLifecycleAction(this.ops["CompleteLifecycleAction"].apply(partialParams));
    }
    createAutoScalingGroup(partialParams) {
        return this.client.createAutoScalingGroup(this.ops["CreateAutoScalingGroup"].apply(partialParams));
    }
    createLaunchConfiguration(partialParams) {
        return this.client.createLaunchConfiguration(this.ops["CreateLaunchConfiguration"].apply(partialParams));
    }
    createOrUpdateTags(partialParams) {
        return this.client.createOrUpdateTags(this.ops["CreateOrUpdateTags"].apply(partialParams));
    }
    deleteAutoScalingGroup(partialParams) {
        return this.client.deleteAutoScalingGroup(this.ops["DeleteAutoScalingGroup"].apply(partialParams));
    }
    deleteLaunchConfiguration(partialParams) {
        return this.client.deleteLaunchConfiguration(this.ops["DeleteLaunchConfiguration"].apply(partialParams));
    }
    deleteLifecycleHook(partialParams) {
        return this.client.deleteLifecycleHook(this.ops["DeleteLifecycleHook"].apply(partialParams));
    }
    deleteNotificationConfiguration(partialParams) {
        return this.client.deleteNotificationConfiguration(this.ops["DeleteNotificationConfiguration"].apply(partialParams));
    }
    deletePolicy(partialParams) {
        return this.client.deletePolicy(this.ops["DeletePolicy"].apply(partialParams));
    }
    deleteScheduledAction(partialParams) {
        return this.client.deleteScheduledAction(this.ops["DeleteScheduledAction"].apply(partialParams));
    }
    deleteTags(partialParams) {
        return this.client.deleteTags(this.ops["DeleteTags"].apply(partialParams));
    }
    deleteWarmPool(partialParams) {
        return this.client.deleteWarmPool(this.ops["DeleteWarmPool"].apply(partialParams));
    }
    describeInstanceRefreshes(partialParams) {
        return this.client.describeInstanceRefreshes(this.ops["DescribeInstanceRefreshes"].apply(partialParams));
    }
    describeLifecycleHooks(partialParams) {
        return this.client.describeLifecycleHooks(this.ops["DescribeLifecycleHooks"].apply(partialParams));
    }
    describeLoadBalancerTargetGroups(partialParams) {
        return this.client.describeLoadBalancerTargetGroups(this.ops["DescribeLoadBalancerTargetGroups"].apply(partialParams));
    }
    describeLoadBalancers(partialParams) {
        return this.client.describeLoadBalancers(this.ops["DescribeLoadBalancers"].apply(partialParams));
    }
    describeWarmPool(partialParams) {
        return this.client.describeWarmPool(this.ops["DescribeWarmPool"].apply(partialParams));
    }
    detachInstances(partialParams) {
        return this.client.detachInstances(this.ops["DetachInstances"].apply(partialParams));
    }
    detachLoadBalancerTargetGroups(partialParams) {
        return this.client.detachLoadBalancerTargetGroups(this.ops["DetachLoadBalancerTargetGroups"].apply(partialParams));
    }
    detachLoadBalancers(partialParams) {
        return this.client.detachLoadBalancers(this.ops["DetachLoadBalancers"].apply(partialParams));
    }
    disableMetricsCollection(partialParams) {
        return this.client.disableMetricsCollection(this.ops["DisableMetricsCollection"].apply(partialParams));
    }
    enableMetricsCollection(partialParams) {
        return this.client.enableMetricsCollection(this.ops["EnableMetricsCollection"].apply(partialParams));
    }
    enterStandby(partialParams) {
        return this.client.enterStandby(this.ops["EnterStandby"].apply(partialParams));
    }
    executePolicy(partialParams) {
        return this.client.executePolicy(this.ops["ExecutePolicy"].apply(partialParams));
    }
    exitStandby(partialParams) {
        return this.client.exitStandby(this.ops["ExitStandby"].apply(partialParams));
    }
    getPredictiveScalingForecast(partialParams) {
        return this.client.getPredictiveScalingForecast(this.ops["GetPredictiveScalingForecast"].apply(partialParams));
    }
    putLifecycleHook(partialParams) {
        return this.client.putLifecycleHook(this.ops["PutLifecycleHook"].apply(partialParams));
    }
    putNotificationConfiguration(partialParams) {
        return this.client.putNotificationConfiguration(this.ops["PutNotificationConfiguration"].apply(partialParams));
    }
    putScalingPolicy(partialParams) {
        return this.client.putScalingPolicy(this.ops["PutScalingPolicy"].apply(partialParams));
    }
    putScheduledUpdateGroupAction(partialParams) {
        return this.client.putScheduledUpdateGroupAction(this.ops["PutScheduledUpdateGroupAction"].apply(partialParams));
    }
    putWarmPool(partialParams) {
        return this.client.putWarmPool(this.ops["PutWarmPool"].apply(partialParams));
    }
    recordLifecycleActionHeartbeat(partialParams) {
        return this.client.recordLifecycleActionHeartbeat(this.ops["RecordLifecycleActionHeartbeat"].apply(partialParams));
    }
    resumeProcesses(partialParams) {
        return this.client.resumeProcesses(this.ops["ResumeProcesses"].apply(partialParams));
    }
    setDesiredCapacity(partialParams) {
        return this.client.setDesiredCapacity(this.ops["SetDesiredCapacity"].apply(partialParams));
    }
    setInstanceHealth(partialParams) {
        return this.client.setInstanceHealth(this.ops["SetInstanceHealth"].apply(partialParams));
    }
    setInstanceProtection(partialParams) {
        return this.client.setInstanceProtection(this.ops["SetInstanceProtection"].apply(partialParams));
    }
    startInstanceRefresh(partialParams) {
        return this.client.startInstanceRefresh(this.ops["StartInstanceRefresh"].apply(partialParams));
    }
    suspendProcesses(partialParams) {
        return this.client.suspendProcesses(this.ops["SuspendProcesses"].apply(partialParams));
    }
    terminateInstanceInAutoScalingGroup(partialParams) {
        return this.client.terminateInstanceInAutoScalingGroup(this.ops["TerminateInstanceInAutoScalingGroup"].apply(partialParams));
    }
    updateAutoScalingGroup(partialParams) {
        return this.client.updateAutoScalingGroup(this.ops["UpdateAutoScalingGroup"].apply(partialParams));
    }
}
exports.default = Resource;
