// out/inspector/AssessmentTarget.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.inspector.AssessmentTarget {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Inspector();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/inspector-2016-02-16.normal.json"), this.client);
    }
    addAttributesToFindings(partialParams) {
        return this.client.addAttributesToFindings(this.ops["AddAttributesToFindings"].apply(partialParams));
    }
    createAssessmentTarget(partialParams) {
        return this.client.createAssessmentTarget(this.ops["CreateAssessmentTarget"].apply(partialParams));
    }
    createAssessmentTemplate(partialParams) {
        return this.client.createAssessmentTemplate(this.ops["CreateAssessmentTemplate"].apply(partialParams));
    }
    createExclusionsPreview(partialParams) {
        return this.client.createExclusionsPreview(this.ops["CreateExclusionsPreview"].apply(partialParams));
    }
    createResourceGroup(partialParams) {
        return this.client.createResourceGroup(this.ops["CreateResourceGroup"].apply(partialParams));
    }
    deleteAssessmentRun(partialParams) {
        return this.client.deleteAssessmentRun(this.ops["DeleteAssessmentRun"].apply(partialParams));
    }
    deleteAssessmentTarget(partialParams) {
        return this.client.deleteAssessmentTarget(this.ops["DeleteAssessmentTarget"].apply(partialParams));
    }
    deleteAssessmentTemplate(partialParams) {
        return this.client.deleteAssessmentTemplate(this.ops["DeleteAssessmentTemplate"].apply(partialParams));
    }
    describeAssessmentRuns(partialParams) {
        return this.client.describeAssessmentRuns(this.ops["DescribeAssessmentRuns"].apply(partialParams));
    }
    describeAssessmentTargets(partialParams) {
        return this.client.describeAssessmentTargets(this.ops["DescribeAssessmentTargets"].apply(partialParams));
    }
    describeAssessmentTemplates(partialParams) {
        return this.client.describeAssessmentTemplates(this.ops["DescribeAssessmentTemplates"].apply(partialParams));
    }
    describeExclusions(partialParams) {
        return this.client.describeExclusions(this.ops["DescribeExclusions"].apply(partialParams));
    }
    describeFindings(partialParams) {
        return this.client.describeFindings(this.ops["DescribeFindings"].apply(partialParams));
    }
    describeResourceGroups(partialParams) {
        return this.client.describeResourceGroups(this.ops["DescribeResourceGroups"].apply(partialParams));
    }
    describeRulesPackages(partialParams) {
        return this.client.describeRulesPackages(this.ops["DescribeRulesPackages"].apply(partialParams));
    }
    getAssessmentReport(partialParams) {
        return this.client.getAssessmentReport(this.ops["GetAssessmentReport"].apply(partialParams));
    }
    getExclusionsPreview(partialParams) {
        return this.client.getExclusionsPreview(this.ops["GetExclusionsPreview"].apply(partialParams));
    }
    getTelemetryMetadata(partialParams) {
        return this.client.getTelemetryMetadata(this.ops["GetTelemetryMetadata"].apply(partialParams));
    }
    listAssessmentRunAgents(partialParams) {
        return this.client.listAssessmentRunAgents(this.ops["ListAssessmentRunAgents"].apply(partialParams));
    }
    listExclusions(partialParams) {
        return this.client.listExclusions(this.ops["ListExclusions"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    previewAgents(partialParams) {
        return this.client.previewAgents(this.ops["PreviewAgents"].apply(partialParams));
    }
    registerCrossAccountAccessRole(partialParams) {
        return this.client.registerCrossAccountAccessRole(this.ops["RegisterCrossAccountAccessRole"].apply(partialParams));
    }
    removeAttributesFromFindings(partialParams) {
        return this.client.removeAttributesFromFindings(this.ops["RemoveAttributesFromFindings"].apply(partialParams));
    }
    setTagsForResource(partialParams) {
        return this.client.setTagsForResource(this.ops["SetTagsForResource"].apply(partialParams));
    }
    startAssessmentRun(partialParams) {
        return this.client.startAssessmentRun(this.ops["StartAssessmentRun"].apply(partialParams));
    }
    stopAssessmentRun(partialParams) {
        return this.client.stopAssessmentRun(this.ops["StopAssessmentRun"].apply(partialParams));
    }
    subscribeToEvent(partialParams) {
        return this.client.subscribeToEvent(this.ops["SubscribeToEvent"].apply(partialParams));
    }
    unsubscribeFromEvent(partialParams) {
        return this.client.unsubscribeFromEvent(this.ops["UnsubscribeFromEvent"].apply(partialParams));
    }
    updateAssessmentTarget(partialParams) {
        return this.client.updateAssessmentTarget(this.ops["UpdateAssessmentTarget"].apply(partialParams));
    }
}
exports.default = Resource;
