// out/devicefarm/Project.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.devicefarm.Project {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.DeviceFarm();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/devicefarm-2015-06-23.normal.json"), this.client);
    }
    createDevicePool(partialParams) {
        return this.client.createDevicePool(this.ops["CreateDevicePool"].apply(partialParams));
    }
    createInstanceProfile(partialParams) {
        return this.client.createInstanceProfile(this.ops["CreateInstanceProfile"].apply(partialParams));
    }
    createNetworkProfile(partialParams) {
        return this.client.createNetworkProfile(this.ops["CreateNetworkProfile"].apply(partialParams));
    }
    createProject(partialParams) {
        return this.client.createProject(this.ops["CreateProject"].apply(partialParams));
    }
    createRemoteAccessSession(partialParams) {
        return this.client.createRemoteAccessSession(this.ops["CreateRemoteAccessSession"].apply(partialParams));
    }
    createTestGridProject(partialParams) {
        return this.client.createTestGridProject(this.ops["CreateTestGridProject"].apply(partialParams));
    }
    createTestGridUrl(partialParams) {
        return this.client.createTestGridUrl(this.ops["CreateTestGridUrl"].apply(partialParams));
    }
    createUpload(partialParams) {
        return this.client.createUpload(this.ops["CreateUpload"].apply(partialParams));
    }
    createVPCEConfiguration(partialParams) {
        return this.client.createVPCEConfiguration(this.ops["CreateVPCEConfiguration"].apply(partialParams));
    }
    deleteDevicePool(partialParams) {
        return this.client.deleteDevicePool(this.ops["DeleteDevicePool"].apply(partialParams));
    }
    deleteInstanceProfile(partialParams) {
        return this.client.deleteInstanceProfile(this.ops["DeleteInstanceProfile"].apply(partialParams));
    }
    deleteNetworkProfile(partialParams) {
        return this.client.deleteNetworkProfile(this.ops["DeleteNetworkProfile"].apply(partialParams));
    }
    deleteProject(partialParams) {
        return this.client.deleteProject(this.ops["DeleteProject"].apply(partialParams));
    }
    deleteRemoteAccessSession(partialParams) {
        return this.client.deleteRemoteAccessSession(this.ops["DeleteRemoteAccessSession"].apply(partialParams));
    }
    deleteRun(partialParams) {
        return this.client.deleteRun(this.ops["DeleteRun"].apply(partialParams));
    }
    deleteTestGridProject(partialParams) {
        return this.client.deleteTestGridProject(this.ops["DeleteTestGridProject"].apply(partialParams));
    }
    deleteUpload(partialParams) {
        return this.client.deleteUpload(this.ops["DeleteUpload"].apply(partialParams));
    }
    deleteVPCEConfiguration(partialParams) {
        return this.client.deleteVPCEConfiguration(this.ops["DeleteVPCEConfiguration"].apply(partialParams));
    }
    getDevice(partialParams) {
        return this.client.getDevice(this.ops["GetDevice"].apply(partialParams));
    }
    getDeviceInstance(partialParams) {
        return this.client.getDeviceInstance(this.ops["GetDeviceInstance"].apply(partialParams));
    }
    getDevicePool(partialParams) {
        return this.client.getDevicePool(this.ops["GetDevicePool"].apply(partialParams));
    }
    getDevicePoolCompatibility(partialParams) {
        return this.client.getDevicePoolCompatibility(this.ops["GetDevicePoolCompatibility"].apply(partialParams));
    }
    getInstanceProfile(partialParams) {
        return this.client.getInstanceProfile(this.ops["GetInstanceProfile"].apply(partialParams));
    }
    getJob(partialParams) {
        return this.client.getJob(this.ops["GetJob"].apply(partialParams));
    }
    getNetworkProfile(partialParams) {
        return this.client.getNetworkProfile(this.ops["GetNetworkProfile"].apply(partialParams));
    }
    getProject(partialParams) {
        return this.client.getProject(this.ops["GetProject"].apply(partialParams));
    }
    getRemoteAccessSession(partialParams) {
        return this.client.getRemoteAccessSession(this.ops["GetRemoteAccessSession"].apply(partialParams));
    }
    getRun(partialParams) {
        return this.client.getRun(this.ops["GetRun"].apply(partialParams));
    }
    getSuite(partialParams) {
        return this.client.getSuite(this.ops["GetSuite"].apply(partialParams));
    }
    getTest(partialParams) {
        return this.client.getTest(this.ops["GetTest"].apply(partialParams));
    }
    getTestGridProject(partialParams) {
        return this.client.getTestGridProject(this.ops["GetTestGridProject"].apply(partialParams));
    }
    getUpload(partialParams) {
        return this.client.getUpload(this.ops["GetUpload"].apply(partialParams));
    }
    getVPCEConfiguration(partialParams) {
        return this.client.getVPCEConfiguration(this.ops["GetVPCEConfiguration"].apply(partialParams));
    }
    installToRemoteAccessSession(partialParams) {
        return this.client.installToRemoteAccessSession(this.ops["InstallToRemoteAccessSession"].apply(partialParams));
    }
    listArtifacts(partialParams) {
        return this.client.listArtifacts(this.ops["ListArtifacts"].apply(partialParams));
    }
    listDevicePools(partialParams) {
        return this.client.listDevicePools(this.ops["ListDevicePools"].apply(partialParams));
    }
    listJobs(partialParams) {
        return this.client.listJobs(this.ops["ListJobs"].apply(partialParams));
    }
    listNetworkProfiles(partialParams) {
        return this.client.listNetworkProfiles(this.ops["ListNetworkProfiles"].apply(partialParams));
    }
    listRemoteAccessSessions(partialParams) {
        return this.client.listRemoteAccessSessions(this.ops["ListRemoteAccessSessions"].apply(partialParams));
    }
    listRuns(partialParams) {
        return this.client.listRuns(this.ops["ListRuns"].apply(partialParams));
    }
    listSamples(partialParams) {
        return this.client.listSamples(this.ops["ListSamples"].apply(partialParams));
    }
    listSuites(partialParams) {
        return this.client.listSuites(this.ops["ListSuites"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTestGridSessionActions(partialParams) {
        return this.client.listTestGridSessionActions(this.ops["ListTestGridSessionActions"].apply(partialParams));
    }
    listTestGridSessionArtifacts(partialParams) {
        return this.client.listTestGridSessionArtifacts(this.ops["ListTestGridSessionArtifacts"].apply(partialParams));
    }
    listTestGridSessions(partialParams) {
        return this.client.listTestGridSessions(this.ops["ListTestGridSessions"].apply(partialParams));
    }
    listTests(partialParams) {
        return this.client.listTests(this.ops["ListTests"].apply(partialParams));
    }
    listUniqueProblems(partialParams) {
        return this.client.listUniqueProblems(this.ops["ListUniqueProblems"].apply(partialParams));
    }
    listUploads(partialParams) {
        return this.client.listUploads(this.ops["ListUploads"].apply(partialParams));
    }
    purchaseOffering(partialParams) {
        return this.client.purchaseOffering(this.ops["PurchaseOffering"].apply(partialParams));
    }
    renewOffering(partialParams) {
        return this.client.renewOffering(this.ops["RenewOffering"].apply(partialParams));
    }
    scheduleRun(partialParams) {
        return this.client.scheduleRun(this.ops["ScheduleRun"].apply(partialParams));
    }
    stopJob(partialParams) {
        return this.client.stopJob(this.ops["StopJob"].apply(partialParams));
    }
    stopRemoteAccessSession(partialParams) {
        return this.client.stopRemoteAccessSession(this.ops["StopRemoteAccessSession"].apply(partialParams));
    }
    stopRun(partialParams) {
        return this.client.stopRun(this.ops["StopRun"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateDeviceInstance(partialParams) {
        return this.client.updateDeviceInstance(this.ops["UpdateDeviceInstance"].apply(partialParams));
    }
    updateDevicePool(partialParams) {
        return this.client.updateDevicePool(this.ops["UpdateDevicePool"].apply(partialParams));
    }
    updateInstanceProfile(partialParams) {
        return this.client.updateInstanceProfile(this.ops["UpdateInstanceProfile"].apply(partialParams));
    }
    updateNetworkProfile(partialParams) {
        return this.client.updateNetworkProfile(this.ops["UpdateNetworkProfile"].apply(partialParams));
    }
    updateProject(partialParams) {
        return this.client.updateProject(this.ops["UpdateProject"].apply(partialParams));
    }
    updateTestGridProject(partialParams) {
        return this.client.updateTestGridProject(this.ops["UpdateTestGridProject"].apply(partialParams));
    }
    updateUpload(partialParams) {
        return this.client.updateUpload(this.ops["UpdateUpload"].apply(partialParams));
    }
    updateVPCEConfiguration(partialParams) {
        return this.client.updateVPCEConfiguration(this.ops["UpdateVPCEConfiguration"].apply(partialParams));
    }
}
exports.default = Resource;
