// out/glue/Registry.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.glue.Registry {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Glue();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/glue-2017-03-31.normal.json"), this.client);
    }
    batchCreatePartition(partialParams) {
        return this.client.batchCreatePartition(this.ops["BatchCreatePartition"].apply(partialParams));
    }
    batchDeleteConnection(partialParams) {
        return this.client.batchDeleteConnection(this.ops["BatchDeleteConnection"].apply(partialParams));
    }
    batchDeletePartition(partialParams) {
        return this.client.batchDeletePartition(this.ops["BatchDeletePartition"].apply(partialParams));
    }
    batchDeleteTable(partialParams) {
        return this.client.batchDeleteTable(this.ops["BatchDeleteTable"].apply(partialParams));
    }
    batchDeleteTableVersion(partialParams) {
        return this.client.batchDeleteTableVersion(this.ops["BatchDeleteTableVersion"].apply(partialParams));
    }
    batchGetBlueprints(partialParams) {
        return this.client.batchGetBlueprints(this.ops["BatchGetBlueprints"].apply(partialParams));
    }
    batchGetCrawlers(partialParams) {
        return this.client.batchGetCrawlers(this.ops["BatchGetCrawlers"].apply(partialParams));
    }
    batchGetDevEndpoints(partialParams) {
        return this.client.batchGetDevEndpoints(this.ops["BatchGetDevEndpoints"].apply(partialParams));
    }
    batchGetJobs(partialParams) {
        return this.client.batchGetJobs(this.ops["BatchGetJobs"].apply(partialParams));
    }
    batchGetPartition(partialParams) {
        return this.client.batchGetPartition(this.ops["BatchGetPartition"].apply(partialParams));
    }
    batchGetTriggers(partialParams) {
        return this.client.batchGetTriggers(this.ops["BatchGetTriggers"].apply(partialParams));
    }
    batchGetWorkflows(partialParams) {
        return this.client.batchGetWorkflows(this.ops["BatchGetWorkflows"].apply(partialParams));
    }
    batchStopJobRun(partialParams) {
        return this.client.batchStopJobRun(this.ops["BatchStopJobRun"].apply(partialParams));
    }
    batchUpdatePartition(partialParams) {
        return this.client.batchUpdatePartition(this.ops["BatchUpdatePartition"].apply(partialParams));
    }
    cancelMLTaskRun(partialParams) {
        return this.client.cancelMLTaskRun(this.ops["CancelMLTaskRun"].apply(partialParams));
    }
    checkSchemaVersionValidity(partialParams) {
        return this.client.checkSchemaVersionValidity(this.ops["CheckSchemaVersionValidity"].apply(partialParams));
    }
    createBlueprint(partialParams) {
        return this.client.createBlueprint(this.ops["CreateBlueprint"].apply(partialParams));
    }
    createConnection(partialParams) {
        return this.client.createConnection(this.ops["CreateConnection"].apply(partialParams));
    }
    createCrawler(partialParams) {
        return this.client.createCrawler(this.ops["CreateCrawler"].apply(partialParams));
    }
    createDatabase(partialParams) {
        return this.client.createDatabase(this.ops["CreateDatabase"].apply(partialParams));
    }
    createDevEndpoint(partialParams) {
        return this.client.createDevEndpoint(this.ops["CreateDevEndpoint"].apply(partialParams));
    }
    createJob(partialParams) {
        return this.client.createJob(this.ops["CreateJob"].apply(partialParams));
    }
    createMLTransform(partialParams) {
        return this.client.createMLTransform(this.ops["CreateMLTransform"].apply(partialParams));
    }
    createPartition(partialParams) {
        return this.client.createPartition(this.ops["CreatePartition"].apply(partialParams));
    }
    createPartitionIndex(partialParams) {
        return this.client.createPartitionIndex(this.ops["CreatePartitionIndex"].apply(partialParams));
    }
    createRegistry(partialParams) {
        return this.client.createRegistry(this.ops["CreateRegistry"].apply(partialParams));
    }
    createSchema(partialParams) {
        return this.client.createSchema(this.ops["CreateSchema"].apply(partialParams));
    }
    createSecurityConfiguration(partialParams) {
        return this.client.createSecurityConfiguration(this.ops["CreateSecurityConfiguration"].apply(partialParams));
    }
    createTable(partialParams) {
        return this.client.createTable(this.ops["CreateTable"].apply(partialParams));
    }
    createTrigger(partialParams) {
        return this.client.createTrigger(this.ops["CreateTrigger"].apply(partialParams));
    }
    createUserDefinedFunction(partialParams) {
        return this.client.createUserDefinedFunction(this.ops["CreateUserDefinedFunction"].apply(partialParams));
    }
    createWorkflow(partialParams) {
        return this.client.createWorkflow(this.ops["CreateWorkflow"].apply(partialParams));
    }
    deleteBlueprint(partialParams) {
        return this.client.deleteBlueprint(this.ops["DeleteBlueprint"].apply(partialParams));
    }
    deleteClassifier(partialParams) {
        return this.client.deleteClassifier(this.ops["DeleteClassifier"].apply(partialParams));
    }
    deleteColumnStatisticsForPartition(partialParams) {
        return this.client.deleteColumnStatisticsForPartition(this.ops["DeleteColumnStatisticsForPartition"].apply(partialParams));
    }
    deleteColumnStatisticsForTable(partialParams) {
        return this.client.deleteColumnStatisticsForTable(this.ops["DeleteColumnStatisticsForTable"].apply(partialParams));
    }
    deleteConnection(partialParams) {
        return this.client.deleteConnection(this.ops["DeleteConnection"].apply(partialParams));
    }
    deleteCrawler(partialParams) {
        return this.client.deleteCrawler(this.ops["DeleteCrawler"].apply(partialParams));
    }
    deleteDatabase(partialParams) {
        return this.client.deleteDatabase(this.ops["DeleteDatabase"].apply(partialParams));
    }
    deleteDevEndpoint(partialParams) {
        return this.client.deleteDevEndpoint(this.ops["DeleteDevEndpoint"].apply(partialParams));
    }
    deleteJob(partialParams) {
        return this.client.deleteJob(this.ops["DeleteJob"].apply(partialParams));
    }
    deleteMLTransform(partialParams) {
        return this.client.deleteMLTransform(this.ops["DeleteMLTransform"].apply(partialParams));
    }
    deletePartition(partialParams) {
        return this.client.deletePartition(this.ops["DeletePartition"].apply(partialParams));
    }
    deletePartitionIndex(partialParams) {
        return this.client.deletePartitionIndex(this.ops["DeletePartitionIndex"].apply(partialParams));
    }
    deleteRegistry(partialParams) {
        return this.client.deleteRegistry(this.ops["DeleteRegistry"].apply(partialParams));
    }
    deleteSchema(partialParams) {
        return this.client.deleteSchema(this.ops["DeleteSchema"].apply(partialParams));
    }
    deleteSchemaVersions(partialParams) {
        return this.client.deleteSchemaVersions(this.ops["DeleteSchemaVersions"].apply(partialParams));
    }
    deleteSecurityConfiguration(partialParams) {
        return this.client.deleteSecurityConfiguration(this.ops["DeleteSecurityConfiguration"].apply(partialParams));
    }
    deleteTable(partialParams) {
        return this.client.deleteTable(this.ops["DeleteTable"].apply(partialParams));
    }
    deleteTableVersion(partialParams) {
        return this.client.deleteTableVersion(this.ops["DeleteTableVersion"].apply(partialParams));
    }
    deleteTrigger(partialParams) {
        return this.client.deleteTrigger(this.ops["DeleteTrigger"].apply(partialParams));
    }
    deleteUserDefinedFunction(partialParams) {
        return this.client.deleteUserDefinedFunction(this.ops["DeleteUserDefinedFunction"].apply(partialParams));
    }
    deleteWorkflow(partialParams) {
        return this.client.deleteWorkflow(this.ops["DeleteWorkflow"].apply(partialParams));
    }
    getBlueprint(partialParams) {
        return this.client.getBlueprint(this.ops["GetBlueprint"].apply(partialParams));
    }
    getBlueprintRun(partialParams) {
        return this.client.getBlueprintRun(this.ops["GetBlueprintRun"].apply(partialParams));
    }
    getBlueprintRuns(partialParams) {
        return this.client.getBlueprintRuns(this.ops["GetBlueprintRuns"].apply(partialParams));
    }
    getClassifier(partialParams) {
        return this.client.getClassifier(this.ops["GetClassifier"].apply(partialParams));
    }
    getColumnStatisticsForPartition(partialParams) {
        return this.client.getColumnStatisticsForPartition(this.ops["GetColumnStatisticsForPartition"].apply(partialParams));
    }
    getColumnStatisticsForTable(partialParams) {
        return this.client.getColumnStatisticsForTable(this.ops["GetColumnStatisticsForTable"].apply(partialParams));
    }
    getConnection(partialParams) {
        return this.client.getConnection(this.ops["GetConnection"].apply(partialParams));
    }
    getCrawler(partialParams) {
        return this.client.getCrawler(this.ops["GetCrawler"].apply(partialParams));
    }
    getDatabase(partialParams) {
        return this.client.getDatabase(this.ops["GetDatabase"].apply(partialParams));
    }
    getDevEndpoint(partialParams) {
        return this.client.getDevEndpoint(this.ops["GetDevEndpoint"].apply(partialParams));
    }
    getJob(partialParams) {
        return this.client.getJob(this.ops["GetJob"].apply(partialParams));
    }
    getJobBookmark(partialParams) {
        return this.client.getJobBookmark(this.ops["GetJobBookmark"].apply(partialParams));
    }
    getJobRun(partialParams) {
        return this.client.getJobRun(this.ops["GetJobRun"].apply(partialParams));
    }
    getJobRuns(partialParams) {
        return this.client.getJobRuns(this.ops["GetJobRuns"].apply(partialParams));
    }
    getMLTaskRun(partialParams) {
        return this.client.getMLTaskRun(this.ops["GetMLTaskRun"].apply(partialParams));
    }
    getMLTaskRuns(partialParams) {
        return this.client.getMLTaskRuns(this.ops["GetMLTaskRuns"].apply(partialParams));
    }
    getMLTransform(partialParams) {
        return this.client.getMLTransform(this.ops["GetMLTransform"].apply(partialParams));
    }
    getMapping(partialParams) {
        return this.client.getMapping(this.ops["GetMapping"].apply(partialParams));
    }
    getPartition(partialParams) {
        return this.client.getPartition(this.ops["GetPartition"].apply(partialParams));
    }
    getPartitionIndexes(partialParams) {
        return this.client.getPartitionIndexes(this.ops["GetPartitionIndexes"].apply(partialParams));
    }
    getPartitions(partialParams) {
        return this.client.getPartitions(this.ops["GetPartitions"].apply(partialParams));
    }
    getPlan(partialParams) {
        return this.client.getPlan(this.ops["GetPlan"].apply(partialParams));
    }
    getRegistry(partialParams) {
        return this.client.getRegistry(this.ops["GetRegistry"].apply(partialParams));
    }
    getSchema(partialParams) {
        return this.client.getSchema(this.ops["GetSchema"].apply(partialParams));
    }
    getSchemaByDefinition(partialParams) {
        return this.client.getSchemaByDefinition(this.ops["GetSchemaByDefinition"].apply(partialParams));
    }
    getSchemaVersionsDiff(partialParams) {
        return this.client.getSchemaVersionsDiff(this.ops["GetSchemaVersionsDiff"].apply(partialParams));
    }
    getSecurityConfiguration(partialParams) {
        return this.client.getSecurityConfiguration(this.ops["GetSecurityConfiguration"].apply(partialParams));
    }
    getTable(partialParams) {
        return this.client.getTable(this.ops["GetTable"].apply(partialParams));
    }
    getTableVersion(partialParams) {
        return this.client.getTableVersion(this.ops["GetTableVersion"].apply(partialParams));
    }
    getTableVersions(partialParams) {
        return this.client.getTableVersions(this.ops["GetTableVersions"].apply(partialParams));
    }
    getTables(partialParams) {
        return this.client.getTables(this.ops["GetTables"].apply(partialParams));
    }
    getTags(partialParams) {
        return this.client.getTags(this.ops["GetTags"].apply(partialParams));
    }
    getTrigger(partialParams) {
        return this.client.getTrigger(this.ops["GetTrigger"].apply(partialParams));
    }
    getUnfilteredPartitionMetadata(partialParams) {
        return this.client.getUnfilteredPartitionMetadata(this.ops["GetUnfilteredPartitionMetadata"].apply(partialParams));
    }
    getUnfilteredPartitionsMetadata(partialParams) {
        return this.client.getUnfilteredPartitionsMetadata(this.ops["GetUnfilteredPartitionsMetadata"].apply(partialParams));
    }
    getUnfilteredTableMetadata(partialParams) {
        return this.client.getUnfilteredTableMetadata(this.ops["GetUnfilteredTableMetadata"].apply(partialParams));
    }
    getUserDefinedFunction(partialParams) {
        return this.client.getUserDefinedFunction(this.ops["GetUserDefinedFunction"].apply(partialParams));
    }
    getUserDefinedFunctions(partialParams) {
        return this.client.getUserDefinedFunctions(this.ops["GetUserDefinedFunctions"].apply(partialParams));
    }
    getWorkflow(partialParams) {
        return this.client.getWorkflow(this.ops["GetWorkflow"].apply(partialParams));
    }
    getWorkflowRun(partialParams) {
        return this.client.getWorkflowRun(this.ops["GetWorkflowRun"].apply(partialParams));
    }
    getWorkflowRunProperties(partialParams) {
        return this.client.getWorkflowRunProperties(this.ops["GetWorkflowRunProperties"].apply(partialParams));
    }
    getWorkflowRuns(partialParams) {
        return this.client.getWorkflowRuns(this.ops["GetWorkflowRuns"].apply(partialParams));
    }
    listSchemaVersions(partialParams) {
        return this.client.listSchemaVersions(this.ops["ListSchemaVersions"].apply(partialParams));
    }
    putDataCatalogEncryptionSettings(partialParams) {
        return this.client.putDataCatalogEncryptionSettings(this.ops["PutDataCatalogEncryptionSettings"].apply(partialParams));
    }
    putResourcePolicy(partialParams) {
        return this.client.putResourcePolicy(this.ops["PutResourcePolicy"].apply(partialParams));
    }
    putSchemaVersionMetadata(partialParams) {
        return this.client.putSchemaVersionMetadata(this.ops["PutSchemaVersionMetadata"].apply(partialParams));
    }
    putWorkflowRunProperties(partialParams) {
        return this.client.putWorkflowRunProperties(this.ops["PutWorkflowRunProperties"].apply(partialParams));
    }
    registerSchemaVersion(partialParams) {
        return this.client.registerSchemaVersion(this.ops["RegisterSchemaVersion"].apply(partialParams));
    }
    removeSchemaVersionMetadata(partialParams) {
        return this.client.removeSchemaVersionMetadata(this.ops["RemoveSchemaVersionMetadata"].apply(partialParams));
    }
    resetJobBookmark(partialParams) {
        return this.client.resetJobBookmark(this.ops["ResetJobBookmark"].apply(partialParams));
    }
    resumeWorkflowRun(partialParams) {
        return this.client.resumeWorkflowRun(this.ops["ResumeWorkflowRun"].apply(partialParams));
    }
    startBlueprintRun(partialParams) {
        return this.client.startBlueprintRun(this.ops["StartBlueprintRun"].apply(partialParams));
    }
    startCrawler(partialParams) {
        return this.client.startCrawler(this.ops["StartCrawler"].apply(partialParams));
    }
    startCrawlerSchedule(partialParams) {
        return this.client.startCrawlerSchedule(this.ops["StartCrawlerSchedule"].apply(partialParams));
    }
    startExportLabelsTaskRun(partialParams) {
        return this.client.startExportLabelsTaskRun(this.ops["StartExportLabelsTaskRun"].apply(partialParams));
    }
    startImportLabelsTaskRun(partialParams) {
        return this.client.startImportLabelsTaskRun(this.ops["StartImportLabelsTaskRun"].apply(partialParams));
    }
    startJobRun(partialParams) {
        return this.client.startJobRun(this.ops["StartJobRun"].apply(partialParams));
    }
    startMLEvaluationTaskRun(partialParams) {
        return this.client.startMLEvaluationTaskRun(this.ops["StartMLEvaluationTaskRun"].apply(partialParams));
    }
    startMLLabelingSetGenerationTaskRun(partialParams) {
        return this.client.startMLLabelingSetGenerationTaskRun(this.ops["StartMLLabelingSetGenerationTaskRun"].apply(partialParams));
    }
    startTrigger(partialParams) {
        return this.client.startTrigger(this.ops["StartTrigger"].apply(partialParams));
    }
    startWorkflowRun(partialParams) {
        return this.client.startWorkflowRun(this.ops["StartWorkflowRun"].apply(partialParams));
    }
    stopCrawler(partialParams) {
        return this.client.stopCrawler(this.ops["StopCrawler"].apply(partialParams));
    }
    stopCrawlerSchedule(partialParams) {
        return this.client.stopCrawlerSchedule(this.ops["StopCrawlerSchedule"].apply(partialParams));
    }
    stopTrigger(partialParams) {
        return this.client.stopTrigger(this.ops["StopTrigger"].apply(partialParams));
    }
    stopWorkflowRun(partialParams) {
        return this.client.stopWorkflowRun(this.ops["StopWorkflowRun"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateBlueprint(partialParams) {
        return this.client.updateBlueprint(this.ops["UpdateBlueprint"].apply(partialParams));
    }
    updateColumnStatisticsForPartition(partialParams) {
        return this.client.updateColumnStatisticsForPartition(this.ops["UpdateColumnStatisticsForPartition"].apply(partialParams));
    }
    updateColumnStatisticsForTable(partialParams) {
        return this.client.updateColumnStatisticsForTable(this.ops["UpdateColumnStatisticsForTable"].apply(partialParams));
    }
    updateConnection(partialParams) {
        return this.client.updateConnection(this.ops["UpdateConnection"].apply(partialParams));
    }
    updateCrawler(partialParams) {
        return this.client.updateCrawler(this.ops["UpdateCrawler"].apply(partialParams));
    }
    updateCrawlerSchedule(partialParams) {
        return this.client.updateCrawlerSchedule(this.ops["UpdateCrawlerSchedule"].apply(partialParams));
    }
    updateDatabase(partialParams) {
        return this.client.updateDatabase(this.ops["UpdateDatabase"].apply(partialParams));
    }
    updateDevEndpoint(partialParams) {
        return this.client.updateDevEndpoint(this.ops["UpdateDevEndpoint"].apply(partialParams));
    }
    updateJob(partialParams) {
        return this.client.updateJob(this.ops["UpdateJob"].apply(partialParams));
    }
    updateMLTransform(partialParams) {
        return this.client.updateMLTransform(this.ops["UpdateMLTransform"].apply(partialParams));
    }
    updatePartition(partialParams) {
        return this.client.updatePartition(this.ops["UpdatePartition"].apply(partialParams));
    }
    updateRegistry(partialParams) {
        return this.client.updateRegistry(this.ops["UpdateRegistry"].apply(partialParams));
    }
    updateSchema(partialParams) {
        return this.client.updateSchema(this.ops["UpdateSchema"].apply(partialParams));
    }
    updateTable(partialParams) {
        return this.client.updateTable(this.ops["UpdateTable"].apply(partialParams));
    }
    updateTrigger(partialParams) {
        return this.client.updateTrigger(this.ops["UpdateTrigger"].apply(partialParams));
    }
    updateUserDefinedFunction(partialParams) {
        return this.client.updateUserDefinedFunction(this.ops["UpdateUserDefinedFunction"].apply(partialParams));
    }
    updateWorkflow(partialParams) {
        return this.client.updateWorkflow(this.ops["UpdateWorkflow"].apply(partialParams));
    }
}
exports.default = Resource;
