// out/elasticbeanstalk/Application.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.elasticbeanstalk.Application {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ElasticBeanstalk();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/elasticbeanstalk-2010-12-01.normal.json"), this.client);
    }
    applyEnvironmentManagedAction(partialParams) {
        return this.client.applyEnvironmentManagedAction(this.ops["ApplyEnvironmentManagedAction"].apply(partialParams));
    }
    associateEnvironmentOperationsRole(partialParams) {
        return this.client.associateEnvironmentOperationsRole(this.ops["AssociateEnvironmentOperationsRole"].apply(partialParams));
    }
    checkDNSAvailability(partialParams) {
        return this.client.checkDNSAvailability(this.ops["CheckDNSAvailability"].apply(partialParams));
    }
    createApplication(partialParams) {
        return this.client.createApplication(this.ops["CreateApplication"].apply(partialParams));
    }
    createApplicationVersion(partialParams) {
        return this.client.createApplicationVersion(this.ops["CreateApplicationVersion"].apply(partialParams));
    }
    createConfigurationTemplate(partialParams) {
        return this.client.createConfigurationTemplate(this.ops["CreateConfigurationTemplate"].apply(partialParams));
    }
    createEnvironment(partialParams) {
        return this.client.createEnvironment(this.ops["CreateEnvironment"].apply(partialParams));
    }
    createPlatformVersion(partialParams) {
        return this.client.createPlatformVersion(this.ops["CreatePlatformVersion"].apply(partialParams));
    }
    deleteApplication(partialParams) {
        return this.client.deleteApplication(this.ops["DeleteApplication"].apply(partialParams));
    }
    deleteApplicationVersion(partialParams) {
        return this.client.deleteApplicationVersion(this.ops["DeleteApplicationVersion"].apply(partialParams));
    }
    deleteConfigurationTemplate(partialParams) {
        return this.client.deleteConfigurationTemplate(this.ops["DeleteConfigurationTemplate"].apply(partialParams));
    }
    deleteEnvironmentConfiguration(partialParams) {
        return this.client.deleteEnvironmentConfiguration(this.ops["DeleteEnvironmentConfiguration"].apply(partialParams));
    }
    describeConfigurationSettings(partialParams) {
        return this.client.describeConfigurationSettings(this.ops["DescribeConfigurationSettings"].apply(partialParams));
    }
    disassociateEnvironmentOperationsRole(partialParams) {
        return this.client.disassociateEnvironmentOperationsRole(this.ops["DisassociateEnvironmentOperationsRole"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    requestEnvironmentInfo(partialParams) {
        return this.client.requestEnvironmentInfo(this.ops["RequestEnvironmentInfo"].apply(partialParams));
    }
    retrieveEnvironmentInfo(partialParams) {
        return this.client.retrieveEnvironmentInfo(this.ops["RetrieveEnvironmentInfo"].apply(partialParams));
    }
    updateApplication(partialParams) {
        return this.client.updateApplication(this.ops["UpdateApplication"].apply(partialParams));
    }
    updateApplicationResourceLifecycle(partialParams) {
        return this.client.updateApplicationResourceLifecycle(this.ops["UpdateApplicationResourceLifecycle"].apply(partialParams));
    }
    updateApplicationVersion(partialParams) {
        return this.client.updateApplicationVersion(this.ops["UpdateApplicationVersion"].apply(partialParams));
    }
    updateConfigurationTemplate(partialParams) {
        return this.client.updateConfigurationTemplate(this.ops["UpdateConfigurationTemplate"].apply(partialParams));
    }
    updateTagsForResource(partialParams) {
        return this.client.updateTagsForResource(this.ops["UpdateTagsForResource"].apply(partialParams));
    }
    validateConfigurationSettings(partialParams) {
        return this.client.validateConfigurationSettings(this.ops["ValidateConfigurationSettings"].apply(partialParams));
    }
}
exports.default = Resource;
