// out/storagegateway/CachesIscsiVolume.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.storagegateway.CachesIscsiVolume {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.StorageGateway();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/storagegateway-2013-06-30.normal.json"), this.client);
    }
    activateGateway(partialParams) {
        return this.client.activateGateway(this.ops["ActivateGateway"].apply(partialParams));
    }
    addCache(partialParams) {
        return this.client.addCache(this.ops["AddCache"].apply(partialParams));
    }
    addTagsToResource(partialParams) {
        return this.client.addTagsToResource(this.ops["AddTagsToResource"].apply(partialParams));
    }
    addUploadBuffer(partialParams) {
        return this.client.addUploadBuffer(this.ops["AddUploadBuffer"].apply(partialParams));
    }
    addWorkingStorage(partialParams) {
        return this.client.addWorkingStorage(this.ops["AddWorkingStorage"].apply(partialParams));
    }
    assignTapePool(partialParams) {
        return this.client.assignTapePool(this.ops["AssignTapePool"].apply(partialParams));
    }
    associateFileSystem(partialParams) {
        return this.client.associateFileSystem(this.ops["AssociateFileSystem"].apply(partialParams));
    }
    attachVolume(partialParams) {
        return this.client.attachVolume(this.ops["AttachVolume"].apply(partialParams));
    }
    cancelArchival(partialParams) {
        return this.client.cancelArchival(this.ops["CancelArchival"].apply(partialParams));
    }
    cancelRetrieval(partialParams) {
        return this.client.cancelRetrieval(this.ops["CancelRetrieval"].apply(partialParams));
    }
    createCachediSCSIVolume(partialParams) {
        return this.client.createCachediSCSIVolume(this.ops["CreateCachediSCSIVolume"].apply(partialParams));
    }
    createNFSFileShare(partialParams) {
        return this.client.createNFSFileShare(this.ops["CreateNFSFileShare"].apply(partialParams));
    }
    createSMBFileShare(partialParams) {
        return this.client.createSMBFileShare(this.ops["CreateSMBFileShare"].apply(partialParams));
    }
    createSnapshot(partialParams) {
        return this.client.createSnapshot(this.ops["CreateSnapshot"].apply(partialParams));
    }
    createSnapshotFromVolumeRecoveryPoint(partialParams) {
        return this.client.createSnapshotFromVolumeRecoveryPoint(this.ops["CreateSnapshotFromVolumeRecoveryPoint"].apply(partialParams));
    }
    createStorediSCSIVolume(partialParams) {
        return this.client.createStorediSCSIVolume(this.ops["CreateStorediSCSIVolume"].apply(partialParams));
    }
    createTapePool(partialParams) {
        return this.client.createTapePool(this.ops["CreateTapePool"].apply(partialParams));
    }
    createTapeWithBarcode(partialParams) {
        return this.client.createTapeWithBarcode(this.ops["CreateTapeWithBarcode"].apply(partialParams));
    }
    createTapes(partialParams) {
        return this.client.createTapes(this.ops["CreateTapes"].apply(partialParams));
    }
    deleteAutomaticTapeCreationPolicy(partialParams) {
        return this.client.deleteAutomaticTapeCreationPolicy(this.ops["DeleteAutomaticTapeCreationPolicy"].apply(partialParams));
    }
    deleteBandwidthRateLimit(partialParams) {
        return this.client.deleteBandwidthRateLimit(this.ops["DeleteBandwidthRateLimit"].apply(partialParams));
    }
    deleteChapCredentials(partialParams) {
        return this.client.deleteChapCredentials(this.ops["DeleteChapCredentials"].apply(partialParams));
    }
    deleteFileShare(partialParams) {
        return this.client.deleteFileShare(this.ops["DeleteFileShare"].apply(partialParams));
    }
    deleteGateway(partialParams) {
        return this.client.deleteGateway(this.ops["DeleteGateway"].apply(partialParams));
    }
    deleteSnapshotSchedule(partialParams) {
        return this.client.deleteSnapshotSchedule(this.ops["DeleteSnapshotSchedule"].apply(partialParams));
    }
    deleteTape(partialParams) {
        return this.client.deleteTape(this.ops["DeleteTape"].apply(partialParams));
    }
    deleteTapeArchive(partialParams) {
        return this.client.deleteTapeArchive(this.ops["DeleteTapeArchive"].apply(partialParams));
    }
    deleteTapePool(partialParams) {
        return this.client.deleteTapePool(this.ops["DeleteTapePool"].apply(partialParams));
    }
    deleteVolume(partialParams) {
        return this.client.deleteVolume(this.ops["DeleteVolume"].apply(partialParams));
    }
    describeAvailabilityMonitorTest(partialParams) {
        return this.client.describeAvailabilityMonitorTest(this.ops["DescribeAvailabilityMonitorTest"].apply(partialParams));
    }
    describeBandwidthRateLimit(partialParams) {
        return this.client.describeBandwidthRateLimit(this.ops["DescribeBandwidthRateLimit"].apply(partialParams));
    }
    describeBandwidthRateLimitSchedule(partialParams) {
        return this.client.describeBandwidthRateLimitSchedule(this.ops["DescribeBandwidthRateLimitSchedule"].apply(partialParams));
    }
    describeCache(partialParams) {
        return this.client.describeCache(this.ops["DescribeCache"].apply(partialParams));
    }
    describeCachediSCSIVolumes(partialParams) {
        return this.client.describeCachediSCSIVolumes(this.ops["DescribeCachediSCSIVolumes"].apply(partialParams));
    }
    describeChapCredentials(partialParams) {
        return this.client.describeChapCredentials(this.ops["DescribeChapCredentials"].apply(partialParams));
    }
    describeFileSystemAssociations(partialParams) {
        return this.client.describeFileSystemAssociations(this.ops["DescribeFileSystemAssociations"].apply(partialParams));
    }
    describeGatewayInformation(partialParams) {
        return this.client.describeGatewayInformation(this.ops["DescribeGatewayInformation"].apply(partialParams));
    }
    describeMaintenanceStartTime(partialParams) {
        return this.client.describeMaintenanceStartTime(this.ops["DescribeMaintenanceStartTime"].apply(partialParams));
    }
    describeNFSFileShares(partialParams) {
        return this.client.describeNFSFileShares(this.ops["DescribeNFSFileShares"].apply(partialParams));
    }
    describeSMBFileShares(partialParams) {
        return this.client.describeSMBFileShares(this.ops["DescribeSMBFileShares"].apply(partialParams));
    }
    describeSMBSettings(partialParams) {
        return this.client.describeSMBSettings(this.ops["DescribeSMBSettings"].apply(partialParams));
    }
    describeSnapshotSchedule(partialParams) {
        return this.client.describeSnapshotSchedule(this.ops["DescribeSnapshotSchedule"].apply(partialParams));
    }
    describeStorediSCSIVolumes(partialParams) {
        return this.client.describeStorediSCSIVolumes(this.ops["DescribeStorediSCSIVolumes"].apply(partialParams));
    }
    describeTapeRecoveryPoints(partialParams) {
        return this.client.describeTapeRecoveryPoints(this.ops["DescribeTapeRecoveryPoints"].apply(partialParams));
    }
    describeTapes(partialParams) {
        return this.client.describeTapes(this.ops["DescribeTapes"].apply(partialParams));
    }
    describeUploadBuffer(partialParams) {
        return this.client.describeUploadBuffer(this.ops["DescribeUploadBuffer"].apply(partialParams));
    }
    describeVTLDevices(partialParams) {
        return this.client.describeVTLDevices(this.ops["DescribeVTLDevices"].apply(partialParams));
    }
    describeWorkingStorage(partialParams) {
        return this.client.describeWorkingStorage(this.ops["DescribeWorkingStorage"].apply(partialParams));
    }
    detachVolume(partialParams) {
        return this.client.detachVolume(this.ops["DetachVolume"].apply(partialParams));
    }
    disableGateway(partialParams) {
        return this.client.disableGateway(this.ops["DisableGateway"].apply(partialParams));
    }
    disassociateFileSystem(partialParams) {
        return this.client.disassociateFileSystem(this.ops["DisassociateFileSystem"].apply(partialParams));
    }
    joinDomain(partialParams) {
        return this.client.joinDomain(this.ops["JoinDomain"].apply(partialParams));
    }
    listLocalDisks(partialParams) {
        return this.client.listLocalDisks(this.ops["ListLocalDisks"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listVolumeInitiators(partialParams) {
        return this.client.listVolumeInitiators(this.ops["ListVolumeInitiators"].apply(partialParams));
    }
    listVolumeRecoveryPoints(partialParams) {
        return this.client.listVolumeRecoveryPoints(this.ops["ListVolumeRecoveryPoints"].apply(partialParams));
    }
    notifyWhenUploaded(partialParams) {
        return this.client.notifyWhenUploaded(this.ops["NotifyWhenUploaded"].apply(partialParams));
    }
    refreshCache(partialParams) {
        return this.client.refreshCache(this.ops["RefreshCache"].apply(partialParams));
    }
    removeTagsFromResource(partialParams) {
        return this.client.removeTagsFromResource(this.ops["RemoveTagsFromResource"].apply(partialParams));
    }
    resetCache(partialParams) {
        return this.client.resetCache(this.ops["ResetCache"].apply(partialParams));
    }
    retrieveTapeArchive(partialParams) {
        return this.client.retrieveTapeArchive(this.ops["RetrieveTapeArchive"].apply(partialParams));
    }
    retrieveTapeRecoveryPoint(partialParams) {
        return this.client.retrieveTapeRecoveryPoint(this.ops["RetrieveTapeRecoveryPoint"].apply(partialParams));
    }
    setLocalConsolePassword(partialParams) {
        return this.client.setLocalConsolePassword(this.ops["SetLocalConsolePassword"].apply(partialParams));
    }
    setSMBGuestPassword(partialParams) {
        return this.client.setSMBGuestPassword(this.ops["SetSMBGuestPassword"].apply(partialParams));
    }
    shutdownGateway(partialParams) {
        return this.client.shutdownGateway(this.ops["ShutdownGateway"].apply(partialParams));
    }
    startAvailabilityMonitorTest(partialParams) {
        return this.client.startAvailabilityMonitorTest(this.ops["StartAvailabilityMonitorTest"].apply(partialParams));
    }
    startGateway(partialParams) {
        return this.client.startGateway(this.ops["StartGateway"].apply(partialParams));
    }
    updateAutomaticTapeCreationPolicy(partialParams) {
        return this.client.updateAutomaticTapeCreationPolicy(this.ops["UpdateAutomaticTapeCreationPolicy"].apply(partialParams));
    }
    updateBandwidthRateLimit(partialParams) {
        return this.client.updateBandwidthRateLimit(this.ops["UpdateBandwidthRateLimit"].apply(partialParams));
    }
    updateBandwidthRateLimitSchedule(partialParams) {
        return this.client.updateBandwidthRateLimitSchedule(this.ops["UpdateBandwidthRateLimitSchedule"].apply(partialParams));
    }
    updateChapCredentials(partialParams) {
        return this.client.updateChapCredentials(this.ops["UpdateChapCredentials"].apply(partialParams));
    }
    updateFileSystemAssociation(partialParams) {
        return this.client.updateFileSystemAssociation(this.ops["UpdateFileSystemAssociation"].apply(partialParams));
    }
    updateGatewayInformation(partialParams) {
        return this.client.updateGatewayInformation(this.ops["UpdateGatewayInformation"].apply(partialParams));
    }
    updateGatewaySoftwareNow(partialParams) {
        return this.client.updateGatewaySoftwareNow(this.ops["UpdateGatewaySoftwareNow"].apply(partialParams));
    }
    updateMaintenanceStartTime(partialParams) {
        return this.client.updateMaintenanceStartTime(this.ops["UpdateMaintenanceStartTime"].apply(partialParams));
    }
    updateNFSFileShare(partialParams) {
        return this.client.updateNFSFileShare(this.ops["UpdateNFSFileShare"].apply(partialParams));
    }
    updateSMBFileShare(partialParams) {
        return this.client.updateSMBFileShare(this.ops["UpdateSMBFileShare"].apply(partialParams));
    }
    updateSMBFileShareVisibility(partialParams) {
        return this.client.updateSMBFileShareVisibility(this.ops["UpdateSMBFileShareVisibility"].apply(partialParams));
    }
    updateSMBLocalGroups(partialParams) {
        return this.client.updateSMBLocalGroups(this.ops["UpdateSMBLocalGroups"].apply(partialParams));
    }
    updateSMBSecurityStrategy(partialParams) {
        return this.client.updateSMBSecurityStrategy(this.ops["UpdateSMBSecurityStrategy"].apply(partialParams));
    }
    updateSnapshotSchedule(partialParams) {
        return this.client.updateSnapshotSchedule(this.ops["UpdateSnapshotSchedule"].apply(partialParams));
    }
    updateVTLDeviceType(partialParams) {
        return this.client.updateVTLDeviceType(this.ops["UpdateVTLDeviceType"].apply(partialParams));
    }
}
exports.default = Resource;
