// out/appstream/DirectoryConfig.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.appstream.DirectoryConfig {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.AppStream();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/appstream-2016-12-01.normal.json"), this.client);
    }
    associateApplicationFleet(partialParams) {
        return this.client.associateApplicationFleet(this.ops["AssociateApplicationFleet"].apply(partialParams));
    }
    associateApplicationToEntitlement(partialParams) {
        return this.client.associateApplicationToEntitlement(this.ops["AssociateApplicationToEntitlement"].apply(partialParams));
    }
    associateFleet(partialParams) {
        return this.client.associateFleet(this.ops["AssociateFleet"].apply(partialParams));
    }
    batchAssociateUserStack(partialParams) {
        return this.client.batchAssociateUserStack(this.ops["BatchAssociateUserStack"].apply(partialParams));
    }
    batchDisassociateUserStack(partialParams) {
        return this.client.batchDisassociateUserStack(this.ops["BatchDisassociateUserStack"].apply(partialParams));
    }
    copyImage(partialParams) {
        return this.client.copyImage(this.ops["CopyImage"].apply(partialParams));
    }
    createAppBlock(partialParams) {
        return this.client.createAppBlock(this.ops["CreateAppBlock"].apply(partialParams));
    }
    createApplication(partialParams) {
        return this.client.createApplication(this.ops["CreateApplication"].apply(partialParams));
    }
    createDirectoryConfig(partialParams) {
        return this.client.createDirectoryConfig(this.ops["CreateDirectoryConfig"].apply(partialParams));
    }
    createEntitlement(partialParams) {
        return this.client.createEntitlement(this.ops["CreateEntitlement"].apply(partialParams));
    }
    createFleet(partialParams) {
        return this.client.createFleet(this.ops["CreateFleet"].apply(partialParams));
    }
    createImageBuilder(partialParams) {
        return this.client.createImageBuilder(this.ops["CreateImageBuilder"].apply(partialParams));
    }
    createImageBuilderStreamingURL(partialParams) {
        return this.client.createImageBuilderStreamingURL(this.ops["CreateImageBuilderStreamingURL"].apply(partialParams));
    }
    createStack(partialParams) {
        return this.client.createStack(this.ops["CreateStack"].apply(partialParams));
    }
    createStreamingURL(partialParams) {
        return this.client.createStreamingURL(this.ops["CreateStreamingURL"].apply(partialParams));
    }
    createUpdatedImage(partialParams) {
        return this.client.createUpdatedImage(this.ops["CreateUpdatedImage"].apply(partialParams));
    }
    createUser(partialParams) {
        return this.client.createUser(this.ops["CreateUser"].apply(partialParams));
    }
    deleteAppBlock(partialParams) {
        return this.client.deleteAppBlock(this.ops["DeleteAppBlock"].apply(partialParams));
    }
    deleteApplication(partialParams) {
        return this.client.deleteApplication(this.ops["DeleteApplication"].apply(partialParams));
    }
    deleteDirectoryConfig(partialParams) {
        return this.client.deleteDirectoryConfig(this.ops["DeleteDirectoryConfig"].apply(partialParams));
    }
    deleteEntitlement(partialParams) {
        return this.client.deleteEntitlement(this.ops["DeleteEntitlement"].apply(partialParams));
    }
    deleteFleet(partialParams) {
        return this.client.deleteFleet(this.ops["DeleteFleet"].apply(partialParams));
    }
    deleteImage(partialParams) {
        return this.client.deleteImage(this.ops["DeleteImage"].apply(partialParams));
    }
    deleteImageBuilder(partialParams) {
        return this.client.deleteImageBuilder(this.ops["DeleteImageBuilder"].apply(partialParams));
    }
    deleteImagePermissions(partialParams) {
        return this.client.deleteImagePermissions(this.ops["DeleteImagePermissions"].apply(partialParams));
    }
    deleteStack(partialParams) {
        return this.client.deleteStack(this.ops["DeleteStack"].apply(partialParams));
    }
    deleteUser(partialParams) {
        return this.client.deleteUser(this.ops["DeleteUser"].apply(partialParams));
    }
    describeEntitlements(partialParams) {
        return this.client.describeEntitlements(this.ops["DescribeEntitlements"].apply(partialParams));
    }
    describeImagePermissions(partialParams) {
        return this.client.describeImagePermissions(this.ops["DescribeImagePermissions"].apply(partialParams));
    }
    describeSessions(partialParams) {
        return this.client.describeSessions(this.ops["DescribeSessions"].apply(partialParams));
    }
    describeUsers(partialParams) {
        return this.client.describeUsers(this.ops["DescribeUsers"].apply(partialParams));
    }
    disableUser(partialParams) {
        return this.client.disableUser(this.ops["DisableUser"].apply(partialParams));
    }
    disassociateApplicationFleet(partialParams) {
        return this.client.disassociateApplicationFleet(this.ops["DisassociateApplicationFleet"].apply(partialParams));
    }
    disassociateApplicationFromEntitlement(partialParams) {
        return this.client.disassociateApplicationFromEntitlement(this.ops["DisassociateApplicationFromEntitlement"].apply(partialParams));
    }
    disassociateFleet(partialParams) {
        return this.client.disassociateFleet(this.ops["DisassociateFleet"].apply(partialParams));
    }
    enableUser(partialParams) {
        return this.client.enableUser(this.ops["EnableUser"].apply(partialParams));
    }
    expireSession(partialParams) {
        return this.client.expireSession(this.ops["ExpireSession"].apply(partialParams));
    }
    listAssociatedFleets(partialParams) {
        return this.client.listAssociatedFleets(this.ops["ListAssociatedFleets"].apply(partialParams));
    }
    listAssociatedStacks(partialParams) {
        return this.client.listAssociatedStacks(this.ops["ListAssociatedStacks"].apply(partialParams));
    }
    listEntitledApplications(partialParams) {
        return this.client.listEntitledApplications(this.ops["ListEntitledApplications"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    startFleet(partialParams) {
        return this.client.startFleet(this.ops["StartFleet"].apply(partialParams));
    }
    startImageBuilder(partialParams) {
        return this.client.startImageBuilder(this.ops["StartImageBuilder"].apply(partialParams));
    }
    stopFleet(partialParams) {
        return this.client.stopFleet(this.ops["StopFleet"].apply(partialParams));
    }
    stopImageBuilder(partialParams) {
        return this.client.stopImageBuilder(this.ops["StopImageBuilder"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApplication(partialParams) {
        return this.client.updateApplication(this.ops["UpdateApplication"].apply(partialParams));
    }
    updateDirectoryConfig(partialParams) {
        return this.client.updateDirectoryConfig(this.ops["UpdateDirectoryConfig"].apply(partialParams));
    }
    updateEntitlement(partialParams) {
        return this.client.updateEntitlement(this.ops["UpdateEntitlement"].apply(partialParams));
    }
    updateImagePermissions(partialParams) {
        return this.client.updateImagePermissions(this.ops["UpdateImagePermissions"].apply(partialParams));
    }
    updateStack(partialParams) {
        return this.client.updateStack(this.ops["UpdateStack"].apply(partialParams));
    }
}
exports.default = Resource;
