// out/lightsail/StaticIpAttachment.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.lightsail.StaticIpAttachment {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Lightsail();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/lightsail-2016-11-28.normal.json"), this.client);
    }
    allocateStaticIp(partialParams) {
        return this.client.allocateStaticIp(this.ops["AllocateStaticIp"].apply(partialParams));
    }
    attachCertificateToDistribution(partialParams) {
        return this.client.attachCertificateToDistribution(this.ops["AttachCertificateToDistribution"].apply(partialParams));
    }
    attachDisk(partialParams) {
        return this.client.attachDisk(this.ops["AttachDisk"].apply(partialParams));
    }
    attachInstancesToLoadBalancer(partialParams) {
        return this.client.attachInstancesToLoadBalancer(this.ops["AttachInstancesToLoadBalancer"].apply(partialParams));
    }
    attachLoadBalancerTlsCertificate(partialParams) {
        return this.client.attachLoadBalancerTlsCertificate(this.ops["AttachLoadBalancerTlsCertificate"].apply(partialParams));
    }
    attachStaticIp(partialParams) {
        return this.client.attachStaticIp(this.ops["AttachStaticIp"].apply(partialParams));
    }
    closeInstancePublicPorts(partialParams) {
        return this.client.closeInstancePublicPorts(this.ops["CloseInstancePublicPorts"].apply(partialParams));
    }
    copySnapshot(partialParams) {
        return this.client.copySnapshot(this.ops["CopySnapshot"].apply(partialParams));
    }
    createBucket(partialParams) {
        return this.client.createBucket(this.ops["CreateBucket"].apply(partialParams));
    }
    createBucketAccessKey(partialParams) {
        return this.client.createBucketAccessKey(this.ops["CreateBucketAccessKey"].apply(partialParams));
    }
    createCertificate(partialParams) {
        return this.client.createCertificate(this.ops["CreateCertificate"].apply(partialParams));
    }
    createCloudFormationStack(partialParams) {
        return this.client.createCloudFormationStack(this.ops["CreateCloudFormationStack"].apply(partialParams));
    }
    createContactMethod(partialParams) {
        return this.client.createContactMethod(this.ops["CreateContactMethod"].apply(partialParams));
    }
    createContainerService(partialParams) {
        return this.client.createContainerService(this.ops["CreateContainerService"].apply(partialParams));
    }
    createContainerServiceDeployment(partialParams) {
        return this.client.createContainerServiceDeployment(this.ops["CreateContainerServiceDeployment"].apply(partialParams));
    }
    createDisk(partialParams) {
        return this.client.createDisk(this.ops["CreateDisk"].apply(partialParams));
    }
    createDiskFromSnapshot(partialParams) {
        return this.client.createDiskFromSnapshot(this.ops["CreateDiskFromSnapshot"].apply(partialParams));
    }
    createDiskSnapshot(partialParams) {
        return this.client.createDiskSnapshot(this.ops["CreateDiskSnapshot"].apply(partialParams));
    }
    createDistribution(partialParams) {
        return this.client.createDistribution(this.ops["CreateDistribution"].apply(partialParams));
    }
    createDomain(partialParams) {
        return this.client.createDomain(this.ops["CreateDomain"].apply(partialParams));
    }
    createDomainEntry(partialParams) {
        return this.client.createDomainEntry(this.ops["CreateDomainEntry"].apply(partialParams));
    }
    createInstanceSnapshot(partialParams) {
        return this.client.createInstanceSnapshot(this.ops["CreateInstanceSnapshot"].apply(partialParams));
    }
    createInstances(partialParams) {
        return this.client.createInstances(this.ops["CreateInstances"].apply(partialParams));
    }
    createInstancesFromSnapshot(partialParams) {
        return this.client.createInstancesFromSnapshot(this.ops["CreateInstancesFromSnapshot"].apply(partialParams));
    }
    createKeyPair(partialParams) {
        return this.client.createKeyPair(this.ops["CreateKeyPair"].apply(partialParams));
    }
    createLoadBalancer(partialParams) {
        return this.client.createLoadBalancer(this.ops["CreateLoadBalancer"].apply(partialParams));
    }
    createLoadBalancerTlsCertificate(partialParams) {
        return this.client.createLoadBalancerTlsCertificate(this.ops["CreateLoadBalancerTlsCertificate"].apply(partialParams));
    }
    createRelationalDatabase(partialParams) {
        return this.client.createRelationalDatabase(this.ops["CreateRelationalDatabase"].apply(partialParams));
    }
    createRelationalDatabaseFromSnapshot(partialParams) {
        return this.client.createRelationalDatabaseFromSnapshot(this.ops["CreateRelationalDatabaseFromSnapshot"].apply(partialParams));
    }
    createRelationalDatabaseSnapshot(partialParams) {
        return this.client.createRelationalDatabaseSnapshot(this.ops["CreateRelationalDatabaseSnapshot"].apply(partialParams));
    }
    deleteAlarm(partialParams) {
        return this.client.deleteAlarm(this.ops["DeleteAlarm"].apply(partialParams));
    }
    deleteAutoSnapshot(partialParams) {
        return this.client.deleteAutoSnapshot(this.ops["DeleteAutoSnapshot"].apply(partialParams));
    }
    deleteBucket(partialParams) {
        return this.client.deleteBucket(this.ops["DeleteBucket"].apply(partialParams));
    }
    deleteBucketAccessKey(partialParams) {
        return this.client.deleteBucketAccessKey(this.ops["DeleteBucketAccessKey"].apply(partialParams));
    }
    deleteCertificate(partialParams) {
        return this.client.deleteCertificate(this.ops["DeleteCertificate"].apply(partialParams));
    }
    deleteContactMethod(partialParams) {
        return this.client.deleteContactMethod(this.ops["DeleteContactMethod"].apply(partialParams));
    }
    deleteContainerImage(partialParams) {
        return this.client.deleteContainerImage(this.ops["DeleteContainerImage"].apply(partialParams));
    }
    deleteContainerService(partialParams) {
        return this.client.deleteContainerService(this.ops["DeleteContainerService"].apply(partialParams));
    }
    deleteDisk(partialParams) {
        return this.client.deleteDisk(this.ops["DeleteDisk"].apply(partialParams));
    }
    deleteDiskSnapshot(partialParams) {
        return this.client.deleteDiskSnapshot(this.ops["DeleteDiskSnapshot"].apply(partialParams));
    }
    deleteDomain(partialParams) {
        return this.client.deleteDomain(this.ops["DeleteDomain"].apply(partialParams));
    }
    deleteDomainEntry(partialParams) {
        return this.client.deleteDomainEntry(this.ops["DeleteDomainEntry"].apply(partialParams));
    }
    deleteInstance(partialParams) {
        return this.client.deleteInstance(this.ops["DeleteInstance"].apply(partialParams));
    }
    deleteInstanceSnapshot(partialParams) {
        return this.client.deleteInstanceSnapshot(this.ops["DeleteInstanceSnapshot"].apply(partialParams));
    }
    deleteKeyPair(partialParams) {
        return this.client.deleteKeyPair(this.ops["DeleteKeyPair"].apply(partialParams));
    }
    deleteKnownHostKeys(partialParams) {
        return this.client.deleteKnownHostKeys(this.ops["DeleteKnownHostKeys"].apply(partialParams));
    }
    deleteLoadBalancer(partialParams) {
        return this.client.deleteLoadBalancer(this.ops["DeleteLoadBalancer"].apply(partialParams));
    }
    deleteLoadBalancerTlsCertificate(partialParams) {
        return this.client.deleteLoadBalancerTlsCertificate(this.ops["DeleteLoadBalancerTlsCertificate"].apply(partialParams));
    }
    deleteRelationalDatabase(partialParams) {
        return this.client.deleteRelationalDatabase(this.ops["DeleteRelationalDatabase"].apply(partialParams));
    }
    deleteRelationalDatabaseSnapshot(partialParams) {
        return this.client.deleteRelationalDatabaseSnapshot(this.ops["DeleteRelationalDatabaseSnapshot"].apply(partialParams));
    }
    detachCertificateFromDistribution(partialParams) {
        return this.client.detachCertificateFromDistribution(this.ops["DetachCertificateFromDistribution"].apply(partialParams));
    }
    detachDisk(partialParams) {
        return this.client.detachDisk(this.ops["DetachDisk"].apply(partialParams));
    }
    detachInstancesFromLoadBalancer(partialParams) {
        return this.client.detachInstancesFromLoadBalancer(this.ops["DetachInstancesFromLoadBalancer"].apply(partialParams));
    }
    detachStaticIp(partialParams) {
        return this.client.detachStaticIp(this.ops["DetachStaticIp"].apply(partialParams));
    }
    disableAddOn(partialParams) {
        return this.client.disableAddOn(this.ops["DisableAddOn"].apply(partialParams));
    }
    enableAddOn(partialParams) {
        return this.client.enableAddOn(this.ops["EnableAddOn"].apply(partialParams));
    }
    exportSnapshot(partialParams) {
        return this.client.exportSnapshot(this.ops["ExportSnapshot"].apply(partialParams));
    }
    getAutoSnapshots(partialParams) {
        return this.client.getAutoSnapshots(this.ops["GetAutoSnapshots"].apply(partialParams));
    }
    getBucketAccessKeys(partialParams) {
        return this.client.getBucketAccessKeys(this.ops["GetBucketAccessKeys"].apply(partialParams));
    }
    getBucketMetricData(partialParams) {
        return this.client.getBucketMetricData(this.ops["GetBucketMetricData"].apply(partialParams));
    }
    getContainerImages(partialParams) {
        return this.client.getContainerImages(this.ops["GetContainerImages"].apply(partialParams));
    }
    getContainerLog(partialParams) {
        return this.client.getContainerLog(this.ops["GetContainerLog"].apply(partialParams));
    }
    getContainerServiceDeployments(partialParams) {
        return this.client.getContainerServiceDeployments(this.ops["GetContainerServiceDeployments"].apply(partialParams));
    }
    getContainerServiceMetricData(partialParams) {
        return this.client.getContainerServiceMetricData(this.ops["GetContainerServiceMetricData"].apply(partialParams));
    }
    getDisk(partialParams) {
        return this.client.getDisk(this.ops["GetDisk"].apply(partialParams));
    }
    getDiskSnapshot(partialParams) {
        return this.client.getDiskSnapshot(this.ops["GetDiskSnapshot"].apply(partialParams));
    }
    getDistributionMetricData(partialParams) {
        return this.client.getDistributionMetricData(this.ops["GetDistributionMetricData"].apply(partialParams));
    }
    getDomain(partialParams) {
        return this.client.getDomain(this.ops["GetDomain"].apply(partialParams));
    }
    getInstance(partialParams) {
        return this.client.getInstance(this.ops["GetInstance"].apply(partialParams));
    }
    getInstanceAccessDetails(partialParams) {
        return this.client.getInstanceAccessDetails(this.ops["GetInstanceAccessDetails"].apply(partialParams));
    }
    getInstanceMetricData(partialParams) {
        return this.client.getInstanceMetricData(this.ops["GetInstanceMetricData"].apply(partialParams));
    }
    getInstancePortStates(partialParams) {
        return this.client.getInstancePortStates(this.ops["GetInstancePortStates"].apply(partialParams));
    }
    getInstanceSnapshot(partialParams) {
        return this.client.getInstanceSnapshot(this.ops["GetInstanceSnapshot"].apply(partialParams));
    }
    getInstanceState(partialParams) {
        return this.client.getInstanceState(this.ops["GetInstanceState"].apply(partialParams));
    }
    getKeyPair(partialParams) {
        return this.client.getKeyPair(this.ops["GetKeyPair"].apply(partialParams));
    }
    getLoadBalancer(partialParams) {
        return this.client.getLoadBalancer(this.ops["GetLoadBalancer"].apply(partialParams));
    }
    getLoadBalancerMetricData(partialParams) {
        return this.client.getLoadBalancerMetricData(this.ops["GetLoadBalancerMetricData"].apply(partialParams));
    }
    getLoadBalancerTlsCertificates(partialParams) {
        return this.client.getLoadBalancerTlsCertificates(this.ops["GetLoadBalancerTlsCertificates"].apply(partialParams));
    }
    getOperation(partialParams) {
        return this.client.getOperation(this.ops["GetOperation"].apply(partialParams));
    }
    getOperationsForResource(partialParams) {
        return this.client.getOperationsForResource(this.ops["GetOperationsForResource"].apply(partialParams));
    }
    getRelationalDatabase(partialParams) {
        return this.client.getRelationalDatabase(this.ops["GetRelationalDatabase"].apply(partialParams));
    }
    getRelationalDatabaseEvents(partialParams) {
        return this.client.getRelationalDatabaseEvents(this.ops["GetRelationalDatabaseEvents"].apply(partialParams));
    }
    getRelationalDatabaseLogEvents(partialParams) {
        return this.client.getRelationalDatabaseLogEvents(this.ops["GetRelationalDatabaseLogEvents"].apply(partialParams));
    }
    getRelationalDatabaseLogStreams(partialParams) {
        return this.client.getRelationalDatabaseLogStreams(this.ops["GetRelationalDatabaseLogStreams"].apply(partialParams));
    }
    getRelationalDatabaseMasterUserPassword(partialParams) {
        return this.client.getRelationalDatabaseMasterUserPassword(this.ops["GetRelationalDatabaseMasterUserPassword"].apply(partialParams));
    }
    getRelationalDatabaseMetricData(partialParams) {
        return this.client.getRelationalDatabaseMetricData(this.ops["GetRelationalDatabaseMetricData"].apply(partialParams));
    }
    getRelationalDatabaseParameters(partialParams) {
        return this.client.getRelationalDatabaseParameters(this.ops["GetRelationalDatabaseParameters"].apply(partialParams));
    }
    getRelationalDatabaseSnapshot(partialParams) {
        return this.client.getRelationalDatabaseSnapshot(this.ops["GetRelationalDatabaseSnapshot"].apply(partialParams));
    }
    getStaticIp(partialParams) {
        return this.client.getStaticIp(this.ops["GetStaticIp"].apply(partialParams));
    }
    importKeyPair(partialParams) {
        return this.client.importKeyPair(this.ops["ImportKeyPair"].apply(partialParams));
    }
    openInstancePublicPorts(partialParams) {
        return this.client.openInstancePublicPorts(this.ops["OpenInstancePublicPorts"].apply(partialParams));
    }
    putAlarm(partialParams) {
        return this.client.putAlarm(this.ops["PutAlarm"].apply(partialParams));
    }
    putInstancePublicPorts(partialParams) {
        return this.client.putInstancePublicPorts(this.ops["PutInstancePublicPorts"].apply(partialParams));
    }
    rebootInstance(partialParams) {
        return this.client.rebootInstance(this.ops["RebootInstance"].apply(partialParams));
    }
    rebootRelationalDatabase(partialParams) {
        return this.client.rebootRelationalDatabase(this.ops["RebootRelationalDatabase"].apply(partialParams));
    }
    registerContainerImage(partialParams) {
        return this.client.registerContainerImage(this.ops["RegisterContainerImage"].apply(partialParams));
    }
    releaseStaticIp(partialParams) {
        return this.client.releaseStaticIp(this.ops["ReleaseStaticIp"].apply(partialParams));
    }
    sendContactMethodVerification(partialParams) {
        return this.client.sendContactMethodVerification(this.ops["SendContactMethodVerification"].apply(partialParams));
    }
    setIpAddressType(partialParams) {
        return this.client.setIpAddressType(this.ops["SetIpAddressType"].apply(partialParams));
    }
    setResourceAccessForBucket(partialParams) {
        return this.client.setResourceAccessForBucket(this.ops["SetResourceAccessForBucket"].apply(partialParams));
    }
    startInstance(partialParams) {
        return this.client.startInstance(this.ops["StartInstance"].apply(partialParams));
    }
    startRelationalDatabase(partialParams) {
        return this.client.startRelationalDatabase(this.ops["StartRelationalDatabase"].apply(partialParams));
    }
    stopInstance(partialParams) {
        return this.client.stopInstance(this.ops["StopInstance"].apply(partialParams));
    }
    stopRelationalDatabase(partialParams) {
        return this.client.stopRelationalDatabase(this.ops["StopRelationalDatabase"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    testAlarm(partialParams) {
        return this.client.testAlarm(this.ops["TestAlarm"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateBucket(partialParams) {
        return this.client.updateBucket(this.ops["UpdateBucket"].apply(partialParams));
    }
    updateBucketBundle(partialParams) {
        return this.client.updateBucketBundle(this.ops["UpdateBucketBundle"].apply(partialParams));
    }
    updateContainerService(partialParams) {
        return this.client.updateContainerService(this.ops["UpdateContainerService"].apply(partialParams));
    }
    updateDistribution(partialParams) {
        return this.client.updateDistribution(this.ops["UpdateDistribution"].apply(partialParams));
    }
    updateDomainEntry(partialParams) {
        return this.client.updateDomainEntry(this.ops["UpdateDomainEntry"].apply(partialParams));
    }
    updateLoadBalancerAttribute(partialParams) {
        return this.client.updateLoadBalancerAttribute(this.ops["UpdateLoadBalancerAttribute"].apply(partialParams));
    }
    updateRelationalDatabase(partialParams) {
        return this.client.updateRelationalDatabase(this.ops["UpdateRelationalDatabase"].apply(partialParams));
    }
    updateRelationalDatabaseParameters(partialParams) {
        return this.client.updateRelationalDatabaseParameters(this.ops["UpdateRelationalDatabaseParameters"].apply(partialParams));
    }
}
exports.default = Resource;
