// out/wafv2/IpSet.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.wafv2.IpSet {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.WAFV2();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/wafv2-2019-07-29.normal.json"), this.client);
    }
    associateWebACL(partialParams) {
        return this.client.associateWebACL(this.ops["AssociateWebACL"].apply(partialParams));
    }
    checkCapacity(partialParams) {
        return this.client.checkCapacity(this.ops["CheckCapacity"].apply(partialParams));
    }
    createIPSet(partialParams) {
        return this.client.createIPSet(this.ops["CreateIPSet"].apply(partialParams));
    }
    createRegexPatternSet(partialParams) {
        return this.client.createRegexPatternSet(this.ops["CreateRegexPatternSet"].apply(partialParams));
    }
    createRuleGroup(partialParams) {
        return this.client.createRuleGroup(this.ops["CreateRuleGroup"].apply(partialParams));
    }
    createWebACL(partialParams) {
        return this.client.createWebACL(this.ops["CreateWebACL"].apply(partialParams));
    }
    deleteFirewallManagerRuleGroups(partialParams) {
        return this.client.deleteFirewallManagerRuleGroups(this.ops["DeleteFirewallManagerRuleGroups"].apply(partialParams));
    }
    deleteIPSet(partialParams) {
        return this.client.deleteIPSet(this.ops["DeleteIPSet"].apply(partialParams));
    }
    deleteLoggingConfiguration(partialParams) {
        return this.client.deleteLoggingConfiguration(this.ops["DeleteLoggingConfiguration"].apply(partialParams));
    }
    deletePermissionPolicy(partialParams) {
        return this.client.deletePermissionPolicy(this.ops["DeletePermissionPolicy"].apply(partialParams));
    }
    deleteRegexPatternSet(partialParams) {
        return this.client.deleteRegexPatternSet(this.ops["DeleteRegexPatternSet"].apply(partialParams));
    }
    deleteRuleGroup(partialParams) {
        return this.client.deleteRuleGroup(this.ops["DeleteRuleGroup"].apply(partialParams));
    }
    deleteWebACL(partialParams) {
        return this.client.deleteWebACL(this.ops["DeleteWebACL"].apply(partialParams));
    }
    describeManagedRuleGroup(partialParams) {
        return this.client.describeManagedRuleGroup(this.ops["DescribeManagedRuleGroup"].apply(partialParams));
    }
    disassociateWebACL(partialParams) {
        return this.client.disassociateWebACL(this.ops["DisassociateWebACL"].apply(partialParams));
    }
    generateMobileSdkReleaseUrl(partialParams) {
        return this.client.generateMobileSdkReleaseUrl(this.ops["GenerateMobileSdkReleaseUrl"].apply(partialParams));
    }
    getIPSet(partialParams) {
        return this.client.getIPSet(this.ops["GetIPSet"].apply(partialParams));
    }
    getLoggingConfiguration(partialParams) {
        return this.client.getLoggingConfiguration(this.ops["GetLoggingConfiguration"].apply(partialParams));
    }
    getManagedRuleSet(partialParams) {
        return this.client.getManagedRuleSet(this.ops["GetManagedRuleSet"].apply(partialParams));
    }
    getMobileSdkRelease(partialParams) {
        return this.client.getMobileSdkRelease(this.ops["GetMobileSdkRelease"].apply(partialParams));
    }
    getPermissionPolicy(partialParams) {
        return this.client.getPermissionPolicy(this.ops["GetPermissionPolicy"].apply(partialParams));
    }
    getRateBasedStatementManagedKeys(partialParams) {
        return this.client.getRateBasedStatementManagedKeys(this.ops["GetRateBasedStatementManagedKeys"].apply(partialParams));
    }
    getRegexPatternSet(partialParams) {
        return this.client.getRegexPatternSet(this.ops["GetRegexPatternSet"].apply(partialParams));
    }
    getSampledRequests(partialParams) {
        return this.client.getSampledRequests(this.ops["GetSampledRequests"].apply(partialParams));
    }
    getWebACL(partialParams) {
        return this.client.getWebACL(this.ops["GetWebACL"].apply(partialParams));
    }
    getWebACLForResource(partialParams) {
        return this.client.getWebACLForResource(this.ops["GetWebACLForResource"].apply(partialParams));
    }
    listAvailableManagedRuleGroupVersions(partialParams) {
        return this.client.listAvailableManagedRuleGroupVersions(this.ops["ListAvailableManagedRuleGroupVersions"].apply(partialParams));
    }
    listAvailableManagedRuleGroups(partialParams) {
        return this.client.listAvailableManagedRuleGroups(this.ops["ListAvailableManagedRuleGroups"].apply(partialParams));
    }
    listIPSets(partialParams) {
        return this.client.listIPSets(this.ops["ListIPSets"].apply(partialParams));
    }
    listLoggingConfigurations(partialParams) {
        return this.client.listLoggingConfigurations(this.ops["ListLoggingConfigurations"].apply(partialParams));
    }
    listManagedRuleSets(partialParams) {
        return this.client.listManagedRuleSets(this.ops["ListManagedRuleSets"].apply(partialParams));
    }
    listMobileSdkReleases(partialParams) {
        return this.client.listMobileSdkReleases(this.ops["ListMobileSdkReleases"].apply(partialParams));
    }
    listRegexPatternSets(partialParams) {
        return this.client.listRegexPatternSets(this.ops["ListRegexPatternSets"].apply(partialParams));
    }
    listResourcesForWebACL(partialParams) {
        return this.client.listResourcesForWebACL(this.ops["ListResourcesForWebACL"].apply(partialParams));
    }
    listRuleGroups(partialParams) {
        return this.client.listRuleGroups(this.ops["ListRuleGroups"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listWebACLs(partialParams) {
        return this.client.listWebACLs(this.ops["ListWebACLs"].apply(partialParams));
    }
    putLoggingConfiguration(partialParams) {
        return this.client.putLoggingConfiguration(this.ops["PutLoggingConfiguration"].apply(partialParams));
    }
    putManagedRuleSetVersions(partialParams) {
        return this.client.putManagedRuleSetVersions(this.ops["PutManagedRuleSetVersions"].apply(partialParams));
    }
    putPermissionPolicy(partialParams) {
        return this.client.putPermissionPolicy(this.ops["PutPermissionPolicy"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateIPSet(partialParams) {
        return this.client.updateIPSet(this.ops["UpdateIPSet"].apply(partialParams));
    }
    updateManagedRuleSetVersionExpiryDate(partialParams) {
        return this.client.updateManagedRuleSetVersionExpiryDate(this.ops["UpdateManagedRuleSetVersionExpiryDate"].apply(partialParams));
    }
    updateRegexPatternSet(partialParams) {
        return this.client.updateRegexPatternSet(this.ops["UpdateRegexPatternSet"].apply(partialParams));
    }
    updateRuleGroup(partialParams) {
        return this.client.updateRuleGroup(this.ops["UpdateRuleGroup"].apply(partialParams));
    }
    updateWebACL(partialParams) {
        return this.client.updateWebACL(this.ops["UpdateWebACL"].apply(partialParams));
    }
}
exports.default = Resource;
