// out/opsworks/Permission.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.opsworks.Permission {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.OpsWorks();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/opsworks-2013-02-18.normal.json"), this.client);
    }
    assignInstance(partialParams) {
        return this.client.assignInstance(this.ops["AssignInstance"].apply(partialParams));
    }
    assignVolume(partialParams) {
        return this.client.assignVolume(this.ops["AssignVolume"].apply(partialParams));
    }
    associateElasticIp(partialParams) {
        return this.client.associateElasticIp(this.ops["AssociateElasticIp"].apply(partialParams));
    }
    attachElasticLoadBalancer(partialParams) {
        return this.client.attachElasticLoadBalancer(this.ops["AttachElasticLoadBalancer"].apply(partialParams));
    }
    cloneStack(partialParams) {
        return this.client.cloneStack(this.ops["CloneStack"].apply(partialParams));
    }
    createApp(partialParams) {
        return this.client.createApp(this.ops["CreateApp"].apply(partialParams));
    }
    createDeployment(partialParams) {
        return this.client.createDeployment(this.ops["CreateDeployment"].apply(partialParams));
    }
    createInstance(partialParams) {
        return this.client.createInstance(this.ops["CreateInstance"].apply(partialParams));
    }
    createLayer(partialParams) {
        return this.client.createLayer(this.ops["CreateLayer"].apply(partialParams));
    }
    createStack(partialParams) {
        return this.client.createStack(this.ops["CreateStack"].apply(partialParams));
    }
    createUserProfile(partialParams) {
        return this.client.createUserProfile(this.ops["CreateUserProfile"].apply(partialParams));
    }
    deleteApp(partialParams) {
        return this.client.deleteApp(this.ops["DeleteApp"].apply(partialParams));
    }
    deleteInstance(partialParams) {
        return this.client.deleteInstance(this.ops["DeleteInstance"].apply(partialParams));
    }
    deleteLayer(partialParams) {
        return this.client.deleteLayer(this.ops["DeleteLayer"].apply(partialParams));
    }
    deleteStack(partialParams) {
        return this.client.deleteStack(this.ops["DeleteStack"].apply(partialParams));
    }
    deleteUserProfile(partialParams) {
        return this.client.deleteUserProfile(this.ops["DeleteUserProfile"].apply(partialParams));
    }
    deregisterEcsCluster(partialParams) {
        return this.client.deregisterEcsCluster(this.ops["DeregisterEcsCluster"].apply(partialParams));
    }
    deregisterElasticIp(partialParams) {
        return this.client.deregisterElasticIp(this.ops["DeregisterElasticIp"].apply(partialParams));
    }
    deregisterInstance(partialParams) {
        return this.client.deregisterInstance(this.ops["DeregisterInstance"].apply(partialParams));
    }
    deregisterRdsDbInstance(partialParams) {
        return this.client.deregisterRdsDbInstance(this.ops["DeregisterRdsDbInstance"].apply(partialParams));
    }
    deregisterVolume(partialParams) {
        return this.client.deregisterVolume(this.ops["DeregisterVolume"].apply(partialParams));
    }
    describeLoadBasedAutoScaling(partialParams) {
        return this.client.describeLoadBasedAutoScaling(this.ops["DescribeLoadBasedAutoScaling"].apply(partialParams));
    }
    describeRdsDbInstances(partialParams) {
        return this.client.describeRdsDbInstances(this.ops["DescribeRdsDbInstances"].apply(partialParams));
    }
    describeStackProvisioningParameters(partialParams) {
        return this.client.describeStackProvisioningParameters(this.ops["DescribeStackProvisioningParameters"].apply(partialParams));
    }
    describeStackSummary(partialParams) {
        return this.client.describeStackSummary(this.ops["DescribeStackSummary"].apply(partialParams));
    }
    describeTimeBasedAutoScaling(partialParams) {
        return this.client.describeTimeBasedAutoScaling(this.ops["DescribeTimeBasedAutoScaling"].apply(partialParams));
    }
    detachElasticLoadBalancer(partialParams) {
        return this.client.detachElasticLoadBalancer(this.ops["DetachElasticLoadBalancer"].apply(partialParams));
    }
    disassociateElasticIp(partialParams) {
        return this.client.disassociateElasticIp(this.ops["DisassociateElasticIp"].apply(partialParams));
    }
    getHostnameSuggestion(partialParams) {
        return this.client.getHostnameSuggestion(this.ops["GetHostnameSuggestion"].apply(partialParams));
    }
    grantAccess(partialParams) {
        return this.client.grantAccess(this.ops["GrantAccess"].apply(partialParams));
    }
    listTags(partialParams) {
        return this.client.listTags(this.ops["ListTags"].apply(partialParams));
    }
    rebootInstance(partialParams) {
        return this.client.rebootInstance(this.ops["RebootInstance"].apply(partialParams));
    }
    registerEcsCluster(partialParams) {
        return this.client.registerEcsCluster(this.ops["RegisterEcsCluster"].apply(partialParams));
    }
    registerElasticIp(partialParams) {
        return this.client.registerElasticIp(this.ops["RegisterElasticIp"].apply(partialParams));
    }
    registerInstance(partialParams) {
        return this.client.registerInstance(this.ops["RegisterInstance"].apply(partialParams));
    }
    registerRdsDbInstance(partialParams) {
        return this.client.registerRdsDbInstance(this.ops["RegisterRdsDbInstance"].apply(partialParams));
    }
    registerVolume(partialParams) {
        return this.client.registerVolume(this.ops["RegisterVolume"].apply(partialParams));
    }
    setLoadBasedAutoScaling(partialParams) {
        return this.client.setLoadBasedAutoScaling(this.ops["SetLoadBasedAutoScaling"].apply(partialParams));
    }
    setPermission(partialParams) {
        return this.client.setPermission(this.ops["SetPermission"].apply(partialParams));
    }
    setTimeBasedAutoScaling(partialParams) {
        return this.client.setTimeBasedAutoScaling(this.ops["SetTimeBasedAutoScaling"].apply(partialParams));
    }
    startInstance(partialParams) {
        return this.client.startInstance(this.ops["StartInstance"].apply(partialParams));
    }
    startStack(partialParams) {
        return this.client.startStack(this.ops["StartStack"].apply(partialParams));
    }
    stopInstance(partialParams) {
        return this.client.stopInstance(this.ops["StopInstance"].apply(partialParams));
    }
    stopStack(partialParams) {
        return this.client.stopStack(this.ops["StopStack"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    unassignInstance(partialParams) {
        return this.client.unassignInstance(this.ops["UnassignInstance"].apply(partialParams));
    }
    unassignVolume(partialParams) {
        return this.client.unassignVolume(this.ops["UnassignVolume"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApp(partialParams) {
        return this.client.updateApp(this.ops["UpdateApp"].apply(partialParams));
    }
    updateElasticIp(partialParams) {
        return this.client.updateElasticIp(this.ops["UpdateElasticIp"].apply(partialParams));
    }
    updateInstance(partialParams) {
        return this.client.updateInstance(this.ops["UpdateInstance"].apply(partialParams));
    }
    updateLayer(partialParams) {
        return this.client.updateLayer(this.ops["UpdateLayer"].apply(partialParams));
    }
    updateRdsDbInstance(partialParams) {
        return this.client.updateRdsDbInstance(this.ops["UpdateRdsDbInstance"].apply(partialParams));
    }
    updateStack(partialParams) {
        return this.client.updateStack(this.ops["UpdateStack"].apply(partialParams));
    }
    updateUserProfile(partialParams) {
        return this.client.updateUserProfile(this.ops["UpdateUserProfile"].apply(partialParams));
    }
    updateVolume(partialParams) {
        return this.client.updateVolume(this.ops["UpdateVolume"].apply(partialParams));
    }
}
exports.default = Resource;
