// out/codepipeline/Webhook.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.codepipeline.Webhook {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.CodePipeline();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/codepipeline-2015-07-09.normal.json"), this.client);
    }
    acknowledgeJob(partialParams) {
        return this.client.acknowledgeJob(this.ops["AcknowledgeJob"].apply(partialParams));
    }
    acknowledgeThirdPartyJob(partialParams) {
        return this.client.acknowledgeThirdPartyJob(this.ops["AcknowledgeThirdPartyJob"].apply(partialParams));
    }
    createCustomActionType(partialParams) {
        return this.client.createCustomActionType(this.ops["CreateCustomActionType"].apply(partialParams));
    }
    createPipeline(partialParams) {
        return this.client.createPipeline(this.ops["CreatePipeline"].apply(partialParams));
    }
    deleteCustomActionType(partialParams) {
        return this.client.deleteCustomActionType(this.ops["DeleteCustomActionType"].apply(partialParams));
    }
    deletePipeline(partialParams) {
        return this.client.deletePipeline(this.ops["DeletePipeline"].apply(partialParams));
    }
    deleteWebhook(partialParams) {
        return this.client.deleteWebhook(this.ops["DeleteWebhook"].apply(partialParams));
    }
    disableStageTransition(partialParams) {
        return this.client.disableStageTransition(this.ops["DisableStageTransition"].apply(partialParams));
    }
    enableStageTransition(partialParams) {
        return this.client.enableStageTransition(this.ops["EnableStageTransition"].apply(partialParams));
    }
    getActionType(partialParams) {
        return this.client.getActionType(this.ops["GetActionType"].apply(partialParams));
    }
    getJobDetails(partialParams) {
        return this.client.getJobDetails(this.ops["GetJobDetails"].apply(partialParams));
    }
    getPipeline(partialParams) {
        return this.client.getPipeline(this.ops["GetPipeline"].apply(partialParams));
    }
    getPipelineExecution(partialParams) {
        return this.client.getPipelineExecution(this.ops["GetPipelineExecution"].apply(partialParams));
    }
    getPipelineState(partialParams) {
        return this.client.getPipelineState(this.ops["GetPipelineState"].apply(partialParams));
    }
    getThirdPartyJobDetails(partialParams) {
        return this.client.getThirdPartyJobDetails(this.ops["GetThirdPartyJobDetails"].apply(partialParams));
    }
    listActionExecutions(partialParams) {
        return this.client.listActionExecutions(this.ops["ListActionExecutions"].apply(partialParams));
    }
    listPipelineExecutions(partialParams) {
        return this.client.listPipelineExecutions(this.ops["ListPipelineExecutions"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    pollForJobs(partialParams) {
        return this.client.pollForJobs(this.ops["PollForJobs"].apply(partialParams));
    }
    pollForThirdPartyJobs(partialParams) {
        return this.client.pollForThirdPartyJobs(this.ops["PollForThirdPartyJobs"].apply(partialParams));
    }
    putActionRevision(partialParams) {
        return this.client.putActionRevision(this.ops["PutActionRevision"].apply(partialParams));
    }
    putApprovalResult(partialParams) {
        return this.client.putApprovalResult(this.ops["PutApprovalResult"].apply(partialParams));
    }
    putJobFailureResult(partialParams) {
        return this.client.putJobFailureResult(this.ops["PutJobFailureResult"].apply(partialParams));
    }
    putJobSuccessResult(partialParams) {
        return this.client.putJobSuccessResult(this.ops["PutJobSuccessResult"].apply(partialParams));
    }
    putThirdPartyJobFailureResult(partialParams) {
        return this.client.putThirdPartyJobFailureResult(this.ops["PutThirdPartyJobFailureResult"].apply(partialParams));
    }
    putThirdPartyJobSuccessResult(partialParams) {
        return this.client.putThirdPartyJobSuccessResult(this.ops["PutThirdPartyJobSuccessResult"].apply(partialParams));
    }
    putWebhook(partialParams) {
        return this.client.putWebhook(this.ops["PutWebhook"].apply(partialParams));
    }
    retryStageExecution(partialParams) {
        return this.client.retryStageExecution(this.ops["RetryStageExecution"].apply(partialParams));
    }
    startPipelineExecution(partialParams) {
        return this.client.startPipelineExecution(this.ops["StartPipelineExecution"].apply(partialParams));
    }
    stopPipelineExecution(partialParams) {
        return this.client.stopPipelineExecution(this.ops["StopPipelineExecution"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateActionType(partialParams) {
        return this.client.updateActionType(this.ops["UpdateActionType"].apply(partialParams));
    }
    updatePipeline(partialParams) {
        return this.client.updatePipeline(this.ops["UpdatePipeline"].apply(partialParams));
    }
}
exports.default = Resource;
