// out/codedeploy/DeploymentGroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.codedeploy.DeploymentGroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.CodeDeploy();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/codedeploy-2014-10-06.normal.json"), this.client);
    }
    addTagsToOnPremisesInstances(partialParams) {
        return this.client.addTagsToOnPremisesInstances(this.ops["AddTagsToOnPremisesInstances"].apply(partialParams));
    }
    batchGetApplicationRevisions(partialParams) {
        return this.client.batchGetApplicationRevisions(this.ops["BatchGetApplicationRevisions"].apply(partialParams));
    }
    batchGetApplications(partialParams) {
        return this.client.batchGetApplications(this.ops["BatchGetApplications"].apply(partialParams));
    }
    batchGetDeploymentGroups(partialParams) {
        return this.client.batchGetDeploymentGroups(this.ops["BatchGetDeploymentGroups"].apply(partialParams));
    }
    batchGetDeploymentInstances(partialParams) {
        return this.client.batchGetDeploymentInstances(this.ops["BatchGetDeploymentInstances"].apply(partialParams));
    }
    batchGetDeployments(partialParams) {
        return this.client.batchGetDeployments(this.ops["BatchGetDeployments"].apply(partialParams));
    }
    batchGetOnPremisesInstances(partialParams) {
        return this.client.batchGetOnPremisesInstances(this.ops["BatchGetOnPremisesInstances"].apply(partialParams));
    }
    createApplication(partialParams) {
        return this.client.createApplication(this.ops["CreateApplication"].apply(partialParams));
    }
    createDeployment(partialParams) {
        return this.client.createDeployment(this.ops["CreateDeployment"].apply(partialParams));
    }
    createDeploymentConfig(partialParams) {
        return this.client.createDeploymentConfig(this.ops["CreateDeploymentConfig"].apply(partialParams));
    }
    createDeploymentGroup(partialParams) {
        return this.client.createDeploymentGroup(this.ops["CreateDeploymentGroup"].apply(partialParams));
    }
    deleteApplication(partialParams) {
        return this.client.deleteApplication(this.ops["DeleteApplication"].apply(partialParams));
    }
    deleteDeploymentConfig(partialParams) {
        return this.client.deleteDeploymentConfig(this.ops["DeleteDeploymentConfig"].apply(partialParams));
    }
    deleteDeploymentGroup(partialParams) {
        return this.client.deleteDeploymentGroup(this.ops["DeleteDeploymentGroup"].apply(partialParams));
    }
    deregisterOnPremisesInstance(partialParams) {
        return this.client.deregisterOnPremisesInstance(this.ops["DeregisterOnPremisesInstance"].apply(partialParams));
    }
    getApplication(partialParams) {
        return this.client.getApplication(this.ops["GetApplication"].apply(partialParams));
    }
    getApplicationRevision(partialParams) {
        return this.client.getApplicationRevision(this.ops["GetApplicationRevision"].apply(partialParams));
    }
    getDeployment(partialParams) {
        return this.client.getDeployment(this.ops["GetDeployment"].apply(partialParams));
    }
    getDeploymentConfig(partialParams) {
        return this.client.getDeploymentConfig(this.ops["GetDeploymentConfig"].apply(partialParams));
    }
    getDeploymentGroup(partialParams) {
        return this.client.getDeploymentGroup(this.ops["GetDeploymentGroup"].apply(partialParams));
    }
    getDeploymentInstance(partialParams) {
        return this.client.getDeploymentInstance(this.ops["GetDeploymentInstance"].apply(partialParams));
    }
    getOnPremisesInstance(partialParams) {
        return this.client.getOnPremisesInstance(this.ops["GetOnPremisesInstance"].apply(partialParams));
    }
    listApplicationRevisions(partialParams) {
        return this.client.listApplicationRevisions(this.ops["ListApplicationRevisions"].apply(partialParams));
    }
    listDeploymentGroups(partialParams) {
        return this.client.listDeploymentGroups(this.ops["ListDeploymentGroups"].apply(partialParams));
    }
    listDeploymentInstances(partialParams) {
        return this.client.listDeploymentInstances(this.ops["ListDeploymentInstances"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    registerApplicationRevision(partialParams) {
        return this.client.registerApplicationRevision(this.ops["RegisterApplicationRevision"].apply(partialParams));
    }
    registerOnPremisesInstance(partialParams) {
        return this.client.registerOnPremisesInstance(this.ops["RegisterOnPremisesInstance"].apply(partialParams));
    }
    removeTagsFromOnPremisesInstances(partialParams) {
        return this.client.removeTagsFromOnPremisesInstances(this.ops["RemoveTagsFromOnPremisesInstances"].apply(partialParams));
    }
    stopDeployment(partialParams) {
        return this.client.stopDeployment(this.ops["StopDeployment"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateDeploymentGroup(partialParams) {
        return this.client.updateDeploymentGroup(this.ops["UpdateDeploymentGroup"].apply(partialParams));
    }
}
exports.default = Resource;
