// out/servicecatalog/ServiceAction.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.servicecatalog.ServiceAction {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ServiceCatalog();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/servicecatalog-2015-12-10.normal.json"), this.client);
    }
    acceptPortfolioShare(partialParams) {
        return this.client.acceptPortfolioShare(this.ops["AcceptPortfolioShare"].apply(partialParams));
    }
    associateBudgetWithResource(partialParams) {
        return this.client.associateBudgetWithResource(this.ops["AssociateBudgetWithResource"].apply(partialParams));
    }
    associatePrincipalWithPortfolio(partialParams) {
        return this.client.associatePrincipalWithPortfolio(this.ops["AssociatePrincipalWithPortfolio"].apply(partialParams));
    }
    associateProductWithPortfolio(partialParams) {
        return this.client.associateProductWithPortfolio(this.ops["AssociateProductWithPortfolio"].apply(partialParams));
    }
    associateServiceActionWithProvisioningArtifact(partialParams) {
        return this.client.associateServiceActionWithProvisioningArtifact(this.ops["AssociateServiceActionWithProvisioningArtifact"].apply(partialParams));
    }
    associateTagOptionWithResource(partialParams) {
        return this.client.associateTagOptionWithResource(this.ops["AssociateTagOptionWithResource"].apply(partialParams));
    }
    batchAssociateServiceActionWithProvisioningArtifact(partialParams) {
        return this.client.batchAssociateServiceActionWithProvisioningArtifact(this.ops["BatchAssociateServiceActionWithProvisioningArtifact"].apply(partialParams));
    }
    batchDisassociateServiceActionFromProvisioningArtifact(partialParams) {
        return this.client.batchDisassociateServiceActionFromProvisioningArtifact(this.ops["BatchDisassociateServiceActionFromProvisioningArtifact"].apply(partialParams));
    }
    copyProduct(partialParams) {
        return this.client.copyProduct(this.ops["CopyProduct"].apply(partialParams));
    }
    createConstraint(partialParams) {
        return this.client.createConstraint(this.ops["CreateConstraint"].apply(partialParams));
    }
    createPortfolio(partialParams) {
        return this.client.createPortfolio(this.ops["CreatePortfolio"].apply(partialParams));
    }
    createPortfolioShare(partialParams) {
        return this.client.createPortfolioShare(this.ops["CreatePortfolioShare"].apply(partialParams));
    }
    createProduct(partialParams) {
        return this.client.createProduct(this.ops["CreateProduct"].apply(partialParams));
    }
    createProvisionedProductPlan(partialParams) {
        return this.client.createProvisionedProductPlan(this.ops["CreateProvisionedProductPlan"].apply(partialParams));
    }
    createProvisioningArtifact(partialParams) {
        return this.client.createProvisioningArtifact(this.ops["CreateProvisioningArtifact"].apply(partialParams));
    }
    createServiceAction(partialParams) {
        return this.client.createServiceAction(this.ops["CreateServiceAction"].apply(partialParams));
    }
    createTagOption(partialParams) {
        return this.client.createTagOption(this.ops["CreateTagOption"].apply(partialParams));
    }
    deleteConstraint(partialParams) {
        return this.client.deleteConstraint(this.ops["DeleteConstraint"].apply(partialParams));
    }
    deletePortfolio(partialParams) {
        return this.client.deletePortfolio(this.ops["DeletePortfolio"].apply(partialParams));
    }
    deletePortfolioShare(partialParams) {
        return this.client.deletePortfolioShare(this.ops["DeletePortfolioShare"].apply(partialParams));
    }
    deleteProduct(partialParams) {
        return this.client.deleteProduct(this.ops["DeleteProduct"].apply(partialParams));
    }
    deleteProvisionedProductPlan(partialParams) {
        return this.client.deleteProvisionedProductPlan(this.ops["DeleteProvisionedProductPlan"].apply(partialParams));
    }
    deleteProvisioningArtifact(partialParams) {
        return this.client.deleteProvisioningArtifact(this.ops["DeleteProvisioningArtifact"].apply(partialParams));
    }
    deleteServiceAction(partialParams) {
        return this.client.deleteServiceAction(this.ops["DeleteServiceAction"].apply(partialParams));
    }
    deleteTagOption(partialParams) {
        return this.client.deleteTagOption(this.ops["DeleteTagOption"].apply(partialParams));
    }
    describeConstraint(partialParams) {
        return this.client.describeConstraint(this.ops["DescribeConstraint"].apply(partialParams));
    }
    describeCopyProductStatus(partialParams) {
        return this.client.describeCopyProductStatus(this.ops["DescribeCopyProductStatus"].apply(partialParams));
    }
    describePortfolio(partialParams) {
        return this.client.describePortfolio(this.ops["DescribePortfolio"].apply(partialParams));
    }
    describePortfolioShareStatus(partialParams) {
        return this.client.describePortfolioShareStatus(this.ops["DescribePortfolioShareStatus"].apply(partialParams));
    }
    describePortfolioShares(partialParams) {
        return this.client.describePortfolioShares(this.ops["DescribePortfolioShares"].apply(partialParams));
    }
    describeProductView(partialParams) {
        return this.client.describeProductView(this.ops["DescribeProductView"].apply(partialParams));
    }
    describeProvisionedProductPlan(partialParams) {
        return this.client.describeProvisionedProductPlan(this.ops["DescribeProvisionedProductPlan"].apply(partialParams));
    }
    describeRecord(partialParams) {
        return this.client.describeRecord(this.ops["DescribeRecord"].apply(partialParams));
    }
    describeServiceAction(partialParams) {
        return this.client.describeServiceAction(this.ops["DescribeServiceAction"].apply(partialParams));
    }
    describeServiceActionExecutionParameters(partialParams) {
        return this.client.describeServiceActionExecutionParameters(this.ops["DescribeServiceActionExecutionParameters"].apply(partialParams));
    }
    describeTagOption(partialParams) {
        return this.client.describeTagOption(this.ops["DescribeTagOption"].apply(partialParams));
    }
    disassociateBudgetFromResource(partialParams) {
        return this.client.disassociateBudgetFromResource(this.ops["DisassociateBudgetFromResource"].apply(partialParams));
    }
    disassociatePrincipalFromPortfolio(partialParams) {
        return this.client.disassociatePrincipalFromPortfolio(this.ops["DisassociatePrincipalFromPortfolio"].apply(partialParams));
    }
    disassociateProductFromPortfolio(partialParams) {
        return this.client.disassociateProductFromPortfolio(this.ops["DisassociateProductFromPortfolio"].apply(partialParams));
    }
    disassociateServiceActionFromProvisioningArtifact(partialParams) {
        return this.client.disassociateServiceActionFromProvisioningArtifact(this.ops["DisassociateServiceActionFromProvisioningArtifact"].apply(partialParams));
    }
    disassociateTagOptionFromResource(partialParams) {
        return this.client.disassociateTagOptionFromResource(this.ops["DisassociateTagOptionFromResource"].apply(partialParams));
    }
    executeProvisionedProductPlan(partialParams) {
        return this.client.executeProvisionedProductPlan(this.ops["ExecuteProvisionedProductPlan"].apply(partialParams));
    }
    executeProvisionedProductServiceAction(partialParams) {
        return this.client.executeProvisionedProductServiceAction(this.ops["ExecuteProvisionedProductServiceAction"].apply(partialParams));
    }
    importAsProvisionedProduct(partialParams) {
        return this.client.importAsProvisionedProduct(this.ops["ImportAsProvisionedProduct"].apply(partialParams));
    }
    listBudgetsForResource(partialParams) {
        return this.client.listBudgetsForResource(this.ops["ListBudgetsForResource"].apply(partialParams));
    }
    listConstraintsForPortfolio(partialParams) {
        return this.client.listConstraintsForPortfolio(this.ops["ListConstraintsForPortfolio"].apply(partialParams));
    }
    listLaunchPaths(partialParams) {
        return this.client.listLaunchPaths(this.ops["ListLaunchPaths"].apply(partialParams));
    }
    listOrganizationPortfolioAccess(partialParams) {
        return this.client.listOrganizationPortfolioAccess(this.ops["ListOrganizationPortfolioAccess"].apply(partialParams));
    }
    listPortfolioAccess(partialParams) {
        return this.client.listPortfolioAccess(this.ops["ListPortfolioAccess"].apply(partialParams));
    }
    listPortfoliosForProduct(partialParams) {
        return this.client.listPortfoliosForProduct(this.ops["ListPortfoliosForProduct"].apply(partialParams));
    }
    listPrincipalsForPortfolio(partialParams) {
        return this.client.listPrincipalsForPortfolio(this.ops["ListPrincipalsForPortfolio"].apply(partialParams));
    }
    listProvisioningArtifacts(partialParams) {
        return this.client.listProvisioningArtifacts(this.ops["ListProvisioningArtifacts"].apply(partialParams));
    }
    listProvisioningArtifactsForServiceAction(partialParams) {
        return this.client.listProvisioningArtifactsForServiceAction(this.ops["ListProvisioningArtifactsForServiceAction"].apply(partialParams));
    }
    listResourcesForTagOption(partialParams) {
        return this.client.listResourcesForTagOption(this.ops["ListResourcesForTagOption"].apply(partialParams));
    }
    listServiceActionsForProvisioningArtifact(partialParams) {
        return this.client.listServiceActionsForProvisioningArtifact(this.ops["ListServiceActionsForProvisioningArtifact"].apply(partialParams));
    }
    listStackInstancesForProvisionedProduct(partialParams) {
        return this.client.listStackInstancesForProvisionedProduct(this.ops["ListStackInstancesForProvisionedProduct"].apply(partialParams));
    }
    provisionProduct(partialParams) {
        return this.client.provisionProduct(this.ops["ProvisionProduct"].apply(partialParams));
    }
    rejectPortfolioShare(partialParams) {
        return this.client.rejectPortfolioShare(this.ops["RejectPortfolioShare"].apply(partialParams));
    }
    terminateProvisionedProduct(partialParams) {
        return this.client.terminateProvisionedProduct(this.ops["TerminateProvisionedProduct"].apply(partialParams));
    }
    updateConstraint(partialParams) {
        return this.client.updateConstraint(this.ops["UpdateConstraint"].apply(partialParams));
    }
    updatePortfolio(partialParams) {
        return this.client.updatePortfolio(this.ops["UpdatePortfolio"].apply(partialParams));
    }
    updatePortfolioShare(partialParams) {
        return this.client.updatePortfolioShare(this.ops["UpdatePortfolioShare"].apply(partialParams));
    }
    updateProduct(partialParams) {
        return this.client.updateProduct(this.ops["UpdateProduct"].apply(partialParams));
    }
    updateProvisionedProduct(partialParams) {
        return this.client.updateProvisionedProduct(this.ops["UpdateProvisionedProduct"].apply(partialParams));
    }
    updateProvisionedProductProperties(partialParams) {
        return this.client.updateProvisionedProductProperties(this.ops["UpdateProvisionedProductProperties"].apply(partialParams));
    }
    updateProvisioningArtifact(partialParams) {
        return this.client.updateProvisioningArtifact(this.ops["UpdateProvisioningArtifact"].apply(partialParams));
    }
    updateServiceAction(partialParams) {
        return this.client.updateServiceAction(this.ops["UpdateServiceAction"].apply(partialParams));
    }
    updateTagOption(partialParams) {
        return this.client.updateTagOption(this.ops["UpdateTagOption"].apply(partialParams));
    }
}
exports.default = Resource;
