// out/ec2/VpnGatewayRoutePropagation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.ec2.VpnGatewayRoutePropagation {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.EC2();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/ec2-2016-11-15.normal.json"), this.client);
    }
    acceptReservedInstancesExchangeQuote(partialParams) {
        return this.client.acceptReservedInstancesExchangeQuote(this.ops["AcceptReservedInstancesExchangeQuote"].apply(partialParams));
    }
    acceptTransitGatewayPeeringAttachment(partialParams) {
        return this.client.acceptTransitGatewayPeeringAttachment(this.ops["AcceptTransitGatewayPeeringAttachment"].apply(partialParams));
    }
    acceptTransitGatewayVpcAttachment(partialParams) {
        return this.client.acceptTransitGatewayVpcAttachment(this.ops["AcceptTransitGatewayVpcAttachment"].apply(partialParams));
    }
    acceptVpcEndpointConnections(partialParams) {
        return this.client.acceptVpcEndpointConnections(this.ops["AcceptVpcEndpointConnections"].apply(partialParams));
    }
    advertiseByoipCidr(partialParams) {
        return this.client.advertiseByoipCidr(this.ops["AdvertiseByoipCidr"].apply(partialParams));
    }
    allocateHosts(partialParams) {
        return this.client.allocateHosts(this.ops["AllocateHosts"].apply(partialParams));
    }
    allocateIpamPoolCidr(partialParams) {
        return this.client.allocateIpamPoolCidr(this.ops["AllocateIpamPoolCidr"].apply(partialParams));
    }
    applySecurityGroupsToClientVpnTargetNetwork(partialParams) {
        return this.client.applySecurityGroupsToClientVpnTargetNetwork(this.ops["ApplySecurityGroupsToClientVpnTargetNetwork"].apply(partialParams));
    }
    assignIpv6Addresses(partialParams) {
        return this.client.assignIpv6Addresses(this.ops["AssignIpv6Addresses"].apply(partialParams));
    }
    assignPrivateIpAddresses(partialParams) {
        return this.client.assignPrivateIpAddresses(this.ops["AssignPrivateIpAddresses"].apply(partialParams));
    }
    associateClientVpnTargetNetwork(partialParams) {
        return this.client.associateClientVpnTargetNetwork(this.ops["AssociateClientVpnTargetNetwork"].apply(partialParams));
    }
    associateDhcpOptions(partialParams) {
        return this.client.associateDhcpOptions(this.ops["AssociateDhcpOptions"].apply(partialParams));
    }
    associateIamInstanceProfile(partialParams) {
        return this.client.associateIamInstanceProfile(this.ops["AssociateIamInstanceProfile"].apply(partialParams));
    }
    associateInstanceEventWindow(partialParams) {
        return this.client.associateInstanceEventWindow(this.ops["AssociateInstanceEventWindow"].apply(partialParams));
    }
    associateRouteTable(partialParams) {
        return this.client.associateRouteTable(this.ops["AssociateRouteTable"].apply(partialParams));
    }
    associateSubnetCidrBlock(partialParams) {
        return this.client.associateSubnetCidrBlock(this.ops["AssociateSubnetCidrBlock"].apply(partialParams));
    }
    associateTransitGatewayRouteTable(partialParams) {
        return this.client.associateTransitGatewayRouteTable(this.ops["AssociateTransitGatewayRouteTable"].apply(partialParams));
    }
    associateTrunkInterface(partialParams) {
        return this.client.associateTrunkInterface(this.ops["AssociateTrunkInterface"].apply(partialParams));
    }
    associateVpcCidrBlock(partialParams) {
        return this.client.associateVpcCidrBlock(this.ops["AssociateVpcCidrBlock"].apply(partialParams));
    }
    attachClassicLinkVpc(partialParams) {
        return this.client.attachClassicLinkVpc(this.ops["AttachClassicLinkVpc"].apply(partialParams));
    }
    attachInternetGateway(partialParams) {
        return this.client.attachInternetGateway(this.ops["AttachInternetGateway"].apply(partialParams));
    }
    attachNetworkInterface(partialParams) {
        return this.client.attachNetworkInterface(this.ops["AttachNetworkInterface"].apply(partialParams));
    }
    attachVolume(partialParams) {
        return this.client.attachVolume(this.ops["AttachVolume"].apply(partialParams));
    }
    attachVpnGateway(partialParams) {
        return this.client.attachVpnGateway(this.ops["AttachVpnGateway"].apply(partialParams));
    }
    authorizeClientVpnIngress(partialParams) {
        return this.client.authorizeClientVpnIngress(this.ops["AuthorizeClientVpnIngress"].apply(partialParams));
    }
    authorizeSecurityGroupEgress(partialParams) {
        return this.client.authorizeSecurityGroupEgress(this.ops["AuthorizeSecurityGroupEgress"].apply(partialParams));
    }
    bundleInstance(partialParams) {
        return this.client.bundleInstance(this.ops["BundleInstance"].apply(partialParams));
    }
    cancelBundleTask(partialParams) {
        return this.client.cancelBundleTask(this.ops["CancelBundleTask"].apply(partialParams));
    }
    cancelCapacityReservation(partialParams) {
        return this.client.cancelCapacityReservation(this.ops["CancelCapacityReservation"].apply(partialParams));
    }
    cancelCapacityReservationFleets(partialParams) {
        return this.client.cancelCapacityReservationFleets(this.ops["CancelCapacityReservationFleets"].apply(partialParams));
    }
    cancelConversionTask(partialParams) {
        return this.client.cancelConversionTask(this.ops["CancelConversionTask"].apply(partialParams));
    }
    cancelExportTask(partialParams) {
        return this.client.cancelExportTask(this.ops["CancelExportTask"].apply(partialParams));
    }
    cancelReservedInstancesListing(partialParams) {
        return this.client.cancelReservedInstancesListing(this.ops["CancelReservedInstancesListing"].apply(partialParams));
    }
    cancelSpotFleetRequests(partialParams) {
        return this.client.cancelSpotFleetRequests(this.ops["CancelSpotFleetRequests"].apply(partialParams));
    }
    cancelSpotInstanceRequests(partialParams) {
        return this.client.cancelSpotInstanceRequests(this.ops["CancelSpotInstanceRequests"].apply(partialParams));
    }
    confirmProductInstance(partialParams) {
        return this.client.confirmProductInstance(this.ops["ConfirmProductInstance"].apply(partialParams));
    }
    copyFpgaImage(partialParams) {
        return this.client.copyFpgaImage(this.ops["CopyFpgaImage"].apply(partialParams));
    }
    copyImage(partialParams) {
        return this.client.copyImage(this.ops["CopyImage"].apply(partialParams));
    }
    copySnapshot(partialParams) {
        return this.client.copySnapshot(this.ops["CopySnapshot"].apply(partialParams));
    }
    createCapacityReservation(partialParams) {
        return this.client.createCapacityReservation(this.ops["CreateCapacityReservation"].apply(partialParams));
    }
    createCapacityReservationFleet(partialParams) {
        return this.client.createCapacityReservationFleet(this.ops["CreateCapacityReservationFleet"].apply(partialParams));
    }
    createCarrierGateway(partialParams) {
        return this.client.createCarrierGateway(this.ops["CreateCarrierGateway"].apply(partialParams));
    }
    createClientVpnEndpoint(partialParams) {
        return this.client.createClientVpnEndpoint(this.ops["CreateClientVpnEndpoint"].apply(partialParams));
    }
    createClientVpnRoute(partialParams) {
        return this.client.createClientVpnRoute(this.ops["CreateClientVpnRoute"].apply(partialParams));
    }
    createCustomerGateway(partialParams) {
        return this.client.createCustomerGateway(this.ops["CreateCustomerGateway"].apply(partialParams));
    }
    createDefaultSubnet(partialParams) {
        return this.client.createDefaultSubnet(this.ops["CreateDefaultSubnet"].apply(partialParams));
    }
    createDhcpOptions(partialParams) {
        return this.client.createDhcpOptions(this.ops["CreateDhcpOptions"].apply(partialParams));
    }
    createEgressOnlyInternetGateway(partialParams) {
        return this.client.createEgressOnlyInternetGateway(this.ops["CreateEgressOnlyInternetGateway"].apply(partialParams));
    }
    createFleet(partialParams) {
        return this.client.createFleet(this.ops["CreateFleet"].apply(partialParams));
    }
    createFlowLogs(partialParams) {
        return this.client.createFlowLogs(this.ops["CreateFlowLogs"].apply(partialParams));
    }
    createFpgaImage(partialParams) {
        return this.client.createFpgaImage(this.ops["CreateFpgaImage"].apply(partialParams));
    }
    createImage(partialParams) {
        return this.client.createImage(this.ops["CreateImage"].apply(partialParams));
    }
    createInstanceExportTask(partialParams) {
        return this.client.createInstanceExportTask(this.ops["CreateInstanceExportTask"].apply(partialParams));
    }
    createIpamPool(partialParams) {
        return this.client.createIpamPool(this.ops["CreateIpamPool"].apply(partialParams));
    }
    createIpamScope(partialParams) {
        return this.client.createIpamScope(this.ops["CreateIpamScope"].apply(partialParams));
    }
    createKeyPair(partialParams) {
        return this.client.createKeyPair(this.ops["CreateKeyPair"].apply(partialParams));
    }
    createLaunchTemplate(partialParams) {
        return this.client.createLaunchTemplate(this.ops["CreateLaunchTemplate"].apply(partialParams));
    }
    createLaunchTemplateVersion(partialParams) {
        return this.client.createLaunchTemplateVersion(this.ops["CreateLaunchTemplateVersion"].apply(partialParams));
    }
    createLocalGatewayRoute(partialParams) {
        return this.client.createLocalGatewayRoute(this.ops["CreateLocalGatewayRoute"].apply(partialParams));
    }
    createLocalGatewayRouteTableVpcAssociation(partialParams) {
        return this.client.createLocalGatewayRouteTableVpcAssociation(this.ops["CreateLocalGatewayRouteTableVpcAssociation"].apply(partialParams));
    }
    createManagedPrefixList(partialParams) {
        return this.client.createManagedPrefixList(this.ops["CreateManagedPrefixList"].apply(partialParams));
    }
    createNatGateway(partialParams) {
        return this.client.createNatGateway(this.ops["CreateNatGateway"].apply(partialParams));
    }
    createNetworkAcl(partialParams) {
        return this.client.createNetworkAcl(this.ops["CreateNetworkAcl"].apply(partialParams));
    }
    createNetworkAclEntry(partialParams) {
        return this.client.createNetworkAclEntry(this.ops["CreateNetworkAclEntry"].apply(partialParams));
    }
    createNetworkInsightsAccessScope(partialParams) {
        return this.client.createNetworkInsightsAccessScope(this.ops["CreateNetworkInsightsAccessScope"].apply(partialParams));
    }
    createNetworkInsightsPath(partialParams) {
        return this.client.createNetworkInsightsPath(this.ops["CreateNetworkInsightsPath"].apply(partialParams));
    }
    createNetworkInterface(partialParams) {
        return this.client.createNetworkInterface(this.ops["CreateNetworkInterface"].apply(partialParams));
    }
    createNetworkInterfacePermission(partialParams) {
        return this.client.createNetworkInterfacePermission(this.ops["CreateNetworkInterfacePermission"].apply(partialParams));
    }
    createReplaceRootVolumeTask(partialParams) {
        return this.client.createReplaceRootVolumeTask(this.ops["CreateReplaceRootVolumeTask"].apply(partialParams));
    }
    createReservedInstancesListing(partialParams) {
        return this.client.createReservedInstancesListing(this.ops["CreateReservedInstancesListing"].apply(partialParams));
    }
    createRestoreImageTask(partialParams) {
        return this.client.createRestoreImageTask(this.ops["CreateRestoreImageTask"].apply(partialParams));
    }
    createRoute(partialParams) {
        return this.client.createRoute(this.ops["CreateRoute"].apply(partialParams));
    }
    createRouteTable(partialParams) {
        return this.client.createRouteTable(this.ops["CreateRouteTable"].apply(partialParams));
    }
    createSecurityGroup(partialParams) {
        return this.client.createSecurityGroup(this.ops["CreateSecurityGroup"].apply(partialParams));
    }
    createSnapshot(partialParams) {
        return this.client.createSnapshot(this.ops["CreateSnapshot"].apply(partialParams));
    }
    createSnapshots(partialParams) {
        return this.client.createSnapshots(this.ops["CreateSnapshots"].apply(partialParams));
    }
    createSpotDatafeedSubscription(partialParams) {
        return this.client.createSpotDatafeedSubscription(this.ops["CreateSpotDatafeedSubscription"].apply(partialParams));
    }
    createStoreImageTask(partialParams) {
        return this.client.createStoreImageTask(this.ops["CreateStoreImageTask"].apply(partialParams));
    }
    createSubnet(partialParams) {
        return this.client.createSubnet(this.ops["CreateSubnet"].apply(partialParams));
    }
    createSubnetCidrReservation(partialParams) {
        return this.client.createSubnetCidrReservation(this.ops["CreateSubnetCidrReservation"].apply(partialParams));
    }
    createTags(partialParams) {
        return this.client.createTags(this.ops["CreateTags"].apply(partialParams));
    }
    createTrafficMirrorFilterRule(partialParams) {
        return this.client.createTrafficMirrorFilterRule(this.ops["CreateTrafficMirrorFilterRule"].apply(partialParams));
    }
    createTrafficMirrorSession(partialParams) {
        return this.client.createTrafficMirrorSession(this.ops["CreateTrafficMirrorSession"].apply(partialParams));
    }
    createTransitGatewayConnect(partialParams) {
        return this.client.createTransitGatewayConnect(this.ops["CreateTransitGatewayConnect"].apply(partialParams));
    }
    createTransitGatewayConnectPeer(partialParams) {
        return this.client.createTransitGatewayConnectPeer(this.ops["CreateTransitGatewayConnectPeer"].apply(partialParams));
    }
    createTransitGatewayMulticastDomain(partialParams) {
        return this.client.createTransitGatewayMulticastDomain(this.ops["CreateTransitGatewayMulticastDomain"].apply(partialParams));
    }
    createTransitGatewayPeeringAttachment(partialParams) {
        return this.client.createTransitGatewayPeeringAttachment(this.ops["CreateTransitGatewayPeeringAttachment"].apply(partialParams));
    }
    createTransitGatewayPrefixListReference(partialParams) {
        return this.client.createTransitGatewayPrefixListReference(this.ops["CreateTransitGatewayPrefixListReference"].apply(partialParams));
    }
    createTransitGatewayRoute(partialParams) {
        return this.client.createTransitGatewayRoute(this.ops["CreateTransitGatewayRoute"].apply(partialParams));
    }
    createTransitGatewayRouteTable(partialParams) {
        return this.client.createTransitGatewayRouteTable(this.ops["CreateTransitGatewayRouteTable"].apply(partialParams));
    }
    createTransitGatewayVpcAttachment(partialParams) {
        return this.client.createTransitGatewayVpcAttachment(this.ops["CreateTransitGatewayVpcAttachment"].apply(partialParams));
    }
    createVolume(partialParams) {
        return this.client.createVolume(this.ops["CreateVolume"].apply(partialParams));
    }
    createVpcEndpoint(partialParams) {
        return this.client.createVpcEndpoint(this.ops["CreateVpcEndpoint"].apply(partialParams));
    }
    createVpcEndpointConnectionNotification(partialParams) {
        return this.client.createVpcEndpointConnectionNotification(this.ops["CreateVpcEndpointConnectionNotification"].apply(partialParams));
    }
    createVpnConnection(partialParams) {
        return this.client.createVpnConnection(this.ops["CreateVpnConnection"].apply(partialParams));
    }
    createVpnConnectionRoute(partialParams) {
        return this.client.createVpnConnectionRoute(this.ops["CreateVpnConnectionRoute"].apply(partialParams));
    }
    createVpnGateway(partialParams) {
        return this.client.createVpnGateway(this.ops["CreateVpnGateway"].apply(partialParams));
    }
    deleteCarrierGateway(partialParams) {
        return this.client.deleteCarrierGateway(this.ops["DeleteCarrierGateway"].apply(partialParams));
    }
    deleteClientVpnEndpoint(partialParams) {
        return this.client.deleteClientVpnEndpoint(this.ops["DeleteClientVpnEndpoint"].apply(partialParams));
    }
    deleteClientVpnRoute(partialParams) {
        return this.client.deleteClientVpnRoute(this.ops["DeleteClientVpnRoute"].apply(partialParams));
    }
    deleteCustomerGateway(partialParams) {
        return this.client.deleteCustomerGateway(this.ops["DeleteCustomerGateway"].apply(partialParams));
    }
    deleteDhcpOptions(partialParams) {
        return this.client.deleteDhcpOptions(this.ops["DeleteDhcpOptions"].apply(partialParams));
    }
    deleteEgressOnlyInternetGateway(partialParams) {
        return this.client.deleteEgressOnlyInternetGateway(this.ops["DeleteEgressOnlyInternetGateway"].apply(partialParams));
    }
    deleteFleets(partialParams) {
        return this.client.deleteFleets(this.ops["DeleteFleets"].apply(partialParams));
    }
    deleteFlowLogs(partialParams) {
        return this.client.deleteFlowLogs(this.ops["DeleteFlowLogs"].apply(partialParams));
    }
    deleteFpgaImage(partialParams) {
        return this.client.deleteFpgaImage(this.ops["DeleteFpgaImage"].apply(partialParams));
    }
    deleteInstanceEventWindow(partialParams) {
        return this.client.deleteInstanceEventWindow(this.ops["DeleteInstanceEventWindow"].apply(partialParams));
    }
    deleteInternetGateway(partialParams) {
        return this.client.deleteInternetGateway(this.ops["DeleteInternetGateway"].apply(partialParams));
    }
    deleteIpam(partialParams) {
        return this.client.deleteIpam(this.ops["DeleteIpam"].apply(partialParams));
    }
    deleteIpamPool(partialParams) {
        return this.client.deleteIpamPool(this.ops["DeleteIpamPool"].apply(partialParams));
    }
    deleteIpamScope(partialParams) {
        return this.client.deleteIpamScope(this.ops["DeleteIpamScope"].apply(partialParams));
    }
    deleteLaunchTemplateVersions(partialParams) {
        return this.client.deleteLaunchTemplateVersions(this.ops["DeleteLaunchTemplateVersions"].apply(partialParams));
    }
    deleteLocalGatewayRoute(partialParams) {
        return this.client.deleteLocalGatewayRoute(this.ops["DeleteLocalGatewayRoute"].apply(partialParams));
    }
    deleteLocalGatewayRouteTableVpcAssociation(partialParams) {
        return this.client.deleteLocalGatewayRouteTableVpcAssociation(this.ops["DeleteLocalGatewayRouteTableVpcAssociation"].apply(partialParams));
    }
    deleteManagedPrefixList(partialParams) {
        return this.client.deleteManagedPrefixList(this.ops["DeleteManagedPrefixList"].apply(partialParams));
    }
    deleteNatGateway(partialParams) {
        return this.client.deleteNatGateway(this.ops["DeleteNatGateway"].apply(partialParams));
    }
    deleteNetworkAcl(partialParams) {
        return this.client.deleteNetworkAcl(this.ops["DeleteNetworkAcl"].apply(partialParams));
    }
    deleteNetworkAclEntry(partialParams) {
        return this.client.deleteNetworkAclEntry(this.ops["DeleteNetworkAclEntry"].apply(partialParams));
    }
    deleteNetworkInsightsAccessScope(partialParams) {
        return this.client.deleteNetworkInsightsAccessScope(this.ops["DeleteNetworkInsightsAccessScope"].apply(partialParams));
    }
    deleteNetworkInsightsAccessScopeAnalysis(partialParams) {
        return this.client.deleteNetworkInsightsAccessScopeAnalysis(this.ops["DeleteNetworkInsightsAccessScopeAnalysis"].apply(partialParams));
    }
    deleteNetworkInsightsAnalysis(partialParams) {
        return this.client.deleteNetworkInsightsAnalysis(this.ops["DeleteNetworkInsightsAnalysis"].apply(partialParams));
    }
    deleteNetworkInsightsPath(partialParams) {
        return this.client.deleteNetworkInsightsPath(this.ops["DeleteNetworkInsightsPath"].apply(partialParams));
    }
    deleteNetworkInterface(partialParams) {
        return this.client.deleteNetworkInterface(this.ops["DeleteNetworkInterface"].apply(partialParams));
    }
    deleteNetworkInterfacePermission(partialParams) {
        return this.client.deleteNetworkInterfacePermission(this.ops["DeleteNetworkInterfacePermission"].apply(partialParams));
    }
    deletePlacementGroup(partialParams) {
        return this.client.deletePlacementGroup(this.ops["DeletePlacementGroup"].apply(partialParams));
    }
    deletePublicIpv4Pool(partialParams) {
        return this.client.deletePublicIpv4Pool(this.ops["DeletePublicIpv4Pool"].apply(partialParams));
    }
    deleteQueuedReservedInstances(partialParams) {
        return this.client.deleteQueuedReservedInstances(this.ops["DeleteQueuedReservedInstances"].apply(partialParams));
    }
    deleteRoute(partialParams) {
        return this.client.deleteRoute(this.ops["DeleteRoute"].apply(partialParams));
    }
    deleteRouteTable(partialParams) {
        return this.client.deleteRouteTable(this.ops["DeleteRouteTable"].apply(partialParams));
    }
    deleteSnapshot(partialParams) {
        return this.client.deleteSnapshot(this.ops["DeleteSnapshot"].apply(partialParams));
    }
    deleteSubnet(partialParams) {
        return this.client.deleteSubnet(this.ops["DeleteSubnet"].apply(partialParams));
    }
    deleteSubnetCidrReservation(partialParams) {
        return this.client.deleteSubnetCidrReservation(this.ops["DeleteSubnetCidrReservation"].apply(partialParams));
    }
    deleteTags(partialParams) {
        return this.client.deleteTags(this.ops["DeleteTags"].apply(partialParams));
    }
    deleteTrafficMirrorFilter(partialParams) {
        return this.client.deleteTrafficMirrorFilter(this.ops["DeleteTrafficMirrorFilter"].apply(partialParams));
    }
    deleteTrafficMirrorFilterRule(partialParams) {
        return this.client.deleteTrafficMirrorFilterRule(this.ops["DeleteTrafficMirrorFilterRule"].apply(partialParams));
    }
    deleteTrafficMirrorSession(partialParams) {
        return this.client.deleteTrafficMirrorSession(this.ops["DeleteTrafficMirrorSession"].apply(partialParams));
    }
    deleteTrafficMirrorTarget(partialParams) {
        return this.client.deleteTrafficMirrorTarget(this.ops["DeleteTrafficMirrorTarget"].apply(partialParams));
    }
    deleteTransitGateway(partialParams) {
        return this.client.deleteTransitGateway(this.ops["DeleteTransitGateway"].apply(partialParams));
    }
    deleteTransitGatewayConnect(partialParams) {
        return this.client.deleteTransitGatewayConnect(this.ops["DeleteTransitGatewayConnect"].apply(partialParams));
    }
    deleteTransitGatewayConnectPeer(partialParams) {
        return this.client.deleteTransitGatewayConnectPeer(this.ops["DeleteTransitGatewayConnectPeer"].apply(partialParams));
    }
    deleteTransitGatewayMulticastDomain(partialParams) {
        return this.client.deleteTransitGatewayMulticastDomain(this.ops["DeleteTransitGatewayMulticastDomain"].apply(partialParams));
    }
    deleteTransitGatewayPeeringAttachment(partialParams) {
        return this.client.deleteTransitGatewayPeeringAttachment(this.ops["DeleteTransitGatewayPeeringAttachment"].apply(partialParams));
    }
    deleteTransitGatewayPrefixListReference(partialParams) {
        return this.client.deleteTransitGatewayPrefixListReference(this.ops["DeleteTransitGatewayPrefixListReference"].apply(partialParams));
    }
    deleteTransitGatewayRoute(partialParams) {
        return this.client.deleteTransitGatewayRoute(this.ops["DeleteTransitGatewayRoute"].apply(partialParams));
    }
    deleteTransitGatewayRouteTable(partialParams) {
        return this.client.deleteTransitGatewayRouteTable(this.ops["DeleteTransitGatewayRouteTable"].apply(partialParams));
    }
    deleteTransitGatewayVpcAttachment(partialParams) {
        return this.client.deleteTransitGatewayVpcAttachment(this.ops["DeleteTransitGatewayVpcAttachment"].apply(partialParams));
    }
    deleteVolume(partialParams) {
        return this.client.deleteVolume(this.ops["DeleteVolume"].apply(partialParams));
    }
    deleteVpc(partialParams) {
        return this.client.deleteVpc(this.ops["DeleteVpc"].apply(partialParams));
    }
    deleteVpcEndpointConnectionNotifications(partialParams) {
        return this.client.deleteVpcEndpointConnectionNotifications(this.ops["DeleteVpcEndpointConnectionNotifications"].apply(partialParams));
    }
    deleteVpcEndpointServiceConfigurations(partialParams) {
        return this.client.deleteVpcEndpointServiceConfigurations(this.ops["DeleteVpcEndpointServiceConfigurations"].apply(partialParams));
    }
    deleteVpcEndpoints(partialParams) {
        return this.client.deleteVpcEndpoints(this.ops["DeleteVpcEndpoints"].apply(partialParams));
    }
    deleteVpcPeeringConnection(partialParams) {
        return this.client.deleteVpcPeeringConnection(this.ops["DeleteVpcPeeringConnection"].apply(partialParams));
    }
    deleteVpnConnection(partialParams) {
        return this.client.deleteVpnConnection(this.ops["DeleteVpnConnection"].apply(partialParams));
    }
    deleteVpnConnectionRoute(partialParams) {
        return this.client.deleteVpnConnectionRoute(this.ops["DeleteVpnConnectionRoute"].apply(partialParams));
    }
    deleteVpnGateway(partialParams) {
        return this.client.deleteVpnGateway(this.ops["DeleteVpnGateway"].apply(partialParams));
    }
    deprovisionByoipCidr(partialParams) {
        return this.client.deprovisionByoipCidr(this.ops["DeprovisionByoipCidr"].apply(partialParams));
    }
    deprovisionIpamPoolCidr(partialParams) {
        return this.client.deprovisionIpamPoolCidr(this.ops["DeprovisionIpamPoolCidr"].apply(partialParams));
    }
    deprovisionPublicIpv4PoolCidr(partialParams) {
        return this.client.deprovisionPublicIpv4PoolCidr(this.ops["DeprovisionPublicIpv4PoolCidr"].apply(partialParams));
    }
    deregisterImage(partialParams) {
        return this.client.deregisterImage(this.ops["DeregisterImage"].apply(partialParams));
    }
    describeByoipCidrs(partialParams) {
        return this.client.describeByoipCidrs(this.ops["DescribeByoipCidrs"].apply(partialParams));
    }
    describeClientVpnAuthorizationRules(partialParams) {
        return this.client.describeClientVpnAuthorizationRules(this.ops["DescribeClientVpnAuthorizationRules"].apply(partialParams));
    }
    describeClientVpnConnections(partialParams) {
        return this.client.describeClientVpnConnections(this.ops["DescribeClientVpnConnections"].apply(partialParams));
    }
    describeClientVpnRoutes(partialParams) {
        return this.client.describeClientVpnRoutes(this.ops["DescribeClientVpnRoutes"].apply(partialParams));
    }
    describeClientVpnTargetNetworks(partialParams) {
        return this.client.describeClientVpnTargetNetworks(this.ops["DescribeClientVpnTargetNetworks"].apply(partialParams));
    }
    describeFleetHistory(partialParams) {
        return this.client.describeFleetHistory(this.ops["DescribeFleetHistory"].apply(partialParams));
    }
    describeFleetInstances(partialParams) {
        return this.client.describeFleetInstances(this.ops["DescribeFleetInstances"].apply(partialParams));
    }
    describeFpgaImageAttribute(partialParams) {
        return this.client.describeFpgaImageAttribute(this.ops["DescribeFpgaImageAttribute"].apply(partialParams));
    }
    describeIdentityIdFormat(partialParams) {
        return this.client.describeIdentityIdFormat(this.ops["DescribeIdentityIdFormat"].apply(partialParams));
    }
    describeImageAttribute(partialParams) {
        return this.client.describeImageAttribute(this.ops["DescribeImageAttribute"].apply(partialParams));
    }
    describeInstanceAttribute(partialParams) {
        return this.client.describeInstanceAttribute(this.ops["DescribeInstanceAttribute"].apply(partialParams));
    }
    describeNetworkInterfaceAttribute(partialParams) {
        return this.client.describeNetworkInterfaceAttribute(this.ops["DescribeNetworkInterfaceAttribute"].apply(partialParams));
    }
    describeScheduledInstanceAvailability(partialParams) {
        return this.client.describeScheduledInstanceAvailability(this.ops["DescribeScheduledInstanceAvailability"].apply(partialParams));
    }
    describeSecurityGroupReferences(partialParams) {
        return this.client.describeSecurityGroupReferences(this.ops["DescribeSecurityGroupReferences"].apply(partialParams));
    }
    describeSnapshotAttribute(partialParams) {
        return this.client.describeSnapshotAttribute(this.ops["DescribeSnapshotAttribute"].apply(partialParams));
    }
    describeSpotFleetInstances(partialParams) {
        return this.client.describeSpotFleetInstances(this.ops["DescribeSpotFleetInstances"].apply(partialParams));
    }
    describeSpotFleetRequestHistory(partialParams) {
        return this.client.describeSpotFleetRequestHistory(this.ops["DescribeSpotFleetRequestHistory"].apply(partialParams));
    }
    describeStaleSecurityGroups(partialParams) {
        return this.client.describeStaleSecurityGroups(this.ops["DescribeStaleSecurityGroups"].apply(partialParams));
    }
    describeVolumeAttribute(partialParams) {
        return this.client.describeVolumeAttribute(this.ops["DescribeVolumeAttribute"].apply(partialParams));
    }
    describeVpcAttribute(partialParams) {
        return this.client.describeVpcAttribute(this.ops["DescribeVpcAttribute"].apply(partialParams));
    }
    describeVpcEndpointServicePermissions(partialParams) {
        return this.client.describeVpcEndpointServicePermissions(this.ops["DescribeVpcEndpointServicePermissions"].apply(partialParams));
    }
    detachClassicLinkVpc(partialParams) {
        return this.client.detachClassicLinkVpc(this.ops["DetachClassicLinkVpc"].apply(partialParams));
    }
    detachInternetGateway(partialParams) {
        return this.client.detachInternetGateway(this.ops["DetachInternetGateway"].apply(partialParams));
    }
    detachNetworkInterface(partialParams) {
        return this.client.detachNetworkInterface(this.ops["DetachNetworkInterface"].apply(partialParams));
    }
    detachVolume(partialParams) {
        return this.client.detachVolume(this.ops["DetachVolume"].apply(partialParams));
    }
    detachVpnGateway(partialParams) {
        return this.client.detachVpnGateway(this.ops["DetachVpnGateway"].apply(partialParams));
    }
    disableFastLaunch(partialParams) {
        return this.client.disableFastLaunch(this.ops["DisableFastLaunch"].apply(partialParams));
    }
    disableFastSnapshotRestores(partialParams) {
        return this.client.disableFastSnapshotRestores(this.ops["DisableFastSnapshotRestores"].apply(partialParams));
    }
    disableImageDeprecation(partialParams) {
        return this.client.disableImageDeprecation(this.ops["DisableImageDeprecation"].apply(partialParams));
    }
    disableIpamOrganizationAdminAccount(partialParams) {
        return this.client.disableIpamOrganizationAdminAccount(this.ops["DisableIpamOrganizationAdminAccount"].apply(partialParams));
    }
    disableTransitGatewayRouteTablePropagation(partialParams) {
        return this.client.disableTransitGatewayRouteTablePropagation(this.ops["DisableTransitGatewayRouteTablePropagation"].apply(partialParams));
    }
    disableVgwRoutePropagation(partialParams) {
        return this.client.disableVgwRoutePropagation(this.ops["DisableVgwRoutePropagation"].apply(partialParams));
    }
    disableVpcClassicLink(partialParams) {
        return this.client.disableVpcClassicLink(this.ops["DisableVpcClassicLink"].apply(partialParams));
    }
    disassociateClientVpnTargetNetwork(partialParams) {
        return this.client.disassociateClientVpnTargetNetwork(this.ops["DisassociateClientVpnTargetNetwork"].apply(partialParams));
    }
    disassociateIamInstanceProfile(partialParams) {
        return this.client.disassociateIamInstanceProfile(this.ops["DisassociateIamInstanceProfile"].apply(partialParams));
    }
    disassociateInstanceEventWindow(partialParams) {
        return this.client.disassociateInstanceEventWindow(this.ops["DisassociateInstanceEventWindow"].apply(partialParams));
    }
    disassociateRouteTable(partialParams) {
        return this.client.disassociateRouteTable(this.ops["DisassociateRouteTable"].apply(partialParams));
    }
    disassociateSubnetCidrBlock(partialParams) {
        return this.client.disassociateSubnetCidrBlock(this.ops["DisassociateSubnetCidrBlock"].apply(partialParams));
    }
    disassociateTransitGatewayRouteTable(partialParams) {
        return this.client.disassociateTransitGatewayRouteTable(this.ops["DisassociateTransitGatewayRouteTable"].apply(partialParams));
    }
    disassociateTrunkInterface(partialParams) {
        return this.client.disassociateTrunkInterface(this.ops["DisassociateTrunkInterface"].apply(partialParams));
    }
    disassociateVpcCidrBlock(partialParams) {
        return this.client.disassociateVpcCidrBlock(this.ops["DisassociateVpcCidrBlock"].apply(partialParams));
    }
    enableFastLaunch(partialParams) {
        return this.client.enableFastLaunch(this.ops["EnableFastLaunch"].apply(partialParams));
    }
    enableFastSnapshotRestores(partialParams) {
        return this.client.enableFastSnapshotRestores(this.ops["EnableFastSnapshotRestores"].apply(partialParams));
    }
    enableImageDeprecation(partialParams) {
        return this.client.enableImageDeprecation(this.ops["EnableImageDeprecation"].apply(partialParams));
    }
    enableIpamOrganizationAdminAccount(partialParams) {
        return this.client.enableIpamOrganizationAdminAccount(this.ops["EnableIpamOrganizationAdminAccount"].apply(partialParams));
    }
    enableTransitGatewayRouteTablePropagation(partialParams) {
        return this.client.enableTransitGatewayRouteTablePropagation(this.ops["EnableTransitGatewayRouteTablePropagation"].apply(partialParams));
    }
    enableVgwRoutePropagation(partialParams) {
        return this.client.enableVgwRoutePropagation(this.ops["EnableVgwRoutePropagation"].apply(partialParams));
    }
    enableVolumeIO(partialParams) {
        return this.client.enableVolumeIO(this.ops["EnableVolumeIO"].apply(partialParams));
    }
    enableVpcClassicLink(partialParams) {
        return this.client.enableVpcClassicLink(this.ops["EnableVpcClassicLink"].apply(partialParams));
    }
    exportClientVpnClientCertificateRevocationList(partialParams) {
        return this.client.exportClientVpnClientCertificateRevocationList(this.ops["ExportClientVpnClientCertificateRevocationList"].apply(partialParams));
    }
    exportClientVpnClientConfiguration(partialParams) {
        return this.client.exportClientVpnClientConfiguration(this.ops["ExportClientVpnClientConfiguration"].apply(partialParams));
    }
    exportImage(partialParams) {
        return this.client.exportImage(this.ops["ExportImage"].apply(partialParams));
    }
    exportTransitGatewayRoutes(partialParams) {
        return this.client.exportTransitGatewayRoutes(this.ops["ExportTransitGatewayRoutes"].apply(partialParams));
    }
    getAssociatedIpv6PoolCidrs(partialParams) {
        return this.client.getAssociatedIpv6PoolCidrs(this.ops["GetAssociatedIpv6PoolCidrs"].apply(partialParams));
    }
    getCapacityReservationUsage(partialParams) {
        return this.client.getCapacityReservationUsage(this.ops["GetCapacityReservationUsage"].apply(partialParams));
    }
    getCoipPoolUsage(partialParams) {
        return this.client.getCoipPoolUsage(this.ops["GetCoipPoolUsage"].apply(partialParams));
    }
    getConsoleOutput(partialParams) {
        return this.client.getConsoleOutput(this.ops["GetConsoleOutput"].apply(partialParams));
    }
    getConsoleScreenshot(partialParams) {
        return this.client.getConsoleScreenshot(this.ops["GetConsoleScreenshot"].apply(partialParams));
    }
    getDefaultCreditSpecification(partialParams) {
        return this.client.getDefaultCreditSpecification(this.ops["GetDefaultCreditSpecification"].apply(partialParams));
    }
    getFlowLogsIntegrationTemplate(partialParams) {
        return this.client.getFlowLogsIntegrationTemplate(this.ops["GetFlowLogsIntegrationTemplate"].apply(partialParams));
    }
    getGroupsForCapacityReservation(partialParams) {
        return this.client.getGroupsForCapacityReservation(this.ops["GetGroupsForCapacityReservation"].apply(partialParams));
    }
    getHostReservationPurchasePreview(partialParams) {
        return this.client.getHostReservationPurchasePreview(this.ops["GetHostReservationPurchasePreview"].apply(partialParams));
    }
    getInstanceTypesFromInstanceRequirements(partialParams) {
        return this.client.getInstanceTypesFromInstanceRequirements(this.ops["GetInstanceTypesFromInstanceRequirements"].apply(partialParams));
    }
    getIpamAddressHistory(partialParams) {
        return this.client.getIpamAddressHistory(this.ops["GetIpamAddressHistory"].apply(partialParams));
    }
    getIpamPoolAllocations(partialParams) {
        return this.client.getIpamPoolAllocations(this.ops["GetIpamPoolAllocations"].apply(partialParams));
    }
    getIpamPoolCidrs(partialParams) {
        return this.client.getIpamPoolCidrs(this.ops["GetIpamPoolCidrs"].apply(partialParams));
    }
    getIpamResourceCidrs(partialParams) {
        return this.client.getIpamResourceCidrs(this.ops["GetIpamResourceCidrs"].apply(partialParams));
    }
    getLaunchTemplateData(partialParams) {
        return this.client.getLaunchTemplateData(this.ops["GetLaunchTemplateData"].apply(partialParams));
    }
    getManagedPrefixListAssociations(partialParams) {
        return this.client.getManagedPrefixListAssociations(this.ops["GetManagedPrefixListAssociations"].apply(partialParams));
    }
    getManagedPrefixListEntries(partialParams) {
        return this.client.getManagedPrefixListEntries(this.ops["GetManagedPrefixListEntries"].apply(partialParams));
    }
    getNetworkInsightsAccessScopeAnalysisFindings(partialParams) {
        return this.client.getNetworkInsightsAccessScopeAnalysisFindings(this.ops["GetNetworkInsightsAccessScopeAnalysisFindings"].apply(partialParams));
    }
    getNetworkInsightsAccessScopeContent(partialParams) {
        return this.client.getNetworkInsightsAccessScopeContent(this.ops["GetNetworkInsightsAccessScopeContent"].apply(partialParams));
    }
    getPasswordData(partialParams) {
        return this.client.getPasswordData(this.ops["GetPasswordData"].apply(partialParams));
    }
    getReservedInstancesExchangeQuote(partialParams) {
        return this.client.getReservedInstancesExchangeQuote(this.ops["GetReservedInstancesExchangeQuote"].apply(partialParams));
    }
    getSpotPlacementScores(partialParams) {
        return this.client.getSpotPlacementScores(this.ops["GetSpotPlacementScores"].apply(partialParams));
    }
    getSubnetCidrReservations(partialParams) {
        return this.client.getSubnetCidrReservations(this.ops["GetSubnetCidrReservations"].apply(partialParams));
    }
    getTransitGatewayAttachmentPropagations(partialParams) {
        return this.client.getTransitGatewayAttachmentPropagations(this.ops["GetTransitGatewayAttachmentPropagations"].apply(partialParams));
    }
    getTransitGatewayPrefixListReferences(partialParams) {
        return this.client.getTransitGatewayPrefixListReferences(this.ops["GetTransitGatewayPrefixListReferences"].apply(partialParams));
    }
    getTransitGatewayRouteTableAssociations(partialParams) {
        return this.client.getTransitGatewayRouteTableAssociations(this.ops["GetTransitGatewayRouteTableAssociations"].apply(partialParams));
    }
    getTransitGatewayRouteTablePropagations(partialParams) {
        return this.client.getTransitGatewayRouteTablePropagations(this.ops["GetTransitGatewayRouteTablePropagations"].apply(partialParams));
    }
    getVpnConnectionDeviceSampleConfiguration(partialParams) {
        return this.client.getVpnConnectionDeviceSampleConfiguration(this.ops["GetVpnConnectionDeviceSampleConfiguration"].apply(partialParams));
    }
    importClientVpnClientCertificateRevocationList(partialParams) {
        return this.client.importClientVpnClientCertificateRevocationList(this.ops["ImportClientVpnClientCertificateRevocationList"].apply(partialParams));
    }
    importInstance(partialParams) {
        return this.client.importInstance(this.ops["ImportInstance"].apply(partialParams));
    }
    importKeyPair(partialParams) {
        return this.client.importKeyPair(this.ops["ImportKeyPair"].apply(partialParams));
    }
    importVolume(partialParams) {
        return this.client.importVolume(this.ops["ImportVolume"].apply(partialParams));
    }
    modifyAddressAttribute(partialParams) {
        return this.client.modifyAddressAttribute(this.ops["ModifyAddressAttribute"].apply(partialParams));
    }
    modifyAvailabilityZoneGroup(partialParams) {
        return this.client.modifyAvailabilityZoneGroup(this.ops["ModifyAvailabilityZoneGroup"].apply(partialParams));
    }
    modifyCapacityReservation(partialParams) {
        return this.client.modifyCapacityReservation(this.ops["ModifyCapacityReservation"].apply(partialParams));
    }
    modifyCapacityReservationFleet(partialParams) {
        return this.client.modifyCapacityReservationFleet(this.ops["ModifyCapacityReservationFleet"].apply(partialParams));
    }
    modifyClientVpnEndpoint(partialParams) {
        return this.client.modifyClientVpnEndpoint(this.ops["ModifyClientVpnEndpoint"].apply(partialParams));
    }
    modifyDefaultCreditSpecification(partialParams) {
        return this.client.modifyDefaultCreditSpecification(this.ops["ModifyDefaultCreditSpecification"].apply(partialParams));
    }
    modifyEbsDefaultKmsKeyId(partialParams) {
        return this.client.modifyEbsDefaultKmsKeyId(this.ops["ModifyEbsDefaultKmsKeyId"].apply(partialParams));
    }
    modifyFleet(partialParams) {
        return this.client.modifyFleet(this.ops["ModifyFleet"].apply(partialParams));
    }
    modifyFpgaImageAttribute(partialParams) {
        return this.client.modifyFpgaImageAttribute(this.ops["ModifyFpgaImageAttribute"].apply(partialParams));
    }
    modifyHosts(partialParams) {
        return this.client.modifyHosts(this.ops["ModifyHosts"].apply(partialParams));
    }
    modifyIdFormat(partialParams) {
        return this.client.modifyIdFormat(this.ops["ModifyIdFormat"].apply(partialParams));
    }
    modifyIdentityIdFormat(partialParams) {
        return this.client.modifyIdentityIdFormat(this.ops["ModifyIdentityIdFormat"].apply(partialParams));
    }
    modifyImageAttribute(partialParams) {
        return this.client.modifyImageAttribute(this.ops["ModifyImageAttribute"].apply(partialParams));
    }
    modifyInstanceAttribute(partialParams) {
        return this.client.modifyInstanceAttribute(this.ops["ModifyInstanceAttribute"].apply(partialParams));
    }
    modifyInstanceCapacityReservationAttributes(partialParams) {
        return this.client.modifyInstanceCapacityReservationAttributes(this.ops["ModifyInstanceCapacityReservationAttributes"].apply(partialParams));
    }
    modifyInstanceCreditSpecification(partialParams) {
        return this.client.modifyInstanceCreditSpecification(this.ops["ModifyInstanceCreditSpecification"].apply(partialParams));
    }
    modifyInstanceEventStartTime(partialParams) {
        return this.client.modifyInstanceEventStartTime(this.ops["ModifyInstanceEventStartTime"].apply(partialParams));
    }
    modifyInstanceEventWindow(partialParams) {
        return this.client.modifyInstanceEventWindow(this.ops["ModifyInstanceEventWindow"].apply(partialParams));
    }
    modifyInstanceMetadataOptions(partialParams) {
        return this.client.modifyInstanceMetadataOptions(this.ops["ModifyInstanceMetadataOptions"].apply(partialParams));
    }
    modifyInstancePlacement(partialParams) {
        return this.client.modifyInstancePlacement(this.ops["ModifyInstancePlacement"].apply(partialParams));
    }
    modifyIpam(partialParams) {
        return this.client.modifyIpam(this.ops["ModifyIpam"].apply(partialParams));
    }
    modifyIpamPool(partialParams) {
        return this.client.modifyIpamPool(this.ops["ModifyIpamPool"].apply(partialParams));
    }
    modifyIpamResourceCidr(partialParams) {
        return this.client.modifyIpamResourceCidr(this.ops["ModifyIpamResourceCidr"].apply(partialParams));
    }
    modifyIpamScope(partialParams) {
        return this.client.modifyIpamScope(this.ops["ModifyIpamScope"].apply(partialParams));
    }
    modifyManagedPrefixList(partialParams) {
        return this.client.modifyManagedPrefixList(this.ops["ModifyManagedPrefixList"].apply(partialParams));
    }
    modifyNetworkInterfaceAttribute(partialParams) {
        return this.client.modifyNetworkInterfaceAttribute(this.ops["ModifyNetworkInterfaceAttribute"].apply(partialParams));
    }
    modifyReservedInstances(partialParams) {
        return this.client.modifyReservedInstances(this.ops["ModifyReservedInstances"].apply(partialParams));
    }
    modifySecurityGroupRules(partialParams) {
        return this.client.modifySecurityGroupRules(this.ops["ModifySecurityGroupRules"].apply(partialParams));
    }
    modifySnapshotAttribute(partialParams) {
        return this.client.modifySnapshotAttribute(this.ops["ModifySnapshotAttribute"].apply(partialParams));
    }
    modifySnapshotTier(partialParams) {
        return this.client.modifySnapshotTier(this.ops["ModifySnapshotTier"].apply(partialParams));
    }
    modifySpotFleetRequest(partialParams) {
        return this.client.modifySpotFleetRequest(this.ops["ModifySpotFleetRequest"].apply(partialParams));
    }
    modifySubnetAttribute(partialParams) {
        return this.client.modifySubnetAttribute(this.ops["ModifySubnetAttribute"].apply(partialParams));
    }
    modifyTrafficMirrorFilterNetworkServices(partialParams) {
        return this.client.modifyTrafficMirrorFilterNetworkServices(this.ops["ModifyTrafficMirrorFilterNetworkServices"].apply(partialParams));
    }
    modifyTrafficMirrorFilterRule(partialParams) {
        return this.client.modifyTrafficMirrorFilterRule(this.ops["ModifyTrafficMirrorFilterRule"].apply(partialParams));
    }
    modifyTrafficMirrorSession(partialParams) {
        return this.client.modifyTrafficMirrorSession(this.ops["ModifyTrafficMirrorSession"].apply(partialParams));
    }
    modifyTransitGateway(partialParams) {
        return this.client.modifyTransitGateway(this.ops["ModifyTransitGateway"].apply(partialParams));
    }
    modifyTransitGatewayPrefixListReference(partialParams) {
        return this.client.modifyTransitGatewayPrefixListReference(this.ops["ModifyTransitGatewayPrefixListReference"].apply(partialParams));
    }
    modifyTransitGatewayVpcAttachment(partialParams) {
        return this.client.modifyTransitGatewayVpcAttachment(this.ops["ModifyTransitGatewayVpcAttachment"].apply(partialParams));
    }
    modifyVolume(partialParams) {
        return this.client.modifyVolume(this.ops["ModifyVolume"].apply(partialParams));
    }
    modifyVolumeAttribute(partialParams) {
        return this.client.modifyVolumeAttribute(this.ops["ModifyVolumeAttribute"].apply(partialParams));
    }
    modifyVpcAttribute(partialParams) {
        return this.client.modifyVpcAttribute(this.ops["ModifyVpcAttribute"].apply(partialParams));
    }
    modifyVpcEndpoint(partialParams) {
        return this.client.modifyVpcEndpoint(this.ops["ModifyVpcEndpoint"].apply(partialParams));
    }
    modifyVpcEndpointConnectionNotification(partialParams) {
        return this.client.modifyVpcEndpointConnectionNotification(this.ops["ModifyVpcEndpointConnectionNotification"].apply(partialParams));
    }
    modifyVpcEndpointServiceConfiguration(partialParams) {
        return this.client.modifyVpcEndpointServiceConfiguration(this.ops["ModifyVpcEndpointServiceConfiguration"].apply(partialParams));
    }
    modifyVpcEndpointServicePayerResponsibility(partialParams) {
        return this.client.modifyVpcEndpointServicePayerResponsibility(this.ops["ModifyVpcEndpointServicePayerResponsibility"].apply(partialParams));
    }
    modifyVpcEndpointServicePermissions(partialParams) {
        return this.client.modifyVpcEndpointServicePermissions(this.ops["ModifyVpcEndpointServicePermissions"].apply(partialParams));
    }
    modifyVpcPeeringConnectionOptions(partialParams) {
        return this.client.modifyVpcPeeringConnectionOptions(this.ops["ModifyVpcPeeringConnectionOptions"].apply(partialParams));
    }
    modifyVpcTenancy(partialParams) {
        return this.client.modifyVpcTenancy(this.ops["ModifyVpcTenancy"].apply(partialParams));
    }
    modifyVpnConnection(partialParams) {
        return this.client.modifyVpnConnection(this.ops["ModifyVpnConnection"].apply(partialParams));
    }
    modifyVpnConnectionOptions(partialParams) {
        return this.client.modifyVpnConnectionOptions(this.ops["ModifyVpnConnectionOptions"].apply(partialParams));
    }
    modifyVpnTunnelCertificate(partialParams) {
        return this.client.modifyVpnTunnelCertificate(this.ops["ModifyVpnTunnelCertificate"].apply(partialParams));
    }
    modifyVpnTunnelOptions(partialParams) {
        return this.client.modifyVpnTunnelOptions(this.ops["ModifyVpnTunnelOptions"].apply(partialParams));
    }
    monitorInstances(partialParams) {
        return this.client.monitorInstances(this.ops["MonitorInstances"].apply(partialParams));
    }
    moveAddressToVpc(partialParams) {
        return this.client.moveAddressToVpc(this.ops["MoveAddressToVpc"].apply(partialParams));
    }
    moveByoipCidrToIpam(partialParams) {
        return this.client.moveByoipCidrToIpam(this.ops["MoveByoipCidrToIpam"].apply(partialParams));
    }
    provisionByoipCidr(partialParams) {
        return this.client.provisionByoipCidr(this.ops["ProvisionByoipCidr"].apply(partialParams));
    }
    provisionIpamPoolCidr(partialParams) {
        return this.client.provisionIpamPoolCidr(this.ops["ProvisionIpamPoolCidr"].apply(partialParams));
    }
    provisionPublicIpv4PoolCidr(partialParams) {
        return this.client.provisionPublicIpv4PoolCidr(this.ops["ProvisionPublicIpv4PoolCidr"].apply(partialParams));
    }
    purchaseHostReservation(partialParams) {
        return this.client.purchaseHostReservation(this.ops["PurchaseHostReservation"].apply(partialParams));
    }
    purchaseReservedInstancesOffering(partialParams) {
        return this.client.purchaseReservedInstancesOffering(this.ops["PurchaseReservedInstancesOffering"].apply(partialParams));
    }
    purchaseScheduledInstances(partialParams) {
        return this.client.purchaseScheduledInstances(this.ops["PurchaseScheduledInstances"].apply(partialParams));
    }
    rebootInstances(partialParams) {
        return this.client.rebootInstances(this.ops["RebootInstances"].apply(partialParams));
    }
    registerImage(partialParams) {
        return this.client.registerImage(this.ops["RegisterImage"].apply(partialParams));
    }
    rejectTransitGatewayPeeringAttachment(partialParams) {
        return this.client.rejectTransitGatewayPeeringAttachment(this.ops["RejectTransitGatewayPeeringAttachment"].apply(partialParams));
    }
    rejectTransitGatewayVpcAttachment(partialParams) {
        return this.client.rejectTransitGatewayVpcAttachment(this.ops["RejectTransitGatewayVpcAttachment"].apply(partialParams));
    }
    rejectVpcEndpointConnections(partialParams) {
        return this.client.rejectVpcEndpointConnections(this.ops["RejectVpcEndpointConnections"].apply(partialParams));
    }
    rejectVpcPeeringConnection(partialParams) {
        return this.client.rejectVpcPeeringConnection(this.ops["RejectVpcPeeringConnection"].apply(partialParams));
    }
    releaseHosts(partialParams) {
        return this.client.releaseHosts(this.ops["ReleaseHosts"].apply(partialParams));
    }
    releaseIpamPoolAllocation(partialParams) {
        return this.client.releaseIpamPoolAllocation(this.ops["ReleaseIpamPoolAllocation"].apply(partialParams));
    }
    replaceIamInstanceProfileAssociation(partialParams) {
        return this.client.replaceIamInstanceProfileAssociation(this.ops["ReplaceIamInstanceProfileAssociation"].apply(partialParams));
    }
    replaceNetworkAclAssociation(partialParams) {
        return this.client.replaceNetworkAclAssociation(this.ops["ReplaceNetworkAclAssociation"].apply(partialParams));
    }
    replaceNetworkAclEntry(partialParams) {
        return this.client.replaceNetworkAclEntry(this.ops["ReplaceNetworkAclEntry"].apply(partialParams));
    }
    replaceRoute(partialParams) {
        return this.client.replaceRoute(this.ops["ReplaceRoute"].apply(partialParams));
    }
    replaceRouteTableAssociation(partialParams) {
        return this.client.replaceRouteTableAssociation(this.ops["ReplaceRouteTableAssociation"].apply(partialParams));
    }
    replaceTransitGatewayRoute(partialParams) {
        return this.client.replaceTransitGatewayRoute(this.ops["ReplaceTransitGatewayRoute"].apply(partialParams));
    }
    reportInstanceStatus(partialParams) {
        return this.client.reportInstanceStatus(this.ops["ReportInstanceStatus"].apply(partialParams));
    }
    requestSpotFleet(partialParams) {
        return this.client.requestSpotFleet(this.ops["RequestSpotFleet"].apply(partialParams));
    }
    resetAddressAttribute(partialParams) {
        return this.client.resetAddressAttribute(this.ops["ResetAddressAttribute"].apply(partialParams));
    }
    resetFpgaImageAttribute(partialParams) {
        return this.client.resetFpgaImageAttribute(this.ops["ResetFpgaImageAttribute"].apply(partialParams));
    }
    resetImageAttribute(partialParams) {
        return this.client.resetImageAttribute(this.ops["ResetImageAttribute"].apply(partialParams));
    }
    resetInstanceAttribute(partialParams) {
        return this.client.resetInstanceAttribute(this.ops["ResetInstanceAttribute"].apply(partialParams));
    }
    resetNetworkInterfaceAttribute(partialParams) {
        return this.client.resetNetworkInterfaceAttribute(this.ops["ResetNetworkInterfaceAttribute"].apply(partialParams));
    }
    resetSnapshotAttribute(partialParams) {
        return this.client.resetSnapshotAttribute(this.ops["ResetSnapshotAttribute"].apply(partialParams));
    }
    restoreAddressToClassic(partialParams) {
        return this.client.restoreAddressToClassic(this.ops["RestoreAddressToClassic"].apply(partialParams));
    }
    restoreImageFromRecycleBin(partialParams) {
        return this.client.restoreImageFromRecycleBin(this.ops["RestoreImageFromRecycleBin"].apply(partialParams));
    }
    restoreManagedPrefixListVersion(partialParams) {
        return this.client.restoreManagedPrefixListVersion(this.ops["RestoreManagedPrefixListVersion"].apply(partialParams));
    }
    restoreSnapshotFromRecycleBin(partialParams) {
        return this.client.restoreSnapshotFromRecycleBin(this.ops["RestoreSnapshotFromRecycleBin"].apply(partialParams));
    }
    restoreSnapshotTier(partialParams) {
        return this.client.restoreSnapshotTier(this.ops["RestoreSnapshotTier"].apply(partialParams));
    }
    revokeClientVpnIngress(partialParams) {
        return this.client.revokeClientVpnIngress(this.ops["RevokeClientVpnIngress"].apply(partialParams));
    }
    revokeSecurityGroupEgress(partialParams) {
        return this.client.revokeSecurityGroupEgress(this.ops["RevokeSecurityGroupEgress"].apply(partialParams));
    }
    runInstances(partialParams) {
        return this.client.runInstances(this.ops["RunInstances"].apply(partialParams));
    }
    runScheduledInstances(partialParams) {
        return this.client.runScheduledInstances(this.ops["RunScheduledInstances"].apply(partialParams));
    }
    searchLocalGatewayRoutes(partialParams) {
        return this.client.searchLocalGatewayRoutes(this.ops["SearchLocalGatewayRoutes"].apply(partialParams));
    }
    searchTransitGatewayRoutes(partialParams) {
        return this.client.searchTransitGatewayRoutes(this.ops["SearchTransitGatewayRoutes"].apply(partialParams));
    }
    sendDiagnosticInterrupt(partialParams) {
        return this.client.sendDiagnosticInterrupt(this.ops["SendDiagnosticInterrupt"].apply(partialParams));
    }
    startInstances(partialParams) {
        return this.client.startInstances(this.ops["StartInstances"].apply(partialParams));
    }
    startNetworkInsightsAccessScopeAnalysis(partialParams) {
        return this.client.startNetworkInsightsAccessScopeAnalysis(this.ops["StartNetworkInsightsAccessScopeAnalysis"].apply(partialParams));
    }
    startNetworkInsightsAnalysis(partialParams) {
        return this.client.startNetworkInsightsAnalysis(this.ops["StartNetworkInsightsAnalysis"].apply(partialParams));
    }
    startVpcEndpointServicePrivateDnsVerification(partialParams) {
        return this.client.startVpcEndpointServicePrivateDnsVerification(this.ops["StartVpcEndpointServicePrivateDnsVerification"].apply(partialParams));
    }
    stopInstances(partialParams) {
        return this.client.stopInstances(this.ops["StopInstances"].apply(partialParams));
    }
    terminateClientVpnConnections(partialParams) {
        return this.client.terminateClientVpnConnections(this.ops["TerminateClientVpnConnections"].apply(partialParams));
    }
    terminateInstances(partialParams) {
        return this.client.terminateInstances(this.ops["TerminateInstances"].apply(partialParams));
    }
    unassignIpv6Addresses(partialParams) {
        return this.client.unassignIpv6Addresses(this.ops["UnassignIpv6Addresses"].apply(partialParams));
    }
    unassignPrivateIpAddresses(partialParams) {
        return this.client.unassignPrivateIpAddresses(this.ops["UnassignPrivateIpAddresses"].apply(partialParams));
    }
    unmonitorInstances(partialParams) {
        return this.client.unmonitorInstances(this.ops["UnmonitorInstances"].apply(partialParams));
    }
    withdrawByoipCidr(partialParams) {
        return this.client.withdrawByoipCidr(this.ops["WithdrawByoipCidr"].apply(partialParams));
    }
}
exports.default = Resource;
