// out/ssm/MaintenanceWindowTarget.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.ssm.MaintenanceWindowTarget {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.SSM();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/ssm-2014-11-06.normal.json"), this.client);
    }
    addTagsToResource(partialParams) {
        return this.client.addTagsToResource(this.ops["AddTagsToResource"].apply(partialParams));
    }
    associateOpsItemRelatedItem(partialParams) {
        return this.client.associateOpsItemRelatedItem(this.ops["AssociateOpsItemRelatedItem"].apply(partialParams));
    }
    cancelCommand(partialParams) {
        return this.client.cancelCommand(this.ops["CancelCommand"].apply(partialParams));
    }
    cancelMaintenanceWindowExecution(partialParams) {
        return this.client.cancelMaintenanceWindowExecution(this.ops["CancelMaintenanceWindowExecution"].apply(partialParams));
    }
    createActivation(partialParams) {
        return this.client.createActivation(this.ops["CreateActivation"].apply(partialParams));
    }
    createAssociation(partialParams) {
        return this.client.createAssociation(this.ops["CreateAssociation"].apply(partialParams));
    }
    createAssociationBatch(partialParams) {
        return this.client.createAssociationBatch(this.ops["CreateAssociationBatch"].apply(partialParams));
    }
    createDocument(partialParams) {
        return this.client.createDocument(this.ops["CreateDocument"].apply(partialParams));
    }
    createMaintenanceWindow(partialParams) {
        return this.client.createMaintenanceWindow(this.ops["CreateMaintenanceWindow"].apply(partialParams));
    }
    createOpsItem(partialParams) {
        return this.client.createOpsItem(this.ops["CreateOpsItem"].apply(partialParams));
    }
    createOpsMetadata(partialParams) {
        return this.client.createOpsMetadata(this.ops["CreateOpsMetadata"].apply(partialParams));
    }
    createPatchBaseline(partialParams) {
        return this.client.createPatchBaseline(this.ops["CreatePatchBaseline"].apply(partialParams));
    }
    createResourceDataSync(partialParams) {
        return this.client.createResourceDataSync(this.ops["CreateResourceDataSync"].apply(partialParams));
    }
    deleteActivation(partialParams) {
        return this.client.deleteActivation(this.ops["DeleteActivation"].apply(partialParams));
    }
    deleteDocument(partialParams) {
        return this.client.deleteDocument(this.ops["DeleteDocument"].apply(partialParams));
    }
    deleteInventory(partialParams) {
        return this.client.deleteInventory(this.ops["DeleteInventory"].apply(partialParams));
    }
    deleteMaintenanceWindow(partialParams) {
        return this.client.deleteMaintenanceWindow(this.ops["DeleteMaintenanceWindow"].apply(partialParams));
    }
    deleteOpsMetadata(partialParams) {
        return this.client.deleteOpsMetadata(this.ops["DeleteOpsMetadata"].apply(partialParams));
    }
    deleteParameter(partialParams) {
        return this.client.deleteParameter(this.ops["DeleteParameter"].apply(partialParams));
    }
    deleteParameters(partialParams) {
        return this.client.deleteParameters(this.ops["DeleteParameters"].apply(partialParams));
    }
    deletePatchBaseline(partialParams) {
        return this.client.deletePatchBaseline(this.ops["DeletePatchBaseline"].apply(partialParams));
    }
    deleteResourceDataSync(partialParams) {
        return this.client.deleteResourceDataSync(this.ops["DeleteResourceDataSync"].apply(partialParams));
    }
    deregisterManagedInstance(partialParams) {
        return this.client.deregisterManagedInstance(this.ops["DeregisterManagedInstance"].apply(partialParams));
    }
    deregisterPatchBaselineForPatchGroup(partialParams) {
        return this.client.deregisterPatchBaselineForPatchGroup(this.ops["DeregisterPatchBaselineForPatchGroup"].apply(partialParams));
    }
    deregisterTargetFromMaintenanceWindow(partialParams) {
        return this.client.deregisterTargetFromMaintenanceWindow(this.ops["DeregisterTargetFromMaintenanceWindow"].apply(partialParams));
    }
    deregisterTaskFromMaintenanceWindow(partialParams) {
        return this.client.deregisterTaskFromMaintenanceWindow(this.ops["DeregisterTaskFromMaintenanceWindow"].apply(partialParams));
    }
    describeAssociationExecutionTargets(partialParams) {
        return this.client.describeAssociationExecutionTargets(this.ops["DescribeAssociationExecutionTargets"].apply(partialParams));
    }
    describeAssociationExecutions(partialParams) {
        return this.client.describeAssociationExecutions(this.ops["DescribeAssociationExecutions"].apply(partialParams));
    }
    describeAutomationStepExecutions(partialParams) {
        return this.client.describeAutomationStepExecutions(this.ops["DescribeAutomationStepExecutions"].apply(partialParams));
    }
    describeDocument(partialParams) {
        return this.client.describeDocument(this.ops["DescribeDocument"].apply(partialParams));
    }
    describeDocumentPermission(partialParams) {
        return this.client.describeDocumentPermission(this.ops["DescribeDocumentPermission"].apply(partialParams));
    }
    describeEffectiveInstanceAssociations(partialParams) {
        return this.client.describeEffectiveInstanceAssociations(this.ops["DescribeEffectiveInstanceAssociations"].apply(partialParams));
    }
    describeEffectivePatchesForPatchBaseline(partialParams) {
        return this.client.describeEffectivePatchesForPatchBaseline(this.ops["DescribeEffectivePatchesForPatchBaseline"].apply(partialParams));
    }
    describeInstanceAssociationsStatus(partialParams) {
        return this.client.describeInstanceAssociationsStatus(this.ops["DescribeInstanceAssociationsStatus"].apply(partialParams));
    }
    describeInstancePatchStates(partialParams) {
        return this.client.describeInstancePatchStates(this.ops["DescribeInstancePatchStates"].apply(partialParams));
    }
    describeInstancePatchStatesForPatchGroup(partialParams) {
        return this.client.describeInstancePatchStatesForPatchGroup(this.ops["DescribeInstancePatchStatesForPatchGroup"].apply(partialParams));
    }
    describeInstancePatches(partialParams) {
        return this.client.describeInstancePatches(this.ops["DescribeInstancePatches"].apply(partialParams));
    }
    describeMaintenanceWindowExecutionTaskInvocations(partialParams) {
        return this.client.describeMaintenanceWindowExecutionTaskInvocations(this.ops["DescribeMaintenanceWindowExecutionTaskInvocations"].apply(partialParams));
    }
    describeMaintenanceWindowExecutionTasks(partialParams) {
        return this.client.describeMaintenanceWindowExecutionTasks(this.ops["DescribeMaintenanceWindowExecutionTasks"].apply(partialParams));
    }
    describeMaintenanceWindowExecutions(partialParams) {
        return this.client.describeMaintenanceWindowExecutions(this.ops["DescribeMaintenanceWindowExecutions"].apply(partialParams));
    }
    describeMaintenanceWindowTargets(partialParams) {
        return this.client.describeMaintenanceWindowTargets(this.ops["DescribeMaintenanceWindowTargets"].apply(partialParams));
    }
    describeMaintenanceWindowTasks(partialParams) {
        return this.client.describeMaintenanceWindowTasks(this.ops["DescribeMaintenanceWindowTasks"].apply(partialParams));
    }
    describeMaintenanceWindowsForTarget(partialParams) {
        return this.client.describeMaintenanceWindowsForTarget(this.ops["DescribeMaintenanceWindowsForTarget"].apply(partialParams));
    }
    describePatchGroupState(partialParams) {
        return this.client.describePatchGroupState(this.ops["DescribePatchGroupState"].apply(partialParams));
    }
    describePatchProperties(partialParams) {
        return this.client.describePatchProperties(this.ops["DescribePatchProperties"].apply(partialParams));
    }
    describeSessions(partialParams) {
        return this.client.describeSessions(this.ops["DescribeSessions"].apply(partialParams));
    }
    disassociateOpsItemRelatedItem(partialParams) {
        return this.client.disassociateOpsItemRelatedItem(this.ops["DisassociateOpsItemRelatedItem"].apply(partialParams));
    }
    getAutomationExecution(partialParams) {
        return this.client.getAutomationExecution(this.ops["GetAutomationExecution"].apply(partialParams));
    }
    getCalendarState(partialParams) {
        return this.client.getCalendarState(this.ops["GetCalendarState"].apply(partialParams));
    }
    getCommandInvocation(partialParams) {
        return this.client.getCommandInvocation(this.ops["GetCommandInvocation"].apply(partialParams));
    }
    getConnectionStatus(partialParams) {
        return this.client.getConnectionStatus(this.ops["GetConnectionStatus"].apply(partialParams));
    }
    getDeployablePatchSnapshotForInstance(partialParams) {
        return this.client.getDeployablePatchSnapshotForInstance(this.ops["GetDeployablePatchSnapshotForInstance"].apply(partialParams));
    }
    getDocument(partialParams) {
        return this.client.getDocument(this.ops["GetDocument"].apply(partialParams));
    }
    getMaintenanceWindow(partialParams) {
        return this.client.getMaintenanceWindow(this.ops["GetMaintenanceWindow"].apply(partialParams));
    }
    getMaintenanceWindowExecution(partialParams) {
        return this.client.getMaintenanceWindowExecution(this.ops["GetMaintenanceWindowExecution"].apply(partialParams));
    }
    getMaintenanceWindowExecutionTask(partialParams) {
        return this.client.getMaintenanceWindowExecutionTask(this.ops["GetMaintenanceWindowExecutionTask"].apply(partialParams));
    }
    getMaintenanceWindowExecutionTaskInvocation(partialParams) {
        return this.client.getMaintenanceWindowExecutionTaskInvocation(this.ops["GetMaintenanceWindowExecutionTaskInvocation"].apply(partialParams));
    }
    getMaintenanceWindowTask(partialParams) {
        return this.client.getMaintenanceWindowTask(this.ops["GetMaintenanceWindowTask"].apply(partialParams));
    }
    getOpsItem(partialParams) {
        return this.client.getOpsItem(this.ops["GetOpsItem"].apply(partialParams));
    }
    getOpsMetadata(partialParams) {
        return this.client.getOpsMetadata(this.ops["GetOpsMetadata"].apply(partialParams));
    }
    getParameter(partialParams) {
        return this.client.getParameter(this.ops["GetParameter"].apply(partialParams));
    }
    getParameterHistory(partialParams) {
        return this.client.getParameterHistory(this.ops["GetParameterHistory"].apply(partialParams));
    }
    getParameters(partialParams) {
        return this.client.getParameters(this.ops["GetParameters"].apply(partialParams));
    }
    getParametersByPath(partialParams) {
        return this.client.getParametersByPath(this.ops["GetParametersByPath"].apply(partialParams));
    }
    getPatchBaseline(partialParams) {
        return this.client.getPatchBaseline(this.ops["GetPatchBaseline"].apply(partialParams));
    }
    getPatchBaselineForPatchGroup(partialParams) {
        return this.client.getPatchBaselineForPatchGroup(this.ops["GetPatchBaselineForPatchGroup"].apply(partialParams));
    }
    getServiceSetting(partialParams) {
        return this.client.getServiceSetting(this.ops["GetServiceSetting"].apply(partialParams));
    }
    labelParameterVersion(partialParams) {
        return this.client.labelParameterVersion(this.ops["LabelParameterVersion"].apply(partialParams));
    }
    listAssociationVersions(partialParams) {
        return this.client.listAssociationVersions(this.ops["ListAssociationVersions"].apply(partialParams));
    }
    listDocumentMetadataHistory(partialParams) {
        return this.client.listDocumentMetadataHistory(this.ops["ListDocumentMetadataHistory"].apply(partialParams));
    }
    listDocumentVersions(partialParams) {
        return this.client.listDocumentVersions(this.ops["ListDocumentVersions"].apply(partialParams));
    }
    listInventoryEntries(partialParams) {
        return this.client.listInventoryEntries(this.ops["ListInventoryEntries"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    modifyDocumentPermission(partialParams) {
        return this.client.modifyDocumentPermission(this.ops["ModifyDocumentPermission"].apply(partialParams));
    }
    putComplianceItems(partialParams) {
        return this.client.putComplianceItems(this.ops["PutComplianceItems"].apply(partialParams));
    }
    putInventory(partialParams) {
        return this.client.putInventory(this.ops["PutInventory"].apply(partialParams));
    }
    putParameter(partialParams) {
        return this.client.putParameter(this.ops["PutParameter"].apply(partialParams));
    }
    registerDefaultPatchBaseline(partialParams) {
        return this.client.registerDefaultPatchBaseline(this.ops["RegisterDefaultPatchBaseline"].apply(partialParams));
    }
    registerPatchBaselineForPatchGroup(partialParams) {
        return this.client.registerPatchBaselineForPatchGroup(this.ops["RegisterPatchBaselineForPatchGroup"].apply(partialParams));
    }
    registerTargetWithMaintenanceWindow(partialParams) {
        return this.client.registerTargetWithMaintenanceWindow(this.ops["RegisterTargetWithMaintenanceWindow"].apply(partialParams));
    }
    registerTaskWithMaintenanceWindow(partialParams) {
        return this.client.registerTaskWithMaintenanceWindow(this.ops["RegisterTaskWithMaintenanceWindow"].apply(partialParams));
    }
    removeTagsFromResource(partialParams) {
        return this.client.removeTagsFromResource(this.ops["RemoveTagsFromResource"].apply(partialParams));
    }
    resetServiceSetting(partialParams) {
        return this.client.resetServiceSetting(this.ops["ResetServiceSetting"].apply(partialParams));
    }
    resumeSession(partialParams) {
        return this.client.resumeSession(this.ops["ResumeSession"].apply(partialParams));
    }
    sendAutomationSignal(partialParams) {
        return this.client.sendAutomationSignal(this.ops["SendAutomationSignal"].apply(partialParams));
    }
    sendCommand(partialParams) {
        return this.client.sendCommand(this.ops["SendCommand"].apply(partialParams));
    }
    startAssociationsOnce(partialParams) {
        return this.client.startAssociationsOnce(this.ops["StartAssociationsOnce"].apply(partialParams));
    }
    startAutomationExecution(partialParams) {
        return this.client.startAutomationExecution(this.ops["StartAutomationExecution"].apply(partialParams));
    }
    startChangeRequestExecution(partialParams) {
        return this.client.startChangeRequestExecution(this.ops["StartChangeRequestExecution"].apply(partialParams));
    }
    startSession(partialParams) {
        return this.client.startSession(this.ops["StartSession"].apply(partialParams));
    }
    stopAutomationExecution(partialParams) {
        return this.client.stopAutomationExecution(this.ops["StopAutomationExecution"].apply(partialParams));
    }
    terminateSession(partialParams) {
        return this.client.terminateSession(this.ops["TerminateSession"].apply(partialParams));
    }
    unlabelParameterVersion(partialParams) {
        return this.client.unlabelParameterVersion(this.ops["UnlabelParameterVersion"].apply(partialParams));
    }
    updateAssociation(partialParams) {
        return this.client.updateAssociation(this.ops["UpdateAssociation"].apply(partialParams));
    }
    updateAssociationStatus(partialParams) {
        return this.client.updateAssociationStatus(this.ops["UpdateAssociationStatus"].apply(partialParams));
    }
    updateDocument(partialParams) {
        return this.client.updateDocument(this.ops["UpdateDocument"].apply(partialParams));
    }
    updateDocumentDefaultVersion(partialParams) {
        return this.client.updateDocumentDefaultVersion(this.ops["UpdateDocumentDefaultVersion"].apply(partialParams));
    }
    updateDocumentMetadata(partialParams) {
        return this.client.updateDocumentMetadata(this.ops["UpdateDocumentMetadata"].apply(partialParams));
    }
    updateMaintenanceWindow(partialParams) {
        return this.client.updateMaintenanceWindow(this.ops["UpdateMaintenanceWindow"].apply(partialParams));
    }
    updateMaintenanceWindowTarget(partialParams) {
        return this.client.updateMaintenanceWindowTarget(this.ops["UpdateMaintenanceWindowTarget"].apply(partialParams));
    }
    updateMaintenanceWindowTask(partialParams) {
        return this.client.updateMaintenanceWindowTask(this.ops["UpdateMaintenanceWindowTask"].apply(partialParams));
    }
    updateManagedInstanceRole(partialParams) {
        return this.client.updateManagedInstanceRole(this.ops["UpdateManagedInstanceRole"].apply(partialParams));
    }
    updateOpsItem(partialParams) {
        return this.client.updateOpsItem(this.ops["UpdateOpsItem"].apply(partialParams));
    }
    updateOpsMetadata(partialParams) {
        return this.client.updateOpsMetadata(this.ops["UpdateOpsMetadata"].apply(partialParams));
    }
    updatePatchBaseline(partialParams) {
        return this.client.updatePatchBaseline(this.ops["UpdatePatchBaseline"].apply(partialParams));
    }
    updateResourceDataSync(partialParams) {
        return this.client.updateResourceDataSync(this.ops["UpdateResourceDataSync"].apply(partialParams));
    }
    updateServiceSetting(partialParams) {
        return this.client.updateServiceSetting(this.ops["UpdateServiceSetting"].apply(partialParams));
    }
}
exports.default = Resource;
