// out/sagemaker/Workforce.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.sagemaker.Workforce {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.SageMaker();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/sagemaker-2017-07-24.normal.json"), this.client);
    }
    addAssociation(partialParams) {
        return this.client.addAssociation(this.ops["AddAssociation"].apply(partialParams));
    }
    addTags(partialParams) {
        return this.client.addTags(this.ops["AddTags"].apply(partialParams));
    }
    associateTrialComponent(partialParams) {
        return this.client.associateTrialComponent(this.ops["AssociateTrialComponent"].apply(partialParams));
    }
    batchDescribeModelPackage(partialParams) {
        return this.client.batchDescribeModelPackage(this.ops["BatchDescribeModelPackage"].apply(partialParams));
    }
    createAction(partialParams) {
        return this.client.createAction(this.ops["CreateAction"].apply(partialParams));
    }
    createAlgorithm(partialParams) {
        return this.client.createAlgorithm(this.ops["CreateAlgorithm"].apply(partialParams));
    }
    createApp(partialParams) {
        return this.client.createApp(this.ops["CreateApp"].apply(partialParams));
    }
    createAppImageConfig(partialParams) {
        return this.client.createAppImageConfig(this.ops["CreateAppImageConfig"].apply(partialParams));
    }
    createArtifact(partialParams) {
        return this.client.createArtifact(this.ops["CreateArtifact"].apply(partialParams));
    }
    createAutoMLJob(partialParams) {
        return this.client.createAutoMLJob(this.ops["CreateAutoMLJob"].apply(partialParams));
    }
    createCodeRepository(partialParams) {
        return this.client.createCodeRepository(this.ops["CreateCodeRepository"].apply(partialParams));
    }
    createCompilationJob(partialParams) {
        return this.client.createCompilationJob(this.ops["CreateCompilationJob"].apply(partialParams));
    }
    createContext(partialParams) {
        return this.client.createContext(this.ops["CreateContext"].apply(partialParams));
    }
    createDataQualityJobDefinition(partialParams) {
        return this.client.createDataQualityJobDefinition(this.ops["CreateDataQualityJobDefinition"].apply(partialParams));
    }
    createDeviceFleet(partialParams) {
        return this.client.createDeviceFleet(this.ops["CreateDeviceFleet"].apply(partialParams));
    }
    createDomain(partialParams) {
        return this.client.createDomain(this.ops["CreateDomain"].apply(partialParams));
    }
    createEdgePackagingJob(partialParams) {
        return this.client.createEdgePackagingJob(this.ops["CreateEdgePackagingJob"].apply(partialParams));
    }
    createEndpoint(partialParams) {
        return this.client.createEndpoint(this.ops["CreateEndpoint"].apply(partialParams));
    }
    createEndpointConfig(partialParams) {
        return this.client.createEndpointConfig(this.ops["CreateEndpointConfig"].apply(partialParams));
    }
    createExperiment(partialParams) {
        return this.client.createExperiment(this.ops["CreateExperiment"].apply(partialParams));
    }
    createFeatureGroup(partialParams) {
        return this.client.createFeatureGroup(this.ops["CreateFeatureGroup"].apply(partialParams));
    }
    createFlowDefinition(partialParams) {
        return this.client.createFlowDefinition(this.ops["CreateFlowDefinition"].apply(partialParams));
    }
    createHumanTaskUi(partialParams) {
        return this.client.createHumanTaskUi(this.ops["CreateHumanTaskUi"].apply(partialParams));
    }
    createHyperParameterTuningJob(partialParams) {
        return this.client.createHyperParameterTuningJob(this.ops["CreateHyperParameterTuningJob"].apply(partialParams));
    }
    createImage(partialParams) {
        return this.client.createImage(this.ops["CreateImage"].apply(partialParams));
    }
    createImageVersion(partialParams) {
        return this.client.createImageVersion(this.ops["CreateImageVersion"].apply(partialParams));
    }
    createInferenceRecommendationsJob(partialParams) {
        return this.client.createInferenceRecommendationsJob(this.ops["CreateInferenceRecommendationsJob"].apply(partialParams));
    }
    createLabelingJob(partialParams) {
        return this.client.createLabelingJob(this.ops["CreateLabelingJob"].apply(partialParams));
    }
    createModel(partialParams) {
        return this.client.createModel(this.ops["CreateModel"].apply(partialParams));
    }
    createModelBiasJobDefinition(partialParams) {
        return this.client.createModelBiasJobDefinition(this.ops["CreateModelBiasJobDefinition"].apply(partialParams));
    }
    createModelExplainabilityJobDefinition(partialParams) {
        return this.client.createModelExplainabilityJobDefinition(this.ops["CreateModelExplainabilityJobDefinition"].apply(partialParams));
    }
    createModelPackageGroup(partialParams) {
        return this.client.createModelPackageGroup(this.ops["CreateModelPackageGroup"].apply(partialParams));
    }
    createModelQualityJobDefinition(partialParams) {
        return this.client.createModelQualityJobDefinition(this.ops["CreateModelQualityJobDefinition"].apply(partialParams));
    }
    createMonitoringSchedule(partialParams) {
        return this.client.createMonitoringSchedule(this.ops["CreateMonitoringSchedule"].apply(partialParams));
    }
    createNotebookInstance(partialParams) {
        return this.client.createNotebookInstance(this.ops["CreateNotebookInstance"].apply(partialParams));
    }
    createNotebookInstanceLifecycleConfig(partialParams) {
        return this.client.createNotebookInstanceLifecycleConfig(this.ops["CreateNotebookInstanceLifecycleConfig"].apply(partialParams));
    }
    createPipeline(partialParams) {
        return this.client.createPipeline(this.ops["CreatePipeline"].apply(partialParams));
    }
    createPresignedDomainUrl(partialParams) {
        return this.client.createPresignedDomainUrl(this.ops["CreatePresignedDomainUrl"].apply(partialParams));
    }
    createPresignedNotebookInstanceUrl(partialParams) {
        return this.client.createPresignedNotebookInstanceUrl(this.ops["CreatePresignedNotebookInstanceUrl"].apply(partialParams));
    }
    createProcessingJob(partialParams) {
        return this.client.createProcessingJob(this.ops["CreateProcessingJob"].apply(partialParams));
    }
    createProject(partialParams) {
        return this.client.createProject(this.ops["CreateProject"].apply(partialParams));
    }
    createStudioLifecycleConfig(partialParams) {
        return this.client.createStudioLifecycleConfig(this.ops["CreateStudioLifecycleConfig"].apply(partialParams));
    }
    createTrainingJob(partialParams) {
        return this.client.createTrainingJob(this.ops["CreateTrainingJob"].apply(partialParams));
    }
    createTransformJob(partialParams) {
        return this.client.createTransformJob(this.ops["CreateTransformJob"].apply(partialParams));
    }
    createTrial(partialParams) {
        return this.client.createTrial(this.ops["CreateTrial"].apply(partialParams));
    }
    createTrialComponent(partialParams) {
        return this.client.createTrialComponent(this.ops["CreateTrialComponent"].apply(partialParams));
    }
    createUserProfile(partialParams) {
        return this.client.createUserProfile(this.ops["CreateUserProfile"].apply(partialParams));
    }
    createWorkforce(partialParams) {
        return this.client.createWorkforce(this.ops["CreateWorkforce"].apply(partialParams));
    }
    createWorkteam(partialParams) {
        return this.client.createWorkteam(this.ops["CreateWorkteam"].apply(partialParams));
    }
    deleteAction(partialParams) {
        return this.client.deleteAction(this.ops["DeleteAction"].apply(partialParams));
    }
    deleteAlgorithm(partialParams) {
        return this.client.deleteAlgorithm(this.ops["DeleteAlgorithm"].apply(partialParams));
    }
    deleteApp(partialParams) {
        return this.client.deleteApp(this.ops["DeleteApp"].apply(partialParams));
    }
    deleteAppImageConfig(partialParams) {
        return this.client.deleteAppImageConfig(this.ops["DeleteAppImageConfig"].apply(partialParams));
    }
    deleteAssociation(partialParams) {
        return this.client.deleteAssociation(this.ops["DeleteAssociation"].apply(partialParams));
    }
    deleteCodeRepository(partialParams) {
        return this.client.deleteCodeRepository(this.ops["DeleteCodeRepository"].apply(partialParams));
    }
    deleteContext(partialParams) {
        return this.client.deleteContext(this.ops["DeleteContext"].apply(partialParams));
    }
    deleteDataQualityJobDefinition(partialParams) {
        return this.client.deleteDataQualityJobDefinition(this.ops["DeleteDataQualityJobDefinition"].apply(partialParams));
    }
    deleteDeviceFleet(partialParams) {
        return this.client.deleteDeviceFleet(this.ops["DeleteDeviceFleet"].apply(partialParams));
    }
    deleteDomain(partialParams) {
        return this.client.deleteDomain(this.ops["DeleteDomain"].apply(partialParams));
    }
    deleteEndpoint(partialParams) {
        return this.client.deleteEndpoint(this.ops["DeleteEndpoint"].apply(partialParams));
    }
    deleteEndpointConfig(partialParams) {
        return this.client.deleteEndpointConfig(this.ops["DeleteEndpointConfig"].apply(partialParams));
    }
    deleteExperiment(partialParams) {
        return this.client.deleteExperiment(this.ops["DeleteExperiment"].apply(partialParams));
    }
    deleteFeatureGroup(partialParams) {
        return this.client.deleteFeatureGroup(this.ops["DeleteFeatureGroup"].apply(partialParams));
    }
    deleteFlowDefinition(partialParams) {
        return this.client.deleteFlowDefinition(this.ops["DeleteFlowDefinition"].apply(partialParams));
    }
    deleteHumanTaskUi(partialParams) {
        return this.client.deleteHumanTaskUi(this.ops["DeleteHumanTaskUi"].apply(partialParams));
    }
    deleteImage(partialParams) {
        return this.client.deleteImage(this.ops["DeleteImage"].apply(partialParams));
    }
    deleteImageVersion(partialParams) {
        return this.client.deleteImageVersion(this.ops["DeleteImageVersion"].apply(partialParams));
    }
    deleteModel(partialParams) {
        return this.client.deleteModel(this.ops["DeleteModel"].apply(partialParams));
    }
    deleteModelBiasJobDefinition(partialParams) {
        return this.client.deleteModelBiasJobDefinition(this.ops["DeleteModelBiasJobDefinition"].apply(partialParams));
    }
    deleteModelExplainabilityJobDefinition(partialParams) {
        return this.client.deleteModelExplainabilityJobDefinition(this.ops["DeleteModelExplainabilityJobDefinition"].apply(partialParams));
    }
    deleteModelPackage(partialParams) {
        return this.client.deleteModelPackage(this.ops["DeleteModelPackage"].apply(partialParams));
    }
    deleteModelPackageGroup(partialParams) {
        return this.client.deleteModelPackageGroup(this.ops["DeleteModelPackageGroup"].apply(partialParams));
    }
    deleteModelPackageGroupPolicy(partialParams) {
        return this.client.deleteModelPackageGroupPolicy(this.ops["DeleteModelPackageGroupPolicy"].apply(partialParams));
    }
    deleteModelQualityJobDefinition(partialParams) {
        return this.client.deleteModelQualityJobDefinition(this.ops["DeleteModelQualityJobDefinition"].apply(partialParams));
    }
    deleteMonitoringSchedule(partialParams) {
        return this.client.deleteMonitoringSchedule(this.ops["DeleteMonitoringSchedule"].apply(partialParams));
    }
    deleteNotebookInstance(partialParams) {
        return this.client.deleteNotebookInstance(this.ops["DeleteNotebookInstance"].apply(partialParams));
    }
    deleteNotebookInstanceLifecycleConfig(partialParams) {
        return this.client.deleteNotebookInstanceLifecycleConfig(this.ops["DeleteNotebookInstanceLifecycleConfig"].apply(partialParams));
    }
    deletePipeline(partialParams) {
        return this.client.deletePipeline(this.ops["DeletePipeline"].apply(partialParams));
    }
    deleteProject(partialParams) {
        return this.client.deleteProject(this.ops["DeleteProject"].apply(partialParams));
    }
    deleteStudioLifecycleConfig(partialParams) {
        return this.client.deleteStudioLifecycleConfig(this.ops["DeleteStudioLifecycleConfig"].apply(partialParams));
    }
    deleteTags(partialParams) {
        return this.client.deleteTags(this.ops["DeleteTags"].apply(partialParams));
    }
    deleteTrial(partialParams) {
        return this.client.deleteTrial(this.ops["DeleteTrial"].apply(partialParams));
    }
    deleteTrialComponent(partialParams) {
        return this.client.deleteTrialComponent(this.ops["DeleteTrialComponent"].apply(partialParams));
    }
    deleteUserProfile(partialParams) {
        return this.client.deleteUserProfile(this.ops["DeleteUserProfile"].apply(partialParams));
    }
    deleteWorkforce(partialParams) {
        return this.client.deleteWorkforce(this.ops["DeleteWorkforce"].apply(partialParams));
    }
    deleteWorkteam(partialParams) {
        return this.client.deleteWorkteam(this.ops["DeleteWorkteam"].apply(partialParams));
    }
    deregisterDevices(partialParams) {
        return this.client.deregisterDevices(this.ops["DeregisterDevices"].apply(partialParams));
    }
    describeAction(partialParams) {
        return this.client.describeAction(this.ops["DescribeAction"].apply(partialParams));
    }
    describeAlgorithm(partialParams) {
        return this.client.describeAlgorithm(this.ops["DescribeAlgorithm"].apply(partialParams));
    }
    describeApp(partialParams) {
        return this.client.describeApp(this.ops["DescribeApp"].apply(partialParams));
    }
    describeAppImageConfig(partialParams) {
        return this.client.describeAppImageConfig(this.ops["DescribeAppImageConfig"].apply(partialParams));
    }
    describeArtifact(partialParams) {
        return this.client.describeArtifact(this.ops["DescribeArtifact"].apply(partialParams));
    }
    describeAutoMLJob(partialParams) {
        return this.client.describeAutoMLJob(this.ops["DescribeAutoMLJob"].apply(partialParams));
    }
    describeCodeRepository(partialParams) {
        return this.client.describeCodeRepository(this.ops["DescribeCodeRepository"].apply(partialParams));
    }
    describeCompilationJob(partialParams) {
        return this.client.describeCompilationJob(this.ops["DescribeCompilationJob"].apply(partialParams));
    }
    describeContext(partialParams) {
        return this.client.describeContext(this.ops["DescribeContext"].apply(partialParams));
    }
    describeDataQualityJobDefinition(partialParams) {
        return this.client.describeDataQualityJobDefinition(this.ops["DescribeDataQualityJobDefinition"].apply(partialParams));
    }
    describeDevice(partialParams) {
        return this.client.describeDevice(this.ops["DescribeDevice"].apply(partialParams));
    }
    describeDeviceFleet(partialParams) {
        return this.client.describeDeviceFleet(this.ops["DescribeDeviceFleet"].apply(partialParams));
    }
    describeDomain(partialParams) {
        return this.client.describeDomain(this.ops["DescribeDomain"].apply(partialParams));
    }
    describeEdgePackagingJob(partialParams) {
        return this.client.describeEdgePackagingJob(this.ops["DescribeEdgePackagingJob"].apply(partialParams));
    }
    describeEndpoint(partialParams) {
        return this.client.describeEndpoint(this.ops["DescribeEndpoint"].apply(partialParams));
    }
    describeEndpointConfig(partialParams) {
        return this.client.describeEndpointConfig(this.ops["DescribeEndpointConfig"].apply(partialParams));
    }
    describeExperiment(partialParams) {
        return this.client.describeExperiment(this.ops["DescribeExperiment"].apply(partialParams));
    }
    describeFeatureGroup(partialParams) {
        return this.client.describeFeatureGroup(this.ops["DescribeFeatureGroup"].apply(partialParams));
    }
    describeFlowDefinition(partialParams) {
        return this.client.describeFlowDefinition(this.ops["DescribeFlowDefinition"].apply(partialParams));
    }
    describeHumanTaskUi(partialParams) {
        return this.client.describeHumanTaskUi(this.ops["DescribeHumanTaskUi"].apply(partialParams));
    }
    describeHyperParameterTuningJob(partialParams) {
        return this.client.describeHyperParameterTuningJob(this.ops["DescribeHyperParameterTuningJob"].apply(partialParams));
    }
    describeImage(partialParams) {
        return this.client.describeImage(this.ops["DescribeImage"].apply(partialParams));
    }
    describeImageVersion(partialParams) {
        return this.client.describeImageVersion(this.ops["DescribeImageVersion"].apply(partialParams));
    }
    describeInferenceRecommendationsJob(partialParams) {
        return this.client.describeInferenceRecommendationsJob(this.ops["DescribeInferenceRecommendationsJob"].apply(partialParams));
    }
    describeLabelingJob(partialParams) {
        return this.client.describeLabelingJob(this.ops["DescribeLabelingJob"].apply(partialParams));
    }
    describeLineageGroup(partialParams) {
        return this.client.describeLineageGroup(this.ops["DescribeLineageGroup"].apply(partialParams));
    }
    describeModel(partialParams) {
        return this.client.describeModel(this.ops["DescribeModel"].apply(partialParams));
    }
    describeModelBiasJobDefinition(partialParams) {
        return this.client.describeModelBiasJobDefinition(this.ops["DescribeModelBiasJobDefinition"].apply(partialParams));
    }
    describeModelExplainabilityJobDefinition(partialParams) {
        return this.client.describeModelExplainabilityJobDefinition(this.ops["DescribeModelExplainabilityJobDefinition"].apply(partialParams));
    }
    describeModelPackage(partialParams) {
        return this.client.describeModelPackage(this.ops["DescribeModelPackage"].apply(partialParams));
    }
    describeModelPackageGroup(partialParams) {
        return this.client.describeModelPackageGroup(this.ops["DescribeModelPackageGroup"].apply(partialParams));
    }
    describeModelQualityJobDefinition(partialParams) {
        return this.client.describeModelQualityJobDefinition(this.ops["DescribeModelQualityJobDefinition"].apply(partialParams));
    }
    describeMonitoringSchedule(partialParams) {
        return this.client.describeMonitoringSchedule(this.ops["DescribeMonitoringSchedule"].apply(partialParams));
    }
    describeNotebookInstance(partialParams) {
        return this.client.describeNotebookInstance(this.ops["DescribeNotebookInstance"].apply(partialParams));
    }
    describeNotebookInstanceLifecycleConfig(partialParams) {
        return this.client.describeNotebookInstanceLifecycleConfig(this.ops["DescribeNotebookInstanceLifecycleConfig"].apply(partialParams));
    }
    describePipeline(partialParams) {
        return this.client.describePipeline(this.ops["DescribePipeline"].apply(partialParams));
    }
    describePipelineDefinitionForExecution(partialParams) {
        return this.client.describePipelineDefinitionForExecution(this.ops["DescribePipelineDefinitionForExecution"].apply(partialParams));
    }
    describePipelineExecution(partialParams) {
        return this.client.describePipelineExecution(this.ops["DescribePipelineExecution"].apply(partialParams));
    }
    describeProcessingJob(partialParams) {
        return this.client.describeProcessingJob(this.ops["DescribeProcessingJob"].apply(partialParams));
    }
    describeProject(partialParams) {
        return this.client.describeProject(this.ops["DescribeProject"].apply(partialParams));
    }
    describeStudioLifecycleConfig(partialParams) {
        return this.client.describeStudioLifecycleConfig(this.ops["DescribeStudioLifecycleConfig"].apply(partialParams));
    }
    describeSubscribedWorkteam(partialParams) {
        return this.client.describeSubscribedWorkteam(this.ops["DescribeSubscribedWorkteam"].apply(partialParams));
    }
    describeTrainingJob(partialParams) {
        return this.client.describeTrainingJob(this.ops["DescribeTrainingJob"].apply(partialParams));
    }
    describeTransformJob(partialParams) {
        return this.client.describeTransformJob(this.ops["DescribeTransformJob"].apply(partialParams));
    }
    describeTrial(partialParams) {
        return this.client.describeTrial(this.ops["DescribeTrial"].apply(partialParams));
    }
    describeTrialComponent(partialParams) {
        return this.client.describeTrialComponent(this.ops["DescribeTrialComponent"].apply(partialParams));
    }
    describeUserProfile(partialParams) {
        return this.client.describeUserProfile(this.ops["DescribeUserProfile"].apply(partialParams));
    }
    describeWorkforce(partialParams) {
        return this.client.describeWorkforce(this.ops["DescribeWorkforce"].apply(partialParams));
    }
    describeWorkteam(partialParams) {
        return this.client.describeWorkteam(this.ops["DescribeWorkteam"].apply(partialParams));
    }
    disassociateTrialComponent(partialParams) {
        return this.client.disassociateTrialComponent(this.ops["DisassociateTrialComponent"].apply(partialParams));
    }
    getDeviceFleetReport(partialParams) {
        return this.client.getDeviceFleetReport(this.ops["GetDeviceFleetReport"].apply(partialParams));
    }
    getLineageGroupPolicy(partialParams) {
        return this.client.getLineageGroupPolicy(this.ops["GetLineageGroupPolicy"].apply(partialParams));
    }
    getModelPackageGroupPolicy(partialParams) {
        return this.client.getModelPackageGroupPolicy(this.ops["GetModelPackageGroupPolicy"].apply(partialParams));
    }
    getSearchSuggestions(partialParams) {
        return this.client.getSearchSuggestions(this.ops["GetSearchSuggestions"].apply(partialParams));
    }
    listCandidatesForAutoMLJob(partialParams) {
        return this.client.listCandidatesForAutoMLJob(this.ops["ListCandidatesForAutoMLJob"].apply(partialParams));
    }
    listImageVersions(partialParams) {
        return this.client.listImageVersions(this.ops["ListImageVersions"].apply(partialParams));
    }
    listLabelingJobsForWorkteam(partialParams) {
        return this.client.listLabelingJobsForWorkteam(this.ops["ListLabelingJobsForWorkteam"].apply(partialParams));
    }
    listPipelineExecutions(partialParams) {
        return this.client.listPipelineExecutions(this.ops["ListPipelineExecutions"].apply(partialParams));
    }
    listPipelineParametersForExecution(partialParams) {
        return this.client.listPipelineParametersForExecution(this.ops["ListPipelineParametersForExecution"].apply(partialParams));
    }
    listTags(partialParams) {
        return this.client.listTags(this.ops["ListTags"].apply(partialParams));
    }
    listTrainingJobsForHyperParameterTuningJob(partialParams) {
        return this.client.listTrainingJobsForHyperParameterTuningJob(this.ops["ListTrainingJobsForHyperParameterTuningJob"].apply(partialParams));
    }
    putModelPackageGroupPolicy(partialParams) {
        return this.client.putModelPackageGroupPolicy(this.ops["PutModelPackageGroupPolicy"].apply(partialParams));
    }
    queryLineage(partialParams) {
        return this.client.queryLineage(this.ops["QueryLineage"].apply(partialParams));
    }
    registerDevices(partialParams) {
        return this.client.registerDevices(this.ops["RegisterDevices"].apply(partialParams));
    }
    renderUiTemplate(partialParams) {
        return this.client.renderUiTemplate(this.ops["RenderUiTemplate"].apply(partialParams));
    }
    retryPipelineExecution(partialParams) {
        return this.client.retryPipelineExecution(this.ops["RetryPipelineExecution"].apply(partialParams));
    }
    search(partialParams) {
        return this.client.search(this.ops["Search"].apply(partialParams));
    }
    sendPipelineExecutionStepFailure(partialParams) {
        return this.client.sendPipelineExecutionStepFailure(this.ops["SendPipelineExecutionStepFailure"].apply(partialParams));
    }
    sendPipelineExecutionStepSuccess(partialParams) {
        return this.client.sendPipelineExecutionStepSuccess(this.ops["SendPipelineExecutionStepSuccess"].apply(partialParams));
    }
    startMonitoringSchedule(partialParams) {
        return this.client.startMonitoringSchedule(this.ops["StartMonitoringSchedule"].apply(partialParams));
    }
    startNotebookInstance(partialParams) {
        return this.client.startNotebookInstance(this.ops["StartNotebookInstance"].apply(partialParams));
    }
    startPipelineExecution(partialParams) {
        return this.client.startPipelineExecution(this.ops["StartPipelineExecution"].apply(partialParams));
    }
    stopAutoMLJob(partialParams) {
        return this.client.stopAutoMLJob(this.ops["StopAutoMLJob"].apply(partialParams));
    }
    stopCompilationJob(partialParams) {
        return this.client.stopCompilationJob(this.ops["StopCompilationJob"].apply(partialParams));
    }
    stopEdgePackagingJob(partialParams) {
        return this.client.stopEdgePackagingJob(this.ops["StopEdgePackagingJob"].apply(partialParams));
    }
    stopHyperParameterTuningJob(partialParams) {
        return this.client.stopHyperParameterTuningJob(this.ops["StopHyperParameterTuningJob"].apply(partialParams));
    }
    stopInferenceRecommendationsJob(partialParams) {
        return this.client.stopInferenceRecommendationsJob(this.ops["StopInferenceRecommendationsJob"].apply(partialParams));
    }
    stopLabelingJob(partialParams) {
        return this.client.stopLabelingJob(this.ops["StopLabelingJob"].apply(partialParams));
    }
    stopMonitoringSchedule(partialParams) {
        return this.client.stopMonitoringSchedule(this.ops["StopMonitoringSchedule"].apply(partialParams));
    }
    stopNotebookInstance(partialParams) {
        return this.client.stopNotebookInstance(this.ops["StopNotebookInstance"].apply(partialParams));
    }
    stopPipelineExecution(partialParams) {
        return this.client.stopPipelineExecution(this.ops["StopPipelineExecution"].apply(partialParams));
    }
    stopProcessingJob(partialParams) {
        return this.client.stopProcessingJob(this.ops["StopProcessingJob"].apply(partialParams));
    }
    stopTrainingJob(partialParams) {
        return this.client.stopTrainingJob(this.ops["StopTrainingJob"].apply(partialParams));
    }
    stopTransformJob(partialParams) {
        return this.client.stopTransformJob(this.ops["StopTransformJob"].apply(partialParams));
    }
    updateAction(partialParams) {
        return this.client.updateAction(this.ops["UpdateAction"].apply(partialParams));
    }
    updateAppImageConfig(partialParams) {
        return this.client.updateAppImageConfig(this.ops["UpdateAppImageConfig"].apply(partialParams));
    }
    updateArtifact(partialParams) {
        return this.client.updateArtifact(this.ops["UpdateArtifact"].apply(partialParams));
    }
    updateCodeRepository(partialParams) {
        return this.client.updateCodeRepository(this.ops["UpdateCodeRepository"].apply(partialParams));
    }
    updateContext(partialParams) {
        return this.client.updateContext(this.ops["UpdateContext"].apply(partialParams));
    }
    updateDeviceFleet(partialParams) {
        return this.client.updateDeviceFleet(this.ops["UpdateDeviceFleet"].apply(partialParams));
    }
    updateDevices(partialParams) {
        return this.client.updateDevices(this.ops["UpdateDevices"].apply(partialParams));
    }
    updateDomain(partialParams) {
        return this.client.updateDomain(this.ops["UpdateDomain"].apply(partialParams));
    }
    updateEndpoint(partialParams) {
        return this.client.updateEndpoint(this.ops["UpdateEndpoint"].apply(partialParams));
    }
    updateEndpointWeightsAndCapacities(partialParams) {
        return this.client.updateEndpointWeightsAndCapacities(this.ops["UpdateEndpointWeightsAndCapacities"].apply(partialParams));
    }
    updateExperiment(partialParams) {
        return this.client.updateExperiment(this.ops["UpdateExperiment"].apply(partialParams));
    }
    updateImage(partialParams) {
        return this.client.updateImage(this.ops["UpdateImage"].apply(partialParams));
    }
    updateModelPackage(partialParams) {
        return this.client.updateModelPackage(this.ops["UpdateModelPackage"].apply(partialParams));
    }
    updateMonitoringSchedule(partialParams) {
        return this.client.updateMonitoringSchedule(this.ops["UpdateMonitoringSchedule"].apply(partialParams));
    }
    updateNotebookInstance(partialParams) {
        return this.client.updateNotebookInstance(this.ops["UpdateNotebookInstance"].apply(partialParams));
    }
    updateNotebookInstanceLifecycleConfig(partialParams) {
        return this.client.updateNotebookInstanceLifecycleConfig(this.ops["UpdateNotebookInstanceLifecycleConfig"].apply(partialParams));
    }
    updatePipeline(partialParams) {
        return this.client.updatePipeline(this.ops["UpdatePipeline"].apply(partialParams));
    }
    updatePipelineExecution(partialParams) {
        return this.client.updatePipelineExecution(this.ops["UpdatePipelineExecution"].apply(partialParams));
    }
    updateProject(partialParams) {
        return this.client.updateProject(this.ops["UpdateProject"].apply(partialParams));
    }
    updateTrainingJob(partialParams) {
        return this.client.updateTrainingJob(this.ops["UpdateTrainingJob"].apply(partialParams));
    }
    updateTrial(partialParams) {
        return this.client.updateTrial(this.ops["UpdateTrial"].apply(partialParams));
    }
    updateTrialComponent(partialParams) {
        return this.client.updateTrialComponent(this.ops["UpdateTrialComponent"].apply(partialParams));
    }
    updateUserProfile(partialParams) {
        return this.client.updateUserProfile(this.ops["UpdateUserProfile"].apply(partialParams));
    }
    updateWorkforce(partialParams) {
        return this.client.updateWorkforce(this.ops["UpdateWorkforce"].apply(partialParams));
    }
    updateWorkteam(partialParams) {
        return this.client.updateWorkteam(this.ops["UpdateWorkteam"].apply(partialParams));
    }
}
exports.default = Resource;
