// out/mediapackage/Channel.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.mediapackage.Channel {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.MediaPackage();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/mediapackage-2017-10-12.normal.json"), this.client);
    }
    configureLogs(partialParams) {
        return this.client.configureLogs(this.ops["ConfigureLogs"].apply(partialParams));
    }
    createChannel(partialParams) {
        return this.client.createChannel(this.ops["CreateChannel"].apply(partialParams));
    }
    createHarvestJob(partialParams) {
        return this.client.createHarvestJob(this.ops["CreateHarvestJob"].apply(partialParams));
    }
    createOriginEndpoint(partialParams) {
        return this.client.createOriginEndpoint(this.ops["CreateOriginEndpoint"].apply(partialParams));
    }
    deleteChannel(partialParams) {
        return this.client.deleteChannel(this.ops["DeleteChannel"].apply(partialParams));
    }
    deleteOriginEndpoint(partialParams) {
        return this.client.deleteOriginEndpoint(this.ops["DeleteOriginEndpoint"].apply(partialParams));
    }
    describeChannel(partialParams) {
        return this.client.describeChannel(this.ops["DescribeChannel"].apply(partialParams));
    }
    describeHarvestJob(partialParams) {
        return this.client.describeHarvestJob(this.ops["DescribeHarvestJob"].apply(partialParams));
    }
    describeOriginEndpoint(partialParams) {
        return this.client.describeOriginEndpoint(this.ops["DescribeOriginEndpoint"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    rotateChannelCredentials(partialParams) {
        return this.client.rotateChannelCredentials(this.ops["RotateChannelCredentials"].apply(partialParams));
    }
    rotateIngestEndpointCredentials(partialParams) {
        return this.client.rotateIngestEndpointCredentials(this.ops["RotateIngestEndpointCredentials"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateChannel(partialParams) {
        return this.client.updateChannel(this.ops["UpdateChannel"].apply(partialParams));
    }
    updateOriginEndpoint(partialParams) {
        return this.client.updateOriginEndpoint(this.ops["UpdateOriginEndpoint"].apply(partialParams));
    }
}
exports.default = Resource;
