// out/backup/VaultPolicy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.backup.VaultPolicy {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Backup();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/backup-2018-11-15.normal.json"), this.client);
    }
    createBackupPlan(partialParams) {
        return this.client.createBackupPlan(this.ops["CreateBackupPlan"].apply(partialParams));
    }
    createBackupSelection(partialParams) {
        return this.client.createBackupSelection(this.ops["CreateBackupSelection"].apply(partialParams));
    }
    createBackupVault(partialParams) {
        return this.client.createBackupVault(this.ops["CreateBackupVault"].apply(partialParams));
    }
    createFramework(partialParams) {
        return this.client.createFramework(this.ops["CreateFramework"].apply(partialParams));
    }
    createReportPlan(partialParams) {
        return this.client.createReportPlan(this.ops["CreateReportPlan"].apply(partialParams));
    }
    deleteBackupPlan(partialParams) {
        return this.client.deleteBackupPlan(this.ops["DeleteBackupPlan"].apply(partialParams));
    }
    deleteBackupSelection(partialParams) {
        return this.client.deleteBackupSelection(this.ops["DeleteBackupSelection"].apply(partialParams));
    }
    deleteBackupVault(partialParams) {
        return this.client.deleteBackupVault(this.ops["DeleteBackupVault"].apply(partialParams));
    }
    deleteBackupVaultAccessPolicy(partialParams) {
        return this.client.deleteBackupVaultAccessPolicy(this.ops["DeleteBackupVaultAccessPolicy"].apply(partialParams));
    }
    deleteBackupVaultLockConfiguration(partialParams) {
        return this.client.deleteBackupVaultLockConfiguration(this.ops["DeleteBackupVaultLockConfiguration"].apply(partialParams));
    }
    deleteBackupVaultNotifications(partialParams) {
        return this.client.deleteBackupVaultNotifications(this.ops["DeleteBackupVaultNotifications"].apply(partialParams));
    }
    deleteFramework(partialParams) {
        return this.client.deleteFramework(this.ops["DeleteFramework"].apply(partialParams));
    }
    deleteRecoveryPoint(partialParams) {
        return this.client.deleteRecoveryPoint(this.ops["DeleteRecoveryPoint"].apply(partialParams));
    }
    deleteReportPlan(partialParams) {
        return this.client.deleteReportPlan(this.ops["DeleteReportPlan"].apply(partialParams));
    }
    describeBackupJob(partialParams) {
        return this.client.describeBackupJob(this.ops["DescribeBackupJob"].apply(partialParams));
    }
    describeBackupVault(partialParams) {
        return this.client.describeBackupVault(this.ops["DescribeBackupVault"].apply(partialParams));
    }
    describeCopyJob(partialParams) {
        return this.client.describeCopyJob(this.ops["DescribeCopyJob"].apply(partialParams));
    }
    describeFramework(partialParams) {
        return this.client.describeFramework(this.ops["DescribeFramework"].apply(partialParams));
    }
    describeProtectedResource(partialParams) {
        return this.client.describeProtectedResource(this.ops["DescribeProtectedResource"].apply(partialParams));
    }
    describeRecoveryPoint(partialParams) {
        return this.client.describeRecoveryPoint(this.ops["DescribeRecoveryPoint"].apply(partialParams));
    }
    describeReportJob(partialParams) {
        return this.client.describeReportJob(this.ops["DescribeReportJob"].apply(partialParams));
    }
    describeReportPlan(partialParams) {
        return this.client.describeReportPlan(this.ops["DescribeReportPlan"].apply(partialParams));
    }
    describeRestoreJob(partialParams) {
        return this.client.describeRestoreJob(this.ops["DescribeRestoreJob"].apply(partialParams));
    }
    disassociateRecoveryPoint(partialParams) {
        return this.client.disassociateRecoveryPoint(this.ops["DisassociateRecoveryPoint"].apply(partialParams));
    }
    exportBackupPlanTemplate(partialParams) {
        return this.client.exportBackupPlanTemplate(this.ops["ExportBackupPlanTemplate"].apply(partialParams));
    }
    getBackupPlan(partialParams) {
        return this.client.getBackupPlan(this.ops["GetBackupPlan"].apply(partialParams));
    }
    getBackupPlanFromJSON(partialParams) {
        return this.client.getBackupPlanFromJSON(this.ops["GetBackupPlanFromJSON"].apply(partialParams));
    }
    getBackupPlanFromTemplate(partialParams) {
        return this.client.getBackupPlanFromTemplate(this.ops["GetBackupPlanFromTemplate"].apply(partialParams));
    }
    getBackupSelection(partialParams) {
        return this.client.getBackupSelection(this.ops["GetBackupSelection"].apply(partialParams));
    }
    getBackupVaultAccessPolicy(partialParams) {
        return this.client.getBackupVaultAccessPolicy(this.ops["GetBackupVaultAccessPolicy"].apply(partialParams));
    }
    getBackupVaultNotifications(partialParams) {
        return this.client.getBackupVaultNotifications(this.ops["GetBackupVaultNotifications"].apply(partialParams));
    }
    getRecoveryPointRestoreMetadata(partialParams) {
        return this.client.getRecoveryPointRestoreMetadata(this.ops["GetRecoveryPointRestoreMetadata"].apply(partialParams));
    }
    listBackupPlanVersions(partialParams) {
        return this.client.listBackupPlanVersions(this.ops["ListBackupPlanVersions"].apply(partialParams));
    }
    listBackupSelections(partialParams) {
        return this.client.listBackupSelections(this.ops["ListBackupSelections"].apply(partialParams));
    }
    listRecoveryPointsByBackupVault(partialParams) {
        return this.client.listRecoveryPointsByBackupVault(this.ops["ListRecoveryPointsByBackupVault"].apply(partialParams));
    }
    listRecoveryPointsByResource(partialParams) {
        return this.client.listRecoveryPointsByResource(this.ops["ListRecoveryPointsByResource"].apply(partialParams));
    }
    listTags(partialParams) {
        return this.client.listTags(this.ops["ListTags"].apply(partialParams));
    }
    putBackupVaultAccessPolicy(partialParams) {
        return this.client.putBackupVaultAccessPolicy(this.ops["PutBackupVaultAccessPolicy"].apply(partialParams));
    }
    putBackupVaultLockConfiguration(partialParams) {
        return this.client.putBackupVaultLockConfiguration(this.ops["PutBackupVaultLockConfiguration"].apply(partialParams));
    }
    putBackupVaultNotifications(partialParams) {
        return this.client.putBackupVaultNotifications(this.ops["PutBackupVaultNotifications"].apply(partialParams));
    }
    startBackupJob(partialParams) {
        return this.client.startBackupJob(this.ops["StartBackupJob"].apply(partialParams));
    }
    startCopyJob(partialParams) {
        return this.client.startCopyJob(this.ops["StartCopyJob"].apply(partialParams));
    }
    startReportJob(partialParams) {
        return this.client.startReportJob(this.ops["StartReportJob"].apply(partialParams));
    }
    startRestoreJob(partialParams) {
        return this.client.startRestoreJob(this.ops["StartRestoreJob"].apply(partialParams));
    }
    stopBackupJob(partialParams) {
        return this.client.stopBackupJob(this.ops["StopBackupJob"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateBackupPlan(partialParams) {
        return this.client.updateBackupPlan(this.ops["UpdateBackupPlan"].apply(partialParams));
    }
    updateFramework(partialParams) {
        return this.client.updateFramework(this.ops["UpdateFramework"].apply(partialParams));
    }
    updateRecoveryPointLifecycle(partialParams) {
        return this.client.updateRecoveryPointLifecycle(this.ops["UpdateRecoveryPointLifecycle"].apply(partialParams));
    }
    updateReportPlan(partialParams) {
        return this.client.updateReportPlan(this.ops["UpdateReportPlan"].apply(partialParams));
    }
}
exports.default = Resource;
