// out/kms/Alias.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.kms.Alias {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.KMS();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/kms-2014-11-01.normal.json"), this.client);
    }
    cancelKeyDeletion(partialParams) {
        return this.client.cancelKeyDeletion(this.ops["CancelKeyDeletion"].apply(partialParams));
    }
    connectCustomKeyStore(partialParams) {
        return this.client.connectCustomKeyStore(this.ops["ConnectCustomKeyStore"].apply(partialParams));
    }
    createAlias(partialParams) {
        return this.client.createAlias(this.ops["CreateAlias"].apply(partialParams));
    }
    createCustomKeyStore(partialParams) {
        return this.client.createCustomKeyStore(this.ops["CreateCustomKeyStore"].apply(partialParams));
    }
    createGrant(partialParams) {
        return this.client.createGrant(this.ops["CreateGrant"].apply(partialParams));
    }
    decrypt(partialParams) {
        return this.client.decrypt(this.ops["Decrypt"].apply(partialParams));
    }
    deleteAlias(partialParams) {
        return this.client.deleteAlias(this.ops["DeleteAlias"].apply(partialParams));
    }
    deleteCustomKeyStore(partialParams) {
        return this.client.deleteCustomKeyStore(this.ops["DeleteCustomKeyStore"].apply(partialParams));
    }
    deleteImportedKeyMaterial(partialParams) {
        return this.client.deleteImportedKeyMaterial(this.ops["DeleteImportedKeyMaterial"].apply(partialParams));
    }
    describeKey(partialParams) {
        return this.client.describeKey(this.ops["DescribeKey"].apply(partialParams));
    }
    disableKey(partialParams) {
        return this.client.disableKey(this.ops["DisableKey"].apply(partialParams));
    }
    disableKeyRotation(partialParams) {
        return this.client.disableKeyRotation(this.ops["DisableKeyRotation"].apply(partialParams));
    }
    disconnectCustomKeyStore(partialParams) {
        return this.client.disconnectCustomKeyStore(this.ops["DisconnectCustomKeyStore"].apply(partialParams));
    }
    enableKey(partialParams) {
        return this.client.enableKey(this.ops["EnableKey"].apply(partialParams));
    }
    enableKeyRotation(partialParams) {
        return this.client.enableKeyRotation(this.ops["EnableKeyRotation"].apply(partialParams));
    }
    encrypt(partialParams) {
        return this.client.encrypt(this.ops["Encrypt"].apply(partialParams));
    }
    generateDataKey(partialParams) {
        return this.client.generateDataKey(this.ops["GenerateDataKey"].apply(partialParams));
    }
    generateDataKeyPair(partialParams) {
        return this.client.generateDataKeyPair(this.ops["GenerateDataKeyPair"].apply(partialParams));
    }
    generateDataKeyPairWithoutPlaintext(partialParams) {
        return this.client.generateDataKeyPairWithoutPlaintext(this.ops["GenerateDataKeyPairWithoutPlaintext"].apply(partialParams));
    }
    generateDataKeyWithoutPlaintext(partialParams) {
        return this.client.generateDataKeyWithoutPlaintext(this.ops["GenerateDataKeyWithoutPlaintext"].apply(partialParams));
    }
    getKeyPolicy(partialParams) {
        return this.client.getKeyPolicy(this.ops["GetKeyPolicy"].apply(partialParams));
    }
    getKeyRotationStatus(partialParams) {
        return this.client.getKeyRotationStatus(this.ops["GetKeyRotationStatus"].apply(partialParams));
    }
    getParametersForImport(partialParams) {
        return this.client.getParametersForImport(this.ops["GetParametersForImport"].apply(partialParams));
    }
    getPublicKey(partialParams) {
        return this.client.getPublicKey(this.ops["GetPublicKey"].apply(partialParams));
    }
    importKeyMaterial(partialParams) {
        return this.client.importKeyMaterial(this.ops["ImportKeyMaterial"].apply(partialParams));
    }
    listGrants(partialParams) {
        return this.client.listGrants(this.ops["ListGrants"].apply(partialParams));
    }
    listKeyPolicies(partialParams) {
        return this.client.listKeyPolicies(this.ops["ListKeyPolicies"].apply(partialParams));
    }
    listResourceTags(partialParams) {
        return this.client.listResourceTags(this.ops["ListResourceTags"].apply(partialParams));
    }
    listRetirableGrants(partialParams) {
        return this.client.listRetirableGrants(this.ops["ListRetirableGrants"].apply(partialParams));
    }
    putKeyPolicy(partialParams) {
        return this.client.putKeyPolicy(this.ops["PutKeyPolicy"].apply(partialParams));
    }
    reEncrypt(partialParams) {
        return this.client.reEncrypt(this.ops["ReEncrypt"].apply(partialParams));
    }
    replicateKey(partialParams) {
        return this.client.replicateKey(this.ops["ReplicateKey"].apply(partialParams));
    }
    revokeGrant(partialParams) {
        return this.client.revokeGrant(this.ops["RevokeGrant"].apply(partialParams));
    }
    scheduleKeyDeletion(partialParams) {
        return this.client.scheduleKeyDeletion(this.ops["ScheduleKeyDeletion"].apply(partialParams));
    }
    sign(partialParams) {
        return this.client.sign(this.ops["Sign"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAlias(partialParams) {
        return this.client.updateAlias(this.ops["UpdateAlias"].apply(partialParams));
    }
    updateCustomKeyStore(partialParams) {
        return this.client.updateCustomKeyStore(this.ops["UpdateCustomKeyStore"].apply(partialParams));
    }
    updateKeyDescription(partialParams) {
        return this.client.updateKeyDescription(this.ops["UpdateKeyDescription"].apply(partialParams));
    }
    updatePrimaryRegion(partialParams) {
        return this.client.updatePrimaryRegion(this.ops["UpdatePrimaryRegion"].apply(partialParams));
    }
    verify(partialParams) {
        return this.client.verify(this.ops["Verify"].apply(partialParams));
    }
}
exports.default = Resource;
