// out/cloudformation/StackSetInstance.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.cloudformation.StackSetInstance {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.CloudFormation();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/cloudformation-2010-05-15.normal.json"), this.client);
    }
    batchDescribeTypeConfigurations(partialParams) {
        return this.client.batchDescribeTypeConfigurations(this.ops["BatchDescribeTypeConfigurations"].apply(partialParams));
    }
    cancelUpdateStack(partialParams) {
        return this.client.cancelUpdateStack(this.ops["CancelUpdateStack"].apply(partialParams));
    }
    continueUpdateRollback(partialParams) {
        return this.client.continueUpdateRollback(this.ops["ContinueUpdateRollback"].apply(partialParams));
    }
    createChangeSet(partialParams) {
        return this.client.createChangeSet(this.ops["CreateChangeSet"].apply(partialParams));
    }
    createStack(partialParams) {
        return this.client.createStack(this.ops["CreateStack"].apply(partialParams));
    }
    createStackInstances(partialParams) {
        return this.client.createStackInstances(this.ops["CreateStackInstances"].apply(partialParams));
    }
    createStackSet(partialParams) {
        return this.client.createStackSet(this.ops["CreateStackSet"].apply(partialParams));
    }
    deleteChangeSet(partialParams) {
        return this.client.deleteChangeSet(this.ops["DeleteChangeSet"].apply(partialParams));
    }
    deleteStack(partialParams) {
        return this.client.deleteStack(this.ops["DeleteStack"].apply(partialParams));
    }
    deleteStackInstances(partialParams) {
        return this.client.deleteStackInstances(this.ops["DeleteStackInstances"].apply(partialParams));
    }
    deleteStackSet(partialParams) {
        return this.client.deleteStackSet(this.ops["DeleteStackSet"].apply(partialParams));
    }
    describeChangeSet(partialParams) {
        return this.client.describeChangeSet(this.ops["DescribeChangeSet"].apply(partialParams));
    }
    describeChangeSetHooks(partialParams) {
        return this.client.describeChangeSetHooks(this.ops["DescribeChangeSetHooks"].apply(partialParams));
    }
    describeStackDriftDetectionStatus(partialParams) {
        return this.client.describeStackDriftDetectionStatus(this.ops["DescribeStackDriftDetectionStatus"].apply(partialParams));
    }
    describeStackInstance(partialParams) {
        return this.client.describeStackInstance(this.ops["DescribeStackInstance"].apply(partialParams));
    }
    describeStackResource(partialParams) {
        return this.client.describeStackResource(this.ops["DescribeStackResource"].apply(partialParams));
    }
    describeStackResourceDrifts(partialParams) {
        return this.client.describeStackResourceDrifts(this.ops["DescribeStackResourceDrifts"].apply(partialParams));
    }
    describeStackSet(partialParams) {
        return this.client.describeStackSet(this.ops["DescribeStackSet"].apply(partialParams));
    }
    describeStackSetOperation(partialParams) {
        return this.client.describeStackSetOperation(this.ops["DescribeStackSetOperation"].apply(partialParams));
    }
    describeTypeRegistration(partialParams) {
        return this.client.describeTypeRegistration(this.ops["DescribeTypeRegistration"].apply(partialParams));
    }
    detectStackDrift(partialParams) {
        return this.client.detectStackDrift(this.ops["DetectStackDrift"].apply(partialParams));
    }
    detectStackResourceDrift(partialParams) {
        return this.client.detectStackResourceDrift(this.ops["DetectStackResourceDrift"].apply(partialParams));
    }
    detectStackSetDrift(partialParams) {
        return this.client.detectStackSetDrift(this.ops["DetectStackSetDrift"].apply(partialParams));
    }
    executeChangeSet(partialParams) {
        return this.client.executeChangeSet(this.ops["ExecuteChangeSet"].apply(partialParams));
    }
    getStackPolicy(partialParams) {
        return this.client.getStackPolicy(this.ops["GetStackPolicy"].apply(partialParams));
    }
    importStacksToStackSet(partialParams) {
        return this.client.importStacksToStackSet(this.ops["ImportStacksToStackSet"].apply(partialParams));
    }
    listChangeSets(partialParams) {
        return this.client.listChangeSets(this.ops["ListChangeSets"].apply(partialParams));
    }
    listImports(partialParams) {
        return this.client.listImports(this.ops["ListImports"].apply(partialParams));
    }
    listStackInstances(partialParams) {
        return this.client.listStackInstances(this.ops["ListStackInstances"].apply(partialParams));
    }
    listStackResources(partialParams) {
        return this.client.listStackResources(this.ops["ListStackResources"].apply(partialParams));
    }
    listStackSetOperationResults(partialParams) {
        return this.client.listStackSetOperationResults(this.ops["ListStackSetOperationResults"].apply(partialParams));
    }
    listStackSetOperations(partialParams) {
        return this.client.listStackSetOperations(this.ops["ListStackSetOperations"].apply(partialParams));
    }
    recordHandlerProgress(partialParams) {
        return this.client.recordHandlerProgress(this.ops["RecordHandlerProgress"].apply(partialParams));
    }
    registerType(partialParams) {
        return this.client.registerType(this.ops["RegisterType"].apply(partialParams));
    }
    rollbackStack(partialParams) {
        return this.client.rollbackStack(this.ops["RollbackStack"].apply(partialParams));
    }
    setStackPolicy(partialParams) {
        return this.client.setStackPolicy(this.ops["SetStackPolicy"].apply(partialParams));
    }
    setTypeConfiguration(partialParams) {
        return this.client.setTypeConfiguration(this.ops["SetTypeConfiguration"].apply(partialParams));
    }
    signalResource(partialParams) {
        return this.client.signalResource(this.ops["SignalResource"].apply(partialParams));
    }
    stopStackSetOperation(partialParams) {
        return this.client.stopStackSetOperation(this.ops["StopStackSetOperation"].apply(partialParams));
    }
    updateStack(partialParams) {
        return this.client.updateStack(this.ops["UpdateStack"].apply(partialParams));
    }
    updateStackInstances(partialParams) {
        return this.client.updateStackInstances(this.ops["UpdateStackInstances"].apply(partialParams));
    }
    updateStackSet(partialParams) {
        return this.client.updateStackSet(this.ops["UpdateStackSet"].apply(partialParams));
    }
    updateTerminationProtection(partialParams) {
        return this.client.updateTerminationProtection(this.ops["UpdateTerminationProtection"].apply(partialParams));
    }
}
exports.default = Resource;
