// out/ecs/Cluster.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.ecs.Cluster {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ECS();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/ecs-2014-11-13.normal.json"), this.client);
    }
    createCapacityProvider(partialParams) {
        return this.client.createCapacityProvider(this.ops["CreateCapacityProvider"].apply(partialParams));
    }
    createService(partialParams) {
        return this.client.createService(this.ops["CreateService"].apply(partialParams));
    }
    createTaskSet(partialParams) {
        return this.client.createTaskSet(this.ops["CreateTaskSet"].apply(partialParams));
    }
    deleteAccountSetting(partialParams) {
        return this.client.deleteAccountSetting(this.ops["DeleteAccountSetting"].apply(partialParams));
    }
    deleteAttributes(partialParams) {
        return this.client.deleteAttributes(this.ops["DeleteAttributes"].apply(partialParams));
    }
    deleteCapacityProvider(partialParams) {
        return this.client.deleteCapacityProvider(this.ops["DeleteCapacityProvider"].apply(partialParams));
    }
    deleteCluster(partialParams) {
        return this.client.deleteCluster(this.ops["DeleteCluster"].apply(partialParams));
    }
    deleteService(partialParams) {
        return this.client.deleteService(this.ops["DeleteService"].apply(partialParams));
    }
    deleteTaskSet(partialParams) {
        return this.client.deleteTaskSet(this.ops["DeleteTaskSet"].apply(partialParams));
    }
    deregisterContainerInstance(partialParams) {
        return this.client.deregisterContainerInstance(this.ops["DeregisterContainerInstance"].apply(partialParams));
    }
    deregisterTaskDefinition(partialParams) {
        return this.client.deregisterTaskDefinition(this.ops["DeregisterTaskDefinition"].apply(partialParams));
    }
    describeContainerInstances(partialParams) {
        return this.client.describeContainerInstances(this.ops["DescribeContainerInstances"].apply(partialParams));
    }
    describeServices(partialParams) {
        return this.client.describeServices(this.ops["DescribeServices"].apply(partialParams));
    }
    describeTaskDefinition(partialParams) {
        return this.client.describeTaskDefinition(this.ops["DescribeTaskDefinition"].apply(partialParams));
    }
    describeTaskSets(partialParams) {
        return this.client.describeTaskSets(this.ops["DescribeTaskSets"].apply(partialParams));
    }
    describeTasks(partialParams) {
        return this.client.describeTasks(this.ops["DescribeTasks"].apply(partialParams));
    }
    executeCommand(partialParams) {
        return this.client.executeCommand(this.ops["ExecuteCommand"].apply(partialParams));
    }
    listAttributes(partialParams) {
        return this.client.listAttributes(this.ops["ListAttributes"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putAccountSetting(partialParams) {
        return this.client.putAccountSetting(this.ops["PutAccountSetting"].apply(partialParams));
    }
    putAccountSettingDefault(partialParams) {
        return this.client.putAccountSettingDefault(this.ops["PutAccountSettingDefault"].apply(partialParams));
    }
    putAttributes(partialParams) {
        return this.client.putAttributes(this.ops["PutAttributes"].apply(partialParams));
    }
    putClusterCapacityProviders(partialParams) {
        return this.client.putClusterCapacityProviders(this.ops["PutClusterCapacityProviders"].apply(partialParams));
    }
    registerTaskDefinition(partialParams) {
        return this.client.registerTaskDefinition(this.ops["RegisterTaskDefinition"].apply(partialParams));
    }
    runTask(partialParams) {
        return this.client.runTask(this.ops["RunTask"].apply(partialParams));
    }
    startTask(partialParams) {
        return this.client.startTask(this.ops["StartTask"].apply(partialParams));
    }
    stopTask(partialParams) {
        return this.client.stopTask(this.ops["StopTask"].apply(partialParams));
    }
    submitAttachmentStateChanges(partialParams) {
        return this.client.submitAttachmentStateChanges(this.ops["SubmitAttachmentStateChanges"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateCapacityProvider(partialParams) {
        return this.client.updateCapacityProvider(this.ops["UpdateCapacityProvider"].apply(partialParams));
    }
    updateCluster(partialParams) {
        return this.client.updateCluster(this.ops["UpdateCluster"].apply(partialParams));
    }
    updateClusterSettings(partialParams) {
        return this.client.updateClusterSettings(this.ops["UpdateClusterSettings"].apply(partialParams));
    }
    updateContainerAgent(partialParams) {
        return this.client.updateContainerAgent(this.ops["UpdateContainerAgent"].apply(partialParams));
    }
    updateContainerInstancesState(partialParams) {
        return this.client.updateContainerInstancesState(this.ops["UpdateContainerInstancesState"].apply(partialParams));
    }
    updateService(partialParams) {
        return this.client.updateService(this.ops["UpdateService"].apply(partialParams));
    }
    updateServicePrimaryTaskSet(partialParams) {
        return this.client.updateServicePrimaryTaskSet(this.ops["UpdateServicePrimaryTaskSet"].apply(partialParams));
    }
    updateTaskSet(partialParams) {
        return this.client.updateTaskSet(this.ops["UpdateTaskSet"].apply(partialParams));
    }
}
exports.default = Resource;
