// out/elasticache/GlobalReplicationGroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.elasticache.GlobalReplicationGroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ElastiCache();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/elasticache-2015-02-02.normal.json"), this.client);
    }
    addTagsToResource(partialParams) {
        return this.client.addTagsToResource(this.ops["AddTagsToResource"].apply(partialParams));
    }
    authorizeCacheSecurityGroupIngress(partialParams) {
        return this.client.authorizeCacheSecurityGroupIngress(this.ops["AuthorizeCacheSecurityGroupIngress"].apply(partialParams));
    }
    batchApplyUpdateAction(partialParams) {
        return this.client.batchApplyUpdateAction(this.ops["BatchApplyUpdateAction"].apply(partialParams));
    }
    batchStopUpdateAction(partialParams) {
        return this.client.batchStopUpdateAction(this.ops["BatchStopUpdateAction"].apply(partialParams));
    }
    completeMigration(partialParams) {
        return this.client.completeMigration(this.ops["CompleteMigration"].apply(partialParams));
    }
    copySnapshot(partialParams) {
        return this.client.copySnapshot(this.ops["CopySnapshot"].apply(partialParams));
    }
    createCacheCluster(partialParams) {
        return this.client.createCacheCluster(this.ops["CreateCacheCluster"].apply(partialParams));
    }
    createCacheParameterGroup(partialParams) {
        return this.client.createCacheParameterGroup(this.ops["CreateCacheParameterGroup"].apply(partialParams));
    }
    createCacheSecurityGroup(partialParams) {
        return this.client.createCacheSecurityGroup(this.ops["CreateCacheSecurityGroup"].apply(partialParams));
    }
    createCacheSubnetGroup(partialParams) {
        return this.client.createCacheSubnetGroup(this.ops["CreateCacheSubnetGroup"].apply(partialParams));
    }
    createGlobalReplicationGroup(partialParams) {
        return this.client.createGlobalReplicationGroup(this.ops["CreateGlobalReplicationGroup"].apply(partialParams));
    }
    createReplicationGroup(partialParams) {
        return this.client.createReplicationGroup(this.ops["CreateReplicationGroup"].apply(partialParams));
    }
    createSnapshot(partialParams) {
        return this.client.createSnapshot(this.ops["CreateSnapshot"].apply(partialParams));
    }
    createUser(partialParams) {
        return this.client.createUser(this.ops["CreateUser"].apply(partialParams));
    }
    createUserGroup(partialParams) {
        return this.client.createUserGroup(this.ops["CreateUserGroup"].apply(partialParams));
    }
    decreaseNodeGroupsInGlobalReplicationGroup(partialParams) {
        return this.client.decreaseNodeGroupsInGlobalReplicationGroup(this.ops["DecreaseNodeGroupsInGlobalReplicationGroup"].apply(partialParams));
    }
    decreaseReplicaCount(partialParams) {
        return this.client.decreaseReplicaCount(this.ops["DecreaseReplicaCount"].apply(partialParams));
    }
    deleteCacheCluster(partialParams) {
        return this.client.deleteCacheCluster(this.ops["DeleteCacheCluster"].apply(partialParams));
    }
    deleteCacheParameterGroup(partialParams) {
        return this.client.deleteCacheParameterGroup(this.ops["DeleteCacheParameterGroup"].apply(partialParams));
    }
    deleteCacheSecurityGroup(partialParams) {
        return this.client.deleteCacheSecurityGroup(this.ops["DeleteCacheSecurityGroup"].apply(partialParams));
    }
    deleteCacheSubnetGroup(partialParams) {
        return this.client.deleteCacheSubnetGroup(this.ops["DeleteCacheSubnetGroup"].apply(partialParams));
    }
    deleteGlobalReplicationGroup(partialParams) {
        return this.client.deleteGlobalReplicationGroup(this.ops["DeleteGlobalReplicationGroup"].apply(partialParams));
    }
    deleteReplicationGroup(partialParams) {
        return this.client.deleteReplicationGroup(this.ops["DeleteReplicationGroup"].apply(partialParams));
    }
    deleteSnapshot(partialParams) {
        return this.client.deleteSnapshot(this.ops["DeleteSnapshot"].apply(partialParams));
    }
    deleteUser(partialParams) {
        return this.client.deleteUser(this.ops["DeleteUser"].apply(partialParams));
    }
    deleteUserGroup(partialParams) {
        return this.client.deleteUserGroup(this.ops["DeleteUserGroup"].apply(partialParams));
    }
    describeCacheParameters(partialParams) {
        return this.client.describeCacheParameters(this.ops["DescribeCacheParameters"].apply(partialParams));
    }
    describeEngineDefaultParameters(partialParams) {
        return this.client.describeEngineDefaultParameters(this.ops["DescribeEngineDefaultParameters"].apply(partialParams));
    }
    disassociateGlobalReplicationGroup(partialParams) {
        return this.client.disassociateGlobalReplicationGroup(this.ops["DisassociateGlobalReplicationGroup"].apply(partialParams));
    }
    failoverGlobalReplicationGroup(partialParams) {
        return this.client.failoverGlobalReplicationGroup(this.ops["FailoverGlobalReplicationGroup"].apply(partialParams));
    }
    increaseNodeGroupsInGlobalReplicationGroup(partialParams) {
        return this.client.increaseNodeGroupsInGlobalReplicationGroup(this.ops["IncreaseNodeGroupsInGlobalReplicationGroup"].apply(partialParams));
    }
    increaseReplicaCount(partialParams) {
        return this.client.increaseReplicaCount(this.ops["IncreaseReplicaCount"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    modifyCacheCluster(partialParams) {
        return this.client.modifyCacheCluster(this.ops["ModifyCacheCluster"].apply(partialParams));
    }
    modifyCacheParameterGroup(partialParams) {
        return this.client.modifyCacheParameterGroup(this.ops["ModifyCacheParameterGroup"].apply(partialParams));
    }
    modifyCacheSubnetGroup(partialParams) {
        return this.client.modifyCacheSubnetGroup(this.ops["ModifyCacheSubnetGroup"].apply(partialParams));
    }
    modifyGlobalReplicationGroup(partialParams) {
        return this.client.modifyGlobalReplicationGroup(this.ops["ModifyGlobalReplicationGroup"].apply(partialParams));
    }
    modifyReplicationGroup(partialParams) {
        return this.client.modifyReplicationGroup(this.ops["ModifyReplicationGroup"].apply(partialParams));
    }
    modifyReplicationGroupShardConfiguration(partialParams) {
        return this.client.modifyReplicationGroupShardConfiguration(this.ops["ModifyReplicationGroupShardConfiguration"].apply(partialParams));
    }
    modifyUser(partialParams) {
        return this.client.modifyUser(this.ops["ModifyUser"].apply(partialParams));
    }
    modifyUserGroup(partialParams) {
        return this.client.modifyUserGroup(this.ops["ModifyUserGroup"].apply(partialParams));
    }
    purchaseReservedCacheNodesOffering(partialParams) {
        return this.client.purchaseReservedCacheNodesOffering(this.ops["PurchaseReservedCacheNodesOffering"].apply(partialParams));
    }
    rebalanceSlotsInGlobalReplicationGroup(partialParams) {
        return this.client.rebalanceSlotsInGlobalReplicationGroup(this.ops["RebalanceSlotsInGlobalReplicationGroup"].apply(partialParams));
    }
    rebootCacheCluster(partialParams) {
        return this.client.rebootCacheCluster(this.ops["RebootCacheCluster"].apply(partialParams));
    }
    removeTagsFromResource(partialParams) {
        return this.client.removeTagsFromResource(this.ops["RemoveTagsFromResource"].apply(partialParams));
    }
    resetCacheParameterGroup(partialParams) {
        return this.client.resetCacheParameterGroup(this.ops["ResetCacheParameterGroup"].apply(partialParams));
    }
    revokeCacheSecurityGroupIngress(partialParams) {
        return this.client.revokeCacheSecurityGroupIngress(this.ops["RevokeCacheSecurityGroupIngress"].apply(partialParams));
    }
    startMigration(partialParams) {
        return this.client.startMigration(this.ops["StartMigration"].apply(partialParams));
    }
    testFailover(partialParams) {
        return this.client.testFailover(this.ops["TestFailover"].apply(partialParams));
    }
}
exports.default = Resource;
