// out/macie2/InvitationAccepter.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.macie2.InvitationAccepter {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Macie2();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/macie2-2020-01-01.normal.json"), this.client);
    }
    acceptInvitation(partialParams) {
        return this.client.acceptInvitation(this.ops["AcceptInvitation"].apply(partialParams));
    }
    createClassificationJob(partialParams) {
        return this.client.createClassificationJob(this.ops["CreateClassificationJob"].apply(partialParams));
    }
    createCustomDataIdentifier(partialParams) {
        return this.client.createCustomDataIdentifier(this.ops["CreateCustomDataIdentifier"].apply(partialParams));
    }
    createFindingsFilter(partialParams) {
        return this.client.createFindingsFilter(this.ops["CreateFindingsFilter"].apply(partialParams));
    }
    createInvitations(partialParams) {
        return this.client.createInvitations(this.ops["CreateInvitations"].apply(partialParams));
    }
    createMember(partialParams) {
        return this.client.createMember(this.ops["CreateMember"].apply(partialParams));
    }
    declineInvitations(partialParams) {
        return this.client.declineInvitations(this.ops["DeclineInvitations"].apply(partialParams));
    }
    deleteCustomDataIdentifier(partialParams) {
        return this.client.deleteCustomDataIdentifier(this.ops["DeleteCustomDataIdentifier"].apply(partialParams));
    }
    deleteFindingsFilter(partialParams) {
        return this.client.deleteFindingsFilter(this.ops["DeleteFindingsFilter"].apply(partialParams));
    }
    deleteInvitations(partialParams) {
        return this.client.deleteInvitations(this.ops["DeleteInvitations"].apply(partialParams));
    }
    deleteMember(partialParams) {
        return this.client.deleteMember(this.ops["DeleteMember"].apply(partialParams));
    }
    describeClassificationJob(partialParams) {
        return this.client.describeClassificationJob(this.ops["DescribeClassificationJob"].apply(partialParams));
    }
    disableOrganizationAdminAccount(partialParams) {
        return this.client.disableOrganizationAdminAccount(this.ops["DisableOrganizationAdminAccount"].apply(partialParams));
    }
    disassociateMember(partialParams) {
        return this.client.disassociateMember(this.ops["DisassociateMember"].apply(partialParams));
    }
    enableOrganizationAdminAccount(partialParams) {
        return this.client.enableOrganizationAdminAccount(this.ops["EnableOrganizationAdminAccount"].apply(partialParams));
    }
    getCustomDataIdentifier(partialParams) {
        return this.client.getCustomDataIdentifier(this.ops["GetCustomDataIdentifier"].apply(partialParams));
    }
    getFindingStatistics(partialParams) {
        return this.client.getFindingStatistics(this.ops["GetFindingStatistics"].apply(partialParams));
    }
    getFindings(partialParams) {
        return this.client.getFindings(this.ops["GetFindings"].apply(partialParams));
    }
    getFindingsFilter(partialParams) {
        return this.client.getFindingsFilter(this.ops["GetFindingsFilter"].apply(partialParams));
    }
    getMember(partialParams) {
        return this.client.getMember(this.ops["GetMember"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putClassificationExportConfiguration(partialParams) {
        return this.client.putClassificationExportConfiguration(this.ops["PutClassificationExportConfiguration"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    testCustomDataIdentifier(partialParams) {
        return this.client.testCustomDataIdentifier(this.ops["TestCustomDataIdentifier"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateClassificationJob(partialParams) {
        return this.client.updateClassificationJob(this.ops["UpdateClassificationJob"].apply(partialParams));
    }
    updateFindingsFilter(partialParams) {
        return this.client.updateFindingsFilter(this.ops["UpdateFindingsFilter"].apply(partialParams));
    }
    updateMemberSession(partialParams) {
        return this.client.updateMemberSession(this.ops["UpdateMemberSession"].apply(partialParams));
    }
    updateOrganizationConfiguration(partialParams) {
        return this.client.updateOrganizationConfiguration(this.ops["UpdateOrganizationConfiguration"].apply(partialParams));
    }
}
exports.default = Resource;
