// out/redshift/ParameterGroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.redshift.ParameterGroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Redshift();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/redshift-2012-12-01.normal.json"), this.client);
    }
    acceptReservedNodeExchange(partialParams) {
        return this.client.acceptReservedNodeExchange(this.ops["AcceptReservedNodeExchange"].apply(partialParams));
    }
    addPartner(partialParams) {
        return this.client.addPartner(this.ops["AddPartner"].apply(partialParams));
    }
    associateDataShareConsumer(partialParams) {
        return this.client.associateDataShareConsumer(this.ops["AssociateDataShareConsumer"].apply(partialParams));
    }
    authorizeClusterSecurityGroupIngress(partialParams) {
        return this.client.authorizeClusterSecurityGroupIngress(this.ops["AuthorizeClusterSecurityGroupIngress"].apply(partialParams));
    }
    authorizeDataShare(partialParams) {
        return this.client.authorizeDataShare(this.ops["AuthorizeDataShare"].apply(partialParams));
    }
    authorizeEndpointAccess(partialParams) {
        return this.client.authorizeEndpointAccess(this.ops["AuthorizeEndpointAccess"].apply(partialParams));
    }
    authorizeSnapshotAccess(partialParams) {
        return this.client.authorizeSnapshotAccess(this.ops["AuthorizeSnapshotAccess"].apply(partialParams));
    }
    batchDeleteClusterSnapshots(partialParams) {
        return this.client.batchDeleteClusterSnapshots(this.ops["BatchDeleteClusterSnapshots"].apply(partialParams));
    }
    batchModifyClusterSnapshots(partialParams) {
        return this.client.batchModifyClusterSnapshots(this.ops["BatchModifyClusterSnapshots"].apply(partialParams));
    }
    cancelResize(partialParams) {
        return this.client.cancelResize(this.ops["CancelResize"].apply(partialParams));
    }
    copyClusterSnapshot(partialParams) {
        return this.client.copyClusterSnapshot(this.ops["CopyClusterSnapshot"].apply(partialParams));
    }
    createAuthenticationProfile(partialParams) {
        return this.client.createAuthenticationProfile(this.ops["CreateAuthenticationProfile"].apply(partialParams));
    }
    createCluster(partialParams) {
        return this.client.createCluster(this.ops["CreateCluster"].apply(partialParams));
    }
    createClusterParameterGroup(partialParams) {
        return this.client.createClusterParameterGroup(this.ops["CreateClusterParameterGroup"].apply(partialParams));
    }
    createClusterSecurityGroup(partialParams) {
        return this.client.createClusterSecurityGroup(this.ops["CreateClusterSecurityGroup"].apply(partialParams));
    }
    createClusterSnapshot(partialParams) {
        return this.client.createClusterSnapshot(this.ops["CreateClusterSnapshot"].apply(partialParams));
    }
    createClusterSubnetGroup(partialParams) {
        return this.client.createClusterSubnetGroup(this.ops["CreateClusterSubnetGroup"].apply(partialParams));
    }
    createEndpointAccess(partialParams) {
        return this.client.createEndpointAccess(this.ops["CreateEndpointAccess"].apply(partialParams));
    }
    createEventSubscription(partialParams) {
        return this.client.createEventSubscription(this.ops["CreateEventSubscription"].apply(partialParams));
    }
    createHsmClientCertificate(partialParams) {
        return this.client.createHsmClientCertificate(this.ops["CreateHsmClientCertificate"].apply(partialParams));
    }
    createHsmConfiguration(partialParams) {
        return this.client.createHsmConfiguration(this.ops["CreateHsmConfiguration"].apply(partialParams));
    }
    createScheduledAction(partialParams) {
        return this.client.createScheduledAction(this.ops["CreateScheduledAction"].apply(partialParams));
    }
    createSnapshotCopyGrant(partialParams) {
        return this.client.createSnapshotCopyGrant(this.ops["CreateSnapshotCopyGrant"].apply(partialParams));
    }
    createTags(partialParams) {
        return this.client.createTags(this.ops["CreateTags"].apply(partialParams));
    }
    createUsageLimit(partialParams) {
        return this.client.createUsageLimit(this.ops["CreateUsageLimit"].apply(partialParams));
    }
    deauthorizeDataShare(partialParams) {
        return this.client.deauthorizeDataShare(this.ops["DeauthorizeDataShare"].apply(partialParams));
    }
    deleteAuthenticationProfile(partialParams) {
        return this.client.deleteAuthenticationProfile(this.ops["DeleteAuthenticationProfile"].apply(partialParams));
    }
    deleteCluster(partialParams) {
        return this.client.deleteCluster(this.ops["DeleteCluster"].apply(partialParams));
    }
    deleteClusterParameterGroup(partialParams) {
        return this.client.deleteClusterParameterGroup(this.ops["DeleteClusterParameterGroup"].apply(partialParams));
    }
    deleteClusterSecurityGroup(partialParams) {
        return this.client.deleteClusterSecurityGroup(this.ops["DeleteClusterSecurityGroup"].apply(partialParams));
    }
    deleteClusterSnapshot(partialParams) {
        return this.client.deleteClusterSnapshot(this.ops["DeleteClusterSnapshot"].apply(partialParams));
    }
    deleteClusterSubnetGroup(partialParams) {
        return this.client.deleteClusterSubnetGroup(this.ops["DeleteClusterSubnetGroup"].apply(partialParams));
    }
    deleteEndpointAccess(partialParams) {
        return this.client.deleteEndpointAccess(this.ops["DeleteEndpointAccess"].apply(partialParams));
    }
    deleteEventSubscription(partialParams) {
        return this.client.deleteEventSubscription(this.ops["DeleteEventSubscription"].apply(partialParams));
    }
    deleteHsmClientCertificate(partialParams) {
        return this.client.deleteHsmClientCertificate(this.ops["DeleteHsmClientCertificate"].apply(partialParams));
    }
    deleteHsmConfiguration(partialParams) {
        return this.client.deleteHsmConfiguration(this.ops["DeleteHsmConfiguration"].apply(partialParams));
    }
    deletePartner(partialParams) {
        return this.client.deletePartner(this.ops["DeletePartner"].apply(partialParams));
    }
    deleteScheduledAction(partialParams) {
        return this.client.deleteScheduledAction(this.ops["DeleteScheduledAction"].apply(partialParams));
    }
    deleteSnapshotCopyGrant(partialParams) {
        return this.client.deleteSnapshotCopyGrant(this.ops["DeleteSnapshotCopyGrant"].apply(partialParams));
    }
    deleteSnapshotSchedule(partialParams) {
        return this.client.deleteSnapshotSchedule(this.ops["DeleteSnapshotSchedule"].apply(partialParams));
    }
    deleteTags(partialParams) {
        return this.client.deleteTags(this.ops["DeleteTags"].apply(partialParams));
    }
    deleteUsageLimit(partialParams) {
        return this.client.deleteUsageLimit(this.ops["DeleteUsageLimit"].apply(partialParams));
    }
    describeClusterParameters(partialParams) {
        return this.client.describeClusterParameters(this.ops["DescribeClusterParameters"].apply(partialParams));
    }
    describeDefaultClusterParameters(partialParams) {
        return this.client.describeDefaultClusterParameters(this.ops["DescribeDefaultClusterParameters"].apply(partialParams));
    }
    describeLoggingStatus(partialParams) {
        return this.client.describeLoggingStatus(this.ops["DescribeLoggingStatus"].apply(partialParams));
    }
    describeNodeConfigurationOptions(partialParams) {
        return this.client.describeNodeConfigurationOptions(this.ops["DescribeNodeConfigurationOptions"].apply(partialParams));
    }
    describePartners(partialParams) {
        return this.client.describePartners(this.ops["DescribePartners"].apply(partialParams));
    }
    describeResize(partialParams) {
        return this.client.describeResize(this.ops["DescribeResize"].apply(partialParams));
    }
    disableLogging(partialParams) {
        return this.client.disableLogging(this.ops["DisableLogging"].apply(partialParams));
    }
    disableSnapshotCopy(partialParams) {
        return this.client.disableSnapshotCopy(this.ops["DisableSnapshotCopy"].apply(partialParams));
    }
    disassociateDataShareConsumer(partialParams) {
        return this.client.disassociateDataShareConsumer(this.ops["DisassociateDataShareConsumer"].apply(partialParams));
    }
    enableLogging(partialParams) {
        return this.client.enableLogging(this.ops["EnableLogging"].apply(partialParams));
    }
    enableSnapshotCopy(partialParams) {
        return this.client.enableSnapshotCopy(this.ops["EnableSnapshotCopy"].apply(partialParams));
    }
    getClusterCredentials(partialParams) {
        return this.client.getClusterCredentials(this.ops["GetClusterCredentials"].apply(partialParams));
    }
    getReservedNodeExchangeConfigurationOptions(partialParams) {
        return this.client.getReservedNodeExchangeConfigurationOptions(this.ops["GetReservedNodeExchangeConfigurationOptions"].apply(partialParams));
    }
    getReservedNodeExchangeOfferings(partialParams) {
        return this.client.getReservedNodeExchangeOfferings(this.ops["GetReservedNodeExchangeOfferings"].apply(partialParams));
    }
    modifyAquaConfiguration(partialParams) {
        return this.client.modifyAquaConfiguration(this.ops["ModifyAquaConfiguration"].apply(partialParams));
    }
    modifyAuthenticationProfile(partialParams) {
        return this.client.modifyAuthenticationProfile(this.ops["ModifyAuthenticationProfile"].apply(partialParams));
    }
    modifyCluster(partialParams) {
        return this.client.modifyCluster(this.ops["ModifyCluster"].apply(partialParams));
    }
    modifyClusterDbRevision(partialParams) {
        return this.client.modifyClusterDbRevision(this.ops["ModifyClusterDbRevision"].apply(partialParams));
    }
    modifyClusterIamRoles(partialParams) {
        return this.client.modifyClusterIamRoles(this.ops["ModifyClusterIamRoles"].apply(partialParams));
    }
    modifyClusterMaintenance(partialParams) {
        return this.client.modifyClusterMaintenance(this.ops["ModifyClusterMaintenance"].apply(partialParams));
    }
    modifyClusterParameterGroup(partialParams) {
        return this.client.modifyClusterParameterGroup(this.ops["ModifyClusterParameterGroup"].apply(partialParams));
    }
    modifyClusterSnapshot(partialParams) {
        return this.client.modifyClusterSnapshot(this.ops["ModifyClusterSnapshot"].apply(partialParams));
    }
    modifyClusterSnapshotSchedule(partialParams) {
        return this.client.modifyClusterSnapshotSchedule(this.ops["ModifyClusterSnapshotSchedule"].apply(partialParams));
    }
    modifyClusterSubnetGroup(partialParams) {
        return this.client.modifyClusterSubnetGroup(this.ops["ModifyClusterSubnetGroup"].apply(partialParams));
    }
    modifyEndpointAccess(partialParams) {
        return this.client.modifyEndpointAccess(this.ops["ModifyEndpointAccess"].apply(partialParams));
    }
    modifyEventSubscription(partialParams) {
        return this.client.modifyEventSubscription(this.ops["ModifyEventSubscription"].apply(partialParams));
    }
    modifyScheduledAction(partialParams) {
        return this.client.modifyScheduledAction(this.ops["ModifyScheduledAction"].apply(partialParams));
    }
    modifySnapshotCopyRetentionPeriod(partialParams) {
        return this.client.modifySnapshotCopyRetentionPeriod(this.ops["ModifySnapshotCopyRetentionPeriod"].apply(partialParams));
    }
    modifySnapshotSchedule(partialParams) {
        return this.client.modifySnapshotSchedule(this.ops["ModifySnapshotSchedule"].apply(partialParams));
    }
    modifyUsageLimit(partialParams) {
        return this.client.modifyUsageLimit(this.ops["ModifyUsageLimit"].apply(partialParams));
    }
    pauseCluster(partialParams) {
        return this.client.pauseCluster(this.ops["PauseCluster"].apply(partialParams));
    }
    purchaseReservedNodeOffering(partialParams) {
        return this.client.purchaseReservedNodeOffering(this.ops["PurchaseReservedNodeOffering"].apply(partialParams));
    }
    rebootCluster(partialParams) {
        return this.client.rebootCluster(this.ops["RebootCluster"].apply(partialParams));
    }
    rejectDataShare(partialParams) {
        return this.client.rejectDataShare(this.ops["RejectDataShare"].apply(partialParams));
    }
    resetClusterParameterGroup(partialParams) {
        return this.client.resetClusterParameterGroup(this.ops["ResetClusterParameterGroup"].apply(partialParams));
    }
    resizeCluster(partialParams) {
        return this.client.resizeCluster(this.ops["ResizeCluster"].apply(partialParams));
    }
    restoreFromClusterSnapshot(partialParams) {
        return this.client.restoreFromClusterSnapshot(this.ops["RestoreFromClusterSnapshot"].apply(partialParams));
    }
    restoreTableFromClusterSnapshot(partialParams) {
        return this.client.restoreTableFromClusterSnapshot(this.ops["RestoreTableFromClusterSnapshot"].apply(partialParams));
    }
    resumeCluster(partialParams) {
        return this.client.resumeCluster(this.ops["ResumeCluster"].apply(partialParams));
    }
    revokeClusterSecurityGroupIngress(partialParams) {
        return this.client.revokeClusterSecurityGroupIngress(this.ops["RevokeClusterSecurityGroupIngress"].apply(partialParams));
    }
    revokeSnapshotAccess(partialParams) {
        return this.client.revokeSnapshotAccess(this.ops["RevokeSnapshotAccess"].apply(partialParams));
    }
    rotateEncryptionKey(partialParams) {
        return this.client.rotateEncryptionKey(this.ops["RotateEncryptionKey"].apply(partialParams));
    }
    updatePartnerStatus(partialParams) {
        return this.client.updatePartnerStatus(this.ops["UpdatePartnerStatus"].apply(partialParams));
    }
}
exports.default = Resource;
