// out/docdb/SubnetGroup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.docdb.SubnetGroup {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.DocDB();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/docdb-2014-10-31.normal.json"), this.client);
    }
    addSourceIdentifierToSubscription(partialParams) {
        return this.client.addSourceIdentifierToSubscription(this.ops["AddSourceIdentifierToSubscription"].apply(partialParams));
    }
    addTagsToResource(partialParams) {
        return this.client.addTagsToResource(this.ops["AddTagsToResource"].apply(partialParams));
    }
    applyPendingMaintenanceAction(partialParams) {
        return this.client.applyPendingMaintenanceAction(this.ops["ApplyPendingMaintenanceAction"].apply(partialParams));
    }
    copyDBClusterParameterGroup(partialParams) {
        return this.client.copyDBClusterParameterGroup(this.ops["CopyDBClusterParameterGroup"].apply(partialParams));
    }
    copyDBClusterSnapshot(partialParams) {
        return this.client.copyDBClusterSnapshot(this.ops["CopyDBClusterSnapshot"].apply(partialParams));
    }
    createDBCluster(partialParams) {
        return this.client.createDBCluster(this.ops["CreateDBCluster"].apply(partialParams));
    }
    createDBClusterParameterGroup(partialParams) {
        return this.client.createDBClusterParameterGroup(this.ops["CreateDBClusterParameterGroup"].apply(partialParams));
    }
    createDBClusterSnapshot(partialParams) {
        return this.client.createDBClusterSnapshot(this.ops["CreateDBClusterSnapshot"].apply(partialParams));
    }
    createDBInstance(partialParams) {
        return this.client.createDBInstance(this.ops["CreateDBInstance"].apply(partialParams));
    }
    createDBSubnetGroup(partialParams) {
        return this.client.createDBSubnetGroup(this.ops["CreateDBSubnetGroup"].apply(partialParams));
    }
    createEventSubscription(partialParams) {
        return this.client.createEventSubscription(this.ops["CreateEventSubscription"].apply(partialParams));
    }
    createGlobalCluster(partialParams) {
        return this.client.createGlobalCluster(this.ops["CreateGlobalCluster"].apply(partialParams));
    }
    deleteDBCluster(partialParams) {
        return this.client.deleteDBCluster(this.ops["DeleteDBCluster"].apply(partialParams));
    }
    deleteDBClusterParameterGroup(partialParams) {
        return this.client.deleteDBClusterParameterGroup(this.ops["DeleteDBClusterParameterGroup"].apply(partialParams));
    }
    deleteDBClusterSnapshot(partialParams) {
        return this.client.deleteDBClusterSnapshot(this.ops["DeleteDBClusterSnapshot"].apply(partialParams));
    }
    deleteDBInstance(partialParams) {
        return this.client.deleteDBInstance(this.ops["DeleteDBInstance"].apply(partialParams));
    }
    deleteDBSubnetGroup(partialParams) {
        return this.client.deleteDBSubnetGroup(this.ops["DeleteDBSubnetGroup"].apply(partialParams));
    }
    deleteEventSubscription(partialParams) {
        return this.client.deleteEventSubscription(this.ops["DeleteEventSubscription"].apply(partialParams));
    }
    deleteGlobalCluster(partialParams) {
        return this.client.deleteGlobalCluster(this.ops["DeleteGlobalCluster"].apply(partialParams));
    }
    describeDBClusterParameters(partialParams) {
        return this.client.describeDBClusterParameters(this.ops["DescribeDBClusterParameters"].apply(partialParams));
    }
    describeDBClusterSnapshotAttributes(partialParams) {
        return this.client.describeDBClusterSnapshotAttributes(this.ops["DescribeDBClusterSnapshotAttributes"].apply(partialParams));
    }
    describeEngineDefaultClusterParameters(partialParams) {
        return this.client.describeEngineDefaultClusterParameters(this.ops["DescribeEngineDefaultClusterParameters"].apply(partialParams));
    }
    describeOrderableDBInstanceOptions(partialParams) {
        return this.client.describeOrderableDBInstanceOptions(this.ops["DescribeOrderableDBInstanceOptions"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    modifyDBCluster(partialParams) {
        return this.client.modifyDBCluster(this.ops["ModifyDBCluster"].apply(partialParams));
    }
    modifyDBClusterParameterGroup(partialParams) {
        return this.client.modifyDBClusterParameterGroup(this.ops["ModifyDBClusterParameterGroup"].apply(partialParams));
    }
    modifyDBClusterSnapshotAttribute(partialParams) {
        return this.client.modifyDBClusterSnapshotAttribute(this.ops["ModifyDBClusterSnapshotAttribute"].apply(partialParams));
    }
    modifyDBInstance(partialParams) {
        return this.client.modifyDBInstance(this.ops["ModifyDBInstance"].apply(partialParams));
    }
    modifyDBSubnetGroup(partialParams) {
        return this.client.modifyDBSubnetGroup(this.ops["ModifyDBSubnetGroup"].apply(partialParams));
    }
    modifyEventSubscription(partialParams) {
        return this.client.modifyEventSubscription(this.ops["ModifyEventSubscription"].apply(partialParams));
    }
    modifyGlobalCluster(partialParams) {
        return this.client.modifyGlobalCluster(this.ops["ModifyGlobalCluster"].apply(partialParams));
    }
    rebootDBInstance(partialParams) {
        return this.client.rebootDBInstance(this.ops["RebootDBInstance"].apply(partialParams));
    }
    removeFromGlobalCluster(partialParams) {
        return this.client.removeFromGlobalCluster(this.ops["RemoveFromGlobalCluster"].apply(partialParams));
    }
    removeSourceIdentifierFromSubscription(partialParams) {
        return this.client.removeSourceIdentifierFromSubscription(this.ops["RemoveSourceIdentifierFromSubscription"].apply(partialParams));
    }
    removeTagsFromResource(partialParams) {
        return this.client.removeTagsFromResource(this.ops["RemoveTagsFromResource"].apply(partialParams));
    }
    resetDBClusterParameterGroup(partialParams) {
        return this.client.resetDBClusterParameterGroup(this.ops["ResetDBClusterParameterGroup"].apply(partialParams));
    }
    restoreDBClusterFromSnapshot(partialParams) {
        return this.client.restoreDBClusterFromSnapshot(this.ops["RestoreDBClusterFromSnapshot"].apply(partialParams));
    }
    restoreDBClusterToPointInTime(partialParams) {
        return this.client.restoreDBClusterToPointInTime(this.ops["RestoreDBClusterToPointInTime"].apply(partialParams));
    }
    startDBCluster(partialParams) {
        return this.client.startDBCluster(this.ops["StartDBCluster"].apply(partialParams));
    }
    stopDBCluster(partialParams) {
        return this.client.stopDBCluster(this.ops["StopDBCluster"].apply(partialParams));
    }
}
exports.default = Resource;
