// out/shield/ProtectionHealthCheckAssociation.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.shield.ProtectionHealthCheckAssociation {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Shield();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/shield-2016-06-02.normal.json"), this.client);
    }
    associateDRTLogBucket(partialParams) {
        return this.client.associateDRTLogBucket(this.ops["AssociateDRTLogBucket"].apply(partialParams));
    }
    associateDRTRole(partialParams) {
        return this.client.associateDRTRole(this.ops["AssociateDRTRole"].apply(partialParams));
    }
    associateHealthCheck(partialParams) {
        return this.client.associateHealthCheck(this.ops["AssociateHealthCheck"].apply(partialParams));
    }
    associateProactiveEngagementDetails(partialParams) {
        return this.client.associateProactiveEngagementDetails(this.ops["AssociateProactiveEngagementDetails"].apply(partialParams));
    }
    createProtection(partialParams) {
        return this.client.createProtection(this.ops["CreateProtection"].apply(partialParams));
    }
    createProtectionGroup(partialParams) {
        return this.client.createProtectionGroup(this.ops["CreateProtectionGroup"].apply(partialParams));
    }
    deleteProtection(partialParams) {
        return this.client.deleteProtection(this.ops["DeleteProtection"].apply(partialParams));
    }
    deleteProtectionGroup(partialParams) {
        return this.client.deleteProtectionGroup(this.ops["DeleteProtectionGroup"].apply(partialParams));
    }
    describeAttack(partialParams) {
        return this.client.describeAttack(this.ops["DescribeAttack"].apply(partialParams));
    }
    describeProtectionGroup(partialParams) {
        return this.client.describeProtectionGroup(this.ops["DescribeProtectionGroup"].apply(partialParams));
    }
    disableApplicationLayerAutomaticResponse(partialParams) {
        return this.client.disableApplicationLayerAutomaticResponse(this.ops["DisableApplicationLayerAutomaticResponse"].apply(partialParams));
    }
    disassociateDRTLogBucket(partialParams) {
        return this.client.disassociateDRTLogBucket(this.ops["DisassociateDRTLogBucket"].apply(partialParams));
    }
    disassociateHealthCheck(partialParams) {
        return this.client.disassociateHealthCheck(this.ops["DisassociateHealthCheck"].apply(partialParams));
    }
    enableApplicationLayerAutomaticResponse(partialParams) {
        return this.client.enableApplicationLayerAutomaticResponse(this.ops["EnableApplicationLayerAutomaticResponse"].apply(partialParams));
    }
    listResourcesInProtectionGroup(partialParams) {
        return this.client.listResourcesInProtectionGroup(this.ops["ListResourcesInProtectionGroup"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApplicationLayerAutomaticResponse(partialParams) {
        return this.client.updateApplicationLayerAutomaticResponse(this.ops["UpdateApplicationLayerAutomaticResponse"].apply(partialParams));
    }
    updateProtectionGroup(partialParams) {
        return this.client.updateProtectionGroup(this.ops["UpdateProtectionGroup"].apply(partialParams));
    }
}
exports.default = Resource;
