// out/grafana/WorkspaceSamlConfiguration.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.grafana.WorkspaceSamlConfiguration {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.Grafana();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/grafana-2020-08-18.normal.json"), this.client);
    }
    associateLicense(partialParams) {
        return this.client.associateLicense(this.ops["AssociateLicense"].apply(partialParams));
    }
    createWorkspace(partialParams) {
        return this.client.createWorkspace(this.ops["CreateWorkspace"].apply(partialParams));
    }
    deleteWorkspace(partialParams) {
        return this.client.deleteWorkspace(this.ops["DeleteWorkspace"].apply(partialParams));
    }
    describeWorkspace(partialParams) {
        return this.client.describeWorkspace(this.ops["DescribeWorkspace"].apply(partialParams));
    }
    describeWorkspaceAuthentication(partialParams) {
        return this.client.describeWorkspaceAuthentication(this.ops["DescribeWorkspaceAuthentication"].apply(partialParams));
    }
    disassociateLicense(partialParams) {
        return this.client.disassociateLicense(this.ops["DisassociateLicense"].apply(partialParams));
    }
    listPermissions(partialParams) {
        return this.client.listPermissions(this.ops["ListPermissions"].apply(partialParams));
    }
    updatePermissions(partialParams) {
        return this.client.updatePermissions(this.ops["UpdatePermissions"].apply(partialParams));
    }
    updateWorkspace(partialParams) {
        return this.client.updateWorkspace(this.ops["UpdateWorkspace"].apply(partialParams));
    }
    updateWorkspaceAuthentication(partialParams) {
        return this.client.updateWorkspaceAuthentication(this.ops["UpdateWorkspaceAuthentication"].apply(partialParams));
    }
}
exports.default = Resource;
