// out/cloudsearch/DomainServiceAccessPolicy.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.cloudsearch.DomainServiceAccessPolicy {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.CloudSearch();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/cloudsearch-2013-01-01.normal.json"), this.client);
    }
    buildSuggesters(partialParams) {
        return this.client.buildSuggesters(this.ops["BuildSuggesters"].apply(partialParams));
    }
    createDomain(partialParams) {
        return this.client.createDomain(this.ops["CreateDomain"].apply(partialParams));
    }
    defineAnalysisScheme(partialParams) {
        return this.client.defineAnalysisScheme(this.ops["DefineAnalysisScheme"].apply(partialParams));
    }
    defineExpression(partialParams) {
        return this.client.defineExpression(this.ops["DefineExpression"].apply(partialParams));
    }
    defineIndexField(partialParams) {
        return this.client.defineIndexField(this.ops["DefineIndexField"].apply(partialParams));
    }
    defineSuggester(partialParams) {
        return this.client.defineSuggester(this.ops["DefineSuggester"].apply(partialParams));
    }
    deleteAnalysisScheme(partialParams) {
        return this.client.deleteAnalysisScheme(this.ops["DeleteAnalysisScheme"].apply(partialParams));
    }
    deleteDomain(partialParams) {
        return this.client.deleteDomain(this.ops["DeleteDomain"].apply(partialParams));
    }
    deleteExpression(partialParams) {
        return this.client.deleteExpression(this.ops["DeleteExpression"].apply(partialParams));
    }
    deleteIndexField(partialParams) {
        return this.client.deleteIndexField(this.ops["DeleteIndexField"].apply(partialParams));
    }
    deleteSuggester(partialParams) {
        return this.client.deleteSuggester(this.ops["DeleteSuggester"].apply(partialParams));
    }
    describeAnalysisSchemes(partialParams) {
        return this.client.describeAnalysisSchemes(this.ops["DescribeAnalysisSchemes"].apply(partialParams));
    }
    describeAvailabilityOptions(partialParams) {
        return this.client.describeAvailabilityOptions(this.ops["DescribeAvailabilityOptions"].apply(partialParams));
    }
    describeDomainEndpointOptions(partialParams) {
        return this.client.describeDomainEndpointOptions(this.ops["DescribeDomainEndpointOptions"].apply(partialParams));
    }
    describeExpressions(partialParams) {
        return this.client.describeExpressions(this.ops["DescribeExpressions"].apply(partialParams));
    }
    describeIndexFields(partialParams) {
        return this.client.describeIndexFields(this.ops["DescribeIndexFields"].apply(partialParams));
    }
    describeScalingParameters(partialParams) {
        return this.client.describeScalingParameters(this.ops["DescribeScalingParameters"].apply(partialParams));
    }
    describeServiceAccessPolicies(partialParams) {
        return this.client.describeServiceAccessPolicies(this.ops["DescribeServiceAccessPolicies"].apply(partialParams));
    }
    describeSuggesters(partialParams) {
        return this.client.describeSuggesters(this.ops["DescribeSuggesters"].apply(partialParams));
    }
    indexDocuments(partialParams) {
        return this.client.indexDocuments(this.ops["IndexDocuments"].apply(partialParams));
    }
    updateAvailabilityOptions(partialParams) {
        return this.client.updateAvailabilityOptions(this.ops["UpdateAvailabilityOptions"].apply(partialParams));
    }
    updateDomainEndpointOptions(partialParams) {
        return this.client.updateDomainEndpointOptions(this.ops["UpdateDomainEndpointOptions"].apply(partialParams));
    }
    updateScalingParameters(partialParams) {
        return this.client.updateScalingParameters(this.ops["UpdateScalingParameters"].apply(partialParams));
    }
    updateServiceAccessPolicies(partialParams) {
        return this.client.updateServiceAccessPolicies(this.ops["UpdateServiceAccessPolicies"].apply(partialParams));
    }
}
exports.default = Resource;
