// out/gamelift/Alias.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.gamelift.Alias {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.GameLift();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/gamelift-2015-10-01.normal.json"), this.client);
    }
    acceptMatch(partialParams) {
        return this.client.acceptMatch(this.ops["AcceptMatch"].apply(partialParams));
    }
    claimGameServer(partialParams) {
        return this.client.claimGameServer(this.ops["ClaimGameServer"].apply(partialParams));
    }
    createAlias(partialParams) {
        return this.client.createAlias(this.ops["CreateAlias"].apply(partialParams));
    }
    createFleet(partialParams) {
        return this.client.createFleet(this.ops["CreateFleet"].apply(partialParams));
    }
    createFleetLocations(partialParams) {
        return this.client.createFleetLocations(this.ops["CreateFleetLocations"].apply(partialParams));
    }
    createGameServerGroup(partialParams) {
        return this.client.createGameServerGroup(this.ops["CreateGameServerGroup"].apply(partialParams));
    }
    createGameSession(partialParams) {
        return this.client.createGameSession(this.ops["CreateGameSession"].apply(partialParams));
    }
    createGameSessionQueue(partialParams) {
        return this.client.createGameSessionQueue(this.ops["CreateGameSessionQueue"].apply(partialParams));
    }
    createMatchmakingConfiguration(partialParams) {
        return this.client.createMatchmakingConfiguration(this.ops["CreateMatchmakingConfiguration"].apply(partialParams));
    }
    createMatchmakingRuleSet(partialParams) {
        return this.client.createMatchmakingRuleSet(this.ops["CreateMatchmakingRuleSet"].apply(partialParams));
    }
    createPlayerSession(partialParams) {
        return this.client.createPlayerSession(this.ops["CreatePlayerSession"].apply(partialParams));
    }
    createPlayerSessions(partialParams) {
        return this.client.createPlayerSessions(this.ops["CreatePlayerSessions"].apply(partialParams));
    }
    createVpcPeeringAuthorization(partialParams) {
        return this.client.createVpcPeeringAuthorization(this.ops["CreateVpcPeeringAuthorization"].apply(partialParams));
    }
    createVpcPeeringConnection(partialParams) {
        return this.client.createVpcPeeringConnection(this.ops["CreateVpcPeeringConnection"].apply(partialParams));
    }
    deleteAlias(partialParams) {
        return this.client.deleteAlias(this.ops["DeleteAlias"].apply(partialParams));
    }
    deleteBuild(partialParams) {
        return this.client.deleteBuild(this.ops["DeleteBuild"].apply(partialParams));
    }
    deleteFleet(partialParams) {
        return this.client.deleteFleet(this.ops["DeleteFleet"].apply(partialParams));
    }
    deleteFleetLocations(partialParams) {
        return this.client.deleteFleetLocations(this.ops["DeleteFleetLocations"].apply(partialParams));
    }
    deleteGameServerGroup(partialParams) {
        return this.client.deleteGameServerGroup(this.ops["DeleteGameServerGroup"].apply(partialParams));
    }
    deleteGameSessionQueue(partialParams) {
        return this.client.deleteGameSessionQueue(this.ops["DeleteGameSessionQueue"].apply(partialParams));
    }
    deleteMatchmakingConfiguration(partialParams) {
        return this.client.deleteMatchmakingConfiguration(this.ops["DeleteMatchmakingConfiguration"].apply(partialParams));
    }
    deleteMatchmakingRuleSet(partialParams) {
        return this.client.deleteMatchmakingRuleSet(this.ops["DeleteMatchmakingRuleSet"].apply(partialParams));
    }
    deleteScalingPolicy(partialParams) {
        return this.client.deleteScalingPolicy(this.ops["DeleteScalingPolicy"].apply(partialParams));
    }
    deleteScript(partialParams) {
        return this.client.deleteScript(this.ops["DeleteScript"].apply(partialParams));
    }
    deleteVpcPeeringAuthorization(partialParams) {
        return this.client.deleteVpcPeeringAuthorization(this.ops["DeleteVpcPeeringAuthorization"].apply(partialParams));
    }
    deleteVpcPeeringConnection(partialParams) {
        return this.client.deleteVpcPeeringConnection(this.ops["DeleteVpcPeeringConnection"].apply(partialParams));
    }
    deregisterGameServer(partialParams) {
        return this.client.deregisterGameServer(this.ops["DeregisterGameServer"].apply(partialParams));
    }
    describeAlias(partialParams) {
        return this.client.describeAlias(this.ops["DescribeAlias"].apply(partialParams));
    }
    describeBuild(partialParams) {
        return this.client.describeBuild(this.ops["DescribeBuild"].apply(partialParams));
    }
    describeFleetEvents(partialParams) {
        return this.client.describeFleetEvents(this.ops["DescribeFleetEvents"].apply(partialParams));
    }
    describeFleetLocationAttributes(partialParams) {
        return this.client.describeFleetLocationAttributes(this.ops["DescribeFleetLocationAttributes"].apply(partialParams));
    }
    describeFleetLocationCapacity(partialParams) {
        return this.client.describeFleetLocationCapacity(this.ops["DescribeFleetLocationCapacity"].apply(partialParams));
    }
    describeFleetLocationUtilization(partialParams) {
        return this.client.describeFleetLocationUtilization(this.ops["DescribeFleetLocationUtilization"].apply(partialParams));
    }
    describeFleetPortSettings(partialParams) {
        return this.client.describeFleetPortSettings(this.ops["DescribeFleetPortSettings"].apply(partialParams));
    }
    describeGameServer(partialParams) {
        return this.client.describeGameServer(this.ops["DescribeGameServer"].apply(partialParams));
    }
    describeGameServerGroup(partialParams) {
        return this.client.describeGameServerGroup(this.ops["DescribeGameServerGroup"].apply(partialParams));
    }
    describeGameServerInstances(partialParams) {
        return this.client.describeGameServerInstances(this.ops["DescribeGameServerInstances"].apply(partialParams));
    }
    describeGameSessionPlacement(partialParams) {
        return this.client.describeGameSessionPlacement(this.ops["DescribeGameSessionPlacement"].apply(partialParams));
    }
    describeInstances(partialParams) {
        return this.client.describeInstances(this.ops["DescribeInstances"].apply(partialParams));
    }
    describeMatchmaking(partialParams) {
        return this.client.describeMatchmaking(this.ops["DescribeMatchmaking"].apply(partialParams));
    }
    describeRuntimeConfiguration(partialParams) {
        return this.client.describeRuntimeConfiguration(this.ops["DescribeRuntimeConfiguration"].apply(partialParams));
    }
    describeScalingPolicies(partialParams) {
        return this.client.describeScalingPolicies(this.ops["DescribeScalingPolicies"].apply(partialParams));
    }
    describeScript(partialParams) {
        return this.client.describeScript(this.ops["DescribeScript"].apply(partialParams));
    }
    getGameSessionLogUrl(partialParams) {
        return this.client.getGameSessionLogUrl(this.ops["GetGameSessionLogUrl"].apply(partialParams));
    }
    getInstanceAccess(partialParams) {
        return this.client.getInstanceAccess(this.ops["GetInstanceAccess"].apply(partialParams));
    }
    listGameServers(partialParams) {
        return this.client.listGameServers(this.ops["ListGameServers"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    putScalingPolicy(partialParams) {
        return this.client.putScalingPolicy(this.ops["PutScalingPolicy"].apply(partialParams));
    }
    registerGameServer(partialParams) {
        return this.client.registerGameServer(this.ops["RegisterGameServer"].apply(partialParams));
    }
    requestUploadCredentials(partialParams) {
        return this.client.requestUploadCredentials(this.ops["RequestUploadCredentials"].apply(partialParams));
    }
    resolveAlias(partialParams) {
        return this.client.resolveAlias(this.ops["ResolveAlias"].apply(partialParams));
    }
    resumeGameServerGroup(partialParams) {
        return this.client.resumeGameServerGroup(this.ops["ResumeGameServerGroup"].apply(partialParams));
    }
    startFleetActions(partialParams) {
        return this.client.startFleetActions(this.ops["StartFleetActions"].apply(partialParams));
    }
    startGameSessionPlacement(partialParams) {
        return this.client.startGameSessionPlacement(this.ops["StartGameSessionPlacement"].apply(partialParams));
    }
    startMatchBackfill(partialParams) {
        return this.client.startMatchBackfill(this.ops["StartMatchBackfill"].apply(partialParams));
    }
    startMatchmaking(partialParams) {
        return this.client.startMatchmaking(this.ops["StartMatchmaking"].apply(partialParams));
    }
    stopFleetActions(partialParams) {
        return this.client.stopFleetActions(this.ops["StopFleetActions"].apply(partialParams));
    }
    stopGameSessionPlacement(partialParams) {
        return this.client.stopGameSessionPlacement(this.ops["StopGameSessionPlacement"].apply(partialParams));
    }
    stopMatchmaking(partialParams) {
        return this.client.stopMatchmaking(this.ops["StopMatchmaking"].apply(partialParams));
    }
    suspendGameServerGroup(partialParams) {
        return this.client.suspendGameServerGroup(this.ops["SuspendGameServerGroup"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAlias(partialParams) {
        return this.client.updateAlias(this.ops["UpdateAlias"].apply(partialParams));
    }
    updateBuild(partialParams) {
        return this.client.updateBuild(this.ops["UpdateBuild"].apply(partialParams));
    }
    updateFleetAttributes(partialParams) {
        return this.client.updateFleetAttributes(this.ops["UpdateFleetAttributes"].apply(partialParams));
    }
    updateFleetCapacity(partialParams) {
        return this.client.updateFleetCapacity(this.ops["UpdateFleetCapacity"].apply(partialParams));
    }
    updateFleetPortSettings(partialParams) {
        return this.client.updateFleetPortSettings(this.ops["UpdateFleetPortSettings"].apply(partialParams));
    }
    updateGameServer(partialParams) {
        return this.client.updateGameServer(this.ops["UpdateGameServer"].apply(partialParams));
    }
    updateGameServerGroup(partialParams) {
        return this.client.updateGameServerGroup(this.ops["UpdateGameServerGroup"].apply(partialParams));
    }
    updateGameSession(partialParams) {
        return this.client.updateGameSession(this.ops["UpdateGameSession"].apply(partialParams));
    }
    updateGameSessionQueue(partialParams) {
        return this.client.updateGameSessionQueue(this.ops["UpdateGameSessionQueue"].apply(partialParams));
    }
    updateMatchmakingConfiguration(partialParams) {
        return this.client.updateMatchmakingConfiguration(this.ops["UpdateMatchmakingConfiguration"].apply(partialParams));
    }
    updateRuntimeConfiguration(partialParams) {
        return this.client.updateRuntimeConfiguration(this.ops["UpdateRuntimeConfiguration"].apply(partialParams));
    }
    updateScript(partialParams) {
        return this.client.updateScript(this.ops["UpdateScript"].apply(partialParams));
    }
    validateMatchmakingRuleSet(partialParams) {
        return this.client.validateMatchmakingRuleSet(this.ops["ValidateMatchmakingRuleSet"].apply(partialParams));
    }
}
exports.default = Resource;
