// out/s3/BucketLoggingV2.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.s3.BucketLoggingV2 {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.S3();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/s3-2006-03-01.normal.json"), this.client);
    }
    abortMultipartUpload(partialParams) {
        return this.client.abortMultipartUpload(this.ops["AbortMultipartUpload"].apply(partialParams));
    }
    completeMultipartUpload(partialParams) {
        return this.client.completeMultipartUpload(this.ops["CompleteMultipartUpload"].apply(partialParams));
    }
    copyObject(partialParams) {
        return this.client.copyObject(this.ops["CopyObject"].apply(partialParams));
    }
    createBucket(partialParams) {
        return this.client.createBucket(this.ops["CreateBucket"].apply(partialParams));
    }
    createMultipartUpload(partialParams) {
        return this.client.createMultipartUpload(this.ops["CreateMultipartUpload"].apply(partialParams));
    }
    deleteBucket(partialParams) {
        return this.client.deleteBucket(this.ops["DeleteBucket"].apply(partialParams));
    }
    deleteBucketAnalyticsConfiguration(partialParams) {
        return this.client.deleteBucketAnalyticsConfiguration(this.ops["DeleteBucketAnalyticsConfiguration"].apply(partialParams));
    }
    deleteBucketCors(partialParams) {
        return this.client.deleteBucketCors(this.ops["DeleteBucketCors"].apply(partialParams));
    }
    deleteBucketEncryption(partialParams) {
        return this.client.deleteBucketEncryption(this.ops["DeleteBucketEncryption"].apply(partialParams));
    }
    deleteBucketIntelligentTieringConfiguration(partialParams) {
        return this.client.deleteBucketIntelligentTieringConfiguration(this.ops["DeleteBucketIntelligentTieringConfiguration"].apply(partialParams));
    }
    deleteBucketInventoryConfiguration(partialParams) {
        return this.client.deleteBucketInventoryConfiguration(this.ops["DeleteBucketInventoryConfiguration"].apply(partialParams));
    }
    deleteBucketLifecycle(partialParams) {
        return this.client.deleteBucketLifecycle(this.ops["DeleteBucketLifecycle"].apply(partialParams));
    }
    deleteBucketMetricsConfiguration(partialParams) {
        return this.client.deleteBucketMetricsConfiguration(this.ops["DeleteBucketMetricsConfiguration"].apply(partialParams));
    }
    deleteBucketOwnershipControls(partialParams) {
        return this.client.deleteBucketOwnershipControls(this.ops["DeleteBucketOwnershipControls"].apply(partialParams));
    }
    deleteBucketPolicy(partialParams) {
        return this.client.deleteBucketPolicy(this.ops["DeleteBucketPolicy"].apply(partialParams));
    }
    deleteBucketReplication(partialParams) {
        return this.client.deleteBucketReplication(this.ops["DeleteBucketReplication"].apply(partialParams));
    }
    deleteBucketTagging(partialParams) {
        return this.client.deleteBucketTagging(this.ops["DeleteBucketTagging"].apply(partialParams));
    }
    deleteBucketWebsite(partialParams) {
        return this.client.deleteBucketWebsite(this.ops["DeleteBucketWebsite"].apply(partialParams));
    }
    deleteObject(partialParams) {
        return this.client.deleteObject(this.ops["DeleteObject"].apply(partialParams));
    }
    deleteObjectTagging(partialParams) {
        return this.client.deleteObjectTagging(this.ops["DeleteObjectTagging"].apply(partialParams));
    }
    deleteObjects(partialParams) {
        return this.client.deleteObjects(this.ops["DeleteObjects"].apply(partialParams));
    }
    deletePublicAccessBlock(partialParams) {
        return this.client.deletePublicAccessBlock(this.ops["DeletePublicAccessBlock"].apply(partialParams));
    }
    getBucketAccelerateConfiguration(partialParams) {
        return this.client.getBucketAccelerateConfiguration(this.ops["GetBucketAccelerateConfiguration"].apply(partialParams));
    }
    getBucketAcl(partialParams) {
        return this.client.getBucketAcl(this.ops["GetBucketAcl"].apply(partialParams));
    }
    getBucketAnalyticsConfiguration(partialParams) {
        return this.client.getBucketAnalyticsConfiguration(this.ops["GetBucketAnalyticsConfiguration"].apply(partialParams));
    }
    getBucketCors(partialParams) {
        return this.client.getBucketCors(this.ops["GetBucketCors"].apply(partialParams));
    }
    getBucketEncryption(partialParams) {
        return this.client.getBucketEncryption(this.ops["GetBucketEncryption"].apply(partialParams));
    }
    getBucketIntelligentTieringConfiguration(partialParams) {
        return this.client.getBucketIntelligentTieringConfiguration(this.ops["GetBucketIntelligentTieringConfiguration"].apply(partialParams));
    }
    getBucketInventoryConfiguration(partialParams) {
        return this.client.getBucketInventoryConfiguration(this.ops["GetBucketInventoryConfiguration"].apply(partialParams));
    }
    getBucketLifecycle(partialParams) {
        return this.client.getBucketLifecycle(this.ops["GetBucketLifecycle"].apply(partialParams));
    }
    getBucketLifecycleConfiguration(partialParams) {
        return this.client.getBucketLifecycleConfiguration(this.ops["GetBucketLifecycleConfiguration"].apply(partialParams));
    }
    getBucketLocation(partialParams) {
        return this.client.getBucketLocation(this.ops["GetBucketLocation"].apply(partialParams));
    }
    getBucketLogging(partialParams) {
        return this.client.getBucketLogging(this.ops["GetBucketLogging"].apply(partialParams));
    }
    getBucketMetricsConfiguration(partialParams) {
        return this.client.getBucketMetricsConfiguration(this.ops["GetBucketMetricsConfiguration"].apply(partialParams));
    }
    getBucketNotification(partialParams) {
        return this.client.getBucketNotification(this.ops["GetBucketNotification"].apply(partialParams));
    }
    getBucketNotificationConfiguration(partialParams) {
        return this.client.getBucketNotificationConfiguration(this.ops["GetBucketNotificationConfiguration"].apply(partialParams));
    }
    getBucketOwnershipControls(partialParams) {
        return this.client.getBucketOwnershipControls(this.ops["GetBucketOwnershipControls"].apply(partialParams));
    }
    getBucketPolicy(partialParams) {
        return this.client.getBucketPolicy(this.ops["GetBucketPolicy"].apply(partialParams));
    }
    getBucketPolicyStatus(partialParams) {
        return this.client.getBucketPolicyStatus(this.ops["GetBucketPolicyStatus"].apply(partialParams));
    }
    getBucketReplication(partialParams) {
        return this.client.getBucketReplication(this.ops["GetBucketReplication"].apply(partialParams));
    }
    getBucketRequestPayment(partialParams) {
        return this.client.getBucketRequestPayment(this.ops["GetBucketRequestPayment"].apply(partialParams));
    }
    getBucketTagging(partialParams) {
        return this.client.getBucketTagging(this.ops["GetBucketTagging"].apply(partialParams));
    }
    getBucketVersioning(partialParams) {
        return this.client.getBucketVersioning(this.ops["GetBucketVersioning"].apply(partialParams));
    }
    getBucketWebsite(partialParams) {
        return this.client.getBucketWebsite(this.ops["GetBucketWebsite"].apply(partialParams));
    }
    getObject(partialParams) {
        return this.client.getObject(this.ops["GetObject"].apply(partialParams));
    }
    getObjectAcl(partialParams) {
        return this.client.getObjectAcl(this.ops["GetObjectAcl"].apply(partialParams));
    }
    getObjectLegalHold(partialParams) {
        return this.client.getObjectLegalHold(this.ops["GetObjectLegalHold"].apply(partialParams));
    }
    getObjectLockConfiguration(partialParams) {
        return this.client.getObjectLockConfiguration(this.ops["GetObjectLockConfiguration"].apply(partialParams));
    }
    getObjectRetention(partialParams) {
        return this.client.getObjectRetention(this.ops["GetObjectRetention"].apply(partialParams));
    }
    getObjectTagging(partialParams) {
        return this.client.getObjectTagging(this.ops["GetObjectTagging"].apply(partialParams));
    }
    getObjectTorrent(partialParams) {
        return this.client.getObjectTorrent(this.ops["GetObjectTorrent"].apply(partialParams));
    }
    getPublicAccessBlock(partialParams) {
        return this.client.getPublicAccessBlock(this.ops["GetPublicAccessBlock"].apply(partialParams));
    }
    headBucket(partialParams) {
        return this.client.headBucket(this.ops["HeadBucket"].apply(partialParams));
    }
    headObject(partialParams) {
        return this.client.headObject(this.ops["HeadObject"].apply(partialParams));
    }
    listBucketAnalyticsConfigurations(partialParams) {
        return this.client.listBucketAnalyticsConfigurations(this.ops["ListBucketAnalyticsConfigurations"].apply(partialParams));
    }
    listBucketIntelligentTieringConfigurations(partialParams) {
        return this.client.listBucketIntelligentTieringConfigurations(this.ops["ListBucketIntelligentTieringConfigurations"].apply(partialParams));
    }
    listBucketInventoryConfigurations(partialParams) {
        return this.client.listBucketInventoryConfigurations(this.ops["ListBucketInventoryConfigurations"].apply(partialParams));
    }
    listBucketMetricsConfigurations(partialParams) {
        return this.client.listBucketMetricsConfigurations(this.ops["ListBucketMetricsConfigurations"].apply(partialParams));
    }
    listMultipartUploads(partialParams) {
        return this.client.listMultipartUploads(this.ops["ListMultipartUploads"].apply(partialParams));
    }
    listObjectVersions(partialParams) {
        return this.client.listObjectVersions(this.ops["ListObjectVersions"].apply(partialParams));
    }
    listObjects(partialParams) {
        return this.client.listObjects(this.ops["ListObjects"].apply(partialParams));
    }
    listObjectsV2(partialParams) {
        return this.client.listObjectsV2(this.ops["ListObjectsV2"].apply(partialParams));
    }
    listParts(partialParams) {
        return this.client.listParts(this.ops["ListParts"].apply(partialParams));
    }
    putBucketAccelerateConfiguration(partialParams) {
        return this.client.putBucketAccelerateConfiguration(this.ops["PutBucketAccelerateConfiguration"].apply(partialParams));
    }
    putBucketAcl(partialParams) {
        return this.client.putBucketAcl(this.ops["PutBucketAcl"].apply(partialParams));
    }
    putBucketAnalyticsConfiguration(partialParams) {
        return this.client.putBucketAnalyticsConfiguration(this.ops["PutBucketAnalyticsConfiguration"].apply(partialParams));
    }
    putBucketCors(partialParams) {
        return this.client.putBucketCors(this.ops["PutBucketCors"].apply(partialParams));
    }
    putBucketEncryption(partialParams) {
        return this.client.putBucketEncryption(this.ops["PutBucketEncryption"].apply(partialParams));
    }
    putBucketIntelligentTieringConfiguration(partialParams) {
        return this.client.putBucketIntelligentTieringConfiguration(this.ops["PutBucketIntelligentTieringConfiguration"].apply(partialParams));
    }
    putBucketInventoryConfiguration(partialParams) {
        return this.client.putBucketInventoryConfiguration(this.ops["PutBucketInventoryConfiguration"].apply(partialParams));
    }
    putBucketLifecycle(partialParams) {
        return this.client.putBucketLifecycle(this.ops["PutBucketLifecycle"].apply(partialParams));
    }
    putBucketLifecycleConfiguration(partialParams) {
        return this.client.putBucketLifecycleConfiguration(this.ops["PutBucketLifecycleConfiguration"].apply(partialParams));
    }
    putBucketLogging(partialParams) {
        return this.client.putBucketLogging(this.ops["PutBucketLogging"].apply(partialParams));
    }
    putBucketMetricsConfiguration(partialParams) {
        return this.client.putBucketMetricsConfiguration(this.ops["PutBucketMetricsConfiguration"].apply(partialParams));
    }
    putBucketNotification(partialParams) {
        return this.client.putBucketNotification(this.ops["PutBucketNotification"].apply(partialParams));
    }
    putBucketNotificationConfiguration(partialParams) {
        return this.client.putBucketNotificationConfiguration(this.ops["PutBucketNotificationConfiguration"].apply(partialParams));
    }
    putBucketOwnershipControls(partialParams) {
        return this.client.putBucketOwnershipControls(this.ops["PutBucketOwnershipControls"].apply(partialParams));
    }
    putBucketPolicy(partialParams) {
        return this.client.putBucketPolicy(this.ops["PutBucketPolicy"].apply(partialParams));
    }
    putBucketReplication(partialParams) {
        return this.client.putBucketReplication(this.ops["PutBucketReplication"].apply(partialParams));
    }
    putBucketRequestPayment(partialParams) {
        return this.client.putBucketRequestPayment(this.ops["PutBucketRequestPayment"].apply(partialParams));
    }
    putBucketTagging(partialParams) {
        return this.client.putBucketTagging(this.ops["PutBucketTagging"].apply(partialParams));
    }
    putBucketVersioning(partialParams) {
        return this.client.putBucketVersioning(this.ops["PutBucketVersioning"].apply(partialParams));
    }
    putBucketWebsite(partialParams) {
        return this.client.putBucketWebsite(this.ops["PutBucketWebsite"].apply(partialParams));
    }
    putObject(partialParams) {
        return this.client.putObject(this.ops["PutObject"].apply(partialParams));
    }
    putObjectAcl(partialParams) {
        return this.client.putObjectAcl(this.ops["PutObjectAcl"].apply(partialParams));
    }
    putObjectLegalHold(partialParams) {
        return this.client.putObjectLegalHold(this.ops["PutObjectLegalHold"].apply(partialParams));
    }
    putObjectLockConfiguration(partialParams) {
        return this.client.putObjectLockConfiguration(this.ops["PutObjectLockConfiguration"].apply(partialParams));
    }
    putObjectRetention(partialParams) {
        return this.client.putObjectRetention(this.ops["PutObjectRetention"].apply(partialParams));
    }
    putObjectTagging(partialParams) {
        return this.client.putObjectTagging(this.ops["PutObjectTagging"].apply(partialParams));
    }
    putPublicAccessBlock(partialParams) {
        return this.client.putPublicAccessBlock(this.ops["PutPublicAccessBlock"].apply(partialParams));
    }
    restoreObject(partialParams) {
        return this.client.restoreObject(this.ops["RestoreObject"].apply(partialParams));
    }
    selectObjectContent(partialParams) {
        return this.client.selectObjectContent(this.ops["SelectObjectContent"].apply(partialParams));
    }
    uploadPart(partialParams) {
        return this.client.uploadPart(this.ops["UploadPart"].apply(partialParams));
    }
    uploadPartCopy(partialParams) {
        return this.client.uploadPartCopy(this.ops["UploadPartCopy"].apply(partialParams));
    }
    writeGetObjectResponse(partialParams) {
        return this.client.writeGetObjectResponse(this.ops["WriteGetObjectResponse"].apply(partialParams));
    }
}
exports.default = Resource;
