// out/lakeformation/Resource.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.lakeformation.Resource {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.LakeFormation();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/lakeformation-2017-03-31.normal.json"), this.client);
    }
    addLFTagsToResource(partialParams) {
        return this.client.addLFTagsToResource(this.ops["AddLFTagsToResource"].apply(partialParams));
    }
    batchGrantPermissions(partialParams) {
        return this.client.batchGrantPermissions(this.ops["BatchGrantPermissions"].apply(partialParams));
    }
    batchRevokePermissions(partialParams) {
        return this.client.batchRevokePermissions(this.ops["BatchRevokePermissions"].apply(partialParams));
    }
    cancelTransaction(partialParams) {
        return this.client.cancelTransaction(this.ops["CancelTransaction"].apply(partialParams));
    }
    commitTransaction(partialParams) {
        return this.client.commitTransaction(this.ops["CommitTransaction"].apply(partialParams));
    }
    createDataCellsFilter(partialParams) {
        return this.client.createDataCellsFilter(this.ops["CreateDataCellsFilter"].apply(partialParams));
    }
    createLFTag(partialParams) {
        return this.client.createLFTag(this.ops["CreateLFTag"].apply(partialParams));
    }
    deleteLFTag(partialParams) {
        return this.client.deleteLFTag(this.ops["DeleteLFTag"].apply(partialParams));
    }
    deleteObjectsOnCancel(partialParams) {
        return this.client.deleteObjectsOnCancel(this.ops["DeleteObjectsOnCancel"].apply(partialParams));
    }
    deregisterResource(partialParams) {
        return this.client.deregisterResource(this.ops["DeregisterResource"].apply(partialParams));
    }
    describeResource(partialParams) {
        return this.client.describeResource(this.ops["DescribeResource"].apply(partialParams));
    }
    describeTransaction(partialParams) {
        return this.client.describeTransaction(this.ops["DescribeTransaction"].apply(partialParams));
    }
    getEffectivePermissionsForPath(partialParams) {
        return this.client.getEffectivePermissionsForPath(this.ops["GetEffectivePermissionsForPath"].apply(partialParams));
    }
    getLFTag(partialParams) {
        return this.client.getLFTag(this.ops["GetLFTag"].apply(partialParams));
    }
    getQueryState(partialParams) {
        return this.client.getQueryState(this.ops["GetQueryState"].apply(partialParams));
    }
    getQueryStatistics(partialParams) {
        return this.client.getQueryStatistics(this.ops["GetQueryStatistics"].apply(partialParams));
    }
    getResourceLFTags(partialParams) {
        return this.client.getResourceLFTags(this.ops["GetResourceLFTags"].apply(partialParams));
    }
    getTableObjects(partialParams) {
        return this.client.getTableObjects(this.ops["GetTableObjects"].apply(partialParams));
    }
    getTemporaryGluePartitionCredentials(partialParams) {
        return this.client.getTemporaryGluePartitionCredentials(this.ops["GetTemporaryGluePartitionCredentials"].apply(partialParams));
    }
    getTemporaryGlueTableCredentials(partialParams) {
        return this.client.getTemporaryGlueTableCredentials(this.ops["GetTemporaryGlueTableCredentials"].apply(partialParams));
    }
    getWorkUnitResults(partialParams) {
        return this.client.getWorkUnitResults(this.ops["GetWorkUnitResults"].apply(partialParams));
    }
    getWorkUnits(partialParams) {
        return this.client.getWorkUnits(this.ops["GetWorkUnits"].apply(partialParams));
    }
    grantPermissions(partialParams) {
        return this.client.grantPermissions(this.ops["GrantPermissions"].apply(partialParams));
    }
    listTableStorageOptimizers(partialParams) {
        return this.client.listTableStorageOptimizers(this.ops["ListTableStorageOptimizers"].apply(partialParams));
    }
    putDataLakeSettings(partialParams) {
        return this.client.putDataLakeSettings(this.ops["PutDataLakeSettings"].apply(partialParams));
    }
    registerResource(partialParams) {
        return this.client.registerResource(this.ops["RegisterResource"].apply(partialParams));
    }
    removeLFTagsFromResource(partialParams) {
        return this.client.removeLFTagsFromResource(this.ops["RemoveLFTagsFromResource"].apply(partialParams));
    }
    revokePermissions(partialParams) {
        return this.client.revokePermissions(this.ops["RevokePermissions"].apply(partialParams));
    }
    searchDatabasesByLFTags(partialParams) {
        return this.client.searchDatabasesByLFTags(this.ops["SearchDatabasesByLFTags"].apply(partialParams));
    }
    searchTablesByLFTags(partialParams) {
        return this.client.searchTablesByLFTags(this.ops["SearchTablesByLFTags"].apply(partialParams));
    }
    startQueryPlanning(partialParams) {
        return this.client.startQueryPlanning(this.ops["StartQueryPlanning"].apply(partialParams));
    }
    updateLFTag(partialParams) {
        return this.client.updateLFTag(this.ops["UpdateLFTag"].apply(partialParams));
    }
    updateResource(partialParams) {
        return this.client.updateResource(this.ops["UpdateResource"].apply(partialParams));
    }
    updateTableObjects(partialParams) {
        return this.client.updateTableObjects(this.ops["UpdateTableObjects"].apply(partialParams));
    }
    updateTableStorageOptimizer(partialParams) {
        return this.client.updateTableStorageOptimizer(this.ops["UpdateTableStorageOptimizer"].apply(partialParams));
    }
}
exports.default = Resource;
