// out/apigateway/Integration.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.apigateway.Integration {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.APIGateway();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/apigateway-2015-07-09.normal.json"), this.client);
    }
    createAuthorizer(partialParams) {
        return this.client.createAuthorizer(this.ops["CreateAuthorizer"].apply(partialParams));
    }
    createBasePathMapping(partialParams) {
        return this.client.createBasePathMapping(this.ops["CreateBasePathMapping"].apply(partialParams));
    }
    createDeployment(partialParams) {
        return this.client.createDeployment(this.ops["CreateDeployment"].apply(partialParams));
    }
    createDocumentationPart(partialParams) {
        return this.client.createDocumentationPart(this.ops["CreateDocumentationPart"].apply(partialParams));
    }
    createDocumentationVersion(partialParams) {
        return this.client.createDocumentationVersion(this.ops["CreateDocumentationVersion"].apply(partialParams));
    }
    createDomainName(partialParams) {
        return this.client.createDomainName(this.ops["CreateDomainName"].apply(partialParams));
    }
    createModel(partialParams) {
        return this.client.createModel(this.ops["CreateModel"].apply(partialParams));
    }
    createRequestValidator(partialParams) {
        return this.client.createRequestValidator(this.ops["CreateRequestValidator"].apply(partialParams));
    }
    createResource(partialParams) {
        return this.client.createResource(this.ops["CreateResource"].apply(partialParams));
    }
    createRestApi(partialParams) {
        return this.client.createRestApi(this.ops["CreateRestApi"].apply(partialParams));
    }
    createStage(partialParams) {
        return this.client.createStage(this.ops["CreateStage"].apply(partialParams));
    }
    createUsagePlan(partialParams) {
        return this.client.createUsagePlan(this.ops["CreateUsagePlan"].apply(partialParams));
    }
    createUsagePlanKey(partialParams) {
        return this.client.createUsagePlanKey(this.ops["CreateUsagePlanKey"].apply(partialParams));
    }
    createVpcLink(partialParams) {
        return this.client.createVpcLink(this.ops["CreateVpcLink"].apply(partialParams));
    }
    deleteApiKey(partialParams) {
        return this.client.deleteApiKey(this.ops["DeleteApiKey"].apply(partialParams));
    }
    deleteAuthorizer(partialParams) {
        return this.client.deleteAuthorizer(this.ops["DeleteAuthorizer"].apply(partialParams));
    }
    deleteBasePathMapping(partialParams) {
        return this.client.deleteBasePathMapping(this.ops["DeleteBasePathMapping"].apply(partialParams));
    }
    deleteClientCertificate(partialParams) {
        return this.client.deleteClientCertificate(this.ops["DeleteClientCertificate"].apply(partialParams));
    }
    deleteDeployment(partialParams) {
        return this.client.deleteDeployment(this.ops["DeleteDeployment"].apply(partialParams));
    }
    deleteDocumentationPart(partialParams) {
        return this.client.deleteDocumentationPart(this.ops["DeleteDocumentationPart"].apply(partialParams));
    }
    deleteDocumentationVersion(partialParams) {
        return this.client.deleteDocumentationVersion(this.ops["DeleteDocumentationVersion"].apply(partialParams));
    }
    deleteDomainName(partialParams) {
        return this.client.deleteDomainName(this.ops["DeleteDomainName"].apply(partialParams));
    }
    deleteGatewayResponse(partialParams) {
        return this.client.deleteGatewayResponse(this.ops["DeleteGatewayResponse"].apply(partialParams));
    }
    deleteIntegration(partialParams) {
        return this.client.deleteIntegration(this.ops["DeleteIntegration"].apply(partialParams));
    }
    deleteIntegrationResponse(partialParams) {
        return this.client.deleteIntegrationResponse(this.ops["DeleteIntegrationResponse"].apply(partialParams));
    }
    deleteMethod(partialParams) {
        return this.client.deleteMethod(this.ops["DeleteMethod"].apply(partialParams));
    }
    deleteMethodResponse(partialParams) {
        return this.client.deleteMethodResponse(this.ops["DeleteMethodResponse"].apply(partialParams));
    }
    deleteModel(partialParams) {
        return this.client.deleteModel(this.ops["DeleteModel"].apply(partialParams));
    }
    deleteRequestValidator(partialParams) {
        return this.client.deleteRequestValidator(this.ops["DeleteRequestValidator"].apply(partialParams));
    }
    deleteResource(partialParams) {
        return this.client.deleteResource(this.ops["DeleteResource"].apply(partialParams));
    }
    deleteRestApi(partialParams) {
        return this.client.deleteRestApi(this.ops["DeleteRestApi"].apply(partialParams));
    }
    deleteStage(partialParams) {
        return this.client.deleteStage(this.ops["DeleteStage"].apply(partialParams));
    }
    deleteUsagePlan(partialParams) {
        return this.client.deleteUsagePlan(this.ops["DeleteUsagePlan"].apply(partialParams));
    }
    deleteUsagePlanKey(partialParams) {
        return this.client.deleteUsagePlanKey(this.ops["DeleteUsagePlanKey"].apply(partialParams));
    }
    deleteVpcLink(partialParams) {
        return this.client.deleteVpcLink(this.ops["DeleteVpcLink"].apply(partialParams));
    }
    flushStageAuthorizersCache(partialParams) {
        return this.client.flushStageAuthorizersCache(this.ops["FlushStageAuthorizersCache"].apply(partialParams));
    }
    flushStageCache(partialParams) {
        return this.client.flushStageCache(this.ops["FlushStageCache"].apply(partialParams));
    }
    getApiKey(partialParams) {
        return this.client.getApiKey(this.ops["GetApiKey"].apply(partialParams));
    }
    getAuthorizer(partialParams) {
        return this.client.getAuthorizer(this.ops["GetAuthorizer"].apply(partialParams));
    }
    getAuthorizers(partialParams) {
        return this.client.getAuthorizers(this.ops["GetAuthorizers"].apply(partialParams));
    }
    getBasePathMapping(partialParams) {
        return this.client.getBasePathMapping(this.ops["GetBasePathMapping"].apply(partialParams));
    }
    getBasePathMappings(partialParams) {
        return this.client.getBasePathMappings(this.ops["GetBasePathMappings"].apply(partialParams));
    }
    getClientCertificate(partialParams) {
        return this.client.getClientCertificate(this.ops["GetClientCertificate"].apply(partialParams));
    }
    getDeployment(partialParams) {
        return this.client.getDeployment(this.ops["GetDeployment"].apply(partialParams));
    }
    getDeployments(partialParams) {
        return this.client.getDeployments(this.ops["GetDeployments"].apply(partialParams));
    }
    getDocumentationPart(partialParams) {
        return this.client.getDocumentationPart(this.ops["GetDocumentationPart"].apply(partialParams));
    }
    getDocumentationParts(partialParams) {
        return this.client.getDocumentationParts(this.ops["GetDocumentationParts"].apply(partialParams));
    }
    getDocumentationVersion(partialParams) {
        return this.client.getDocumentationVersion(this.ops["GetDocumentationVersion"].apply(partialParams));
    }
    getDocumentationVersions(partialParams) {
        return this.client.getDocumentationVersions(this.ops["GetDocumentationVersions"].apply(partialParams));
    }
    getDomainName(partialParams) {
        return this.client.getDomainName(this.ops["GetDomainName"].apply(partialParams));
    }
    getExport(partialParams) {
        return this.client.getExport(this.ops["GetExport"].apply(partialParams));
    }
    getGatewayResponse(partialParams) {
        return this.client.getGatewayResponse(this.ops["GetGatewayResponse"].apply(partialParams));
    }
    getGatewayResponses(partialParams) {
        return this.client.getGatewayResponses(this.ops["GetGatewayResponses"].apply(partialParams));
    }
    getIntegration(partialParams) {
        return this.client.getIntegration(this.ops["GetIntegration"].apply(partialParams));
    }
    getIntegrationResponse(partialParams) {
        return this.client.getIntegrationResponse(this.ops["GetIntegrationResponse"].apply(partialParams));
    }
    getMethod(partialParams) {
        return this.client.getMethod(this.ops["GetMethod"].apply(partialParams));
    }
    getMethodResponse(partialParams) {
        return this.client.getMethodResponse(this.ops["GetMethodResponse"].apply(partialParams));
    }
    getModel(partialParams) {
        return this.client.getModel(this.ops["GetModel"].apply(partialParams));
    }
    getModelTemplate(partialParams) {
        return this.client.getModelTemplate(this.ops["GetModelTemplate"].apply(partialParams));
    }
    getModels(partialParams) {
        return this.client.getModels(this.ops["GetModels"].apply(partialParams));
    }
    getRequestValidator(partialParams) {
        return this.client.getRequestValidator(this.ops["GetRequestValidator"].apply(partialParams));
    }
    getRequestValidators(partialParams) {
        return this.client.getRequestValidators(this.ops["GetRequestValidators"].apply(partialParams));
    }
    getResource(partialParams) {
        return this.client.getResource(this.ops["GetResource"].apply(partialParams));
    }
    getResources(partialParams) {
        return this.client.getResources(this.ops["GetResources"].apply(partialParams));
    }
    getRestApi(partialParams) {
        return this.client.getRestApi(this.ops["GetRestApi"].apply(partialParams));
    }
    getSdk(partialParams) {
        return this.client.getSdk(this.ops["GetSdk"].apply(partialParams));
    }
    getSdkType(partialParams) {
        return this.client.getSdkType(this.ops["GetSdkType"].apply(partialParams));
    }
    getStage(partialParams) {
        return this.client.getStage(this.ops["GetStage"].apply(partialParams));
    }
    getStages(partialParams) {
        return this.client.getStages(this.ops["GetStages"].apply(partialParams));
    }
    getTags(partialParams) {
        return this.client.getTags(this.ops["GetTags"].apply(partialParams));
    }
    getUsage(partialParams) {
        return this.client.getUsage(this.ops["GetUsage"].apply(partialParams));
    }
    getUsagePlan(partialParams) {
        return this.client.getUsagePlan(this.ops["GetUsagePlan"].apply(partialParams));
    }
    getUsagePlanKey(partialParams) {
        return this.client.getUsagePlanKey(this.ops["GetUsagePlanKey"].apply(partialParams));
    }
    getUsagePlanKeys(partialParams) {
        return this.client.getUsagePlanKeys(this.ops["GetUsagePlanKeys"].apply(partialParams));
    }
    getVpcLink(partialParams) {
        return this.client.getVpcLink(this.ops["GetVpcLink"].apply(partialParams));
    }
    importApiKeys(partialParams) {
        return this.client.importApiKeys(this.ops["ImportApiKeys"].apply(partialParams));
    }
    importDocumentationParts(partialParams) {
        return this.client.importDocumentationParts(this.ops["ImportDocumentationParts"].apply(partialParams));
    }
    importRestApi(partialParams) {
        return this.client.importRestApi(this.ops["ImportRestApi"].apply(partialParams));
    }
    putGatewayResponse(partialParams) {
        return this.client.putGatewayResponse(this.ops["PutGatewayResponse"].apply(partialParams));
    }
    putIntegration(partialParams) {
        return this.client.putIntegration(this.ops["PutIntegration"].apply(partialParams));
    }
    putIntegrationResponse(partialParams) {
        return this.client.putIntegrationResponse(this.ops["PutIntegrationResponse"].apply(partialParams));
    }
    putMethod(partialParams) {
        return this.client.putMethod(this.ops["PutMethod"].apply(partialParams));
    }
    putMethodResponse(partialParams) {
        return this.client.putMethodResponse(this.ops["PutMethodResponse"].apply(partialParams));
    }
    putRestApi(partialParams) {
        return this.client.putRestApi(this.ops["PutRestApi"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    testInvokeAuthorizer(partialParams) {
        return this.client.testInvokeAuthorizer(this.ops["TestInvokeAuthorizer"].apply(partialParams));
    }
    testInvokeMethod(partialParams) {
        return this.client.testInvokeMethod(this.ops["TestInvokeMethod"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateApiKey(partialParams) {
        return this.client.updateApiKey(this.ops["UpdateApiKey"].apply(partialParams));
    }
    updateAuthorizer(partialParams) {
        return this.client.updateAuthorizer(this.ops["UpdateAuthorizer"].apply(partialParams));
    }
    updateBasePathMapping(partialParams) {
        return this.client.updateBasePathMapping(this.ops["UpdateBasePathMapping"].apply(partialParams));
    }
    updateClientCertificate(partialParams) {
        return this.client.updateClientCertificate(this.ops["UpdateClientCertificate"].apply(partialParams));
    }
    updateDeployment(partialParams) {
        return this.client.updateDeployment(this.ops["UpdateDeployment"].apply(partialParams));
    }
    updateDocumentationPart(partialParams) {
        return this.client.updateDocumentationPart(this.ops["UpdateDocumentationPart"].apply(partialParams));
    }
    updateDocumentationVersion(partialParams) {
        return this.client.updateDocumentationVersion(this.ops["UpdateDocumentationVersion"].apply(partialParams));
    }
    updateDomainName(partialParams) {
        return this.client.updateDomainName(this.ops["UpdateDomainName"].apply(partialParams));
    }
    updateGatewayResponse(partialParams) {
        return this.client.updateGatewayResponse(this.ops["UpdateGatewayResponse"].apply(partialParams));
    }
    updateIntegration(partialParams) {
        return this.client.updateIntegration(this.ops["UpdateIntegration"].apply(partialParams));
    }
    updateIntegrationResponse(partialParams) {
        return this.client.updateIntegrationResponse(this.ops["UpdateIntegrationResponse"].apply(partialParams));
    }
    updateMethod(partialParams) {
        return this.client.updateMethod(this.ops["UpdateMethod"].apply(partialParams));
    }
    updateMethodResponse(partialParams) {
        return this.client.updateMethodResponse(this.ops["UpdateMethodResponse"].apply(partialParams));
    }
    updateModel(partialParams) {
        return this.client.updateModel(this.ops["UpdateModel"].apply(partialParams));
    }
    updateRequestValidator(partialParams) {
        return this.client.updateRequestValidator(this.ops["UpdateRequestValidator"].apply(partialParams));
    }
    updateResource(partialParams) {
        return this.client.updateResource(this.ops["UpdateResource"].apply(partialParams));
    }
    updateRestApi(partialParams) {
        return this.client.updateRestApi(this.ops["UpdateRestApi"].apply(partialParams));
    }
    updateStage(partialParams) {
        return this.client.updateStage(this.ops["UpdateStage"].apply(partialParams));
    }
    updateUsage(partialParams) {
        return this.client.updateUsage(this.ops["UpdateUsage"].apply(partialParams));
    }
    updateUsagePlan(partialParams) {
        return this.client.updateUsagePlan(this.ops["UpdateUsagePlan"].apply(partialParams));
    }
    updateVpcLink(partialParams) {
        return this.client.updateVpcLink(this.ops["UpdateVpcLink"].apply(partialParams));
    }
}
exports.default = Resource;
