// out/quicksight/GroupMembership.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.quicksight.GroupMembership {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.QuickSight();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/quicksight-2018-04-01.normal.json"), this.client);
    }
    cancelIngestion(partialParams) {
        return this.client.cancelIngestion(this.ops["CancelIngestion"].apply(partialParams));
    }
    createAccountCustomization(partialParams) {
        return this.client.createAccountCustomization(this.ops["CreateAccountCustomization"].apply(partialParams));
    }
    createAnalysis(partialParams) {
        return this.client.createAnalysis(this.ops["CreateAnalysis"].apply(partialParams));
    }
    createDashboard(partialParams) {
        return this.client.createDashboard(this.ops["CreateDashboard"].apply(partialParams));
    }
    createDataSet(partialParams) {
        return this.client.createDataSet(this.ops["CreateDataSet"].apply(partialParams));
    }
    createDataSource(partialParams) {
        return this.client.createDataSource(this.ops["CreateDataSource"].apply(partialParams));
    }
    createFolder(partialParams) {
        return this.client.createFolder(this.ops["CreateFolder"].apply(partialParams));
    }
    createFolderMembership(partialParams) {
        return this.client.createFolderMembership(this.ops["CreateFolderMembership"].apply(partialParams));
    }
    createGroup(partialParams) {
        return this.client.createGroup(this.ops["CreateGroup"].apply(partialParams));
    }
    createGroupMembership(partialParams) {
        return this.client.createGroupMembership(this.ops["CreateGroupMembership"].apply(partialParams));
    }
    createIAMPolicyAssignment(partialParams) {
        return this.client.createIAMPolicyAssignment(this.ops["CreateIAMPolicyAssignment"].apply(partialParams));
    }
    createIngestion(partialParams) {
        return this.client.createIngestion(this.ops["CreateIngestion"].apply(partialParams));
    }
    createNamespace(partialParams) {
        return this.client.createNamespace(this.ops["CreateNamespace"].apply(partialParams));
    }
    createTemplate(partialParams) {
        return this.client.createTemplate(this.ops["CreateTemplate"].apply(partialParams));
    }
    createTemplateAlias(partialParams) {
        return this.client.createTemplateAlias(this.ops["CreateTemplateAlias"].apply(partialParams));
    }
    createTheme(partialParams) {
        return this.client.createTheme(this.ops["CreateTheme"].apply(partialParams));
    }
    createThemeAlias(partialParams) {
        return this.client.createThemeAlias(this.ops["CreateThemeAlias"].apply(partialParams));
    }
    deleteAccountCustomization(partialParams) {
        return this.client.deleteAccountCustomization(this.ops["DeleteAccountCustomization"].apply(partialParams));
    }
    deleteAnalysis(partialParams) {
        return this.client.deleteAnalysis(this.ops["DeleteAnalysis"].apply(partialParams));
    }
    deleteDashboard(partialParams) {
        return this.client.deleteDashboard(this.ops["DeleteDashboard"].apply(partialParams));
    }
    deleteDataSet(partialParams) {
        return this.client.deleteDataSet(this.ops["DeleteDataSet"].apply(partialParams));
    }
    deleteDataSource(partialParams) {
        return this.client.deleteDataSource(this.ops["DeleteDataSource"].apply(partialParams));
    }
    deleteFolder(partialParams) {
        return this.client.deleteFolder(this.ops["DeleteFolder"].apply(partialParams));
    }
    deleteFolderMembership(partialParams) {
        return this.client.deleteFolderMembership(this.ops["DeleteFolderMembership"].apply(partialParams));
    }
    deleteGroup(partialParams) {
        return this.client.deleteGroup(this.ops["DeleteGroup"].apply(partialParams));
    }
    deleteGroupMembership(partialParams) {
        return this.client.deleteGroupMembership(this.ops["DeleteGroupMembership"].apply(partialParams));
    }
    deleteIAMPolicyAssignment(partialParams) {
        return this.client.deleteIAMPolicyAssignment(this.ops["DeleteIAMPolicyAssignment"].apply(partialParams));
    }
    deleteNamespace(partialParams) {
        return this.client.deleteNamespace(this.ops["DeleteNamespace"].apply(partialParams));
    }
    deleteTemplate(partialParams) {
        return this.client.deleteTemplate(this.ops["DeleteTemplate"].apply(partialParams));
    }
    deleteTemplateAlias(partialParams) {
        return this.client.deleteTemplateAlias(this.ops["DeleteTemplateAlias"].apply(partialParams));
    }
    deleteTheme(partialParams) {
        return this.client.deleteTheme(this.ops["DeleteTheme"].apply(partialParams));
    }
    deleteThemeAlias(partialParams) {
        return this.client.deleteThemeAlias(this.ops["DeleteThemeAlias"].apply(partialParams));
    }
    deleteUser(partialParams) {
        return this.client.deleteUser(this.ops["DeleteUser"].apply(partialParams));
    }
    deleteUserByPrincipalId(partialParams) {
        return this.client.deleteUserByPrincipalId(this.ops["DeleteUserByPrincipalId"].apply(partialParams));
    }
    describeAccountCustomization(partialParams) {
        return this.client.describeAccountCustomization(this.ops["DescribeAccountCustomization"].apply(partialParams));
    }
    describeAccountSettings(partialParams) {
        return this.client.describeAccountSettings(this.ops["DescribeAccountSettings"].apply(partialParams));
    }
    describeAnalysis(partialParams) {
        return this.client.describeAnalysis(this.ops["DescribeAnalysis"].apply(partialParams));
    }
    describeAnalysisPermissions(partialParams) {
        return this.client.describeAnalysisPermissions(this.ops["DescribeAnalysisPermissions"].apply(partialParams));
    }
    describeDashboard(partialParams) {
        return this.client.describeDashboard(this.ops["DescribeDashboard"].apply(partialParams));
    }
    describeDashboardPermissions(partialParams) {
        return this.client.describeDashboardPermissions(this.ops["DescribeDashboardPermissions"].apply(partialParams));
    }
    describeDataSet(partialParams) {
        return this.client.describeDataSet(this.ops["DescribeDataSet"].apply(partialParams));
    }
    describeDataSetPermissions(partialParams) {
        return this.client.describeDataSetPermissions(this.ops["DescribeDataSetPermissions"].apply(partialParams));
    }
    describeDataSource(partialParams) {
        return this.client.describeDataSource(this.ops["DescribeDataSource"].apply(partialParams));
    }
    describeDataSourcePermissions(partialParams) {
        return this.client.describeDataSourcePermissions(this.ops["DescribeDataSourcePermissions"].apply(partialParams));
    }
    describeFolder(partialParams) {
        return this.client.describeFolder(this.ops["DescribeFolder"].apply(partialParams));
    }
    describeFolderPermissions(partialParams) {
        return this.client.describeFolderPermissions(this.ops["DescribeFolderPermissions"].apply(partialParams));
    }
    describeFolderResolvedPermissions(partialParams) {
        return this.client.describeFolderResolvedPermissions(this.ops["DescribeFolderResolvedPermissions"].apply(partialParams));
    }
    describeGroup(partialParams) {
        return this.client.describeGroup(this.ops["DescribeGroup"].apply(partialParams));
    }
    describeIAMPolicyAssignment(partialParams) {
        return this.client.describeIAMPolicyAssignment(this.ops["DescribeIAMPolicyAssignment"].apply(partialParams));
    }
    describeIngestion(partialParams) {
        return this.client.describeIngestion(this.ops["DescribeIngestion"].apply(partialParams));
    }
    describeIpRestriction(partialParams) {
        return this.client.describeIpRestriction(this.ops["DescribeIpRestriction"].apply(partialParams));
    }
    describeNamespace(partialParams) {
        return this.client.describeNamespace(this.ops["DescribeNamespace"].apply(partialParams));
    }
    describeTemplate(partialParams) {
        return this.client.describeTemplate(this.ops["DescribeTemplate"].apply(partialParams));
    }
    describeTemplateAlias(partialParams) {
        return this.client.describeTemplateAlias(this.ops["DescribeTemplateAlias"].apply(partialParams));
    }
    describeTemplatePermissions(partialParams) {
        return this.client.describeTemplatePermissions(this.ops["DescribeTemplatePermissions"].apply(partialParams));
    }
    describeTheme(partialParams) {
        return this.client.describeTheme(this.ops["DescribeTheme"].apply(partialParams));
    }
    describeThemeAlias(partialParams) {
        return this.client.describeThemeAlias(this.ops["DescribeThemeAlias"].apply(partialParams));
    }
    describeThemePermissions(partialParams) {
        return this.client.describeThemePermissions(this.ops["DescribeThemePermissions"].apply(partialParams));
    }
    describeUser(partialParams) {
        return this.client.describeUser(this.ops["DescribeUser"].apply(partialParams));
    }
    generateEmbedUrlForAnonymousUser(partialParams) {
        return this.client.generateEmbedUrlForAnonymousUser(this.ops["GenerateEmbedUrlForAnonymousUser"].apply(partialParams));
    }
    generateEmbedUrlForRegisteredUser(partialParams) {
        return this.client.generateEmbedUrlForRegisteredUser(this.ops["GenerateEmbedUrlForRegisteredUser"].apply(partialParams));
    }
    getDashboardEmbedUrl(partialParams) {
        return this.client.getDashboardEmbedUrl(this.ops["GetDashboardEmbedUrl"].apply(partialParams));
    }
    getSessionEmbedUrl(partialParams) {
        return this.client.getSessionEmbedUrl(this.ops["GetSessionEmbedUrl"].apply(partialParams));
    }
    listAnalyses(partialParams) {
        return this.client.listAnalyses(this.ops["ListAnalyses"].apply(partialParams));
    }
    listDashboardVersions(partialParams) {
        return this.client.listDashboardVersions(this.ops["ListDashboardVersions"].apply(partialParams));
    }
    listDashboards(partialParams) {
        return this.client.listDashboards(this.ops["ListDashboards"].apply(partialParams));
    }
    listDataSets(partialParams) {
        return this.client.listDataSets(this.ops["ListDataSets"].apply(partialParams));
    }
    listDataSources(partialParams) {
        return this.client.listDataSources(this.ops["ListDataSources"].apply(partialParams));
    }
    listFolderMembers(partialParams) {
        return this.client.listFolderMembers(this.ops["ListFolderMembers"].apply(partialParams));
    }
    listFolders(partialParams) {
        return this.client.listFolders(this.ops["ListFolders"].apply(partialParams));
    }
    listGroupMemberships(partialParams) {
        return this.client.listGroupMemberships(this.ops["ListGroupMemberships"].apply(partialParams));
    }
    listGroups(partialParams) {
        return this.client.listGroups(this.ops["ListGroups"].apply(partialParams));
    }
    listIAMPolicyAssignments(partialParams) {
        return this.client.listIAMPolicyAssignments(this.ops["ListIAMPolicyAssignments"].apply(partialParams));
    }
    listIAMPolicyAssignmentsForUser(partialParams) {
        return this.client.listIAMPolicyAssignmentsForUser(this.ops["ListIAMPolicyAssignmentsForUser"].apply(partialParams));
    }
    listIngestions(partialParams) {
        return this.client.listIngestions(this.ops["ListIngestions"].apply(partialParams));
    }
    listNamespaces(partialParams) {
        return this.client.listNamespaces(this.ops["ListNamespaces"].apply(partialParams));
    }
    listTagsForResource(partialParams) {
        return this.client.listTagsForResource(this.ops["ListTagsForResource"].apply(partialParams));
    }
    listTemplateAliases(partialParams) {
        return this.client.listTemplateAliases(this.ops["ListTemplateAliases"].apply(partialParams));
    }
    listTemplateVersions(partialParams) {
        return this.client.listTemplateVersions(this.ops["ListTemplateVersions"].apply(partialParams));
    }
    listTemplates(partialParams) {
        return this.client.listTemplates(this.ops["ListTemplates"].apply(partialParams));
    }
    listThemeAliases(partialParams) {
        return this.client.listThemeAliases(this.ops["ListThemeAliases"].apply(partialParams));
    }
    listThemeVersions(partialParams) {
        return this.client.listThemeVersions(this.ops["ListThemeVersions"].apply(partialParams));
    }
    listThemes(partialParams) {
        return this.client.listThemes(this.ops["ListThemes"].apply(partialParams));
    }
    listUserGroups(partialParams) {
        return this.client.listUserGroups(this.ops["ListUserGroups"].apply(partialParams));
    }
    listUsers(partialParams) {
        return this.client.listUsers(this.ops["ListUsers"].apply(partialParams));
    }
    registerUser(partialParams) {
        return this.client.registerUser(this.ops["RegisterUser"].apply(partialParams));
    }
    restoreAnalysis(partialParams) {
        return this.client.restoreAnalysis(this.ops["RestoreAnalysis"].apply(partialParams));
    }
    searchAnalyses(partialParams) {
        return this.client.searchAnalyses(this.ops["SearchAnalyses"].apply(partialParams));
    }
    searchDashboards(partialParams) {
        return this.client.searchDashboards(this.ops["SearchDashboards"].apply(partialParams));
    }
    searchFolders(partialParams) {
        return this.client.searchFolders(this.ops["SearchFolders"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateAccountCustomization(partialParams) {
        return this.client.updateAccountCustomization(this.ops["UpdateAccountCustomization"].apply(partialParams));
    }
    updateAccountSettings(partialParams) {
        return this.client.updateAccountSettings(this.ops["UpdateAccountSettings"].apply(partialParams));
    }
    updateAnalysis(partialParams) {
        return this.client.updateAnalysis(this.ops["UpdateAnalysis"].apply(partialParams));
    }
    updateAnalysisPermissions(partialParams) {
        return this.client.updateAnalysisPermissions(this.ops["UpdateAnalysisPermissions"].apply(partialParams));
    }
    updateDashboard(partialParams) {
        return this.client.updateDashboard(this.ops["UpdateDashboard"].apply(partialParams));
    }
    updateDashboardPermissions(partialParams) {
        return this.client.updateDashboardPermissions(this.ops["UpdateDashboardPermissions"].apply(partialParams));
    }
    updateDashboardPublishedVersion(partialParams) {
        return this.client.updateDashboardPublishedVersion(this.ops["UpdateDashboardPublishedVersion"].apply(partialParams));
    }
    updateDataSet(partialParams) {
        return this.client.updateDataSet(this.ops["UpdateDataSet"].apply(partialParams));
    }
    updateDataSetPermissions(partialParams) {
        return this.client.updateDataSetPermissions(this.ops["UpdateDataSetPermissions"].apply(partialParams));
    }
    updateDataSource(partialParams) {
        return this.client.updateDataSource(this.ops["UpdateDataSource"].apply(partialParams));
    }
    updateDataSourcePermissions(partialParams) {
        return this.client.updateDataSourcePermissions(this.ops["UpdateDataSourcePermissions"].apply(partialParams));
    }
    updateFolder(partialParams) {
        return this.client.updateFolder(this.ops["UpdateFolder"].apply(partialParams));
    }
    updateFolderPermissions(partialParams) {
        return this.client.updateFolderPermissions(this.ops["UpdateFolderPermissions"].apply(partialParams));
    }
    updateGroup(partialParams) {
        return this.client.updateGroup(this.ops["UpdateGroup"].apply(partialParams));
    }
    updateIAMPolicyAssignment(partialParams) {
        return this.client.updateIAMPolicyAssignment(this.ops["UpdateIAMPolicyAssignment"].apply(partialParams));
    }
    updateIpRestriction(partialParams) {
        return this.client.updateIpRestriction(this.ops["UpdateIpRestriction"].apply(partialParams));
    }
    updateTemplate(partialParams) {
        return this.client.updateTemplate(this.ops["UpdateTemplate"].apply(partialParams));
    }
    updateTemplateAlias(partialParams) {
        return this.client.updateTemplateAlias(this.ops["UpdateTemplateAlias"].apply(partialParams));
    }
    updateTemplatePermissions(partialParams) {
        return this.client.updateTemplatePermissions(this.ops["UpdateTemplatePermissions"].apply(partialParams));
    }
    updateTheme(partialParams) {
        return this.client.updateTheme(this.ops["UpdateTheme"].apply(partialParams));
    }
    updateThemeAlias(partialParams) {
        return this.client.updateThemeAlias(this.ops["UpdateThemeAlias"].apply(partialParams));
    }
    updateThemePermissions(partialParams) {
        return this.client.updateThemePermissions(this.ops["UpdateThemePermissions"].apply(partialParams));
    }
    updateUser(partialParams) {
        return this.client.updateUser(this.ops["UpdateUser"].apply(partialParams));
    }
}
exports.default = Resource;
