// out/acm/Certificate.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.acm.Certificate {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.ACM();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/acm-2015-12-08.normal.json"), this.client);
    }
    addTagsToCertificate(partialParams) {
        return this.client.addTagsToCertificate(this.ops["AddTagsToCertificate"].apply(partialParams));
    }
    deleteCertificate(partialParams) {
        return this.client.deleteCertificate(this.ops["DeleteCertificate"].apply(partialParams));
    }
    describeCertificate(partialParams) {
        return this.client.describeCertificate(this.ops["DescribeCertificate"].apply(partialParams));
    }
    exportCertificate(partialParams) {
        return this.client.exportCertificate(this.ops["ExportCertificate"].apply(partialParams));
    }
    getCertificate(partialParams) {
        return this.client.getCertificate(this.ops["GetCertificate"].apply(partialParams));
    }
    importCertificate(partialParams) {
        return this.client.importCertificate(this.ops["ImportCertificate"].apply(partialParams));
    }
    listTagsForCertificate(partialParams) {
        return this.client.listTagsForCertificate(this.ops["ListTagsForCertificate"].apply(partialParams));
    }
    putAccountConfiguration(partialParams) {
        return this.client.putAccountConfiguration(this.ops["PutAccountConfiguration"].apply(partialParams));
    }
    removeTagsFromCertificate(partialParams) {
        return this.client.removeTagsFromCertificate(this.ops["RemoveTagsFromCertificate"].apply(partialParams));
    }
    renewCertificate(partialParams) {
        return this.client.renewCertificate(this.ops["RenewCertificate"].apply(partialParams));
    }
    requestCertificate(partialParams) {
        return this.client.requestCertificate(this.ops["RequestCertificate"].apply(partialParams));
    }
    resendValidationEmail(partialParams) {
        return this.client.resendValidationEmail(this.ops["ResendValidationEmail"].apply(partialParams));
    }
    updateCertificateOptions(partialParams) {
        return this.client.updateCertificateOptions(this.ops["UpdateCertificateOptions"].apply(partialParams));
    }
}
exports.default = Resource;
