// out/dynamodb/TableItem.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const aws = __importStar(require("@pulumi/aws"));
const awssdk = __importStar(require("aws-sdk"));
const parse_1 = require("../parse");
class Resource extends aws.dynamodb.TableItem {
    constructor(...args) {
        super(...args);
        this.client = new awssdk.DynamoDB();
        this.ops = (0, parse_1.getResourceOperations)(this, require("../../aws-sdk-js/apis/dynamodb-2012-08-10.normal.json"), this.client);
    }
    batchExecuteStatement(partialParams) {
        return this.client.batchExecuteStatement(this.ops["BatchExecuteStatement"].apply(partialParams));
    }
    batchGetItem(partialParams) {
        return this.client.batchGetItem(this.ops["BatchGetItem"].apply(partialParams));
    }
    batchWriteItem(partialParams) {
        return this.client.batchWriteItem(this.ops["BatchWriteItem"].apply(partialParams));
    }
    createBackup(partialParams) {
        return this.client.createBackup(this.ops["CreateBackup"].apply(partialParams));
    }
    createGlobalTable(partialParams) {
        return this.client.createGlobalTable(this.ops["CreateGlobalTable"].apply(partialParams));
    }
    createTable(partialParams) {
        return this.client.createTable(this.ops["CreateTable"].apply(partialParams));
    }
    deleteBackup(partialParams) {
        return this.client.deleteBackup(this.ops["DeleteBackup"].apply(partialParams));
    }
    deleteItem(partialParams) {
        return this.client.deleteItem(this.ops["DeleteItem"].apply(partialParams));
    }
    deleteTable(partialParams) {
        return this.client.deleteTable(this.ops["DeleteTable"].apply(partialParams));
    }
    describeBackup(partialParams) {
        return this.client.describeBackup(this.ops["DescribeBackup"].apply(partialParams));
    }
    describeContinuousBackups(partialParams) {
        return this.client.describeContinuousBackups(this.ops["DescribeContinuousBackups"].apply(partialParams));
    }
    describeContributorInsights(partialParams) {
        return this.client.describeContributorInsights(this.ops["DescribeContributorInsights"].apply(partialParams));
    }
    describeExport(partialParams) {
        return this.client.describeExport(this.ops["DescribeExport"].apply(partialParams));
    }
    describeGlobalTable(partialParams) {
        return this.client.describeGlobalTable(this.ops["DescribeGlobalTable"].apply(partialParams));
    }
    describeGlobalTableSettings(partialParams) {
        return this.client.describeGlobalTableSettings(this.ops["DescribeGlobalTableSettings"].apply(partialParams));
    }
    describeKinesisStreamingDestination(partialParams) {
        return this.client.describeKinesisStreamingDestination(this.ops["DescribeKinesisStreamingDestination"].apply(partialParams));
    }
    describeTable(partialParams) {
        return this.client.describeTable(this.ops["DescribeTable"].apply(partialParams));
    }
    describeTableReplicaAutoScaling(partialParams) {
        return this.client.describeTableReplicaAutoScaling(this.ops["DescribeTableReplicaAutoScaling"].apply(partialParams));
    }
    describeTimeToLive(partialParams) {
        return this.client.describeTimeToLive(this.ops["DescribeTimeToLive"].apply(partialParams));
    }
    disableKinesisStreamingDestination(partialParams) {
        return this.client.disableKinesisStreamingDestination(this.ops["DisableKinesisStreamingDestination"].apply(partialParams));
    }
    enableKinesisStreamingDestination(partialParams) {
        return this.client.enableKinesisStreamingDestination(this.ops["EnableKinesisStreamingDestination"].apply(partialParams));
    }
    executeStatement(partialParams) {
        return this.client.executeStatement(this.ops["ExecuteStatement"].apply(partialParams));
    }
    executeTransaction(partialParams) {
        return this.client.executeTransaction(this.ops["ExecuteTransaction"].apply(partialParams));
    }
    exportTableToPointInTime(partialParams) {
        return this.client.exportTableToPointInTime(this.ops["ExportTableToPointInTime"].apply(partialParams));
    }
    getItem(partialParams) {
        return this.client.getItem(this.ops["GetItem"].apply(partialParams));
    }
    listTagsOfResource(partialParams) {
        return this.client.listTagsOfResource(this.ops["ListTagsOfResource"].apply(partialParams));
    }
    putItem(partialParams) {
        return this.client.putItem(this.ops["PutItem"].apply(partialParams));
    }
    query(partialParams) {
        return this.client.query(this.ops["Query"].apply(partialParams));
    }
    restoreTableFromBackup(partialParams) {
        return this.client.restoreTableFromBackup(this.ops["RestoreTableFromBackup"].apply(partialParams));
    }
    restoreTableToPointInTime(partialParams) {
        return this.client.restoreTableToPointInTime(this.ops["RestoreTableToPointInTime"].apply(partialParams));
    }
    scan(partialParams) {
        return this.client.scan(this.ops["Scan"].apply(partialParams));
    }
    tagResource(partialParams) {
        return this.client.tagResource(this.ops["TagResource"].apply(partialParams));
    }
    transactGetItems(partialParams) {
        return this.client.transactGetItems(this.ops["TransactGetItems"].apply(partialParams));
    }
    transactWriteItems(partialParams) {
        return this.client.transactWriteItems(this.ops["TransactWriteItems"].apply(partialParams));
    }
    untagResource(partialParams) {
        return this.client.untagResource(this.ops["UntagResource"].apply(partialParams));
    }
    updateContinuousBackups(partialParams) {
        return this.client.updateContinuousBackups(this.ops["UpdateContinuousBackups"].apply(partialParams));
    }
    updateContributorInsights(partialParams) {
        return this.client.updateContributorInsights(this.ops["UpdateContributorInsights"].apply(partialParams));
    }
    updateGlobalTable(partialParams) {
        return this.client.updateGlobalTable(this.ops["UpdateGlobalTable"].apply(partialParams));
    }
    updateGlobalTableSettings(partialParams) {
        return this.client.updateGlobalTableSettings(this.ops["UpdateGlobalTableSettings"].apply(partialParams));
    }
    updateItem(partialParams) {
        return this.client.updateItem(this.ops["UpdateItem"].apply(partialParams));
    }
    updateTable(partialParams) {
        return this.client.updateTable(this.ops["UpdateTable"].apply(partialParams));
    }
    updateTableReplicaAutoScaling(partialParams) {
        return this.client.updateTableReplicaAutoScaling(this.ops["UpdateTableReplicaAutoScaling"].apply(partialParams));
    }
    updateTimeToLive(partialParams) {
        return this.client.updateTimeToLive(this.ops["UpdateTimeToLive"].apply(partialParams));
    }
}
exports.default = Resource;
